---
title: "R Notebook"
output: html_notebook
---

#############################################
################### SETUP ###################
#############################################
### Packages
```{r}
library(Seurat)
library(ggpubr)
library(dplyr)
library(plyr)
library(tidyr)
library(harmony)
library(matrixStats)
library(patchwork)
options(future.globals.maxSize = 8000 * 1024^2)

setwd("~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Oct21")
HumanTFs = read.csv("~/Dropbox/Columbia/Useful Gene Lists/Human TFs.csv", head=F)

`%notin%` = Negate(`%in%`)
DEGs = function(Input){
Input.markers <- FindAllMarkers(Input, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25) 
return(Input.markers)}

load("~/Fetal_OCT_DEC2021_Analysis_NeuronalWorkingFiles.RData")

CheckMarkers = read.csv("OCT_AllCell_Assigns_2021-11-24.csv")
```

### Save and Load Data
```{r}
save(list=c("Fetal.combined", "Fetal.combined_Data", "HypoSTD_UMAP",  "Blood", "VLMC", "Vasculature", "Microglia", "DivProg", "Neurons", "Ependy", "BroadGenes", "HypothalamicProgenSubclass", "WellKnownGenes", "SuspectedGenes", "Zhou_Extras", "BlackshawEtc", "VariousUkn_1", "VariousUkn_2", "VariousUkn_3", "OligList", "AstroMicro", "EpendyTanyList", "EndoList", "MuralPeriVLMCList", "Fetal.integrated.HypothalV2.Harmony_RNA", "All.integrated_DataSubs", "All.integrated_DataSubsV2", "ClusterFunc_Neuro_RNA", "ClusterFunc_All_RNA", "HypothalamicNeurons", "NonHypothalamicNeurons", "Neurons_Seurat", "Neuronal_UMAP", "HumanTFs", "OCT_AllCell_Assigns", "Ependy_Final", "Tany_Final", "RadialGlia", "Astrocytes", "Oligodendro_Final", "OCTOBER_OLIGO_Assigns", "ImmatureOlig1",  "ImmatureOlig2_RECOMBO", "ImmatureOlig3_RECOMBO", "ImmatureOlig4_RECOMBO", "ImmatureOlig5_RECOMBO", "ImmatureOlig6_RECOMBO", "ImmatureOlig7_RECOMBO", "ImmatureOlig8_RECOMBO", "ImmatureOlig9_RECOMBO", "ImmatureOlig10_RECOMBO", "IntermediateOlig_RECOMBO", "MatureOlig_RECOMBO", "OCTOBER_MICROGLIA_Assigns", "Micro_1",  "Micro_2", "Micro_3", "vSMC_Clean",  "EndoArterial_Clean", "EndoVenous_Clean", "Pericytes_Clean", "Dividing_Astro_Final", "Dividing_Oligo_Final", "Dividing_Neuro_Final", "Ependy_Final_V2", "GlialProgen_Clean", "vRG_Clean", "IPC_Clean", "oRG_1_Clean", "oRG_2_Clean", "Astrocytes_Clean", "Ukn_NKX21_Clean", "CheckUMAP", "OCTOBER_EPENDY_Assigns", "Micro_Seurat", "Vasculature_Seurat", "OCTOBER_ENDOMURAL_Assigns", "Ependy_Final_Seu", "SEPTEMBER_ASTRO_Assigns", "OCTOBER_TANY_Assigns", "VLMC_Seurat", "OCTOBER_RG_Assigns"), file = "~/Fetal_OCT_DEC2021_Analysis_BroadWorkingFiles.RData")


save(list=c("Fetal.combined", "Fetal.combined_Data", "HypoSTD_UMAP", "OCT_AllCell_Assigns", "BroadGenes", "HypothalamicProgenSubclass", "WellKnownGenes", "SuspectedGenes", "Zhou_Extras", "BlackshawEtc", "VariousUkn_1", "VariousUkn_2", "VariousUkn_3", "OligList", "AstroMicro", "EpendyTanyList", "EndoList", "MuralPeriVLMCList","CheckUMAP", "Fetal.integrated.HypothalV2.Harmony_RNA", "All.integrated_DataSubs", "All.integrated_DataSubsV2", "ClusterFunc_Neuro_RNA", "ClusterFunc_All_RNA", "HypothalamicNeurons", "NonHypothalamicNeurons", "Neurons_Seurat", "Neuronal_UMAP", "HumanTFs","HypothalamicNeurons_Rest1", "HypothalamicNeurons_Rest2", "HypothalamicNeurons_Rest3", "HypothalamicNeurons_Rest4", "HypothalamicNeurons_Rest5", "HypothalamicNeurons_Rest6", "HypothalamicNeurons_Rest7", "HypothalamicNeurons_Rest8", "NKX21TRH_Clean",  "NHLH2_Clean", "Preoptic_PNOC_NTS_Clean", "SIX3_SST_Clean", "SIX3_Clean", "CCK_RELN_Clean", "VMH2_EBF4_Clean", "PENK_PNOC_Clean", "PVH_2_Clean", "TM_1_Clean", "UKN_PostMM_Clean", "PVH3_Clean",  "UKN_CRABP1", "MN_1_Clean", "UKN_Top_Clean", "Arc_1_Clean", "SCN_1_Clean", "UKN_TAC1", "LH_1", "PVH_4_Clean", "MMPost_Clean", "SIX3_V2_Clean", "TBX3_Clean", "RELN_NRN1_Clean", "MoveToPVH3", "ZI_Clean", "NPAS4_Clean", "DMH_Clean", "Arc_Precursor_Clean", "VMH_1_Clean", "PVH_1_Clean", "OCT_Neuron_Assigns", "HypothalamicNeurons_UMAP", "ExtraArc_Clean", "ExtraONECUT2", "ExtraSIX6_SST_Clean", "ExtraVMH_Clean", "ExtraNRN1_Clean", "Ukn_D_Clean", "Ukn_C_Clean", "Ukn_B_Clean", "Ukn_A_Clean","SIX3_MEIS2_",  "SIX3_SST_OTP","UKN_Top_Cleaned","ClustForLHX6_V7", "SIX3_SST_OTPpt2","Top_PVH_V4", "PNOC_Preoptic", "ForONECUT2", "SIX3_Extra", "AddToLHX6", "AddToPNOC", "AddToMN", "CheckZICs_V2", "NHLH2_CleanV2", "MN_Clean", "CheckZICs_V3","Arc_4", "Arc_5", "AddToPVH3_Clean","AddToPVH3_V2pt2", "PVH_1_CleanV2", "AVP", "OXT_Only", "OXT_MC4R_V4",  "PVH3_Clean_V2", "Arc_4_V3", "MoveToMN3", "PVH_SIM1_Extras", "PVH_SST_Extras", "PVH3_OXT", "PVH3_SIM1_Clean", "PVH3_TRH2", "AVP_Recombo", "PVH1_Rest8", "TRH_Clean", "MC4R_V3", "CRH_Recombo", "GAL_Recombo", "SST_Recombo", "OXT_Recombo", "OCT_Neuron_Assigns_Renamed2", "OCT_Neuron_Assigns_Renamed", "PVH1", "PVH1V2", "TM_NKX21_HDC_SOX2", "TM_HDC", "Arc_TBX3_Clean", "Arc_AGRP_NPY_Clean", "Arc_POMC_Clean_V2Seu", "Arc_NKX21_PCSK1_Clean", "ClustForLEPR_V5", "TM_NKX21_HDC", "AVP_Extra_Clean", "SST_Extra_Clean", "PVH1_Rest7", "ClustForMC4R_V6", "TH_Clean", "ClustForCRH_V2", "PVH_SST", "Arc_5_V2", "PVHCells", "OCT_PREOPTIC_Assigns_Renamed", "PO_Subcluster_1", "PO_Subcluster_2", "PO_Subcluster_3", "PO_Subcluster_4", "PO_Subcluster_5", "PO_Subcluster_6_Recombo", "PO_ForSubcluster_7_V3", "PO_ForSubcluster_8_V2", "PO_ForSubcluster_9_V3", "PO_Rest6", "OCT_LH_Assigns_Renamed", "LH_Cluster1", "LH_Cluster2", "LH_Cluster3", "LH_Cluster4", "LH_Cluster5_V2", "LH_Cluster6_V2", "LH_Cluster7_V2", "LH_Cluster8_V2", "LH_Cluster9", "LH_Cluster10", "OCT_PVH_Assigns_Renamed", "PVH_Sub1_Recombo", "PVH_Sub3", "PVH_Sub2", "PVH_Sub4_Recombo", "PVH_Sub5", "PVH_6", "PVH_7", "PVH_Sub8", "PVH_Sub9_V3", "PVH_Sub10_V2", "PVH_Sub10_Extra", "PVH_Sub11_V5", "PVH_Sub12_Recombo", "PVH_Sub13_V3", "PVH_Sub14", "PVH_Sub15_V2", "PVH_Sub16", "PVH_Sub17", "PVH_Sub18", "PVH_Sub19_Recombo", "PVH_Sub20_Recombo", "PVH_Sub21_Recombo" ,"PVH_Sub22_Recombo", "PVH_Sub23", "OCT_ARC_Assigns_Renamed", "ARC_1", "ARC_2", "ARC_4_V2", "ARC_5_V3", "ARC_6_V3", "ARC_7", "ARC_8", "ARC_9", "ARC_10", "ARC_11_V2", "ARC_12_V2", "ARC_13_V2", "ARC_14_V2", "All_ARCUATE_Rest10", "ARC_15_V2", "ARC_16", "ARC_3_Clean", "ARC_17", "All_ARCUATE", "All_MN", "All_LH", "All_PVH", "All_PREOPTIC",  "All_SCN", "All_ZI", "All_TM", "ZI_Cells_V3", "ARC_Cells_V3", "MoveZI", "MoveToIDTT", "Ukn48_Clean", "MoveToVMH_Comb", "UKN_Top_Cleaned_V2", "Ukn48_Clean_V2", "Arc_Cluster_V4"), file = "~/Fetal_OCT_DEC2021_Analysis_NeuronalWorkingFiles.RData")


save(list=c("Fetal.combined", "Fetal.combined_Data", "HypoSTD_UMAP", "OCT_AllCell_Assigns", "BroadGenes", "HypothalamicProgenSubclass", "WellKnownGenes", "SuspectedGenes", "Zhou_Extras", "BlackshawEtc", "VariousUkn_1", "VariousUkn_2", "VariousUkn_3", "OligList", "AstroMicro", "EpendyTanyList", "EndoList", "MuralPeriVLMCList","CheckUMAP", "Fetal.integrated.HypothalV2.Harmony_RNA", "All.integrated_DataSubs", "All.integrated_DataSubsV2", "ClusterFunc_Neuro_RNA", "ClusterFunc_All_RNA", "HypothalamicNeurons", "NonHypothalamicNeurons",   "HumanTFs", "OCT_Neuron_Assigns", "HypothalamicNeurons_UMAP", "OCT_Neuron_Assigns_Renamed2", "OCT_Neuron_Assigns_Renamed", "OCT_PREOPTIC_Assigns_Renamed", "PO_Subcluster_1", "PO_Subcluster_2", "PO_Subcluster_3", "PO_Subcluster_4", "PO_Subcluster_5", "PO_Subcluster_6_Recombo", "PO_ForSubcluster_7_V3", "PO_ForSubcluster_8_V2", "PO_ForSubcluster_9_V3", "PO_Rest6", "OCT_LH_Assigns_Renamed", "LH_Cluster1", "LH_Cluster2", "LH_Cluster3", "LH_Cluster4", "LH_Cluster5_V2", "LH_Cluster6_V2", "LH_Cluster7_V2", "LH_Cluster8_V2", "LH_Cluster9", "LH_Cluster10", "OCT_PVH_Assigns_Renamed", "PVH_Sub1_Recombo", "PVH_Sub3", "PVH_Sub2", "PVH_Sub4_Recombo", "PVH_Sub5", "PVH_6", "PVH_7", "PVH_Sub8", "PVH_Sub9_V3", "PVH_Sub10_V2", "PVH_Sub10_Extra", "PVH_Sub11_V5", "PVH_Sub12_Recombo", "PVH_Sub13_V3", "PVH_Sub14", "PVH_Sub15_V2", "PVH_Sub16", "PVH_Sub17", "PVH_Sub18", "PVH_Sub19_Recombo", "PVH_Sub20_Recombo", "PVH_Sub21_Recombo" ,"PVH_Sub22_Recombo", "PVH_Sub23", "OCT_ARC_Assigns_Renamed", "ARC_1", "ARC_2", "ARC_4_V2", "ARC_5_V3", "ARC_6_V3", "ARC_7", "ARC_8", "ARC_9", "ARC_10", "ARC_11_V2", "ARC_12_V2", "ARC_13_V2", "ARC_14_V2", "All_ARCUATE_Rest10", "ARC_15_V2", "ARC_16", "ARC_3_Clean", "ARC_17", "All_ARCUATE", "All_MN", "All_LH", "All_PVH", "All_PREOPTIC",  "All_SCN", "All_ZI", "All_TM","OCT_MN_Assigns_Renamed", "MN_3_V2", "MN_2_V3", "MN_1_V2", "MN_4_V2", "MN_4_PT2_V2", "MN_5_V2", "MN_6_V2", "MN_6_pt2", "MN_7_V2", "MN_8", "MN_MN_9_V2", "MN_10", "OCT_TM_Assigns_Renamed", "TM_1", "TM_2", "TM3_Recombo", "Keep_TM4", "TM_5_V2", "TM_6", "OCT_SCN_Assigns_Renamed", "SCN_1", "SCN_2", "SCN_3", "SCN_4_V2", "SCN_5", "SCN_6", "SCN_7_V3", "SCN_8", "SCN_9_V2", "SCN_10_V2", "SCN_11", "OCT_ZI_Assigns_Renamed", "ZI_1_V2", "ZI_2", "ZI_3_V2", "MoveToZI3", "ZI_4_V2", "ZI_5_Recombo", "ZI_6_PT2", "ZI_7", "ZI_8_V2", "ZI_9_V2", "All_ZI_Rest6", "OCT_VMH_Assigns_Renamed", "VMH_1_V3", "VMH_2_V2", "VMH_3_V2", "VMH_4", "VMH_5_V4", "VMH_6_V2", "VMH_7_V2", "VMH_8", "VMH_9_V3", "VMH_10_V2", "VMH_11_V2", "VMH_12", "VMH_13", "OCT_DMH_Assigns_Renamed", "DMH_1_V2_Seu", "DMH_2_V2_Seu", "DMH_3", "OCT_IDTT_Assigns_Renamed", "IDTT_1_V2_Seu", "IDTT_2", "IDTT_3", "All_VMH", "All_DMH", "All_SMN", "All_IDTT", "OCT_SMN_Assigns_Renamed", "SMN_1_V2", "SMN_2", "SMN_3", "SMN_4"), file = "~/Fetal_OCT_DEC2021_Analysis_GranularNeuronalWorkingFiles.RData")


save(list=c("Fetal.combined", "HypothalamicNeurons", 
            "OCT_AllCell_Assigns",  #Checked
            "OCT_Neuron_Assigns_Renamed2", #Checked
            "OCT_PREOPTIC_Assigns_Renamed", #Checked
            "OCT_TM_Assigns_Renamed", #Checked
            "OCT_IDTT_Assigns_Renamed", #Checked
            "OCT_VMH_Assigns_Renamed", #Checked
            "OCT_DMH_Assigns_Renamed", #Checked
            "OCT_ARC_Assigns_Renamed", #Checked
            "OCT_PVH_Assigns_Renamed", #Checked
            "OCT_ZI_Assigns_Renamed", #Checked
            "OCT_LH_Assigns_Renamed", #Checked
            "OCT_MN_Assigns_Renamed", #Checked
            "OCT_SMN_Assigns_Renamed", #Checked 
            "OCTOBER_ENDOMURAL_Assigns", #Checked
            "OCTOBER_EPENDY_Assigns", #Checked
            "SEPTEMBER_ASTRO_Assigns", #Checked
            "OCTOBER_TANY_Assigns", #Checked
            "OCTOBER_OLIGO_Assigns", #Checked
            "OCT_SCN_Assigns_Renamed",
            "All_SMN",
            "OCTOBER_MICROGLIA_Assigns","All_GenRG", "GenRG_SubclusterAssigns", "All_vRG", "vRG_SubclusterAssigns", "All_oRG", "oRG_SubclusterAssigns", "All_IPC", "IPC_SubclusterAssigns",
            "Micro_Seurat", "Ependy_Final_Seu", "Oligodendro_Final", "Tany_Final", "Vasculature_Seurat", "Astrocytes_Clean","All_ARCUATE", "All_MN", "All_LH", "All_PVH", "All_PREOPTIC",  "All_SCN", "All_ZI", "All_TM", "All_SCN", "All_DMH", "All_VMH", "All_IDTT"
            ), file = "~/Fetal_MainSeuratFilesPlusAssignments_DEC21.RData")


paste(sort(unique(OCT_SMN_Assigns_Renamed$Pop)), collapse = ", ")

load("~/Fetal_OCT_DEC2021_Analysis_BroadWorkingFiles.RData")
load("~/Fetal_OCT_DEC2021_Analysis_GranularNeuronalWorkingFiles.RData")
load("~/Fetal_OCT_DEC2021_Analysis_NeuronalWorkingFiles.RData")
load("~/Fetal_MainSeuratFilesPlusAssignments_DEC21.RData")

```

### Set up clustering functions 
```{r}
##Check population on UMAP
CheckInput = row.names(Ependy)
CheckUMAP = function(SeuFilez){
CheckMeta = as.data.frame(c(rep("RoI", length(CheckInput))))
colnames(CheckMeta) = "Pop"
row.names(CheckMeta) = CheckInput
SeuFilez = AddMetaData(SeuFilez, CheckMeta, "CheckMeta")
Idents(SeuFilez) = "CheckMeta"
DimPlot(SeuFilez, reduction="umap")
}

BroadGenes = c("SYT1", "SNAP25", "TUBB3","SYP", #Broad Neuronal 
               "SLC32A1", "DLX1", "DLX2", "DLX5", "GAD1", #GABAergic
           "SLC17A6", "SLC17A8") #Glutaminergic
HypothalamicProgenSubclass =c("VIM", "ASCL1", #Hypothalamic Progenitor - And/or
            "HES1", "MKI67", "FABP7", "TTYH1", "HMGA2", "MEIS2", "SLC1A3", "FAM107A", "NKX2-1", "NHLH2", "DLX1", "OLIG2", "DLX6-AS1", "STMN2", "HOPX", "HES1", "NES", "SOX2", "EGFR", "DLL1", "CLDN10", "FAM107A", "GADD45G", "HES6", "SLC1A2") #Hypothalamic subclasses
WellKnownGenes = c("SIM1", "NKX2-1", "MC4R", "SST", "KISS1", "GHRH", "TH", "TRH", "OXT", "AVP", "CRH", "BDNF", "PCSK1", "ADCYAP1", "GAL","POMC","NPY","AGRP", "HDC", "RBFOX3", "TBX3", "ISL1",  "SIX6",  "OTP", "POU3F2", "NRN1", "GABRA5", "NPTX2", "HCRT", "CARTPT", "FOXG1", "NEUROD6") #Known Hypothalamus
SuspectedGenes = c("SETBP1", "TAC1", "PITX2", "FOXP1", "FOXP2", "FOXB1", "SOX14", "MASP1","PPP1R17", "EOMES", "ELAVL4", "NEUROG1", "NEUROD1", "NEUROD4", "PENK", "RGS16", "CHD7", "NPVF", "RELN", "ZEB1", "SOX3", "LEPR", "PVALB") #Suspected Hypothalamus
Zhou_Extras =c("FEZF2", "LHX1", "VAX1", "ONECUT2", "SIX6", "NKX2-2", "GABRA2", "POU3F4", "FEZF1", "NR5A1", "NPTX2", "GSX1", "LHX9", "PDYN", "HMX2", "BUB3", "DLX6", "EMX2", "LHX5", "BARHL1", "IRX3", "NR4A2", "ARX", "SP9", "TBR1") #Zhou subcatergories of neurons not previously implicated
 ####Known from Blackshaw/ABA I believe           
 BlackshawEtc = c("SIX3", "NPY1R", "CRABP1", "NPAS4", #Arc
 "PCP4", #Arc/LH
 "RASL10A", "NTS", "PMCH", "LHX2", #LH
 "RGS8", #MMPost
"GRP", "VIP", "CCK", #SCN 
"RAMP3", "SEMA3C", "TIAM2", "CHRM2", #Thalamus or Mamillary Area
"ST18", "SYNE2", #Preoptic
"PAX7", #TT
"SP8", "PAX6", #Prethalamus 
"NR2F2", #AntID or Mamil 
"LHX6", #ID
"LHX8", #DMH 
"CITED1", #ARC or PMN 
"CALB2", #AndID or PVH/SON or SMN, 
"PBX1", #AndID or PVH/SON or POA/SCN 
"ZFHX3", #AndID or PVH/SON or Mamil 
"MEIS3", #POA/SCN
"PBX3", #Prethal or PVH/SON
"PENK", #AntID or SMN or LH 
"NHLH2", #Mamil or SON 
"PNOC", "CALB1")  #Blackshaw Extras
 
VariousUkn_1 = c("SRM", "RGS4", "KLHDC8B", "LYPD1", "NEUROD1", "CALCR", "CHD8", "CHN2", "COL19A1", "GALNTL6", "CXCL14", "LPL", "SOX6", "CEP70", "SV2C", "MAOB", "MAOA", "RSU1", "SLC18A2", "CDH11", "CDH23", "HS6ST1", "BTG2", "MYO1B", "HIST1H1E", "GLCCI1", "TOP2A", "GNG5", "PLPP3", "IRF2BPL", "BARHL2", "CORT", "NEB", "ARC", "ATAD2", "CCND2", "CDK6", "CENPU", "FAM111A", "GINS2")

VariousUkn_2 = c('GPR50', "PCSK5", "GRP", "SULF1", "RORB", "PARPBP", "LYPD6","LYPD6B", "GPR61", "NRP1", "SYNPR", "TRHR", "CBLN2", "MET", "ERBB4", "RGS16", "CCND2", "ZMIZ1", "CACHD1", "GALR1", "NTS", "SKOR1", "KIRREL3", "FAM107B", "JUNB", "KLF6", "DLX1", "RAX", "LHX8", # DMH Lee et al 2013)
                 "PMCH", "NRGN", "ZIC1", "EBF3", "GPR101", "CBLN2", "SYNPR", "SYT2", "CPLX1")  

#VariousUkn_3 = c("CORO1C", "CXCR4", "FERMT1", "AKNAD1", "ANGPT2", "FYB", "CRYM", "CXorf57", "FAM92B", "TPPP3", "CSRP2", "FRMD4B", "H3F3C", "NRN1", "PDGFRA", "PMP22", "GFAP", "AQP4", "S100B", "NPAS3", "RFX4", "ETV1", "HEY1", "HIST1H1C", "HIST1H1D", "HIVEP2", "INSM1", "MBNL3", "MED12", "MN1", "NR3C1", "NR5A2", "SATB2", "SGK3", "SOX9", "TCF7L2", "ZNF844")

VariousUkn_3  = c("BCL11B", "HIVEP2", #TF
"RAB3B", "DNAJC6", "DIRAS3", "OLFM3", "SYT6", "S100A10", "FAM84A", "FAM84A", "VSNL1", "HS6ST1", "CTNNB1", "CAMKV", "CADM2", "KIT", "PPP3CA", "SPOCK3", "DYNC1I1", "MAP7D2", "PNCK", "NEFM", "DNAJC12", "ATP8A2", "KCNMB2","HEY1", "FEZF1", "EBF2", "EBF3",  "LHX5", "LHX6", "NR3C1", "ONECUT3", "POU6F2", "PRDM8", "PROX1", "TBR1", "ZIC1", "ZIC2", "ZIC5")

OligList = unique(c("APOD", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9"))
AstroMicro = unique(c("GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B",  "ALDOC", "SLC1A2", "ID3", "ASCL1", "BTG2", "APOE", "C1QB", "AIF1", "P2RY12", "MRC1"))
EpendyTanyList = c("FOXJ1", "RAX", "ADM",  "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "RFX2", "RFX3", "RFX4", "NPAS3", "COL23A1", "CD59", "SLC17A8", "VCAN", "FRZB", "PENK", "COL25A1", "CACNA2D2", "STOML3", "TMEM212", "SLC7A11", "PCP4L1", "PLTP", "TGFB2", "NR2E1", "EPHB1", "P3H2", "NELL2", "FGF10", "RGCC", "GRIA2", "RLBP1", "SIX6", "SCN7A", "MEST", "IGFBP5", "RGS7BP")
EndoList = c("SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1")
MuralPeriVLMCList = c("VTN",  "COL1A1", "COL3A1", "LUM", "DCN", "PTGDS", "AMBP", "RGS5", "ACTA2", "TAGLN", "ABCC9", "KCNJ8") #3 GenMural, 6 VLMC, 2 Peri, 1 SMA



Fetal.integrated.HypothalV2.Harmony_RNA = as.data.frame(t(as.data.frame(Fetal.combined@assays$RNA@counts)))
All.integrated_DataSubs = Fetal.integrated.HypothalV2.Harmony_RNA %>% dplyr::select(SIM1, `NKX2-1`, MC4R, SST, KISS1, GHRH, TH, TRH, OXT, AVP, CRH, BDNF, PCSK1, GAL,POMC,LEPR, NPY,AGRP, TBX3,  SIX6,  OTP, POU3F2, GABRA5, NPTX2, HCRT, CARTPT, ONECUT2, FOXP2, HDC, CCK, LHX6, LHX9, ARC, NRN1, LHX8, PMCH, `NKX2-2`, LHX1, SIX3, LHX8, PMCH, `NKX2-2`, LHX1, SIX3, SEMA3C, PITX2, PPP1R17, NPY1R, FOXG1, NEUROD6, RGS16,  BARHL1, BCL11A, BCL11B, DMRTA2, EBF2, EMX2, HEY1, LMX1A, MEIS2, MESP1, NHLH2, NHLH1, NME2,  `NKX2-2`, NPAS4, NR2F1, NR2F2, NR5A2, NR5A1, ONECUT1, ONECUT3, OTX2, PBX3, PITX2, PRDM8, SOX14, SOX3, ST18, TBR1, TCF7L2, ZBTB16, ZIC1, ZIC2, LHX1, TAC3, BCL11B, HIVEP2,
RAB3B, DNAJC6, DIRAS3, OLFM3, SYT6, S100A10, FAM84A, FAM84A, VSNL1, HS6ST1, CTNNB1, CAMKV, CADM2, KIT, PPP3CA, SPOCK3, DYNC1I1, MAP7D2, PNCK, NEFM, DNAJC12, ATP8A2, KCNMB2)

All.integrated_DataSubsV2 = Fetal.integrated.HypothalV2.Harmony_RNA %>% select(OligList, AstroMicro, EpendyTanyList, EndoList, MuralPeriVLMCList, BroadGenes, HypothalamicProgenSubclass, WellKnownGenes, SuspectedGenes)

   
ClusterFunc_Neuro = function(SeuFile){
Filename = as.character(substitute(SeuFile))

for(y in set.kparam){
for(z in set.dim){
for(v in set.res){
DefaultAssay(SeuFile) = "integrated"
SeuFile <- FindNeighbors(SeuFile, k.param=y, dims=1:z)
SeuFile <- FindClusters(SeuFile, resolution = v)
DefaultAssay(SeuFile) = "SCT"

pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapSML.pdf", sep=""), width=12, height=10)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapLGE.pdf", sep=""), width=25, height=25)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()

###OUTPUT MEAN OF COUNTS
Subset = subset(All.integrated_DataSubs, row.names(All.integrated_DataSubs) %in% colnames(SeuFile))
MetaData = as.data.frame(SeuFile@meta.data$seurat_clusters)
row.names(MetaData) = row.names(SeuFile@meta.data)
SubsetwMeta = merge(Subset, MetaData, by=0)
row.names(SubsetwMeta) = SubsetwMeta$Row.names
SubsetwMeta$Row.names = NULL

SubsetSum = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(sum, na.rm = TRUE) 
write.csv(SubsetSum, paste("ALLFETAL_", Filename, "_SumCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)

SubsetMean = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(mean, na.rm = TRUE) 
write.csv(SubsetMean, paste("ALLFETAL_", Filename, "_MeanCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)


###VLNS
AllVlns = list()

StorePlots = list()
  for( in BroadGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12)) + labs(title = "BroadGenes", subtitle = BroadGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in BroadGenes[2:length(BroadGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BroadGenes)-1)), ncol = 40,  nrow = 1)
      

StorePlots = list()
  for( in HypothalamicProgenSubclass[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "HypothalamicProgenSubclass", subtitle = HypothalamicProgenSubclass[1])
            StorePlots[[x]] = plotA 
            }
  for( in HypothalamicProgenSubclass[2:length(HypothalamicProgenSubclass)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot2"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(HypothalamicProgenSubclass)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in WellKnownGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "WellKnownGenes", subtitle = WellKnownGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in WellKnownGenes[2:length(WellKnownGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot3"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(WellKnownGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in SuspectedGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "SuspectedGenes", subtitle = SuspectedGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in SuspectedGenes[2:length(SuspectedGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot4"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(SuspectedGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in BlackshawEtc[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "BlackshawEtc", subtitle = BlackshawEtc[1])
            StorePlots[[x]] = plotA 
            }
  for( in BlackshawEtc[2:length(BlackshawEtc)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot5"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BlackshawEtc)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in Zhou_Extras[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Zhou_Extras", subtitle = Zhou_Extras[1])
            StorePlots[[x]] = plotA 
            }
  for( in Zhou_Extras[2:length(Zhou_Extras)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot6"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(Zhou_Extras)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in VariousUkn_1[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "VariousUkn", subtitle = VariousUkn_1[1])
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_1[2:length(VariousUkn_1)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot7"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_1)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in VariousUkn_2[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_2[2:length(VariousUkn_2)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot8"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_2)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in VariousUkn_3[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_3[2:length(VariousUkn_3)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot9"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_3)-1)), ncol = 40,  nrow = 1)


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_AllMultiVlns.pdf", sep=""), width=40, height=length(unique(SeuFile@active.ident)))
print(AllVlns)
dev.off()

#FeaturePlot(SeuFile, "POMC", reduction="umap", split.by = "seurat_clusters")


CellNo = as.data.frame(table(SeuFile@meta.data$seurat_clusters))
write.csv(CellNo, paste("ALLFETAL_", Filename, "_counts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)
}}

  
FPList = list()  
FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BroadGenes[1], reduction="umap") + labs(title=paste("BroadGenes", BroadGenes[1]))
for(p in seq(2, length(BroadGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BroadGenes[p], reduction="umap") 
}
FPList[["Plotz1"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[1], reduction="umap") + labs(title=paste("HypothalamicProgenSubclass", HypothalamicProgenSubclass[1], sep=" - "))
for(p in seq(2, length(HypothalamicProgenSubclass), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[p], reduction="umap") 
}
FPList[["Plotz2"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, WellKnownGenes[1], reduction="umap") + labs(title=paste("WellKnownGenes", WellKnownGenes[1]))
for(p in seq(2, length(WellKnownGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, WellKnownGenes[p], reduction="umap") 
}
FPList[["Plotz3"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, SuspectedGenes[1], reduction="umap") + labs(title=paste("SuspectedGenes", SuspectedGenes[1]))
for(p in seq(2, length(SuspectedGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, SuspectedGenes[p], reduction="umap") 
}
FPList[["Plotz4"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)


FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BlackshawEtc[1], reduction="umap") + labs(title=paste("BlackshawEtc", BlackshawEtc[1]))
for(p in seq(2, length(BlackshawEtc), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BlackshawEtc[p], reduction="umap") 
}
FPList[["Plotz5"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, Zhou_Extras[1], reduction="umap") + labs(title=paste("Zhou_Extras", Zhou_Extras[1]))
for(p in seq(2, length(Zhou_Extras), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, Zhou_Extras[p], reduction="umap") 
}
FPList[["Plotz6"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, VariousUkn_1[1], reduction="umap") + labs(title=paste("VariousUkn", VariousUkn_1[1]))
for(p in seq(2, length(VariousUkn_1), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_1[p], reduction="umap") 
}
FPList[["Plotz7"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
for(p in seq(1, length(VariousUkn_2), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_2[p], reduction="umap") 
}
FPList[["Plotz8"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
for(p in seq(1, length(VariousUkn_3), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_3[p], reduction="umap") 
}
FPList[["Plotz19"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)


pdf(paste("ALLFETAL_", Filename, "_dim", z, "_AllFPs.pdf", sep=""), width = 25, height = 45)
print(FPList)
dev.off()
}}

ClusterFunc_Neuro_RNA = function(SeuFile){
Filename = as.character(substitute(SeuFile))

for(y in set.kparam){
for(z in set.dim){
for(v in set.res){
DefaultAssay(SeuFile) = "integrated"
SeuFile <- FindNeighbors(SeuFile, k.param=y, dims=1:z)
SeuFile <- FindClusters(SeuFile, resolution = v)
DefaultAssay(SeuFile) = "RNA"

pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapSML.pdf", sep=""), width=12, height=10)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapLGE.pdf", sep=""), width=25, height=25)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()

###OUTPUT MEAN OF COUNTS
Subset = subset(All.integrated_DataSubs, row.names(All.integrated_DataSubs) %in% colnames(SeuFile))
MetaData = as.data.frame(SeuFile@meta.data$seurat_clusters)
row.names(MetaData) = row.names(SeuFile@meta.data)
SubsetwMeta = merge(Subset, MetaData, by=0)
row.names(SubsetwMeta) = SubsetwMeta$Row.names
SubsetwMeta$Row.names = NULL

SubsetSum = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(sum, na.rm = TRUE) 
write.csv(SubsetSum, paste("ALLFETAL_", Filename, "_SumCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)

SubsetMean = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(mean, na.rm = TRUE) 
write.csv(SubsetMean, paste("ALLFETAL_", Filename, "_MeanCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)


###VLNS
AllVlns = list()

StorePlots = list()
  for( in BroadGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12)) + labs(title = "BroadGenes", subtitle = BroadGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in BroadGenes[2:length(BroadGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BroadGenes)-1)), ncol = 40,  nrow = 1)
      

StorePlots = list()
  for( in HypothalamicProgenSubclass[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "HypothalamicProgenSubclass", subtitle = HypothalamicProgenSubclass[1])
            StorePlots[[x]] = plotA 
            }
  for( in HypothalamicProgenSubclass[2:length(HypothalamicProgenSubclass)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot2"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(HypothalamicProgenSubclass)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in WellKnownGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "WellKnownGenes", subtitle = WellKnownGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in WellKnownGenes[2:length(WellKnownGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot3"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(WellKnownGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in SuspectedGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "SuspectedGenes", subtitle = SuspectedGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in SuspectedGenes[2:length(SuspectedGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot4"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(SuspectedGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in BlackshawEtc[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "BlackshawEtc", subtitle = BlackshawEtc[1])
            StorePlots[[x]] = plotA 
            }
  for( in BlackshawEtc[2:length(BlackshawEtc)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot5"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BlackshawEtc)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in Zhou_Extras[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Zhou_Extras", subtitle = Zhou_Extras[1])
            StorePlots[[x]] = plotA 
            }
  for( in Zhou_Extras[2:length(Zhou_Extras)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot6"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(Zhou_Extras)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in VariousUkn_1[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "VariousUkn", subtitle = VariousUkn_1[1])
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_1[2:length(VariousUkn_1)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot7"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_1)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in VariousUkn_2[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_2[2:length(VariousUkn_2)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot8"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_2)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in VariousUkn_3[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_3[2:length(VariousUkn_3)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot9"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_3)-1)), ncol = 40,  nrow = 1)


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_AllMultiVlns.pdf", sep=""), width=40, height=length(unique(SeuFile@active.ident)))
print(AllVlns)
dev.off()

#FeaturePlot(SeuFile, "POMC", reduction="umap", split.by = "seurat_clusters")


CellNo = as.data.frame(table(SeuFile@meta.data$seurat_clusters))
write.csv(CellNo, paste("ALLFETAL_", Filename, "_counts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)
}}

  
FPList = list()  
FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BroadGenes[1], reduction="umap") + labs(title=paste("BroadGenes", BroadGenes[1]))
for(p in seq(2, length(BroadGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BroadGenes[p], reduction="umap") 
}
FPList[["Plotz1"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[1], reduction="umap") + labs(title=paste("HypothalamicProgenSubclass", HypothalamicProgenSubclass[1], sep=" - "))
for(p in seq(2, length(HypothalamicProgenSubclass), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[p], reduction="umap") 
}
FPList[["Plotz2"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, WellKnownGenes[1], reduction="umap") + labs(title=paste("WellKnownGenes", WellKnownGenes[1]))
for(p in seq(2, length(WellKnownGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, WellKnownGenes[p], reduction="umap") 
}
FPList[["Plotz3"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, SuspectedGenes[1], reduction="umap") + labs(title=paste("SuspectedGenes", SuspectedGenes[1]))
for(p in seq(2, length(SuspectedGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, SuspectedGenes[p], reduction="umap") 
}
FPList[["Plotz4"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)


FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BlackshawEtc[1], reduction="umap") + labs(title=paste("BlackshawEtc", BlackshawEtc[1]))
for(p in seq(2, length(BlackshawEtc), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BlackshawEtc[p], reduction="umap") 
}
FPList[["Plotz5"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, Zhou_Extras[1], reduction="umap") + labs(title=paste("Zhou_Extras", Zhou_Extras[1]))
for(p in seq(2, length(Zhou_Extras), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, Zhou_Extras[p], reduction="umap") 
}
FPList[["Plotz6"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, VariousUkn_1[1], reduction="umap") + labs(title=paste("VariousUkn", VariousUkn_1[1]))
for(p in seq(2, length(VariousUkn_1), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_1[p], reduction="umap") 
}
FPList[["Plotz7"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
for(p in seq(1, length(VariousUkn_2), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_2[p], reduction="umap") 
}
FPList[["Plotz8"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
for(p in seq(1, length(VariousUkn_3), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_3[p], reduction="umap") 
}
FPList[["Plotz19"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)


pdf(paste("ALLFETAL_", Filename, "_dim", z, "_AllFPs.pdf", sep=""), width = 25, height = 45)
print(FPList)
dev.off()
}}


ClusterFunc_All = function(SeuFile){
Filename = as.character(substitute(SeuFile))

for(y in set.kparam){
for(z in set.dim){
for(v in set.res){
DefaultAssay(SeuFile) = "integrated"
SeuFile <- FindNeighbors(SeuFile, k.param=y, dims=1:z)
SeuFile <- FindClusters(SeuFile, resolution = v)
DefaultAssay(SeuFile) = "SCT"

pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapSML.pdf", sep=""), width=12, height=10)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapLGE.pdf", sep=""), width=25, height=25)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()

###OUTPUT MEAN OF COUNTS
Subset = subset(All.integrated_DataSubsV2, row.names(All.integrated_DataSubsV2) %in% colnames(SeuFile))
MetaData = as.data.frame(SeuFile@meta.data$seurat_clusters)
row.names(MetaData) = row.names(SeuFile@meta.data)
SubsetwMeta = merge(Subset, MetaData, by=0)
row.names(SubsetwMeta) = SubsetwMeta$Row.names
SubsetwMeta$Row.names = NULL

SubsetSum = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(sum, na.rm = TRUE) 
write.csv(SubsetSum, paste("ALLFETAL_", Filename, "_SumCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)

SubsetMean = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(mean, na.rm = TRUE) 
write.csv(SubsetMean, paste("ALLFETAL_", Filename, "_MeanCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)


###VLNS
AllVlns = list()

StorePlots = list()
  for( in OligList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Olig", subtitle = OligList[1])
            StorePlots[[x]] = plotA 
            }
  for( in OligList[2:length(OligList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotA"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(OligList)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in AstroMicro[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "AstroMicro", subtitle = AstroMicro[1])
            StorePlots[[x]] = plotA 
            }
  for( in AstroMicro[2:length(AstroMicro)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotB"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(AstroMicro)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in EpendyTanyList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "EpendyTanyList", subtitle = EpendyTanyList[1])
            StorePlots[[x]] = plotA 
            }
  for( in EpendyTanyList[2:length(EpendyTanyList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotC"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(EpendyTanyList)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in EndoList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Endo", subtitle = EndoList[1])
            StorePlots[[x]] = plotA 
            }
  for( in EndoList[2:length(EndoList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotD"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(EndoList)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in MuralPeriVLMCList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "MuralPeriVLMC", subtitle = MuralPeriVLMCList[1])
            StorePlots[[x]] = plotA 
            }
  for( in MuralPeriVLMCList[2:length(MuralPeriVLMCList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotE"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(MuralPeriVLMCList)-1)), ncol = 40,  nrow = 1)





StorePlots = list()
  for( in BroadGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12)) + labs(title = "BroadNeuronal", subtitle = BroadGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in BroadGenes[2:length(BroadGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotF"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BroadGenes)-1)), ncol = 40,  nrow = 1)
      

StorePlots = list()
  for( in HypothalamicProgenSubclass[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "HypothalamicProgen", subtitle = HypothalamicProgenSubclass[1])
            StorePlots[[x]] = plotA 
            }
  for( in HypothalamicProgenSubclass[2:length(HypothalamicProgenSubclass)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotG"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(HypothalamicProgenSubclass)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in WellKnownGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "WellKnownGenes", subtitle = WellKnownGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in WellKnownGenes[2:length(WellKnownGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotH"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(WellKnownGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in SuspectedGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "SuspectedGenes", subtitle = SuspectedGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in SuspectedGenes[2:length(SuspectedGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotI"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(SuspectedGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in BlackshawEtc[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "BlackshawEtc", subtitle = BlackshawEtc[1])
            StorePlots[[x]] = plotA 
            }
  for( in BlackshawEtc[2:length(BlackshawEtc)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotJ"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BlackshawEtc)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in Zhou_Extras[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Zhou_Extras", subtitle = Zhou_Extras[1])
            StorePlots[[x]] = plotA 
            }
  for( in Zhou_Extras[2:length(Zhou_Extras)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotK"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(Zhou_Extras)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in VariousUkn_3[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "LocalizedNeuralTFs", subtitle = Zhou_Extras[1])
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_3[2:length(VariousUkn_3)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotL"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_3)-1)), ncol = 40,  nrow = 1)


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_AllMultiVlns.pdf", sep=""), width=40, height=length(unique(SeuFile@active.ident)))
print(AllVlns)
dev.off()



CellNo = as.data.frame(table(SeuFile@meta.data$seurat_clusters))
write.csv(CellNo, paste("ALLFETAL_", Filename, "_counts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)
}}

  
FPList = list()  
FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, OligList[1], reduction="umap") + labs(title=paste("Olig", OligList[1]))
for(p in seq(2, length(OligList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, OligList[p], reduction="umap") 
}
FPList[["FP1"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, AstroMicro[1], reduction="umap") + labs(title=paste("AstroMicro", AstroMicro[1]))
for(p in seq(2, length(AstroMicro), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, AstroMicro[p], reduction="umap") 
}
FPList[["FP2"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, EpendyTanyList[1], reduction="umap") + labs(title=paste("EpendyTany", EpendyTanyList[1]))
for(p in seq(2, length(EpendyTanyList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, EpendyTanyList[p], reduction="umap") 
}
FPList[["FP3"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, EndoList[1], reduction="umap") + labs(title=paste("EndoList", EndoList[1]))
for(p in seq(2, length(EndoList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, EndoList[p], reduction="umap") 
}
FPList[["FP4"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, MuralPeriVLMCList[1], reduction="umap") + labs(title=paste("MuralPeriVLMC", MuralPeriVLMCList[1]))
for(p in seq(2, length(MuralPeriVLMCList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, MuralPeriVLMCList[p], reduction="umap") 
}
FPList[["FP5"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)
  
FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BroadGenes[1], reduction="umap") + labs(title=paste("BroadGenes", BroadGenes[1]))
for(p in seq(2, length(BroadGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BroadGenes[p], reduction="umap") 
}
FPList[["FP6"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[1], reduction="umap") + labs(title=paste("HypothalamicProgenSubclass", HypothalamicProgenSubclass[1], sep=" - "))
for(p in seq(2, length(HypothalamicProgenSubclass), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[p], reduction="umap") 
}
FPList[["FP7"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, WellKnownGenes[1], reduction="umap") + labs(title=paste("WellKnownGenes", WellKnownGenes[1]))
for(p in seq(2, length(WellKnownGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, WellKnownGenes[p], reduction="umap") 
}
FPList[["F8"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, SuspectedGenes[1], reduction="umap") + labs(title=paste("SuspectedGenes", SuspectedGenes[1]))
for(p in seq(2, length(SuspectedGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, SuspectedGenes[p], reduction="umap") 
}
FPList[["FP9"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)


FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BlackshawEtc[1], reduction="umap") + labs(title=paste("BlackshawEtc", BlackshawEtc[1]))
for(p in seq(2, length(BlackshawEtc), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BlackshawEtc[p], reduction="umap") 
}
FPList[["FP10"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, Zhou_Extras[1], reduction="umap") + labs(title=paste("Zhou_Extras", Zhou_Extras[1]))
for(p in seq(2, length(Zhou_Extras), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, Zhou_Extras[p], reduction="umap") 
}
FPList[["FP11"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, VariousUkn_3[1], reduction="umap") + labs(title=paste("LocalizedNeuronalTFs", VariousUkn_3[1]))
for(p in seq(2, length(VariousUkn_3), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_3[p], reduction="umap") 
}
FPList[["FP12"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

pdf(paste("ALLFETAL_", Filename, "_dim", z, "_AllFPs.pdf", sep=""), width = 25, height = 45)
print(FPList)
dev.off()
}}


ClusterFunc_All_RNA = function(SeuFile){
Filename = as.character(substitute(SeuFile))

for(y in set.kparam){
for(z in set.dim){
for(v in set.res){
DefaultAssay(SeuFile) = "integrated"
SeuFile <- FindNeighbors(SeuFile, k.param=y, dims=1:z)
SeuFile <- FindClusters(SeuFile, resolution = v)
DefaultAssay(SeuFile) = "RNA"

pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapSML.pdf", sep=""), width=12, height=10)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_umapLGE.pdf", sep=""), width=25, height=25)
dimplot = DimPlot(SeuFile, reduction="umap", label=T)
print(dimplot)
dev.off()

###OUTPUT MEAN OF COUNTS
Subset = subset(All.integrated_DataSubsV2, row.names(All.integrated_DataSubsV2) %in% colnames(SeuFile))
MetaData = as.data.frame(SeuFile@meta.data$seurat_clusters)
row.names(MetaData) = row.names(SeuFile@meta.data)
SubsetwMeta = merge(Subset, MetaData, by=0)
row.names(SubsetwMeta) = SubsetwMeta$Row.names
SubsetwMeta$Row.names = NULL

SubsetSum = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(sum, na.rm = TRUE) 
write.csv(SubsetSum, paste("ALLFETAL_", Filename, "_SumCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)

SubsetMean = Subset %>% group_by(SubsetwMeta$`SeuFile@meta.data$seurat_clusters`) %>% summarise_all(mean, na.rm = TRUE) 
write.csv(SubsetMean, paste("ALLFETAL_", Filename, "_MeanCounts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)


###VLNS
AllVlns = list()

StorePlots = list()
  for( in OligList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Olig", subtitle = OligList[1])
            StorePlots[[x]] = plotA 
            }
  for( in OligList[2:length(OligList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotA"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(OligList)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in AstroMicro[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "AstroMicro", subtitle = AstroMicro[1])
            StorePlots[[x]] = plotA 
            }
  for( in AstroMicro[2:length(AstroMicro)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotB"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(AstroMicro)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in EpendyTanyList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "EpendyTanyList", subtitle = EpendyTanyList[1])
            StorePlots[[x]] = plotA 
            }
  for( in EpendyTanyList[2:length(EpendyTanyList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotC"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(EpendyTanyList)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in EndoList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Endo", subtitle = EndoList[1])
            StorePlots[[x]] = plotA 
            }
  for( in EndoList[2:length(EndoList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotD"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(EndoList)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in MuralPeriVLMCList[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "MuralPeriVLMC", subtitle = MuralPeriVLMCList[1])
            StorePlots[[x]] = plotA 
            }
  for( in MuralPeriVLMCList[2:length(MuralPeriVLMCList)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotE"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(MuralPeriVLMCList)-1)), ncol = 40,  nrow = 1)





StorePlots = list()
  for( in BroadGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12)) + labs(title = "BroadNeuronal", subtitle = BroadGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in BroadGenes[2:length(BroadGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotF"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BroadGenes)-1)), ncol = 40,  nrow = 1)
      

StorePlots = list()
  for( in HypothalamicProgenSubclass[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "HypothalamicProgen", subtitle = HypothalamicProgenSubclass[1])
            StorePlots[[x]] = plotA 
            }
  for( in HypothalamicProgenSubclass[2:length(HypothalamicProgenSubclass)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotG"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(HypothalamicProgenSubclass)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in WellKnownGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "WellKnownGenes", subtitle = WellKnownGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in WellKnownGenes[2:length(WellKnownGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotH"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(WellKnownGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in SuspectedGenes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "SuspectedGenes", subtitle = SuspectedGenes[1])
            StorePlots[[x]] = plotA 
            }
  for( in SuspectedGenes[2:length(SuspectedGenes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotI"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(SuspectedGenes)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in BlackshawEtc[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "BlackshawEtc", subtitle = BlackshawEtc[1])
            StorePlots[[x]] = plotA 
            }
  for( in BlackshawEtc[2:length(BlackshawEtc)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotJ"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(BlackshawEtc)-1)), ncol = 40,  nrow = 1)

StorePlots = list()
  for( in Zhou_Extras[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "Zhou_Extras", subtitle = Zhou_Extras[1])
            StorePlots[[x]] = plotA 
            }
  for( in Zhou_Extras[2:length(Zhou_Extras)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotK"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(Zhou_Extras)-1)), ncol = 40,  nrow = 1)


StorePlots = list()
  for( in VariousUkn_3[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))+ labs(title = "LocalizedNeuralTFs", subtitle = Zhou_Extras[1])
            StorePlots[[x]] = plotA 
            }
  for( in VariousUkn_3[2:length(VariousUkn_3)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["PlotL"]] <- ggarrange(plotlist = StorePlots, widths=c(1.4, rep(1, length(VariousUkn_3)-1)), ncol = 40,  nrow = 1)


pdf(paste("ALLFETAL_", Filename, "_res", v, "_k", y, "_dim", z, "_AllMultiVlns.pdf", sep=""), width=40, height=length(unique(SeuFile@active.ident)))
print(AllVlns)
dev.off()



CellNo = as.data.frame(table(SeuFile@meta.data$seurat_clusters))
write.csv(CellNo, paste("ALLFETAL_", Filename, "_counts_k", y, "_dim", z, "_res", v, ".csv", sep=""), row.names = F)
}}

  
FPList = list()  
FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, OligList[1], reduction="umap") + labs(title=paste("Olig", OligList[1]))
for(p in seq(2, length(OligList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, OligList[p], reduction="umap") 
}
FPList[["FP1"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, AstroMicro[1], reduction="umap") + labs(title=paste("AstroMicro", AstroMicro[1]))
for(p in seq(2, length(AstroMicro), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, AstroMicro[p], reduction="umap") 
}
FPList[["FP2"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, EpendyTanyList[1], reduction="umap") + labs(title=paste("EpendyTany", EpendyTanyList[1]))
for(p in seq(2, length(EpendyTanyList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, EpendyTanyList[p], reduction="umap") 
}
FPList[["FP3"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, EndoList[1], reduction="umap") + labs(title=paste("EndoList", EndoList[1]))
for(p in seq(2, length(EndoList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, EndoList[p], reduction="umap") 
}
FPList[["FP4"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, MuralPeriVLMCList[1], reduction="umap") + labs(title=paste("MuralPeriVLMC", MuralPeriVLMCList[1]))
for(p in seq(2, length(MuralPeriVLMCList), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, MuralPeriVLMCList[p], reduction="umap") 
}
FPList[["FP5"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)
  
FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BroadGenes[1], reduction="umap") + labs(title=paste("BroadGenes", BroadGenes[1]))
for(p in seq(2, length(BroadGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BroadGenes[p], reduction="umap") 
}
FPList[["FP6"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[1], reduction="umap") + labs(title=paste("HypothalamicProgenSubclass", HypothalamicProgenSubclass[1], sep=" - "))
for(p in seq(2, length(HypothalamicProgenSubclass), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, HypothalamicProgenSubclass[p], reduction="umap") 
}
FPList[["FP7"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, WellKnownGenes[1], reduction="umap") + labs(title=paste("WellKnownGenes", WellKnownGenes[1]))
for(p in seq(2, length(WellKnownGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, WellKnownGenes[p], reduction="umap") 
}
FPList[["F8"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, SuspectedGenes[1], reduction="umap") + labs(title=paste("SuspectedGenes", SuspectedGenes[1]))
for(p in seq(2, length(SuspectedGenes), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, SuspectedGenes[p], reduction="umap") 
}
FPList[["FP9"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)


FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, BlackshawEtc[1], reduction="umap") + labs(title=paste("BlackshawEtc", BlackshawEtc[1]))
for(p in seq(2, length(BlackshawEtc), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, BlackshawEtc[p], reduction="umap") 
}
FPList[["FP10"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, Zhou_Extras[1], reduction="umap") + labs(title=paste("Zhou_Extras", Zhou_Extras[1]))
for(p in seq(2, length(Zhou_Extras), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, Zhou_Extras[p], reduction="umap") 
}
FPList[["FP11"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

FPSinglePage = list()
FPSinglePage[[1]] = FeaturePlot(SeuFile, VariousUkn_3[1], reduction="umap") + labs(title=paste("LocalizedNeuronalTFs", VariousUkn_3[1]))
for(p in seq(2, length(VariousUkn_3), 1)){
FPSinglePage[[p]] = FeaturePlot(SeuFile, VariousUkn_3[p], reduction="umap") 
}
FPList[["FP12"]] = ggarrange(plotlist = FPSinglePage, ncol=5, nrow=8)

pdf(paste("ALLFETAL_", Filename, "_dim", z, "_AllFPs.pdf", sep=""), width = 25, height = 45)
print(FPList)
dev.off()
}}
```


#############################################
############## QUALITY CONTROL ##############
#############################################
```{r}
Fetal.list.Hypothal = list()
Fetal.list.Hypothal[["CS12"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS12Diencephalon_GRCh38") 
Fetal.list.Hypothal[["CS14"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS14_Dienceph")
#Fetal.list.Hypothal[["CS20_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS20_hypothalamus1") #**Not hypothalamic AT ALL** - High in FOXG1 and NEUROD6 - No 
#Fetal.list.Hypothal[["CS20_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS20_hypothalamus2") #**Not hypothalamic AT ALL** - High in FOXG1 and NEUROD6 - No 
Fetal.list.Hypothal[["CS22_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS22_2_hypothalamus") #High in SIM1 and DLXs - Probably an artifact of having a mostly dorsal dissection
Fetal.list.Hypothal[["CS22"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS22_Hypothalamus") #Tiny bit of FOXG1 but mostly hypothalamic
Fetal.list.Hypothal[["GW14"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW14_Hypo") #No FOXG1 or NEUROD6 - Also not really any TBX3 or SIM1 - Some scattered neuropeptides - try running another set of FPs
Fetal.list.Hypothal[["GW15"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW15_hypothalamus") #No FOXG1 or NEUROD6 - Also not really any TBX3 or SIM1 - Some scattered neuropeptides - try running another set of FPs
#Fetal.list.Hypothal[["GW18_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW18_2_hypothalamus")  #No FOXG1 or NEUROD6 - Also not really any TBX3 or SIM1 - Some scattered neuropeptides - try running another set of FPs. Almost 20K cells and no indication of containing genes of interest - will likley exclude from final analysis
Fetal.list.Hypothal[["GW18_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW18_hypothalamus") #Really nice hypothalamus. Tiny NEUROD6 cluster which can easily be clustered out or labelled
Fetal.list.Hypothal[["GW19"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW19_hypothalamus")
#Fetal.list.Hypothal[["GW20_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW20_31_hypo") #Almost all FOXG1 positive
Fetal.list.Hypothal[["GW20_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW20_34_hypo")
#Fetal.list.Hypothal[["GW20_3"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW20_hypothalamus") #Vast majority of FOXG1 and NEUROD6 positive
Fetal.list.Hypothal[["GW16"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW16_hypo")
#Fetal.list.Hypothal[["GW22_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW22T_hypo2") 
Fetal.list.Hypothal[["GW22_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW22T_hypo")
Fetal.list.Hypothal[["GW25"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW25_3V_hypo")

#### nFeature_RNA Table #### 
QCTable = as.data.frame(matrix(ncol=10, nrow=0))
for( in 1:length(Fetal.list.Hypothal)){ #
Filename= names(Fetal.list.Hypothal)[x]

ScreeingFile = CreateSeuratObject(counts = Fetal.list.Hypothal[[x]], project = Filename, min.cells = 3, min.features = 200)
ScreeingFile[["percent.mt"]] <- PercentageFeatureSet(ScreeingFile, pattern = "^MT-")

pdf(paste("ALLFETAL_", Filename, "_QCData.pdf", sep=""), width=20, height=8)
print(VlnPlot(ScreeingFile, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3))
dev.off()


TableData = as.data.frame(t(c(Filename, median(ScreeingFile$nFeature_RNA), mean(ScreeingFile$nFeature_RNA), sd(ScreeingFile$nFeature_RNA), median(ScreeingFile$nCount_RNA), mean(ScreeingFile$nCount_RNA), sd(ScreeingFile$nCount_RNA), median(ScreeingFile$percent.mt), mean(ScreeingFile$percent.mt), sd(ScreeingFile$percent.mt))))
colnames(TableData) = c("Sample", "nFeature_Median", "nFeature_Mean", "nFeature_SD", "nCount_Median", "nCount_Mean", "nCount_SD", "percent.mt_Median", "percent.mt_Mean", "percent.mt_SD")

QCTable = rbind(QCTable, TableData)
}
write.csv(QCTable, "FetalQCTable.csv")



Fetal.list.HypothalV2 = Fetal.list.Hypothal
Fetal.list.HypothalV2[["CS12"]] = NULL
Fetal.list.HypothalV2[["CS14"]] = NULL
Fetal.list.HypothalV2[["GW14"]] = NULL
Fetal.list.HypothalV2[["GW15"]] = NULL

#### QC Clustering #### 
for(min in c(800, 1000)){
for( in 1:length(Fetal.list.HypothalV2)){ #
Filename= names(Fetal.list.HypothalV2)[x]

ScreeingFile = CreateSeuratObject(counts = Fetal.list.HypothalV2[[x]], project = Filename, min.cells = 3, min.features = min)
ScreeingFile <- NormalizeData(ScreeingFile, normalization.method = "LogNormalize", scale.factor = 10000) 

mt.genes <- grep(pattern = "^MT-",  = row.names(ScreeingFile), value = TRUE)
percent.mito <- Matrix::colSums(ScreeingFile@assays[["RNA"]][mt.genes, ])/Matrix::colSums(ScreeingFile@assays[["RNA"]])

ScreeingFile <- AddMetaData(object = ScreeingFile, metadata = percent.mito, col.name = "percent.mito")
ScreeingFile <- ScaleData(object=ScreeingFile, vars.to.regress = "percent.mito")
ScreeingFile <- FindVariableFeatures(ScreeingFile, selection.method = "vst", nfeatures = 2000)  

ScreeingFile = RunPCA(ScreeingFile, npcs = 20)
ScreeingFile <- RunUMAP(ScreeingFile, dims = 1:20, spread= 5)

DefaultAssay(ScreeingFile) = "RNA"
FPs = FeaturePlot(ScreeingFile, c("SYT1","STMN2", "FOXG1", "NEUROD6", "SOX2", "VIM", "SLC32A1", "SLC17A6", "HDC", "TBX3", "AGRP", "NPY", "SST", "NPY1R","POMC","TCF7L2", "TAC3", "KISS1", "TAC1", "GHRH", "PBX3","PCSK1", "OTP",  "NRN1", "SIM1", "POU3F2", "OXT","PENK", "CRH","AVP","TH","NHLH1","ZBTB16","DMRTA2", "EBF2", "ST18", "PPP1R17", "NKX2-1", "GABRA5", "NPTX2", "ZMAT4", "LHX9", "LHX2", "NKX2-2", "HCRT", "CARTPT", "PITX2", "NTS", "PDYN", "LHX1", "TRH", "MEIS3",  "ONECUT2", "FOXP2", "SYNPR", "SOX14", "SEMA3C", "NKX2-1","RGS16", "SIX6", "HES6",  "HMGA1", "HMGB2", "GNG5", "EEF1G", "MEIS2", "DLX1", "DLX5"), reduction="umap", ncol=6) 

pdf(paste("ALLFETAL_", Filename, "MinGenes", min, "_FeaturePlots.pdf", sep=""), width=30, height=60)
print(FPs)
dev.off()

assign(paste(Filename, "_Seurat_MinGenes", min, sep=""), ScreeingFile)
}}  


save(list=c("CS12_Seurat", "CS14_Seurat", "CS22_Seurat", "CS22_2_Seurat", "GW14_Seurat", "GW15_Seurat", "GW20_2_Seurat", "GW16_Seurat", "GW18_2_Seurat", "GW19_Seurat", "GW22_2_Seurat", "GW25_Seurat"), file = "~/Fetal_Hypo_IndivAnalysis_QC.RData")


####Run any extra genes as FPs #### 
for(p in list(CS12_Seurat, CS14_Seurat, CS20_1_Seurat, CS20_2_Seurat, CS22_Seurat, CS22_2_Seurat, GW14_Seurat, GW15_Seurat, GW20_2_Seurat, GW16_Seurat, GW18_2_Seurat, GW19_Seurat, GW20_1_Seurat, GW20_2_Seurat, GW20_3_Seurat, GW22_1_Seurat, GW22_2_Seurat, GW25_Seurat)){
ScreeingFile = p
Filename = as.character(substitute(ScreeingFile))

DefaultAssay(ScreeingFile) = "RNA"
FPs = FeaturePlot(ScreeingFile, c("SYT1","STMN2", "FOXG1", "NEUROD6","TBX3", "OTP", "SIM1", "POU3F2", "OXT", "POMC", "AGRP", "NPY", "GHRH", "TRH", "CRH", "AVP", "KISS1", "PCSK1", "HCRT", "CARTPT", "NKX2-1"), reduction="umap", ncol=3) 

pdf(paste("ALLFETAL_", Filename, "_FeaturePlots2.pdf", sep=""), width=15, height=30)
print(FPs)
dev.off()
}



#### Output histograms #### 
HistList = list()
for(p in list(CS12_Seurat, CS14_Seurat, CS22_Seurat, CS22_2_Seurat, GW14_Seurat, GW15_Seurat, GW16_Seurat, GW18_2_Seurat, GW19_Seurat, GW20_2_Seurat, GW22_2_Seurat, GW25_Seurat)){
ScreeingFile = p
Meta = ScreeingFile@meta.data
Filename = as.character(unique(Meta$orig.ident))

HistList[[p]] = ggplot(Meta, aes(x=nFeature_RNA)) + 
  geom_histogram(binwidth=30)+ ggtitle(Filename)+ scale_x_continuous(breaks = seq(0, max(Meta$nFeature_RNA), by = 100)) + theme(axis.text. = element_text(angle = 90, vjust = 0.5, hjust=1)) 
}
pdf(paste("ALLFETAL_nFeature_Histogram.pdf", sep=""), width=10, height=6)
print(HistList)
dev.off()

#Sample   TOTAL     Min800    Min1000
#CS12   	2492	    38	      15
#CS14	    4363	    75	      30
#CS22_2   1808	    1806	    1205
#CS22	    1964	    1964	    1632
#GW14	    10889	    104	      37
#GW15	    1161	    4	        2
#GW16 	  21674	    15775	    12828
#GW18_2	  5536	    5115	    3587
#GW19	    3283	    3278	    3269
#GW20_2	  2849	    2841	    2836
#GW22_2	  3085	    3025	    2435
#GW25	    8408	    7494	    6153

```


#############################################
################ INTEGRATION ################
#############################################
### Integration
```{r}
Fetal.list.Hypo = list()
#Fetal.list.Hypo[["CS12"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS12Diencephalon_GRCh38") 
#Fetal.list.Hypo[["CS14"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS14_Dienceph")
###Fetal.list.Hypo[["CS20_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS20_hypothalamus1") #**Not hypothalamic AT ALL** - High in FOXG1 and NEUROD6 - No 
###Fetal.list.Hypo[["CS20_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS20_hypothalamus2") #**Not hypothalamic AT ALL** - High in FOXG1 and NEUROD6 - No 
Fetal.list.Hypo[["CS22_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS22_2_hypothalamus") #High in SIM1 and DLXs - Probably an artifact of having a mostly dorsal dissection
Fetal.list.Hypo[["CS22"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/CS22_Hypothalamus") #Tiny bit of FOXG1 but mostly hypothalamic
#Fetal.list.Hypo[["GW14"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW14_Hypo") #No FOXG1 or NEUROD6 - Also not really any TBX3 or SIM1 - Some scattered neuropeptides - try running another set of FPs
#Fetal.list.Hypo[["GW15"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW15_hypothalamus") #No FOXG1 or NEUROD6 - Also not really any TBX3 or SIM1 - Some scattered neuropeptides - try running another set of FPs
Fetal.list.Hypo[["GW16"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW16_hypo")
#Fetal.list.Hypo[["GW18_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW18_2_hypothalamus")  #No FOXG1 or NEUROD6 - Also not really any TBX3 or SIM1 - Some scattered neuropeptides - try running another set of FPs
Fetal.list.Hypo[["GW18_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW18_hypothalamus") #Really nice hypothalamus. Tiny NEUROD6 cluster which can easily be clustered out or labelled
Fetal.list.Hypo[["GW19"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW19_hypothalamus")
###Fetal.list.Hypo[["GW20_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW20_31_hypo")
Fetal.list.Hypo[["GW20_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW20_34_hypo")
###Fetal.list.Hypo[["GW20_3"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW20_hypothalamus")
#Fetal.list.Hypo[["GW22_1"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW22T_hypo2") 
Fetal.list.Hypo[["GW22_2"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW22T_hypo")
Fetal.list.Hypo[["GW25"]] = Read10X(data.dir = "~/Dropbox/Columbia/Fetal scRNAseq/New Integration - Aug21/Input Data/GW25_3V_hypo")


for( in 1:length(Fetal.list.Hypo)){
File =CreateSeuratObject(counts = Fetal.list.Hypo[[x]], project = names(Fetal.list.Hypo)[x], min.cells = 3, min.features = 1000)
Temp = as.data.frame(File@assays$RNA@counts)
colnames(Temp) = paste0(names(Fetal.list.Hypo)[x], colnames(Temp), sep="_")
Temp =CreateSeuratObject(counts = Temp, project = names(Fetal.list.Hypo)[x], min.cells = 3, min.features = 200)
Temp <- NormalizeData(Temp)
Fetal.list.Hypo[[x]] <- FindVariableFeatures(Temp, do.plot = F, display.progress = F)
}


features <- SelectIntegrationFeatures(object.list = Fetal.list.Hypo)
Fetal.anchors <- FindIntegrationAnchors(object.list = Fetal.list.Hypo, anchor.features = features)
Fetal.combined <- IntegrateData(anchorset = Fetal.anchors)
DefaultAssay(Fetal.combined) <- "integrated"

save(list=c("Fetal.combined"), file = "~/Fetal_HypoExtra_StdIntegration_13OCT21.RData")
MT.genes <- grep(pattern = "^MT-",  = rownames( = Fetal.combined), value = TRUE)
percent.mt <- Matrix::colSums(Fetal.combined@assays[["RNA"]][MT.genes, ])/Matrix::colSums(Fetal.combined@assays[["RNA"]])
Fetal.combined  = AddMetaData(Fetal.combined, percent.mt, "percent.mt")
Fetal.combined <- ScaleData(Fetal.combined, vars.to.regress = "percent.mt")

# Check how integration looks - pretty nice!
save(list=c("Fetal.combined"), file = "~/Fetal_HypoExtra_StdIntegration_13OCT21.RData")
Fetal.combined <- RunPCA(Fetal.combined, npcs = 30, verbose = FALSE)
Fetal.combined <- RunUMAP(Fetal.combined, reduction = "pca", dims = 1:30)

Filename="Fetal.integrated.HypothalV2_13OCT"
pdf(paste("ALLFETAL_", Filename, "_GroupedUMAP.pdf", sep=""), width=10, height=8)
print(DimPlot(Fetal.combined, reduction = "umap", group.by = "orig.ident", label=T))
dev.off()

Idents(Fetal.combined) = "orig.ident"
pdf(paste("ALLFETAL_", Filename, "_SplitUMAP.pdf", sep=""), width=60, height=8)
print(DimPlot(Fetal.combined, reduction = "umap", split.by = "orig.ident"))
dev.off()

DefaultAssay(Fetal.combined) = "RNA"
FPHAR = FeaturePlot(Fetal.combined, unique(c("PDGFRA", "PLLP", "OLIG1", "TNS3", "FYN", "TCF7L2", "PLP1", "SERINC5", "MOG", "MAG", "AQP4", "GJA1", "GFAP", "AGT", "CRYM", "FOXJ1", "RFX2", "SLC38A5", "CLDN5", "FLT1", "RGS5", "KCNJ8", "ABCC9", "IGFBP2", "LUM", "PTGDS", "COL1A1", "COL3A1", "AIF1", "C1QB", "EGR2", "PCNA", "BIRC5", "PIN1", "CSRNP3", "STMN2", "SYT1", "HBA1", 
"HES5", "FGF10", "TJP1", "SOX2", "NES", "VIM", "SYT1", "SNAP25", "TUBB3","SYP", "SLC32A1", "SLC17A6", "SLC17A8", "HES1", "MKI67", "FABP7", "TTYH1", "HMGA2", "MEIS2", "POU3F2", "SIM1",  "TBX3", "NKX2-1", "AGRP",  "SST", "KISS1", "GHRH", "TH", "TRH", "OXT", "AVP", "CRH", "BDNF", "GAL","POMC","NPY", "PCSK1", #Neuronal
                      "APOD", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9", #OLIG
                      "GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "C1QB", "AIF1", "P2RY12", "MRC1", #Astro
                      "FOXJ1", "RAX", "ADM",  "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "RFX2", "RFX3", "RFX4", "NPAS3", "COL23A1", #EpendyTany
                      "SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1", #Endothelial
                      "VTN", "KCNJ1", "MUSTN1",  "COL1A1", "COL3A1", "LUM", "DCN", "PTGDS", "AMBP", "RGS5", "ACTA2" #Mural/SMA/Pericytes/VLMC
                      )), reduction="umap") 

pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=90)
print(FPHAR)
dev.off()


DefaultAssay(Fetal.combined) = "RNA"
FPHAR = FeaturePlot(Fetal.combined, unique(c("HES1", "ID4", "CYR61", "FOS", "GAS1", "UHRF1", "FABP7", "RRM2", "TUBA1B", "RBP1", "PTPRZ1", "DLX2", "CKB", "SALL3", "NEUROG2", "VIM", "TMEM132B", "GADD45G", "INSM1", "SOX11", "DCX", "NEUROD1", "NEUROD2", "EOMES", "ELAVL4", "ABRACL", "NEUROD4", "RSPO3", "SHB", "SLC1A2", "SEZ6L", "NFIX")), reduction="umap") 

pdf(paste("ALLFETAL_", Filename, "_FeaturePlots_RGGENES.pdf", sep=""), width=20, height=40)
print(FPHAR)
dev.off()

FeaturePlot(Fetal.combined, unique(c("DLL1", "DLL3")), reduction="umap") 

#PCDH15", "NKX2-2

DPlist2 = list()
DPlist = list()
FPlist = list()
Filename = "Fetal.combined_20SEP"
for(y in seq(10,50,10)){
DefaultAssay(Fetal.combined) = "integrated"
Fetal.combined = RunPCA(Fetal.combined, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Fetal.combined <- RunUMAP(Fetal.combined, dims = 1:z, spread= s)
DefaultAssay(Fetal.combined) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Fetal.combined, unique(c("PDGFRA", "PLLP", "OLIG1", "TNS3", "FYN", "TCF7L2", "PLP1", "SERINC5", "MOG", "MAG", "AQP4", "GJA1", "GFAP", "AGT", "CRYM", "FOXJ1", "RFX2", "SLC38A5", "CLDN5", "FLT1", "RGS5", "KCNJ8", "ABCC9", "IGFBP2", "LUM", "PTGDS", "COL1A1", "COL3A1", "AIF1", "C1QB", "EGR2", "PCNA", "BIRC5", "PIN1", "CSRNP3", "STMN2", "SYT1", "HBA1", 
  "NES", "VIM", "SYT1", "SNAP25", "TUBB3","SYP", "SLC32A1", "SLC17A6", "SLC17A8", "HES1", "MKI67", "FABP7", "TTYH1", "HMGA2", "MEIS2", "POU3F2", "SIM1",  "TBX3", "NKX2-1", "AGRP",  "SST", "KISS1", "GHRH", "TH", "TRH", "OXT", "AVP", "CRH", "BDNF", "GAL","POMC","NPY", "PCSK1", #Neuronal
                      "APOD", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9", #OLIG
                      "GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "C1QB", "AIF1", "P2RY12", "MRC1", #Astro
                      "FOXJ1", "RAX", "ADM",  "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "RFX2", "RFX3", "RFX4", "NPAS3", "COL23A1", #EpendyTany
                      "SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1", #Endothelial
                      "VTN", "KCNJ1", "MUSTN1",  "COL1A1", "COL3A1", "LUM", "DCN", "PTGDS", "AMBP", "RGS5", "ACTA2" #Mural/SMA/Pericytes/VLMC
                      )), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Fetal.combined, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Fetal.combined, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=90)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()

```

### Clustering of main populations
```{r}
DefaultAssay(Fetal.combined) = "integrated"
Fetal.combined = RunPCA(Fetal.combined, npcs = 50)
Fetal.combined = RunUMAP(Fetal.combined, dims = 1:40, spread= 2)
HypoSTD_UMAP = as.data.frame(Fetal.combined@reductions$umap@cell.embeddings)
Fetal.combined_Data = as.data.frame(t(as.data.frame(Fetal.combined@assays$RNA@counts)))

VLMC = subset(HypoSTD_UMAP, HypoSTD_UMAP$UMAP_1 < -13 & HypoSTD_UMAP$UMAP_2 > 2)
Vasculature = subset(HypoSTD_UMAP, HypoSTD_UMAP$UMAP_1 < -14 & HypoSTD_UMAP$UMAP_2 < 2 & HypoSTD_UMAP$UMAP_2 > -5)
Microglia = subset(HypoSTD_UMAP, HypoSTD_UMAP$UMAP_1 > -14 &  HypoSTD_UMAP$UMAP_1 < -9 & HypoSTD_UMAP$UMAP_2 < -2 & HypoSTD_UMAP$UMAP_2 > -7)
Blood = subset(HypoSTD_UMAP, HypoSTD_UMAP$UMAP_1 > -7 &  HypoSTD_UMAP$UMAP_1 < -1 & HypoSTD_UMAP$UMAP_2 < 0 & HypoSTD_UMAP$UMAP_2 > -7)
DivProg = subset(HypoSTD_UMAP, HypoSTD_UMAP$UMAP_1 > -13 & HypoSTD_UMAP$UMAP_1 < -4 & HypoSTD_UMAP$UMAP_2 > 1 & HypoSTD_UMAP$UMAP_2 < 8)
Neurons = subset(HypoSTD_UMAP,  HypoSTD_UMAP$UMAP_2 < -7)
Ependy = subset(HypoSTD_UMAP, HypoSTD_UMAP$UMAP_2 > 13 & HypoSTD_UMAP$UMAP_1 > 4.5)

Neurons_Seurat = subset(Fetal.combined, cells = row.names(Neurons))
Vasculature_Seurat = subset(Fetal.combined, cells = row.names(Vasculature))
RestCells = subset(Fetal.combined, cells = c(row.names(VLMC), row.names(Vasculature), row.names(Microglia), row.names(Blood), row.names(DivProg), row.names(Neurons), row.names(Ependy)), invert=T)
RestCells2 = subset(Fetal.combined, cells = c(row.names(VLMC), row.names(Vasculature), row.names(Microglia), row.names(Blood), row.names(DivProg), row.names(Neurons), row.names(Ependy)), invert=T)


DivProg_Seu = subset(Fetal.combined, cells = row.names(DivProg))
set.dim=40
set.res = 1
set.kparam = 50
ClusterFunc_All_RNA(DivProg_Seu)
DefaultAssay(DivProg_Seu) = "integrated"
DivProg_Seu <- FindNeighbors(DivProg_Seu, k.param=20, dims=1:40)
DivProg_Seu <- FindClusters(DivProg_Seu, resolution =1)
DivProg_Astro = subset(DivProg_Seu, idents= c(8,10,11))
DivProg_Neuro = subset(DivProg_Seu, idents= c(6,12))
DivProg_Oligo = subset(DivProg_Seu, idents= c(5,9,13))
DivProg_Rest1 = subset(DivProg_Seu, idents= c(5,9,13, 6,12, 8,10,11), invert=T)

set.dim=40
set.res = 2
set.kparam = 3
ClusterFunc_All_RNA(DivProg_Rest1)
DivProg_Rest1 <- FindNeighbors(DivProg_Rest1, k.param=3, dims=1:40)
DivProg_Rest1 <- FindClusters(DivProg_Rest1, resolution =2)
DivProg_Astro_Pt2 = subset(DivProg_Rest1, idents= c(0,1,2,3,4,5,6,13,14,16,17,19,22,23, 24,26,27,33,15,31))
DivProg_Oligo_Pt2 = subset(DivProg_Rest1, idents= c(8,21, 25,32,34)) ####
DivProg_Neuro_Pt2 = subset(DivProg_Rest1, idents= c(12,18,20,28,29,30))
DivProg_Rest2 = subset(DivProg_Rest1, idents= c(0,1,2,3,4,5,6,13,14,16,17,19,22, 24,26,27,33,15,31,8,21,23, 25,32,34,12,18,20,28,29,30), invert=T)
set.dim=40
set.res = 0.05
set.kparam = 30
ClusterFunc_All_RNA(DivProg_Rest2)


DivProg_Rest2 <- FindNeighbors(DivProg_Rest2, k.param=3, dims=1:40)
DivProg_Rest2 <- FindClusters(DivProg_Rest2, resolution =1)
DivProg_Astro_Pt3 = subset(DivProg_Rest2, idents= c(2))
DivProg_Neuro_Pt3 = subset(DivProg_Rest2, idents= c(4,6,7,8))
DivProg_Rest3 = subset(DivProg_Rest2, idents= c(4,6,7,8,2), invert=T)

set.dim=40
set.res = 1
set.kparam = 3
ClusterFunc_All_RNA(DivProg_Rest3)


DivProg_Rest3 <- FindNeighbors(DivProg_Rest3, k.param=3, dims=1:40)
DivProg_Rest3 <- FindClusters(DivProg_Rest3, resolution =1)
DivProg_Astro_Pt4 = subset(DivProg_Rest3, idents= c(4,5))
DivProg_Rest4 = subset(DivProg_Rest3, idents= c(4,5), invert=T)
set.dim=40
set.res = 2
set.kparam = 2
ClusterFunc_All_RNA(DivProg_Rest4)
DivProg_Rest4 <- FindNeighbors(DivProg_Rest4, k.param=2, dims=1:40)
DivProg_Rest4 <- FindClusters(DivProg_Rest4, resolution =2)
DivProg_Astro_Pt5 = subset(DivProg_Rest4, idents= c(8,9,14,15))
DivProg_Oligo_Pt3 = subset(DivProg_Rest4, idents= c(2,12))
DivProg_Neuro_Pt4 = subset(DivProg_Rest4, idents= c(8,9,14,15, 2,12), invert=T)



Dividing_Astro_Final = subset(DivProg_Seu, cells = c(colnames(DivProg_Astro), colnames(DivProg_Astro_Pt2), colnames(DivProg_Astro_Pt3), colnames(DivProg_Astro_Pt4), colnames(DivProg_Astro_Pt5)))
Dividing_Neuro_Final = subset(DivProg_Seu, cells = c(colnames(DivProg_Neuro), colnames(DivProg_Neuro_Pt2), colnames(DivProg_Neuro_Pt3), colnames(DivProg_Neuro_Pt4)))
Dividing_Oligo_Final = subset(DivProg_Seu, cells = c(colnames(DivProg_Oligo), colnames(DivProg_Oligo_Pt2), colnames(DivProg_Oligo_Pt3)))

####Done Dividing


set.dim=40
set.res = 1
set.kparam = 20
ClusterFunc_All_RNA(Vasculature_Seurat)

set.dim=40
set.res = 1
set.kparam = c(30, 250)
ClusterFunc_All_RNA(RestCells)

DefaultAssay(RestCells) = "integrated"
RestCells <- FindNeighbors(RestCells, k.param=120, dims=1:40)
RestCells <- FindClusters(RestCells, resolution =1)
ClustForEpendyTany = subset(RestCells, idents= c(0,2,5))
Oligodendro = subset(RestCells, idents= c(3,7))
RestCells_V2 = subset(RestCells, idents= c(3,7,0,2,5), invert=T)


set.dim=40
set.res = 1
set.kparam = c(3)
ClusterFunc_All_RNA(ClustForEpendyTany)


ClustForEpendyTany <- FindNeighbors(ClustForEpendyTany, k.param=3, dims=1:40)
ClustForEpendyTany <- FindClusters(ClustForEpendyTany, resolution =1)
ClustForTany = subset(ClustForEpendyTany, idents = c(18,4,1,14))
ClustForEpendy = subset(ClustForEpendyTany, idents = c(23,17,2,3))
ClustForEpendyTany_V2= subset(ClustForEpendyTany, idents = c(23,17,2,3,18,4,1,14), invert=T)

set.dim=40
set.res = 10
set.kparam = c(3)
ClusterFunc_All_RNA(ClustForEpendyTany_V2)


ClustForEpendyTany_V2 <- FindNeighbors(ClustForEpendyTany_V2, k.param=3, dims=1:40)
ClustForEpendyTany_V2 <- FindClusters(ClustForEpendyTany_V2, resolution =10)
ClustForTany_Pt2 = subset(ClustForEpendyTany_V2, idents = c(130,110,92,67))
ClustForEpendy_Pt2 = subset(ClustForEpendyTany_V2, idents = c(135,134,128,125,124,119,116,107,104,96,83,80,79,68,67,65,53,49,37,35,34,32,30,28,25,23,20,2))
ClustForEpendyTany_V3 = subset(ClustForEpendyTany_V2, idents = c(130,110,92,67, 135,134,128,125,124, 119,116,107,104,96,83,80,79, 68,67,65, 53, 49, 37, 35, 34, 32, 30, 28,25,23,20,2), invert=T)

set.dim=40
set.res = 1
set.kparam = c(20)
ClusterFunc_All_RNA(ClustForEpendyTany_V3)

ClustForEpendyTany_V3 <- FindNeighbors(ClustForEpendyTany_V3, k.param=20, dims=1:40)
ClustForEpendyTany_V3 <- FindClusters(ClustForEpendyTany_V3, resolution =1)
ClustForTany_Pt3 = subset(ClustForEpendyTany_V3, idents = c(1))

set.dim=40
set.res = 1
set.kparam = c(2)
ClusterFunc_All_RNA(ClustForTany_Pt3)


ClustForTany_Pt3 <- FindNeighbors(ClustForTany_Pt3, k.param=2, dims=1:40)
ClustForTany_Pt3 <- FindClusters(ClustForTany_Pt3, resolution =1)
ClustForTany_Pt3_V2 = subset(ClustForTany_Pt3, idents = c(13, 16, 24, 25, 26, 27, 29, 31, 33, 35, 36, 37, 39, 40, 41, 43, 44, 45, 46, 47, 49, 51, 52, 53, 54, 55, 6, 9, 4, 12, 1, 2, 7,3,5), invert=T)


Tany_Combined = subset(Fetal.combined, cells = c(colnames(ClustForTany_Pt3_V2), colnames(ClustForTany_Pt2), colnames(ClustForTany)))
Ependy_Combined = subset(Fetal.combined, cells = c(colnames(ClustForEpendy), colnames(ClustForEpendy_Pt2), row.names(Ependy)))

CheckInput = colnames(Tany_Combined)
CheckUMAP(Fetal.combined)


CheckInput = colnames(Ependy_Combined)
CheckUMAP(Fetal.combined)

set.dim=40
set.res = 1
set.kparam = c(10)
ClusterFunc_All_RNA(Tany_Combined)
ClusterFunc_All_RNA(Ependy_Combined)



Tany_Combined <- FindNeighbors(Tany_Combined, k.param=10, dims=1:40)
Tany_Combined <- FindClusters(Tany_Combined, resolution =1)
Tany_ToClean = subset(Tany_Combined, idents = c(0,3))
Tany_Clean = subset(Tany_Combined, idents = c(0,3,2), invert=T)
set.dim=40
set.res = 3
set.kparam = c(3)
ClusterFunc_All_RNA(Tany_ToClean)


Tany_ToClean <- FindNeighbors(Tany_ToClean, k.param=3, dims=1:40)
Tany_ToClean <- FindClusters(Tany_ToClean, resolution =3)
Tany_Clean_pt2 = subset(Tany_ToClean, idents = c(0,4,8,10,14,18,19,23,25))
Tany_ToClean_pt2 = subset(Tany_ToClean, idents = c(0,4,8,10,14,18,19,23,25), invert=T)
set.dim=20
set.res = 1
set.kparam = c(5)
ClusterFunc_All_RNA(Tany_ToClean_pt2)


Tany_ToClean_pt2 <- FindNeighbors(Tany_ToClean_pt2, k.param=3, dims=1:20)
Tany_ToClean_pt2 <- FindClusters(Tany_ToClean_pt2, resolution =5)
Tany_ToClean_pt3 = subset(Tany_ToClean_pt2, idents = c(6,11,13,21,12,15,30,2,4,10,16,22,20,28,0))
set.dim=20
set.res = 1
set.kparam = c(2)
ClusterFunc_All_RNA(Tany_ToClean_pt3)


Tany_ToClean_pt3 <- FindNeighbors(Tany_ToClean_pt3, k.param=2, dims=1:20)
Tany_ToClean_pt3 <- FindClusters(Tany_ToClean_pt3, resolution =1)
Tany_Clean_pt3 = subset(Tany_ToClean_pt3, idents = c(0,8,17,20,21,23,26))


Tany_Final = subset(Fetal.combined, cells = c(colnames(Tany_Clean_pt3), colnames(Tany_Clean_pt2), colnames(Tany_Clean)))


Ependy_Combined <- FindNeighbors(Ependy_Combined, k.param=10, dims=1:40)
Ependy_Combined <- FindClusters(Ependy_Combined, resolution =1)
Ependy_Combined_V2 = subset(Ependy_Combined, idents = c(3), invert=T)
Ependy_Final = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Ependy_Combined_V2) & HypoSTD_UMAP$UMAP_2 > 9.5 & HypoSTD_UMAP$UMAP_1 > -1 & row.names(HypoSTD_UMAP) %notin% colnames(Tany_Final))


Ependy_Final_Seu = subset(Fetal.combined, cell = row.names(Ependy_Final))

StrayEpendymalCell = subset(Fetal.combined, cells = c(row.names(Blood),row.names(Microglia), row.names(Vasculature), row.names(Neurons), row.names(VLMC), row.names(DivProg), colnames(RestCells), colnames(RestCells_V2), row.names(Ependy_Final)), invert=T)
                                                      
CheckInput = colnames(StrayEpendymalCell) #Need to clean by coords
CheckUMAP(Fetal.combined)

Ependy_Final_V2 = subset(Fetal.combined, cells = c(row.names(Ependy_Final), colnames(StrayEpendymalCell)))


##Oligodendrocytes
CheckInput = colnames(Oligodendro) #Need to clean by coords
CheckUMAP(RestCells)


MoveToOlig = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(RestCells_V2) & HypoSTD_UMAP$UMAP_2 < 2 & HypoSTD_UMAP$UMAP_1 > 3)
MoveToRG = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Oligodendro) & HypoSTD_UMAP$UMAP_2 > 1 & HypoSTD_UMAP$UMAP_1 < 4)


Oligodendro_V2 = subset(Oligodendro, cells = row.names(MoveToRG), invert=T)
Oligodendro_Final = subset(RestCells, cells = c(colnames(Oligodendro_V2), row.names(MoveToOlig)))

CheckInput = colnames(Oligodendro_Final) #Clean
CheckUMAP(RestCells)


##Radial Glia and Astro Left
RestCells_V3 = subset(RestCells, cells = c(colnames(Oligodendro_Final), colnames(Tany_Final), row.names(Ependy_Final)), invert=T)
CheckInput = colnames(RestCells_V3) 
CheckUMAP(Fetal.combined)
set.dim=40
set.res = 1
set.kparam = c(100)
ClusterFunc_All_RNA(RestCells_V3)

DefaultAssay(RestCells_V3) = "integrated"
RestCells_V3 <- FindNeighbors(RestCells_V3, k.param=100, dims=1:40)
RestCells_V3 <- FindClusters(RestCells_V3, resolution =1)
GlialProgen = subset(RestCells_V3, idents = c(5)) # True Astro #Hopefully can find nice marker
vRG = subset(RestCells_V3, idents = c(1)) # vRG [PAX6/NES]
IPC = subset(RestCells_V3, idents = c(2,4)) # IPC [PreOPC] [EGFR/ASCL1/DLL1/GADD45G]
oRG_1 = subset(RestCells_V3, idents = c(0,8)) #[HOPX]
oRG_2 = subset(RestCells_V3, idents = c(7)) #[HOPX/STMN1] - Neuro Progenitor
Astrocytes = subset(RestCells_V3, idents = c(6)) #High GFP/ID3
Ukn_NKX21 = subset(RestCells_V3, idents = c(3)) #NKX2-1+ve

set.dim=40
set.res = 1
set.kparam = c(100)
ClusterFunc_All_RNA(Ukn_NKX21)


###Clean assignments
GlialProgen_MoveToIPC = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(GlialProgen) & HypoSTD_UMAP$UMAP_1 > 4 & HypoSTD_UMAP$UMAP_2 < 8)
GlialProgen_MoveTo_oRG1 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(GlialProgen) & HypoSTD_UMAP$UMAP_1 < 4 & HypoSTD_UMAP$UMAP_2 < 7)
GlialProgen_MoveTo_vRG = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(GlialProgen) & HypoSTD_UMAP$UMAP_1 > 1 & HypoSTD_UMAP$UMAP_2 > 8 & HypoSTD_UMAP$UMAP_2 < 11)
GlialProgen_MoveTo_Astro = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(GlialProgen) & HypoSTD_UMAP$UMAP_1 < 0 & HypoSTD_UMAP$UMAP_2 > 4 & HypoSTD_UMAP$UMAP_2 < 11)
Keep_GlialProgen = subset(GlialProgen, cells = c(row.names(GlialProgen_MoveToIPC), row.names(GlialProgen_MoveTo_oRG1), row.names(GlialProgen_MoveTo_vRG), row.names(GlialProgen_MoveTo_Astro)), invert=T)

vRG_MoveToGlialProgen = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(vRG) & HypoSTD_UMAP$UMAP_1 < 1 & HypoSTD_UMAP$UMAP_2 > 13)
vRG_MoveTo_oRG1= subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(vRG) & HypoSTD_UMAP$UMAP_2 < 6.5)
Keep_vRG = subset(vRG, cells = c(), invert=T)

IPC_MoveTo_oRG1 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(IPC) & HypoSTD_UMAP$UMAP_1 < 1 & HypoSTD_UMAP$UMAP_2 < 7)
IPC_MoveTo_vRG = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(IPC) & HypoSTD_UMAP$UMAP_1 > 1 & HypoSTD_UMAP$UMAP_2 > 8 & HypoSTD_UMAP$UMAP_2 < 11)
IPC_MoveTo_Astro = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(IPC) & row.names(HypoSTD_UMAP) %notin% row.names(IPC_MoveTo_oRG1) & HypoSTD_UMAP$UMAP_1 < 0 & HypoSTD_UMAP$UMAP_2 > 4 & HypoSTD_UMAP$UMAP_2 < 11)
IPC_MoveToGlialProgen = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(IPC) & HypoSTD_UMAP$UMAP_1 > -1  &  HypoSTD_UMAP$UMAP_2 > 12)
IPC_MoveToNKX21 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(IPC) & row.names(HypoSTD_UMAP) %notin% c(row.names(IPC_MoveTo_Astro), row.names(IPC_MoveToGlialProgen), row.names(IPC_MoveTo_vRG)) &  HypoSTD_UMAP$UMAP_2 > 9 &  HypoSTD_UMAP$UMAP_1 < 2)
Keep_IPC = subset(IPC, cells = c(row.names(IPC_MoveTo_oRG1), row.names(IPC_MoveTo_vRG), row.names(IPC_MoveTo_Astro), row.names(IPC_MoveToGlialProgen), row.names(IPC_MoveToNKX21)), invert=T)

#oRG_1 - Clean

oRG_2_MoveToIPC = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(oRG_2) & HypoSTD_UMAP$UMAP_1 > 3 & HypoSTD_UMAP$UMAP_2 < 8)
oRG_2_MoveTo_oRG1 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(oRG_2) & HypoSTD_UMAP$UMAP_1 < 3  & HypoSTD_UMAP$UMAP_2 > 4.5 & HypoSTD_UMAP$UMAP_2 < 7)
oRG_2_MoveTo_vRG = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(oRG_2) & HypoSTD_UMAP$UMAP_1 > 0 & HypoSTD_UMAP$UMAP_2 > 8 & HypoSTD_UMAP$UMAP_2 < 12)
oRG_2_MoveTo_Astro = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(oRG_2) & HypoSTD_UMAP$UMAP_1 < -0.5 & HypoSTD_UMAP$UMAP_2 > 7 & HypoSTD_UMAP$UMAP_2 < 11)
oRG_2_MoveToGlialProgen = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(oRG_2) & HypoSTD_UMAP$UMAP_1 > -1  &  HypoSTD_UMAP$UMAP_2 > 12)
oRG_2_MoveToNKX21 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(oRG_2) & row.names(HypoSTD_UMAP) %notin% c(row.names(oRG_2_MoveTo_Astro), row.names(oRG_2_MoveToGlialProgen), row.names(oRG_2_MoveTo_vRG)) &  HypoSTD_UMAP$UMAP_2 > 9 &  HypoSTD_UMAP$UMAP_1 < 2)
Keep_oRG_2 = subset(oRG_2, cells = c(row.names(oRG_2_MoveToIPC), row.names(oRG_2_MoveTo_oRG1), row.names(oRG_2_MoveTo_vRG), row.names(oRG_2_MoveTo_Astro), row.names(oRG_2_MoveToNKX21), row.names(oRG_2_MoveToGlialProgen)), invert=T)

Astrocytes_2_MoveToGlialProgen = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Astrocytes) & HypoSTD_UMAP$UMAP_1 > -1  &  HypoSTD_UMAP$UMAP_2 > 12)
Astrocytes_2_MoveTo_vRG = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Astrocytes) & HypoSTD_UMAP$UMAP_1 > 1 & HypoSTD_UMAP$UMAP_1 < 3 & HypoSTD_UMAP$UMAP_2 > 5 & HypoSTD_UMAP$UMAP_2 < 12.5)
Astrocytes_2_MoveToIPC = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Astrocytes) & HypoSTD_UMAP$UMAP_1 > 3 & HypoSTD_UMAP$UMAP_2 < 9)
Astrocytes_2_MoveTo_oRG1 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Astrocytes) & HypoSTD_UMAP$UMAP_2 < 7 & HypoSTD_UMAP$UMAP_1 < 1)
Keep_Astrocytes = subset(Astrocytes, cells = c(row.names(Astrocytes_2_MoveToIPC), row.names(Astrocytes_2_MoveTo_vRG), row.names(Astrocytes_2_MoveToGlialProgen), row.names(Astrocytes_2_MoveTo_oRG1)), invert=T)

Ukn_NKX21_2_MoveToIPC = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Ukn_NKX21) & HypoSTD_UMAP$UMAP_1 > 3 & HypoSTD_UMAP$UMAP_2 < 9)
Ukn_NKX21_MoveTo_vRG = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Ukn_NKX21) & row.names(HypoSTD_UMAP) %notin% row.names(Ukn_NKX21_2_MoveToIPC) & HypoSTD_UMAP$UMAP_1 > 1 & HypoSTD_UMAP$UMAP_1 < 5 & HypoSTD_UMAP$UMAP_2 > 8 & HypoSTD_UMAP$UMAP_2 < 10)
Ukn_NKX21_MoveTo_oRG1 = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Ukn_NKX21) & HypoSTD_UMAP$UMAP_2 < 9 & HypoSTD_UMAP$UMAP_1 < 1)
Ukn_NKX21_MoveToGlialProgen = subset(HypoSTD_UMAP, row.names(HypoSTD_UMAP) %in% colnames(Ukn_NKX21) & HypoSTD_UMAP$UMAP_1 > 1  &  HypoSTD_UMAP$UMAP_2 > 11)
Keep_Ukn_NKX21 = subset(Ukn_NKX21, cells = c(row.names(Ukn_NKX21_MoveTo_vRG), row.names(Ukn_NKX21_2_MoveToIPC), row.names(Ukn_NKX21_MoveTo_oRG1), row.names(Ukn_NKX21_MoveToGlialProgen)), invert=T)


##Recombine
GlialProgen_Clean = subset(RestCells_V3, cells = c(colnames(Keep_GlialProgen), row.names(vRG_MoveToGlialProgen), row.names(IPC_MoveToGlialProgen), row.names(oRG_2_MoveToGlialProgen), row.names(Astrocytes_2_MoveToGlialProgen), row.names(Ukn_NKX21_MoveToGlialProgen)))
vRG_Clean = subset(RestCells_V3, cells = c(colnames(Keep_vRG), row.names(GlialProgen_MoveTo_vRG), row.names(IPC_MoveTo_vRG), row.names(oRG_2_MoveTo_vRG), row.names(Astrocytes_2_MoveTo_vRG), row.names(Ukn_NKX21_MoveTo_vRG)))
IPC_Clean = subset(RestCells_V3, cells = c(colnames(Keep_IPC), row.names(GlialProgen_MoveToIPC), row.names(oRG_2_MoveToIPC), row.names(Astrocytes_2_MoveToIPC), row.names(Ukn_NKX21_2_MoveToIPC)))
oRG_1_Clean = subset(RestCells_V3, cells = c(colnames(oRG_1), row.names(GlialProgen_MoveTo_oRG1), row.names(vRG_MoveTo_oRG1), row.names(IPC_MoveTo_oRG1), row.names(oRG_2_MoveTo_oRG1), row.names(Astrocytes_2_MoveTo_oRG1), row.names(Ukn_NKX21_MoveTo_oRG1)))
oRG_2_Clean = subset(RestCells_V3, cells = c(colnames(Keep_oRG_2)))
Astrocytes_Clean = subset(RestCells_V3, cells = c(colnames(Keep_Astrocytes), row.names(GlialProgen_MoveTo_Astro), row.names(IPC_MoveTo_Astro), row.names(oRG_2_MoveTo_Astro)))
Ukn_NKX21_Clean = subset(RestCells_V3, cells = c(colnames(Keep_Ukn_NKX21), row.names(IPC_MoveToNKX21), row.names(oRG_2_MoveToNKX21)))



DefaultAssay(Fetal.combined) = "RNA"
Fetal.combined_DEGs = DEGs(Fetal.combined)
Fetal.combined_DEGs_Subs = subset(Fetal.combined_DEGs, Fetal.combined_DEGs$p_val_adj < 0.05 & Fetal.combined_DEGs$avg_log2FC > 0.5  & Fetal.combined_DEGs$cluster %in% "[NKX2-1]") #& Fetal.combined_DEGs$gene %in% HumanTFs$V1

DefaultAssay(Fetal.combined) = "RNA"
fplist = list()
for(g in unique(Fetal.combined_DEGs_Subs$gene)){
fplist[[g]] = FeaturePlot(Fetal.combined, g, reduction="umap")
}
pdf("IndivFeatPlots_AllCells_DEGs_NonTF.pdf", width = 10, height = 8)
print(fplist)
dev.off()
```

### Compile Assignments for main populations
```{r}
OCT_AllCell_Assigns = as.data.frame(c(rep("Blood", length(row.names(Blood))),
                                 rep("Microglia", length(row.names(Microglia))), 
                                 rep("Vasculature", length(row.names(Vasculature))), 
                                 rep("Neurons [Mature]", length(row.names(Neurons))), 
                                 rep("Leptomeningeal", length(row.names(VLMC))), 
                                 rep("Oligodendrocytes [Mature]", length(colnames(Oligodendro_Final))), 
                                 rep("Astrocytes [Dividing Progenitors]", length(colnames(Dividing_Astro_Final))), 
                                 rep("Oligodendrocytes [Dividing Progenitors]", length(colnames(Dividing_Oligo_Final))), 
                                 rep("Neurons [Dividing Progenitors]", length(colnames(Dividing_Neuro_Final))), 
                                 rep("Ependymocytes", length(colnames(Ependy_Final_V2))), 
                                 rep("Tanycytes", length(colnames(Tany_Final))), 
                                 rep("RadialGlia_05", length(colnames(GlialProgen_Clean))),
                                 rep("RadialGlia_01 [vRG]", length(colnames(vRG_Clean))), # [PAX6/NES
                                 rep("RadialGlia_04 [IPC]", length(colnames(IPC_Clean))), # [EGFR/ASCL1
                                 rep("RadialGlia_02 [oRG]", length(colnames(oRG_1_Clean))), # [HOPX]
                                 rep("RadialGlia_03 [oRG]", length(colnames(oRG_2_Clean))), #[HOPX/STMN1]
                                 rep("Astrocytes [Mature]", length(colnames(Astrocytes_Clean))),  #GFAP/ID3
                                 rep("RadialGlia_06", length(colnames(Ukn_NKX21_Clean)))))
colnames(OCT_AllCell_Assigns) = "Pop"
row.names(OCT_AllCell_Assigns) = c(row.names(Blood), #row.names(OCT_AllCell_Assigns) #OCT_AllCell_Assigns$Barcs
                            row.names(Microglia),
                            row.names(Vasculature),
                            row.names(Neurons),
                            row.names(VLMC),
                            colnames(Oligodendro_Final),
                            colnames(Dividing_Astro_Final),
                            colnames(Dividing_Oligo_Final),
                            colnames(Dividing_Neuro_Final),
                            colnames(Ependy_Final_V2),
                            colnames(Tany_Final),
                            colnames(GlialProgen_Clean), 
                            colnames(vRG_Clean),
                            colnames(IPC_Clean),
                            colnames(oRG_1_Clean),
                            colnames(oRG_2_Clean),
                            colnames(Astrocytes_Clean),
                            colnames(Ukn_NKX21_Clean))
#OCT_AllCell_Assigns$Dup = duplicated(OCT_AllCell_Assigns$Barcs) | duplicated(OCT_AllCell_Assigns$Barcs, fromLast = T)
#CHECK =  subset(OCT_AllCell_Assigns, OCT_AllCell_Assigns$Dup == T)
#table(CHECK$Pop)
Check = as.data.frame(table(OCT_AllCell_Assigns$Pop))
paste(sort(unique(OCT_AllCell_Assigns$Pop)), collapse=", ")

OCT_AllCell_Assigns$Pop = factor(OCT_AllCell_Assigns$Pop, levels = c("RadialGlia_01 [vRG]", "RadialGlia_02 [oRG]", "RadialGlia_03 [oRG]", "RadialGlia_04 [IPC]", "RadialGlia_05", "RadialGlia_06", "Neurons [Dividing Progenitors]", "Neurons [Mature]", "Oligodendrocytes [Dividing Progenitors]", "Oligodendrocytes [Mature]", "Astrocytes [Dividing Progenitors]", "Astrocytes [Mature]", "Ependymocytes",   "Tanycytes", "Microglia", "Leptomeningeal", "Vasculature", "Blood"))

write.csv(OCT_AllCell_Assigns, paste("OCT_AllCell_Assigns_", Sys.Date(), ".csv", sep=""))
Fetal.combined = AddMetaData(Fetal.combined, OCT_AllCell_Assigns, "OCT_AllCell_Assigns")
Idents(Fetal.combined) = "OCT_AllCell_Assigns"

pdf(paste("ALLFETAL_Assignments_UMAP_27OCT21.pdf", sep=""), width=12, height=8)
print(DimPlot(Fetal.combined, reduction = "umap", label=T))
dev.off()

pdf(paste("ALLFETAL_Assignments_UMAP_20SEP_Small.pdf", sep=""), width=6, height=5)
print(DimPlot(Fetal.combined, reduction = "umap"))
dev.off()

Idents(Fetal.combined) = "orig.ident"
Filename = "OCT_AllCell_Assigns"
pdf(paste("ALLFETAL_", Filename, "_GroupedUMAP.pdf", sep=""), width=10, height=8)
print(DimPlot(Fetal.combined, reduction = "umap", group.by = "orig.ident", label=T))
dev.off()

Idents(Fetal.combined) = "orig.ident"
pdf(paste("ALLFETAL_", Filename, "_SplitUMAP.pdf", sep=""), width=60, height=8)
print(DimPlot(Fetal.combined, reduction = "umap", split.by = "orig.ident"))
dev.off()



###### Choose genes for multiviolin
GeneList = c("GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "ALDOC", "SLC1A2", "ID3", "ASCL1", "APOE", "FOXJ1", "RAX", "ADM",  "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "RFX2", "RFX3", "RFX4", "NPAS3", "COL23A1", "CD59", "SLC17A8", "VCAN", "FRZB", "PENK", "COL25A1", "CACNA2D2", "STOML3", "TMEM212", "PCP4L1", "TM4SF1", "PLTP", "TGFB2", "SLC7A11", "NR2E1", "EPHB1", "P3H2", "NELL2", "GRIA2", "RLBP1", "SIX6", "FGF10", "SCN7A", "FNDC3C1", "MEST", "IGFBP5", "RGCC", "RGS7BP", "CTGF", "SPRR1A", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9", "SYT1", "STMN2", "SLC17A6", "SLC17A8", "VIM", "ASCL1", "NES", "HES1", "MKI67", "FABP7", "TTYH1", "HMGA2", "MEIS2", "SLC1A3", "FAM107A", "NKX2-1", "NHLH2", "DLX1",  "C1QB", "AIF1", "P2RY12", "MRC1", "SELPLG", "CST3", "RPL4", "EEF1A1", "SLC2A5", "CTSB", "APOE", "TMSB4X", "SPARC", "PDGFRA", "COL1A1", "COL1A2", "LUM", "DCN", "FBLN1", "TBX18", "VTN", "OGN", "IGFBP2", "IL33", "PTGDS", "NNAT", "RSPO3", "NOV", "SLC47A1", "SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1", "CCNB1", "KIF4", "CENPE", "CCL19", "MAFB", "MMM1", "LRRC55", "ATP1B1", "CHN2", "DDC", "LYPD1", "CDKN2B", "PLAUR", "MCAM", "PCDH17", "FBLN2", "SEMA3G", "BMX", "ABCB1", "TEK", "ACTA2", "TAGLN", "ADGRF5", "EMCN", "VTN", "CSPRG4", "ATP13A5", "PTH1R", "KCNJ8", "ABCC9", "APN", "CD82", "CHST1", "PLN", "PDGFRB", "RGS5", "IGFBP2", "LUM", "DCN")

DefaultAssay(Fetal.combined) = "RNA"
GeneListIN = subset(GeneList, GeneList %in% row.names(Fetal.combined))
fplist = list()
for(g in GeneListIN){
fplist[[g]] = VlnPlot(Fetal.combined, g, pt.size=0)
}
pdf("IndivVlnPlots_AllCells.pdf", width = 15, height = 8)
print(fplist)
dev.off()



###### Generate Multiviolin
AllCells_PlotOrder = c("SYT1", "OLIG1", "PDGFRA", "OLIG2", "FYN", "AQP4",  "GFAP", "ALDOC", "SLC1A2","FOXJ1", "CCDC153", "RFX2", "CRYM","C1QB", "AIF1", "HES1", "COL1A1","RGS5", "PDGFRB",  "HBA1")

DefaultAssay(Fetal.combined) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Fetal.combined, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16),
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Fetal.combined, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(3.4, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_MultiVln_6OCT21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Fetal.combined@active.ident))*1.5)
print(AllVlns)
dev.off()
```


#############################################
################# NEURONS ###################
#############################################

### Run iterative clustering parameters for Neuron subclustering and remove FOXG1 and NEUROD6
```{r}
DPlist2 = list()
DPlist = list()
FPlist = list()
Filename = "Neurons_14OCT"
for(y in seq(10,50,10)){
DefaultAssay(Neurons_Seurat) = "integrated"
Neurons_Seurat = RunPCA(Neurons_Seurat, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Neurons_Seurat <- RunUMAP(Neurons_Seurat, dims = 1:z, spread= s)
DefaultAssay(Neurons_Seurat) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Neurons_Seurat, unique(c( "SYT1", "FOXG1", "NEUROD6", "TBX3", "SIM1", "POMC", WellKnownGenes, SuspectedGenes)), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Neurons_Seurat, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Neurons_Seurat, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=50)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()


#Want 38
DefaultAssay(Neurons_Seurat) = "integrated"
Neurons_Seurat = RunPCA(Neurons_Seurat, npcs = 50)
Neurons_Seurat <- RunUMAP(Neurons_Seurat, dims = 1:30, spread= 5)
Neuronal_UMAP = as.data.frame(Neurons_Seurat@reductions$umap@cell.embeddings)

#Remove FOXG1 and NEUROD6
set.dim = 30
set.res = 1
set.kparam = c(3)
ClusterFunc_Neuro_RNA(Neurons_Seurat)

Neurons_Seurat <- FindNeighbors(Neurons_Seurat, k.param=3, dims=1:30)
Neurons_Seurat <- FindClusters(Neurons_Seurat, resolution =1)
MostlyFOXG1Neg = subset(Neurons_Seurat, idents= c(0,3,4,5,7,10,13,15,16,19,20,21,22,24,25,26,28,32,33,38))
FOXG1Pos = subset(Neurons_Seurat, idents= c(8,9,12,23,27,35,37, 1,2,6,11,14,17,18,29,30,31,34,36)) 

set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos)

FOXG1Pos <- FindNeighbors(FOXG1Pos, k.param=2, dims=1:30)
FOXG1Pos <- FindClusters(FOXG1Pos, resolution =1)
FOXG1Pos_V2 = subset(FOXG1Pos, idents= c(8, 27, 32, 38, 51, 54, 57, 62, 63, 71, 74, 75, 77, 80, 86, 89, 94, 97, 103, 110, 119, 129, 138, 144, 156, 168, 184, 189, 199, 207, 211, 227, 238, 250, 252, 273, 277, 301, 316, 319, 360, 364, 376, 387, 392, 426, 433, 440, 472, 484, 502, 541, 546, 4, 20, 26, 29, 33, 39, 42, 48, 60, 73, 88, 92, 98, 115, 126, 128, 139, 141, 149, 153, 155, 174, 196, 210, 218, 222, 223, 251, 366, 386, 417, 458, 471, 2, 14, 21, 28, 31, 35, 37, 46, 76, 78, 82, 87, 95, 109, 112, 148, 150, 179, 226, 247, 272, 325, 1, 3, 5, 18, 24, 25, 70, 178, 0, 7, 17, 53, 55, 81, 107, 6, 122, 175, 197, 22, 56, 64, 10, 47, 12, 11, 30, 15, 13, 34, 66, 9))
MostlyFOXG1Neg_Pt2 = subset(FOXG1Pos, idents= c(8, 27, 32, 38, 51, 54, 57, 62, 63, 71, 74, 75, 77, 80, 86, 89, 94, 97, 103, 110, 119, 129, 138, 144, 156, 168, 184, 189, 199, 207, 211, 227, 238, 250, 252, 273, 277, 301, 316, 319, 360, 364, 376, 387, 392, 426, 433, 440, 472, 484, 502, 541, 546, 4, 20, 26, 29, 33, 39, 42, 48, 60, 73, 88, 92, 98, 115, 126, 128, 139, 141, 149, 153, 155, 174, 196, 210, 218, 222, 223, 251, 366, 386, 417, 458, 471, 2, 14, 21, 28, 31, 35, 37, 46, 76, 78, 82, 87, 95, 109, 112, 148, 150, 179, 226, 247, 272, 325, 1, 3, 5, 18, 24, 25, 70, 178, 0, 7, 17, 53, 55, 81, 107, 6, 122, 175, 197, 22, 56, 64, 10, 47, 12, 11, 30, 15, 13, 34, 66, 9), invert=T)


set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_V2)


FOXG1Pos_V2 <- FindNeighbors(FOXG1Pos_V2, k.param=2, dims=1:30)
FOXG1Pos_V2 <- FindClusters(FOXG1Pos_V2, resolution =1)
MostlyFOXG1Neg_Pt3 = subset(FOXG1Pos_V2, idents= c(50, 68, 71, 74, 77, 80, 86, 90, 111, 117, 118, 124, 127, 129, 132, 133, 135, 139, 141, 142, 143, 144, 152, 161, 167, 168, 170, 173, 174, 175, 176, 177, 178, 179, 180, 182, 185, 186, 14, 17, 34, 58, 64,69, 83, 105, 112, 116, 120, 123, 126, 134, 138, 140, 147, 149, 154, 155, 156, 160, 171, 183, 187))
FOXG1Pos_V3 = subset(FOXG1Pos_V2, idents= c(50, 68, 71, 74, 77, 80, 86, 90, 111, 117, 118, 124, 127, 129, 132, 133, 135, 139, 141, 142, 143, 144, 152, 161, 167, 168, 170, 173, 174, 175, 176, 177, 178, 179, 180, 182, 185, 186, 14, 17, 34, 58, 64,69, 83, 105, 112, 116, 120, 123, 126, 134, 138, 140, 147, 149, 154, 155, 156, 160, 171, 183, 187), invert=T)

set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_V3)



FOXG1Pos_V3 <- FindNeighbors(FOXG1Pos_V3, k.param=2, dims=1:30)
FOXG1Pos_V3 <- FindClusters(FOXG1Pos_V3, resolution =1)
MostlyFOXG1Neg_Pt4 = subset(FOXG1Pos_V3, idents= c(47, 59, 95, 98, 102, 106, 107, 114, 121, 122, 129, 134, 5, 56, 68, 81, 94, 101, 108, 126, 130, 132, 135))
FOXG1Pos_V4 = subset(FOXG1Pos_V3, idents= c(47, 59, 95, 98, 102, 106, 107, 114, 121, 122, 129, 134, 5, 56, 68, 81, 94, 101, 108, 126, 130, 132, 135), invert=T)


set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_V4)


FOXG1Pos_V4 <- FindNeighbors(FOXG1Pos_V4, k.param=2, dims=1:30)
FOXG1Pos_V4 <- FindClusters(FOXG1Pos_V4, resolution =1)
MostlyFOXG1Neg_Pt5 = subset(FOXG1Pos_V4, idents= c(93, 101, 104, 106, 111, 113, 130, 131, 133, 135, 137, 8, 14, 29, 58, 70, 96, 100, 110, 112, 120, 132, 134))
FOXG1Pos_V5 = subset(FOXG1Pos_V4, idents= c(93, 101, 104, 106, 111, 113, 130, 131, 133, 135, 137, 8, 14, 29, 58, 70, 96, 100, 110, 112, 120, 132, 134), invert=T)

set.dim = 20
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_V5)




FOXG1Pos_V5 <- FindNeighbors(FOXG1Pos_V5, k.param=2, dims=1:20)
FOXG1Pos_V5 <- FindClusters(FOXG1Pos_V5, resolution =1)
MostlyFOXG1Neg_Pt6 = subset(FOXG1Pos_V5, idents= c(6, 18, 21, 32, 38, 41, 56, 58, 67, 75, 83, 90, 94, 99, 102, 108, 110, 120, 123, 125, 127, 133, 139, 151, 161, 165, 168, 171, 172, 173, 176, 181, 185, 191, 192, 197, 208, 209, 221, 8, 16, 23, 28, 29, 34, 37, 39, 44, 48, 59, 62, 72, 74, 76, 77, 106, 109, 115, 128, 142, 152, 160, 186, 198, 223, 1, 4, 5, 12, 14, 27, 43, 55, 60, 61, 71, 73, 85, 89, 92, 122, 155, 190, 3, 13, 19, 26, 30, 36, 40, 50, 54, 65, 68, 118, 147, 9, 20, 25, 31, 46, 96, 22, 24, 57, 45, 112, 15, 0, 2, 7, 11, 51, 10, 17), invert=T)
FOXG1Pos_V6 = subset(FOXG1Pos_V5, idents= c(6, 18, 21, 32, 38, 41, 56, 58, 67, 75, 83, 90, 94, 99, 102, 108, 110, 120, 123, 125, 127, 133, 139, 151, 161, 165, 168, 171, 172, 173, 176, 181, 185, 191, 192, 197, 208, 209, 221, 8, 16, 23, 28, 29, 34, 37, 39, 44, 48, 59, 62, 72, 74, 76, 77, 106, 109, 115, 128, 142, 152, 160, 186, 198, 223, 1, 4, 5, 12, 14, 27, 43, 55, 60, 61, 71, 73, 85, 89, 92, 122, 155, 190, 3, 13, 19, 26, 30, 36, 40, 50, 54, 65, 68, 118, 147, 9, 20, 25, 31, 46, 96, 22, 24, 57, 45, 112, 15, 0, 2, 7, 11, 51, 10, 17))

set.dim = 10
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_V6)



FOXG1Pos_V6 <- FindNeighbors(FOXG1Pos_V6, k.param=2, dims=1:10)
FOXG1Pos_V6 <- FindClusters(FOXG1Pos_V6, resolution =1)
FOXG1Pos_V7 = subset(FOXG1Pos_V6, idents= c(11, 29, 45, 48, 49, 52, 55, 61, 62, 64, 65, 66, 71, 72, 73, 84, 92, 93, 96, 102, 103, 111, 112, 117, 118, 119, 120, 123, 127, 133, 137, 142, 145, 146,148, 151, 152, 154, 157, 158, 163, 166, 168, 169, 171, 176, 177, 180, 181, 183, 187, 193, 194, 195, 196, 199, 202, 203, 204, 205, 206, 207, 211, 218, 220, 221, 223, 224, 225, 231, 235, 236, 237, 240), invert=T)

FOXG1Neg_Recombo = subset(Neurons_Seurat, cells = colnames(FOXG1Pos_V7), invert=T)
set.dim = 30
set.res = 10
set.kparam = c(3)
ClusterFunc_Neuro_RNA(FOXG1Neg_Recombo)





FOXG1Neg_Recombo <- FindNeighbors(FOXG1Neg_Recombo, k.param=3, dims=1:30)
FOXG1Neg_Recombo <- FindClusters(FOXG1Neg_Recombo, resolution =10)
FOXG1Pos_Pt2 = subset(FOXG1Neg_Recombo, idents= c(190, 139, 140, 29, 75, 145, 79, 36, 84, 90, 15, 95, 96, 154, 16, 4, 105, 160, 7, 163, 28, 32, 122, 168, 170, 38, 72, 130, 195, 3, 83, 85, 175, 137, 49, 177, 94, 97, 183, 104, 148, 184, 65, 40, 20, 119, 70, 43, 151, 153, 45, 189, 129, 89, 166, 63, 102, 66, 46, 142, 144, 111, 120, 191, 60, 41, 127, 44, 93, 181, 108, 185, 82, 114, 88, 123, 171, 150, 176, 14, 157, 76, 161, 138, 10, 143, 146, 11, 186, 33, 200, 71))
FOXG1Neg_Recombo_V2 = subset(FOXG1Neg_Recombo, idents= c(190, 139, 140, 29, 75, 145, 79, 36, 84, 90, 15, 95, 96, 154, 16, 4, 105, 160, 7, 163, 28, 32, 122, 168, 170, 38, 72, 130, 195, 3, 83, 85, 175, 137, 49, 177, 94, 97, 183, 104, 148, 184, 65, 40, 20, 119, 70, 43, 151, 153, 45, 189, 129, 89, 166, 63, 102, 66, 46, 142, 144, 111, 120, 191, 60, 41, 127, 44, 93, 181, 108, 185, 82, 114, 88, 123, 171, 150, 176, 14, 157, 76, 161, 138, 10, 143, 146, 11, 186, 33, 200, 71), invert=T)

set.dim = 30
set.res = 5
set.kparam = c(3)
ClusterFunc_Neuro_RNA(FOXG1Pos_Pt2) 
ClusterFunc_Neuro_RNA(FOXG1Neg_Recombo_V2) #Almost completely clean



FOXG1Pos_Pt2 <- FindNeighbors(FOXG1Pos_Pt2, k.param=3, dims=1:30)
FOXG1Pos_Pt2 <- FindClusters(FOXG1Pos_Pt2, resolution =5)
FOXG1Pos_Pt2_V2 = subset(FOXG1Pos_Pt2, idents= c(77, 87, 88, 91, 16, 25, 42, 66, 68, 73, 74, 79, 89, 92, 7, 27, 36, 39, 52, 55, 76, 80, 84, 86, 90), invert=T) #High FOXG1
FOXG1Neg_Pt3 = subset(FOXG1Pos_Pt2, idents= c(77, 87, 88, 91, 16, 25, 42, 66, 68, 73, 74, 79, 89, 92, 7, 27, 36, 39, 52, 55, 76, 80, 84, 86, 90)) #Low FOXG1


FOXG1Neg_Recombo_V3 = subset(Neurons_Seurat, cells = c(colnames(FOXG1Neg_Pt3), colnames(FOXG1Neg_Recombo_V2)))
ClusterFunc_Neuro_RNA(FOXG1Neg_Recombo_V3) #Mostly clean but still needs some additional cleaning
set.dim = 20
set.res = 10
set.kparam = c(3)
ClusterFunc_Neuro_RNA(FOXG1Pos_Pt2_V2) #Needs considerable cleaning




FOXG1Pos_Pt2_V2 <- FindNeighbors(FOXG1Pos_Pt2_V2, k.param=3, dims=1:20)
FOXG1Pos_Pt2_V2 <- FindClusters(FOXG1Pos_Pt2_V2, resolution =10)
FOXG1Neg_Pt4 = subset(FOXG1Pos_Pt2_V2, idents= c(22,23,46,58,66,75,82,85,87,94,95,105,117,124,127,129,130,131,134,137,138,139,33,36,42,49,55,63,88,93))
FOXG1Pos_Pt2_V3 = subset(FOXG1Pos_Pt2_V2, idents= c(22,23,46,58,66,75,82,85,87,94,95,105,117,124,127,129,130,131,134,137,138,139,33,36,42,49,55,63,88,93), invert=T)
set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_Pt2_V3) 

FOXG1Pos_Pt2_V3 <- FindNeighbors(FOXG1Pos_Pt2_V3, k.param=2, dims=1:30)
FOXG1Pos_Pt2_V3 <- FindClusters(FOXG1Pos_Pt2_V3, resolution =1)
FOXG1Neg_Pt5 = subset(FOXG1Pos_Pt2_V3, idents= c(14, 18, 22, 28, 29, 31, 32, 45, 51, 54, 58, 62, 64, 65, 66, 75, 77, 84, 87, 99, 102, 103, 107, 108, 110, 111, 115, 117, 118, 123, 124, 125, 129, 131, 132, 133, 135, 139, 147, 149, 150, 153, 157, 158, 159, 164, 165, 169, 170, 171, 175, 177, 179, 180, 181, 182, 184, 185, 186, 191, 194, 196, 197, 198, 201, 202, 204, 205, 206, 207, 208, 209, 211, 212, 213, 215, 216, 219, 225, 226, 228, 229, 230, 231, 232, 233, 234, 236, 237, 238, 240, 241, 242, 243, 247, 252, 255, 258, 259, 260, 261, 262, 263, 264, 267, 268, 269, 271, 273, 276, 277, 279, 280, 281, 282, 283, 284, 286, 287, 288, 289, 290, 291, 292, 293, 296, 297, 298, 299, 301, 302, 303, 310, 311, 312, 314, 315, 316, 317, 319, 320, 321, 322, 323, 325, 326, 327, 328, 329, 330, 331, 332, 333, 335, 336, 337, 340, 342, 345, 346, 347, 351, 352, 353, 354, 355, 356, 359, 361, 362, 363, 365, 366, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 381, 382, 383, 385, 388, 391, 394, 15, 20, 21))
FOXG1Pos_Pt2_V4 = subset(FOXG1Pos_Pt2_V3, cells = colnames(FOXG1Neg_Pt5), invert=T)


set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_Pt2_V4) 


FOXG1Pos_Pt2_V4 <- FindNeighbors(FOXG1Pos_Pt2_V4, k.param=2, dims=1:30)
FOXG1Pos_Pt2_V4 <- FindClusters(FOXG1Pos_Pt2_V4, resolution =1)
FOXG1Neg_Pt6 = subset(FOXG1Pos_Pt2_V4, idents= c(18, 49, 57, 80, 82, 96, 97, 99, 131, 135, 143, 146, 160, 166, 174, 179, 182, 183, 184, 189, 194, 198, 203, 12, 22, 23, 27, 38, 43, 44, 46, 50, 53, 54, 55, 56, 59, 60, 61, 65, 66, 67, 76, 77, 84, 87, 90, 91, 92, 93, 94, 95, 100, 101, 106, 108, 109, 110, 111, 112, 113, 114, 116, 118, 119, 123, 125, 126, 127, 128, 129, 130, 134, 136, 137, 138, 139, 141, 144, 145, 147, 148, 150, 153, 154, 156, 157, 161, 163, 164, 165, 167, 168, 169, 170, 172, 173, 175, 177, 180, 181, 185, 186, 188, 190, 191, 192, 193, 195, 196, 197, 199, 200, 201, 202, 204, 205, 206, 207, 208, 209, 211, 212, 213, 214, 215))
FOXG1Pos_Pt2_V5 = subset(FOXG1Pos_Pt2_V4, cells = colnames(FOXG1Neg_Pt6), invert=T)

FOXG1Pos_Recombo = subset(Neurons_Seurat, cells = c(colnames(FOXG1Pos_Pt2_V5), colnames(FOXG1Pos_V7)))
set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_Recombo) 


FOXG1Pos_Recombo <- FindNeighbors(FOXG1Pos_Recombo, k.param=2, dims=1:30)
FOXG1Pos_Recombo <- FindClusters(FOXG1Pos_Recombo, resolution =1)
FOXG1Pos_Recombo_V2 = subset(FOXG1Pos_Recombo, idents= c(80, 87, 114, 117, 135, 145, 149, 157, 162, 165, 166, 169, 172, 176, 182, 184, 191, 64, 72, 83, 90, 96, 106, 109, 134, 146, 147, 151, 171, 177, 181, 183, 185, 186, 188, 192), invert=T)
set.dim = 30
set.res = 1
set.kparam = c(2)
ClusterFunc_Neuro_RNA(FOXG1Pos_Recombo_V2) 

FOXG1Neg_Recombo_V4 = subset(Neurons_Seurat, cells = c(colnames(FOXG1Pos_Recombo_V2)), invert=T)

set.dim = 30
set.res = 1
set.kparam = c(3)
ClusterFunc_Neuro_RNA(FOXG1Neg_Recombo_V4) 

MoveToFOXG1Pos = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(FOXG1Neg_Recombo_V4) & Fetal.combined_Data$FOXG1 > 0 | row.names(Fetal.combined_Data) %in% colnames(FOXG1Neg_Recombo_V4) & Fetal.combined_Data$NEUROD6 > 0)
MoveToFOXG1PosV2 = subset(Neuronal_UMAP, row.names(Neuronal_UMAP) %in% colnames(FOXG1Neg_Recombo_V4) & Neuronal_UMAP$UMAP_2 > 20 | row.names(Neuronal_UMAP) %in% colnames(FOXG1Neg_Recombo_V4) & Neuronal_UMAP$UMAP_2 < -10 & Neuronal_UMAP$UMAP_1 > 20)



FOXG1Neg_Recombo_V5 = subset(FOXG1Neg_Recombo_V4, cells = c(row.names(MoveToFOXG1Pos), row.names(MoveToFOXG1PosV2)), invert=T)
FOXG1Pos_Recombo_V3= subset(Neurons_Seurat, cells = colnames(FOXG1Neg_Recombo_V5), invert=T)
set.dim = 30
set.res = 1
set.kparam = c(5)
ClusterFunc_Neuro_RNA(FOXG1Neg_Recombo_V5) ###SUPER CLEAN
ClusterFunc_Neuro_RNA(FOXG1Pos_Recombo_V3) ###NOW NEED TO PUSH TBX3 and SIM1 back


FOXG1Pos_Recombo_V3 <- FindNeighbors(FOXG1Pos_Recombo_V3, k.param=5, dims=1:30)
FOXG1Pos_Recombo_V3 <- FindClusters(FOXG1Pos_Recombo_V3, resolution =1)
FOXG1Pos_TBX3_SIM1 = subset(FOXG1Pos_Recombo_V3, idents= c(5,13,19,20,17))
FOXG1Pos_ExclTBX3_SIM1 = subset(FOXG1Pos_Recombo_V3, idents= c(5,13,19,20,17, 21), invert=T) 


set.dim = 30
set.res = 3
set.kparam = c(3)
ClusterFunc_Neuro_RNA(FOXG1Pos_ExclTBX3_SIM1) 


FOXG1Pos_ExclTBX3_SIM1 <- FindNeighbors(FOXG1Pos_ExclTBX3_SIM1, k.param=3, dims=1:30)
FOXG1Pos_ExclTBX3_SIM1 <- FindClusters(FOXG1Pos_ExclTBX3_SIM1, resolution =3)
FOXG1Pos_ExclTBX3_SIM1_V2 = subset(FOXG1Pos_ExclTBX3_SIM1, idents= c(4,6,15,16,49,50), invert=T) ####FINAL GROUP OF CELLS TO BE REMOVED


NonHypothalamicNeurons = subset(Neurons_Seurat, cells = colnames(FOXG1Pos_ExclTBX3_SIM1_V2))
HypothalamicNeurons = subset(Neurons_Seurat, cells = colnames(FOXG1Pos_ExclTBX3_SIM1_V2), invert=T)


set.dim = 30
set.res = 1
set.kparam = c(5)
ClusterFunc_Neuro_RNA(NonHypothalamicNeurons) 
ClusterFunc_Neuro_RNA(HypothalamicNeurons) 


OCT_ALLNEURONAL_Assigns_Renamed = as.data.frame(c(rep("Non-Hypothalamic", length(colnames(NonHypothalamicNeurons))), 
                                 rep("Hypothalamic", length(colnames(HypothalamicNeurons))))) 
colnames(OCT_ALLNEURONAL_Assigns_Renamed) = "Pop"
row.names(OCT_ALLNEURONAL_Assigns_Renamed) = c(colnames(NonHypothalamicNeurons), #row.names(OCT_ALLNEURONAL_Assigns_Renamed) #OCT_ALLNEURONAL_Assigns_Renamed$Barcs
                                           colnames(HypothalamicNeurons))

OCT_ALLNEURONAL_Assigns_Renamed$Pop = factor(OCT_ALLNEURONAL_Assigns_Renamed$Pop, levels = c("Hypothalamic", "Non-Hypothalamic"))

Neurons_Seurat = AddMetaData(Neurons_Seurat, OCT_ALLNEURONAL_Assigns_Renamed, "OCT_ALLNEURONAL_Assigns_Renamed")
Idents(Neurons_Seurat) = "OCT_ALLNEURONAL_Assigns_Renamed"

pdf(paste("ALLFETAL_ALLNEURONAL_3DEC21_V2.pdf", sep=""), width=12, height=8)
print(DimPlot(Neurons_Seurat, reduction = "umap"))
dev.off()
```

#Iterative clustering parameters for Hypothalamic Neurons
```{r}
DPlist2 = list()
DPlist = list()
FPlist = list()
Filename = "HypothalamicNeurons"
for(y in seq(10,50,10)){
DefaultAssay(HypothalamicNeurons) = "integrated"
HypothalamicNeurons = RunPCA(HypothalamicNeurons, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
HypothalamicNeurons <- RunUMAP(HypothalamicNeurons, dims = 1:z, spread= s)
DefaultAssay(HypothalamicNeurons) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(HypothalamicNeurons, unique(c( "SYT1", "FOXG1", "NEUROD6", "TBX3", "SIM1", "POMC", WellKnownGenes, SuspectedGenes)), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(HypothalamicNeurons, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(HypothalamicNeurons, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=50)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()
```

#Subclustering neurons
```{r}
#Use version 28 - "PCA 40 _dims 40 _spread 2"
DefaultAssay(HypothalamicNeurons) = "integrated"
HypothalamicNeurons = RunPCA(HypothalamicNeurons, npcs = 40)
HypothalamicNeurons <- RunUMAP(HypothalamicNeurons, dims = 1:40, spread= 2)
HypothalamicNeurons <- FindNeighbors(HypothalamicNeurons, k.param=50, dims=1:40)
HypothalamicNeurons <- FindClusters(HypothalamicNeurons, resolution =1)
HypothalamicNeurons_UMAP = as.data.frame(HypothalamicNeurons@reductions$umap@cell.embeddings)
set.dim = 40
set.res = c(1)
set.kparam = c(500)
ClusterFunc_Neuro_RNA(HypothalamicNeurons)

LH_1 = subset(HypothalamicNeurons_UMAP, HypothalamicNeurons_UMAP$UMAP_2 > 9 & HypothalamicNeurons_UMAP$UMAP_1 > -3)
UKN_TAC1 = subset(HypothalamicNeurons_UMAP, HypothalamicNeurons_UMAP$UMAP_2 > 6 & HypothalamicNeurons_UMAP$UMAP_2 < 9.5 & HypothalamicNeurons_UMAP$UMAP_1 < -3)

HypothalamicNeurons_Rest1 = subset(HypothalamicNeurons, cells = c(row.names(LH_1), row.names(UKN_TAC1)), invert=T)

set.dim = 40
set.res = c(1)
set.kparam = c(50)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest1)

DefaultAssay(HypothalamicNeurons_Rest1) = "integrated"
HypothalamicNeurons_Rest1 <- FindNeighbors(HypothalamicNeurons_Rest1, k.param=50, dims=1:40)
HypothalamicNeurons_Rest1 <- FindClusters(HypothalamicNeurons_Rest1, resolution =1)
PVH_1 = subset(HypothalamicNeurons_Rest1, idents = 3)
VMH_1 = subset(HypothalamicNeurons_Rest1, idents = 4)
Arc_1 = subset(HypothalamicNeurons_Rest1, idents = 7)
SCN_1 = subset(HypothalamicNeurons_Rest1, idents = 8)
TM_1 = subset(HypothalamicNeurons_Rest1, idents = 10)


CheckInput = colnames(TM_1) #Check Populations
CheckUMAP(HypothalamicNeurons)

PVH_1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PVH_1) & HypothalamicNeurons_UMAP$UMAP_1 < -3 & HypothalamicNeurons_UMAP$UMAP_2 < -0.5 | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest1) & HypothalamicNeurons_UMAP$UMAP_1 < -4 & HypothalamicNeurons_UMAP$UMAP_2 < -3)
VMH_1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(VMH_1) & HypothalamicNeurons_UMAP$UMAP_1 < 1 & HypothalamicNeurons_UMAP$UMAP_1 > -4 & HypothalamicNeurons_UMAP$UMAP_2 > 1)
Arc_1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc_1) & HypothalamicNeurons_UMAP$UMAP_1 > 0 & HypothalamicNeurons_UMAP$UMAP_2 > 1 | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest1) & HypothalamicNeurons_UMAP$UMAP_1 > 1.5 & HypothalamicNeurons_UMAP$UMAP_1 < 8 & HypothalamicNeurons_UMAP$UMAP_2 > 2.5)
SCN_1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(SCN_1) & HypothalamicNeurons_UMAP$UMAP_1 > 5 & HypothalamicNeurons_UMAP$UMAP_2 < -0.5)
TM_1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(TM_1) | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest1) &HypothalamicNeurons_UMAP$UMAP_1 > 11 & row.names(HypothalamicNeurons_UMAP) %notin% colnames(Arc_1_Clean) )

intersect(row.names(PVH_1_Clean), colnames(HypothalamicNeurons_Rest2))

HypothalamicNeurons_Rest2 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(PVH_1_Clean), row.names(VMH_1_Clean), row.names(Arc_1_Clean), row.names(SCN_1_Clean)), invert=T) #, row.names(TM_1_Clean) = Exclude manually below

set.dim = 40
set.res = c(1)
set.kparam = 20 #c(5)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest2)

DefaultAssay(HypothalamicNeurons_Rest2) = "integrated"
HypothalamicNeurons_Rest2 <- FindNeighbors(HypothalamicNeurons_Rest2, k.param=20, dims=1:40)
HypothalamicNeurons_Rest2 <- FindClusters(HypothalamicNeurons_Rest2, resolution =1)
PVH_2 = subset(HypothalamicNeurons_Rest2, idents = 10)
TM_1 = subset(HypothalamicNeurons_Rest2, idents = 7)
UKN_PostMM = subset(HypothalamicNeurons_Rest2, idents = 2) #PITX2/SOX14 + LHX1 + FOXB1
UKN_CRABP1 = subset(HypothalamicNeurons_Rest2, idents = 14) #CRABP1
MN_1 = subset(HypothalamicNeurons_Rest2, idents = 9) #ONECUT2
UKN_Top = subset(HypothalamicNeurons_Rest2, idents = 5) #Ukn gene expression but distinct clutstering

set.dim = c(40)
set.res = c(0.5)
set.kparam = c(30) #c(5)
ClusterFunc_Neuro_RNA(UKN_PostMM)

UKN_PostMM <- FindNeighbors(UKN_PostMM, k.param=30, dims=1:40)
UKN_PostMM <- FindClusters(UKN_PostMM, resolution =0.5)
PVH_3 = subset(UKN_PostMM, idents = c(1))
UKN_PostMM_V2 = subset(UKN_PostMM, idents = c(1), invert=T)

CheckInput = colnames(UKN_Top) #Check Populations
CheckUMAP(HypothalamicNeurons_Rest2)

UKN_Top_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(UKN_Top) & HypothalamicNeurons_UMAP$UMAP_2 > 3 & HypothalamicNeurons_UMAP$UMAP_1 > -2 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean)  | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest2) & HypothalamicNeurons_UMAP$UMAP_2 > 3 & HypothalamicNeurons_UMAP$UMAP_1 > -3 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean))
PVH_2_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PVH_2) & HypothalamicNeurons_UMAP$UMAP_2 > 1.5 | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest2) & HypothalamicNeurons_UMAP$UMAP_1 > -6 & HypothalamicNeurons_UMAP$UMAP_1 < -2.8 & HypothalamicNeurons_UMAP$UMAP_2 > 2.5 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(UKN_Top_Clean))
PVH3_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PVH_3) & HypothalamicNeurons_UMAP$UMAP_1 < -3.5)
UKN_PostMM_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(UKN_PostMM_V2) & HypothalamicNeurons_UMAP$UMAP_2 > 0 & HypothalamicNeurons_UMAP$UMAP_1 < -5.6 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(PVH_2_Clean), row.names(PVH3_Clean)) | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest2) & HypothalamicNeurons_UMAP$UMAP_1 < -6 &  HypothalamicNeurons_UMAP$UMAP_2 > 1 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(PVH_2_Clean), row.names(PVH3_Clean)))
MN_1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(MN_1) & HypothalamicNeurons_UMAP$UMAP_2 < -1 & HypothalamicNeurons_UMAP$UMAP_2 > -4 & HypothalamicNeurons_UMAP$UMAP_1 > -4 & HypothalamicNeurons_UMAP$UMAP_1 < 1 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean)| row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest2) & HypothalamicNeurons_UMAP$UMAP_1 > -2.5  & HypothalamicNeurons_UMAP$UMAP_1 < 0 &  HypothalamicNeurons_UMAP$UMAP_2 > -4.5 &  HypothalamicNeurons_UMAP$UMAP_2 < -2.5 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean))


intersect(row.names(PVH_2_Clean), row.names(UKN_PostMM_Clean))


HypothalamicNeurons_Rest3 = subset(HypothalamicNeurons_Rest2, cells = c(row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean), row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean)), invert=T)



set.dim = c(40)
set.res = c(1)
set.kparam = c(5) 
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest3)

HypothalamicNeurons_Rest3 <- FindNeighbors(HypothalamicNeurons_Rest3, k.param=20, dims=1:40)
HypothalamicNeurons_Rest3 <- FindClusters(HypothalamicNeurons_Rest3, resolution =1)
NRN1_ToClean = subset(HypothalamicNeurons_Rest3, idents = c(1,7))
SIX3_ToClean = subset(HypothalamicNeurons_Rest3, idents = c(2))
SIX3_SST_ToClean = subset(HypothalamicNeurons_Rest3, idents = c(5))
PNOC_ToClean = subset(HypothalamicNeurons_Rest3, idents = c(3,6))
NHLH2_ToClean = subset(HypothalamicNeurons_Rest3, idents = c(8))
NKX21TRH_ToClean = subset(HypothalamicNeurons_Rest3, idents = c(10))

CheckInput = colnames(SIX6_ToClean) #Check Populations
CheckUMAP(HypothalamicNeurons_Rest2)


set.kparam = c(30,3) 
ClusterFunc_Neuro_RNA(PNOC_ToClean) #Preoptic, Strong Preoptic
PNOC_ToClean <- FindNeighbors(PNOC_ToClean, k.param=50, dims=1:40)
PNOC_ToClean <- FindClusters(PNOC_ToClean, resolution =1)
Preoptic_PNOC_NTS_ToClean = subset(PNOC_ToClean, idents = c(0))


set.dim = c(40)
set.kparam = c(4,5) 
ClusterFunc_Neuro_RNA(NRN1_ToClean) #SIM1 (3), TRH (3,7), PITX2/FOXP2/FOXP1/PBX3/PDYN (1,7), PENK/PNOC (6), RELN/CCK/FEZF,(2(x2),6) EBF3 (4)
NRN1_ToClean <- FindNeighbors(NRN1_ToClean, k.param=5, dims=1:40)
NRN1_ToClean <- FindClusters(NRN1_ToClean, resolution =1)
CCK_RELN_ToClean= subset(NRN1_ToClean, idents = c(2))
VMH2_EBF4_ToClean = subset(NRN1_ToClean, idents = c(4))
PENK_PNOC = subset(NRN1_ToClean, idents = c(6))
NRN1_ToClean_V2 = subset(NRN1_ToClean, idents = c(2,4,6), invert=T)

set.kparam = c(10,20,40) 
ClusterFunc_Neuro_RNA(NRN1_ToClean_V2) 
NRN1_ToClean_V2 <- FindNeighbors(NRN1_ToClean_V2, k.param=40, dims=1:40)
NRN1_ToClean_V2 <- FindClusters(NRN1_ToClean_V2, resolution =1)
MMPost_ToClean = subset(NRN1_ToClean_V2, idents = c(1))
PVH_4_ToClean= subset(NRN1_ToClean_V2, idents = c(1), invert=T)



VMH2_EBF4_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(VMH2_EBF4_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 < 
  0.5 & HypothalamicNeurons_UMAP$UMAP_1 < -5 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean) | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest3) &  HypothalamicNeurons_UMAP$UMAP_2 < -0.7 & HypothalamicNeurons_UMAP$UMAP_1 < -6.5 & row.names(HypothalamicNeurons_UMAP)   %notin% row.names(TM_1_Clean))
CCK_RELN_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(CCK_RELN_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 < 0.5 
  & HypothalamicNeurons_UMAP$UMAP_1 > -5 & HypothalamicNeurons_UMAP$UMAP_1 < -2.5 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean) | 
    row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest3) &  HypothalamicNeurons_UMAP$UMAP_2 < 0 &HypothalamicNeurons_UMAP$UMAP_2 > -2 
  & HypothalamicNeurons_UMAP$UMAP_1 > -4 & HypothalamicNeurons_UMAP$UMAP_1 < -3.5 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), 
  row.names(SCN_1_Clean)))
NKX21TRH_Clean = subset(HypothalamicNeurons_UMAP,  HypothalamicNeurons_UMAP$UMAP_2 < -2 & HypothalamicNeurons_UMAP$UMAP_1 > 8 & 
  row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), row.names(SCN_1_Clean)))
NHLH2_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(NHLH2_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 < 1 & 
  HypothalamicNeurons_UMAP$UMAP_1 > -7 & HypothalamicNeurons_UMAP$UMAP_1 < -3 & row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean) & 
  row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(TM_1_Clean)) | 
  row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest3) & HypothalamicNeurons_UMAP$UMAP_2 < -1 & HypothalamicNeurons_UMAP$UMAP_1 > -6.5   & HypothalamicNeurons_UMAP$UMAP_1 < -4.5 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), 
  row.names(TM_1_Clean)))
Preoptic_PNOC_NTS_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Preoptic_PNOC_NTS_ToClean) & 
  HypothalamicNeurons_UMAP$UMAP_2 < 1 & HypothalamicNeurons_UMAP$UMAP_2 > -4  & HypothalamicNeurons_UMAP$UMAP_1 > 0 & HypothalamicNeurons_UMAP$UMAP_1 < 5  & 
  row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean) | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest3) & 
  HypothalamicNeurons_UMAP$UMAP_2 < -1 & HypothalamicNeurons_UMAP$UMAP_2 > -3  & HypothalamicNeurons_UMAP$UMAP_1 > 1.8 & HypothalamicNeurons_UMAP$UMAP_1 < 2 &  row.names(HypothalamicNeurons_UMAP) %notin% row.names(TM_1_Clean) )
SIX3_SST_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(SIX3_SST_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 < -2.4    & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(Preoptic_PNOC_NTS_Clean), row.names(TM_1_Clean)))
SIX3_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(SIX3_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 < 0.5 & 
  HypothalamicNeurons_UMAP$UMAP_2 > -4 & HypothalamicNeurons_UMAP$UMAP_1 > 2 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(NKX21TRH_Clean), 
  row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(TM_1_Clean)) | row.names(HypothalamicNeurons_UMAP) %in%   
  colnames(HypothalamicNeurons_Rest3) & HypothalamicNeurons_UMAP$UMAP_2 < -1.8 & HypothalamicNeurons_UMAP$UMAP_1 > 4  & HypothalamicNeurons_UMAP$UMAP_2 > -3 &   row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(NKX21TRH_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean),
  row.names(TM_1_Clean)))
PENK_PNOC_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PENK_PNOC) & HypothalamicNeurons_UMAP$UMAP_2 < 0.5 & 
  HypothalamicNeurons_UMAP$UMAP_2 > -1 & HypothalamicNeurons_UMAP$UMAP_1 < -7 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), 
  row.names(VMH2_EBF4_Clean)) | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest3) & HypothalamicNeurons_UMAP$UMAP_2 < 0.2 &
  HypothalamicNeurons_UMAP$UMAP_2 > -1 & HypothalamicNeurons_UMAP$UMAP_1 < -7.5 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), 
  row.names(VMH2_EBF4_Clean)))
PVH_4_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PVH_4_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 > -2 & 
  HypothalamicNeurons_UMAP$UMAP_2 < 1.5 & HypothalamicNeurons_UMAP$UMAP_1 > -5 &  HypothalamicNeurons_UMAP$UMAP_1 < 1 & row.names(HypothalamicNeurons_UMAP)   
  %notin% c(row.names(TM_1_Clean), row.names(NHLH2_Clean), row.names(CCK_RELN_Clean)) | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest3) &  HypothalamicNeurons_UMAP$UMAP_2 > -1.5 & HypothalamicNeurons_UMAP$UMAP_2 < 1 & HypothalamicNeurons_UMAP$UMAP_1 > -4 &
  HypothalamicNeurons_UMAP$UMAP_1 < -1 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), row.names(NHLH2_Clean), 
  row.names(CCK_RELN_Clean)))
MMPost_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(MMPost_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 > -0.5 & 
  HypothalamicNeurons_UMAP$UMAP_1 > -8 &  HypothalamicNeurons_UMAP$UMAP_1 < -1 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), 
  row.names(NHLH2_Clean), row.names(PVH_4_Clean)) | row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest3) &  
  HypothalamicNeurons_UMAP$UMAP_2 > 0.5 & HypothalamicNeurons_UMAP$UMAP_1 > -7    &  HypothalamicNeurons_UMAP$UMAP_1 < -3 & 
  row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(TM_1_Clean), row.names(NHLH2_Clean), row.names(PVH_4_Clean)))





HypothalamicNeurons_Rest4 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean)), invert=T)


set.dim = 40
set.res = c(1)
set.kparam = c(5)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest4)

MoveToPVH3 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest4) & HypothalamicNeurons_UMAP$UMAP_1 < -2 & HypothalamicNeurons_UMAP$UMAP_2 < 1.8 &  HypothalamicNeurons_UMAP$UMAP_1 < -2.5)


HypothalamicNeurons_Rest5 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean), row.names(MoveToPVH3)), invert=T)

set.dim = 40
set.res = c(1)
set.kparam = c(6)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest5)



HypothalamicNeurons_Rest5 <- FindNeighbors(HypothalamicNeurons_Rest5, k.param=5, dims=1:40)
HypothalamicNeurons_Rest5 <- FindClusters(HypothalamicNeurons_Rest5, resolution =1)
RELN_NRN1_ToClean= subset(HypothalamicNeurons_Rest5, idents = c(6))
TBX3_ToClean= subset(HypothalamicNeurons_Rest5, idents = c(8))
SIX3_V2_ToClean= subset(HypothalamicNeurons_Rest5, idents = c(3,4,7))
NPAS4_ToClean= subset(HypothalamicNeurons_Rest5, idents = c(12))


CheckInput = colnames(NPAS4_ToClean) #Check Populations
CheckUMAP(HypothalamicNeurons_Rest5)






RELN_NRN1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(RELN_NRN1_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 > -3 & 
  HypothalamicNeurons_UMAP$UMAP_2 < 0 & HypothalamicNeurons_UMAP$UMAP_1 > -4 &  HypothalamicNeurons_UMAP$UMAP_1 < 3 | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest5) & HypothalamicNeurons_UMAP$UMAP_2 > -2 & 
  HypothalamicNeurons_UMAP$UMAP_2 < -1 & HypothalamicNeurons_UMAP$UMAP_1 > -2.5 &  HypothalamicNeurons_UMAP$UMAP_1 < 1)


TBX3_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(TBX3_ToClean) & HypothalamicNeurons_UMAP$UMAP_2 > 0 & 
  HypothalamicNeurons_UMAP$UMAP_2 < 3 & HypothalamicNeurons_UMAP$UMAP_1 > 5  | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest5) & HypothalamicNeurons_UMAP$UMAP_2 > 1.8 & 
  HypothalamicNeurons_UMAP$UMAP_2 < 3 & HypothalamicNeurons_UMAP$UMAP_1 > 4.8 | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest5) & HypothalamicNeurons_UMAP$UMAP_2 > .8 & 
  HypothalamicNeurons_UMAP$UMAP_2 < 3 & HypothalamicNeurons_UMAP$UMAP_1 > 6.4 | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest5) & HypothalamicNeurons_UMAP$UMAP_2 > 0 & 
  HypothalamicNeurons_UMAP$UMAP_2 < 3 & HypothalamicNeurons_UMAP$UMAP_1 > 8.4)


SIX3_V2_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(SIX3_V2_ToClean) & 
  HypothalamicNeurons_UMAP$UMAP_2 < -3  | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest5) & HypothalamicNeurons_UMAP$UMAP_2 < -3.5 &  HypothalamicNeurons_UMAP$UMAP_1 > -1 &  HypothalamicNeurons_UMAP$UMAP_1 < 7)

NPAS4_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(NPAS4_ToClean) & 
  HypothalamicNeurons_UMAP$UMAP_2 > 1.2 &  HypothalamicNeurons_UMAP$UMAP_1 < 4 & HypothalamicNeurons_UMAP$UMAP_1 > 0 | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest5) & HypothalamicNeurons_UMAP$UMAP_2 > 1.3 &  HypothalamicNeurons_UMAP$UMAP_1 < 2 & HypothalamicNeurons_UMAP$UMAP_1 > 0.5)



HypothalamicNeurons_Rest6 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean), row.names(MoveToPVH3), row.names(SIX3_V2_Clean), row.names(TBX3_Clean), row.names(RELN_NRN1_Clean), row.names(MoveToPVH3), row.names(NPAS4_Clean)), invert=T)


set.dim = 40
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest6)



HypothalamicNeurons_Rest6 <- FindNeighbors(HypothalamicNeurons_Rest6, k.param=50, dims=1:40)
HypothalamicNeurons_Rest6 <- FindClusters(HypothalamicNeurons_Rest6, resolution =1)
ClustForZI = subset(HypothalamicNeurons_Rest6, idents = c(3))
ClustForRGS4_PDYN_ARC = subset(HypothalamicNeurons_Rest6, idents = c(0,1,4))

set.dim = 40
set.res = c(2)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(ClustForRGS4_PDYN_ARC)



ZI_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ClustForZI) & 
  HypothalamicNeurons_UMAP$UMAP_2 > -2 & HypothalamicNeurons_UMAP$UMAP_2 < 0.5 & HypothalamicNeurons_UMAP$UMAP_1 > 4  | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest6) &   HypothalamicNeurons_UMAP$UMAP_2 > -2 & HypothalamicNeurons_UMAP$UMAP_2 < -1.3 & HypothalamicNeurons_UMAP$UMAP_1 > 6)


ClustForRGS4_PDYN_ARC <- FindNeighbors(ClustForRGS4_PDYN_ARC, k.param=3, dims=1:40)
ClustForRGS4_PDYN_ARC <- FindClusters(ClustForRGS4_PDYN_ARC, resolution =3)
ClustForDMH = subset(ClustForRGS4_PDYN_ARC, idents = c(0, 2, 5, 8, 9, 10, 13, 17, 30, 4, 15, 1, 3, 11, 7, 31, 29, 14, 23))


set.dim = 40
set.res = c(3)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(ClustForDMH)



ClustForDMH <- FindNeighbors(ClustForDMH, k.param=3, dims=1:40)
ClustForDMH <- FindClusters(ClustForDMH, resolution =3)
ClustForDMH_V2 = subset(ClustForDMH, idents = c(10, 17, 2, 5, 16, 23, 22, 1, 14, 6, 12))


set.dim = 40
set.res = c(1)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(ClustForDMH_V2)



ClustForDMH_V2 <- FindNeighbors(ClustForDMH_V2, k.param=2, dims=1:40)
ClustForDMH_V2 <- FindClusters(ClustForDMH_V2, resolution =1)
ClustForDMH_V3 = subset(ClustForDMH_V2, idents = c(7, 19, 3, 4, 9, 16, 20, 24, 1, 11, 13, 6, 10, 0))


set.dim = 40
set.res = c(0.8)
set.kparam = c(50)
ClusterFunc_Neuro_RNA(ClustForDMH_V3)

ClustForDMH_V3 <- FindNeighbors(ClustForDMH_V3, k.param=50, dims=1:40)
ClustForDMH_V3 <- FindClusters(ClustForDMH_V3, resolution =0.8)
ClustForDMH_V4 = subset(ClustForDMH_V3, idents = c(1))

PullDMH = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest6) & Fetal.combined_Data$PPP1R17 > 0)


DMH_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ClustForDMH_V4) & 
  HypothalamicNeurons_UMAP$UMAP_2 > 0 & HypothalamicNeurons_UMAP$UMAP_2 < 2.5 & HypothalamicNeurons_UMAP$UMAP_1 > 3 & HypothalamicNeurons_UMAP$UMAP_1 < 9 | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(ClustForDMH_V2) &   HypothalamicNeurons_UMAP$UMAP_2 > 0 & HypothalamicNeurons_UMAP$UMAP_2 < 2.5 & HypothalamicNeurons_UMAP$UMAP_1 > 3 & HypothalamicNeurons_UMAP$UMAP_1 < 9 | row.names(HypothalamicNeurons_UMAP) %in% row.names(PullDMH) & 
  HypothalamicNeurons_UMAP$UMAP_2 > 0 & HypothalamicNeurons_UMAP$UMAP_2 < 2.5 & HypothalamicNeurons_UMAP$UMAP_1 > 3 & HypothalamicNeurons_UMAP$UMAP_1 < 9)


CheckInput = row.names(DMH_Clean) #Check Populations
CheckUMAP(HypothalamicNeurons)


DMH_Clean_Seu = subset(HypothalamicNeurons_Rest1, cells = row.names(DMH_Clean))
set.dim = 40
set.res = c(0.8)
set.kparam = c(50)
ClusterFunc_Neuro_RNA(DMH_Clean_Seu)



HypothalamicNeurons_Rest7 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean), row.names(MoveToPVH3), row.names(SIX3_V2_Clean), row.names(TBX3_Clean), row.names(RELN_NRN1_Clean), row.names(MoveToPVH3), row.names(NPAS4_Clean), row.names(DMH_Clean), row.names(ZI_Clean)), invert=T)


set.dim = 40
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest7)
HypothalamicNeurons_Rest7 <- FindNeighbors(HypothalamicNeurons_Rest7, k.param=20, dims=1:40)
HypothalamicNeurons_Rest7 <- FindClusters(HypothalamicNeurons_Rest7, resolution =1)
Arc_Precursor = subset(HypothalamicNeurons_Rest7, idents = c(2))


Arc_Precursor_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc_Precursor) & 
 HypothalamicNeurons_UMAP$UMAP_1 > 2  | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest7) &  HypothalamicNeurons_UMAP$UMAP_2 > -0.7 & HypothalamicNeurons_UMAP$UMAP_2 < 1 & HypothalamicNeurons_UMAP$UMAP_1 > 4  )


HypothalamicNeurons_Rest8 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean), row.names(MoveToPVH3), row.names(SIX3_V2_Clean), row.names(TBX3_Clean), row.names(RELN_NRN1_Clean), row.names(MoveToPVH3), row.names(NPAS4_Clean), row.names(DMH_Clean), row.names(ZI_Clean), row.names(Arc_Precursor_Clean)), invert=T)


set.dim = 40
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest8)


ExtraArc = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest8) & Fetal.combined_Data$GHRH > 1 | row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest8) & Fetal.combined_Data$PCSK1 > 0 | row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest8) & Fetal.combined_Data$`NKX2-1` > 0 | row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest8) & Fetal.combined_Data$TBX3 > 0 | row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest8) & Fetal.combined_Data$POMC > 1)

ExtraArc_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(ExtraArc) & HypothalamicNeurons_UMAP$UMAP_2 > -1) 

ExtraONECUT2 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(HypothalamicNeurons_Rest8) &  HypothalamicNeurons_UMAP$UMAP_2 < -1.2 & HypothalamicNeurons_UMAP$UMAP_1 < -1 ) 


HypothalamicNeurons_Rest9 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean), row.names(MoveToPVH3), row.names(SIX3_V2_Clean), row.names(TBX3_Clean), row.names(RELN_NRN1_Clean), row.names(MoveToPVH3), row.names(NPAS4_Clean), row.names(DMH_Clean), row.names(ZI_Clean), row.names(Arc_Precursor_Clean), row.names(ExtraArc_Clean), row.names(ExtraONECUT2)), invert=T)



ExtraNRN1 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest9) & Fetal.combined_Data$NRN1 > 0)

ExtraNRN1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(ExtraNRN1) &  HypothalamicNeurons_UMAP$UMAP_1 < 1) 

ExtraVMH = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest9) & Fetal.combined_Data$GABRA5 > 0 | row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest9) & Fetal.combined_Data$NPTX2 > 0)

ExtraVMH_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(ExtraVMH) & row.names(HypothalamicNeurons_UMAP) %notin% row.names(ExtraNRN1_Clean) & HypothalamicNeurons_UMAP$UMAP_2 > -1 & HypothalamicNeurons_UMAP$UMAP_1 < 1) 



ExtraSIX6_SST = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest9) & Fetal.combined_Data$SIX6 > 0| row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest9) & Fetal.combined_Data$SST > 0)

ExtraSIX6_SST_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(ExtraSIX6_SST) & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(ExtraNRN1_Clean), row.names(ExtraVMH_Clean)) & HypothalamicNeurons_UMAP$UMAP_2 < -1) 



HypothalamicNeurons_Rest10 = subset(HypothalamicNeurons_Rest1, cells = c(row.names(NKX21TRH_Clean),  row.names(NHLH2_Clean), row.names(Preoptic_PNOC_NTS_Clean), row.names(SIX3_SST_Clean), row.names(SIX3_Clean), row.names(CCK_RELN_Clean), row.names(VMH2_EBF4_Clean), row.names(PENK_PNOC_Clean), row.names(PVH_2_Clean), row.names(TM_1_Clean), row.names(UKN_PostMM_Clean), row.names(PVH3_Clean),  row.names(UKN_CRABP1), row.names(MN_1_Clean), row.names(UKN_Top_Clean),row.names(Arc_1_Clean), row.names(SCN_1_Clean), row.names(UKN_TAC1), row.names(LH_1), row.names(PVH_4_Clean), row.names(MMPost_Clean), row.names(VMH_1_Clean), row.names(PVH_1_Clean), row.names(MoveToPVH3), row.names(SIX3_V2_Clean), row.names(TBX3_Clean), row.names(RELN_NRN1_Clean), row.names(MoveToPVH3), row.names(NPAS4_Clean), row.names(DMH_Clean), row.names(ZI_Clean), row.names(Arc_Precursor_Clean), row.names(ExtraArc_Clean), row.names(ExtraONECUT2), row.names(ExtraSIX6_SST_Clean), row.names(ExtraVMH_Clean), row.names(ExtraNRN1_Clean)), invert=T)


set.dim = c(30)
set.res = c(1)
set.kparam = c(10,20)
ClusterFunc_Neuro_RNA(HypothalamicNeurons_Rest10)
HypothalamicNeurons_Rest10 <- FindNeighbors(HypothalamicNeurons_Rest10, k.param=10, dims=1:30)
HypothalamicNeurons_Rest10 <- FindClusters(HypothalamicNeurons_Rest10, resolution =1)
Ukn_A = subset(HypothalamicNeurons_Rest10, idents = c(4))
Ukn_B = subset(HypothalamicNeurons_Rest10, idents = c(3,5,2))
Ukn_C = subset(HypothalamicNeurons_Rest10, idents = c(0))

CheckInput = colnames(Ukn_C) #Check Populations
CheckUMAP(HypothalamicNeurons_Rest10)


ExtraSIX6_SST = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(Ukn_A) & Fetal.combined_Data$SIX6 > 0| row.names(Fetal.combined_Data) %in% colnames(HypothalamicNeurons_Rest10) & Fetal.combined_Data$SST > 0)

Ukn_A_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Ukn_A) & 
  HypothalamicNeurons_UMAP$UMAP_2 > 0 &  HypothalamicNeurons_UMAP$UMAP_1 > 2  | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest10) &   HypothalamicNeurons_UMAP$UMAP_2 > 0 &  HypothalamicNeurons_UMAP$UMAP_1 > 2.2)

Ukn_B_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Ukn_B) & 
  HypothalamicNeurons_UMAP$UMAP_2 > -1.8 &  HypothalamicNeurons_UMAP$UMAP_1 < 3  & row.names(HypothalamicNeurons_UMAP)  %notin% row.names(Ukn_A_Clean) | row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest10) &   HypothalamicNeurons_UMAP$UMAP_2 > -1.5 &  HypothalamicNeurons_UMAP$UMAP_1 < 2.2  & row.names(HypothalamicNeurons_UMAP) %notin% row.names(Ukn_A_Clean) )


Ukn_C_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Ukn_C) & 
  HypothalamicNeurons_UMAP$UMAP_2 < -1.5 &  HypothalamicNeurons_UMAP$UMAP_1 < 2.8  & row.names(HypothalamicNeurons_UMAP)  %notin% c(row.names(Ukn_A_Clean), row.names(Ukn_B_Clean))| row.names(HypothalamicNeurons_UMAP) %in% 
  colnames(HypothalamicNeurons_Rest10) & HypothalamicNeurons_UMAP$UMAP_2 < -1.5 &  HypothalamicNeurons_UMAP$UMAP_1 < 2.8   & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(Ukn_A_Clean), row.names(Ukn_B_Clean)))


Ukn_D_Clean = subset(HypothalamicNeurons_Rest10, cells = c(row.names(Ukn_A_Clean), row.names(Ukn_B_Clean), row.names(Ukn_C_Clean)), invert=T)



ExtraPNOC = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% row.names(Ukn_A_Clean) & Fetal.combined_Data$PNOC > 0)

Neur40 = subset(HypothalamicNeurons, idents = "Neuron_40 [PO]")

CheckInput = row.names(ExtraPNOC) #Check Populations
#CheckInput = colnames(Neur40) #Check Populations
CheckUMAP(HypothalamicNeurons_Rest1)


UKN_Top_Clean_Seu = subset(HypothalamicNeurons_Rest1, cells = row.names(UKN_Top_Clean))
set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(UKN_Top_Clean_Seu)

UKN_Top_Clean_Seu <- FindNeighbors(UKN_Top_Clean_Seu, k.param=2, dims=1:40)
UKN_Top_Clean_Seu <- FindClusters(UKN_Top_Clean_Seu, resolution =1)
Top_PVH = subset(UKN_Top_Clean_Seu, idents = c(5, 9,  26,  35, 8, 31, 21))

set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(Top_PVH)

Top_PVH <- FindNeighbors(Top_PVH, k.param=2, dims=1:40)
Top_PVH <- FindClusters(Top_PVH, resolution =1)
Top_PVH_V2 = subset(Top_PVH, idents = c(2,7), invert=T)

set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(Top_PVH_V2)

Top_PVH_V2 <- FindNeighbors(Top_PVH_V2, k.param=2, dims=1:40)
Top_PVH_V2 <- FindClusters(Top_PVH_V2, resolution =1)
Top_PVH_V3 = subset(Top_PVH_V2, idents = c(3), invert=T)

set.dim = c(40)
set.res = c(5)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(Top_PVH_V3)


Top_PVH_V3 <- FindNeighbors(Top_PVH_V3, k.param=2, dims=1:40)
Top_PVH_V3 <- FindClusters(Top_PVH_V3, resolution =5)
Top_PVH_V4 = subset(Top_PVH_V3, idents = c(1,5,7,9,11), invert=T)




UKN_Top_Cleaned = subset(UKN_Top_Clean_Seu, cells = colnames(Top_PVH_V4), invert=T)

set.dim = c(40)
set.res = c(1)
set.kparam = c(5)
ClusterFunc_Neuro_RNA(UKN_Top_Cleaned)

ExtraVMH = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(UKN_Top_Cleaned) & Fetal.combined_Data$NPTX2 > 0| row.names(Fetal.combined_Data) %in% colnames(UKN_Top_Cleaned) & Fetal.combined_Data$GABRA5 > 0 | row.names(Fetal.combined_Data) %in% colnames(UKN_Top_Cleaned) & Fetal.combined_Data$ADCYAP1 > 0)

CheckInput = row.names(ExtraVMH)
CheckUMAP(UKN_Top_Cleaned)

DefaultAssay(HypothalamicNeurons) = "integrated"
SIX3Cells = subset(HypothalamicNeurons, cells = c(row.names(SIX3_V2_Clean), row.names(SIX3_Clean), row.names(SIX3_SST_Clean), row.names(ExtraSIX6_SST_Clean)))
set.dim = c(40)
set.res = c(1)
set.kparam = c(50)
#ClusterFunc_Neuro_RNA(SIX3Cells)
SIX3Cells <- FindNeighbors(SIX3Cells, k.param=50, dims=1:40)
SIX3Cells <- FindClusters(SIX3Cells, resolution =1)
SIX3_MEIS2_ = subset(SIX3Cells, idents = c(0))
SIX3_SST_OTP = subset(SIX3Cells, idents = c(2))
SIX3CellsRest1 = subset(SIX3Cells, idents = c(0,2), invert=T)


set.dim = c(40)
set.res = c(1)
set.kparam = c(5,10,20)
ClusterFunc_Neuro_RNA(SIX3_SST_OTP)
SIX3_SST_OTP <- FindNeighbors(SIX3_SST_OTP, k.param=20, dims=1:40)
SIX3_SST_OTP <- FindClusters(SIX3_SST_OTP, resolution =1)
ARC_Cells = subset(SIX3_SST_OTP, idents = c(0,2))

set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ARC_Cells)

MEIS2_Cells = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ARC_Cells) & Fetal.combined_Data$MEIS2 > 0)
ARC_Cells_V2 = subset(ARC_Cells, cells = row.names(MEIS2_Cells), invert=T)

ARC_Cells_V3 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ARC_Cells_V2) & 
  HypothalamicNeurons_UMAP$UMAP_2 < -3.8 &  HypothalamicNeurons_UMAP$UMAP_1 > 2.5)

ZI_Cells_V2 = subset(SIX3_SST_OTP, cells = row.names(ARC_Cells_V3), invert=T)
ZI_Cells_V3 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ZI_Cells_V2) & HypothalamicNeurons_UMAP$UMAP_1 < 4 | row.names(HypothalamicNeurons_UMAP) %in% colnames(ZI_Cells_V2) & HypothalamicNeurons_UMAP$UMAP_1 > 4 & HypothalamicNeurons_UMAP$UMAP_2 < -3)

Ukn_Cells =  subset(SIX3_SST_OTP, cells = c(row.names(ARC_Cells_V3), row.names(ZI_Cells_V3)), invert=T)

Neur47 = subset(HypothalamicNeurons, idents = c("Neuron_47"))


CheckInput = colnames(Neur47)
#CheckInput = colnames(Ukn_Cells)
CheckUMAP(HypothalamicNeurons)


SIX3CellsRest2 = subset(HypothalamicNeurons, cells = c(colnames(SIX3CellsRest1), colnames(Ukn_D_Clean)))
set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(SIX3CellsRest2)
SIX3CellsRest2 <- FindNeighbors(SIX3CellsRest2, k.param=3, dims=1:40)
SIX3CellsRest2 <- FindClusters(SIX3CellsRest2, resolution =3)

ClustForLHX6 = subset(SIX3CellsRest2, idents = c(3, 5, 6, 10, 19, 23, 28, 11, 16, 17, 18, 21, 2, 12, 4, 27, 1, 22, 25, 8))
set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(ClustForLHX6)

ClustForLHX6 <- FindNeighbors(ClustForLHX6, k.param=2, dims=1:40)
ClustForLHX6 <- FindClusters(ClustForLHX6, resolution =1)
ClustForLHX6_V2 = subset(ClustForLHX6, idents = c(5, 6, 8, 9, 10, 13, 14, 15, 19, 26, 30, 32, 34, 39, 1, 2, 29, 47, 0, 12, 22, 25, 38, 4, 16, 11))

set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ClustForLHX6_V2)

ClustForLHX6_V2 <- FindNeighbors(ClustForLHX6_V2, k.param=2, dims=1:40)
ClustForLHX6_V2 <- FindClusters(ClustForLHX6_V2, resolution =1)
ClustForLHX6_V3 = subset(ClustForLHX6_V2, idents = c(17, 1, 9, 20, 22, 26, 4, 14, 21, 15, 28, 13, 19, 25,2,3, 0))

set.dim = c(40)
set.res = c(2)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(ClustForLHX6_V3)


ClustForLHX6_V3 <- FindNeighbors(ClustForLHX6_V3, k.param=2, dims=1:40)
ClustForLHX6_V3 <- FindClusters(ClustForLHX6_V3, resolution =2)
ClustForLHX6_V4 = subset(ClustForLHX6_V3, idents = c(0, 5, 6, 11, 15, 16, 21, 12, 19, 3, 9, 8))

set.dim = c(40)
set.res = c(2)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(ClustForLHX6_V4)

ClustForLHX6_V4 <- FindNeighbors(ClustForLHX6_V4, k.param=2, dims=1:40)
ClustForLHX6_V4 <- FindClusters(ClustForLHX6_V4, resolution =2)
ClustForLHX6_V5 = subset(ClustForLHX6_V4, idents = c(2, 13, 14), invert=T)


ExtraLHX6 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ClustForLHX6_V3) & Fetal.combined_Data$LHX6 > 0)
ClustForLHX6_V6 = subset(HypothalamicNeurons, cells = unique(c(row.names(ExtraLHX6), colnames(ClustForLHX6_V4))))
set.dim = c(40)
set.res = c(0.05)
set.kparam = c(200)
#ClusterFunc_Neuro_RNA(ClustForLHX6_V6)

ClustForLHX6_V7 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ClustForLHX6_V6) & HypothalamicNeurons_UMAP$UMAP_2 < -1.5 ) 

ClustForLHX6_V7_Seu = subset(HypothalamicNeurons, cells = row.names(ClustForLHX6_V7))
set.dim = c(40)
set.res = c(0.05)
set.kparam = c(200)
#ClusterFunc_Neuro_RNA(ClustForLHX6_V7_Seu)

SIX3CellsRest3 = subset(SIX3CellsRest2, cells = row.names(ClustForLHX6_V7), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(10,5)
#ClusterFunc_Neuro_RNA(SIX3CellsRest3)

SIX3CellsRest3 <- FindNeighbors(SIX3CellsRest3, k.param=5, dims=1:40)
SIX3CellsRest3 <- FindClusters(SIX3CellsRest3, resolution =1)
SIX3_SST_OTPpt2 = subset(SIX3CellsRest3, idents = c(0))
PNOC_Preoptic = subset(SIX3CellsRest3, idents = c(0), invert=T)

set.dim = c(40)
set.res = c(0.05)
set.kparam = c(100)
ClusterFunc_Neuro_RNA(Neur42)



Ukn_C_Clean_Seu = subset(HypothalamicNeurons, cells = row.names(Ukn_C_Clean))
set.dim = c(10,20,30)
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(Ukn_C_Clean_Seu)
Ukn_C_Clean_Seu <- FindNeighbors(Ukn_C_Clean_Seu, k.param=10, dims=1:30)
Ukn_C_Clean_Seu <- FindClusters(Ukn_C_Clean_Seu, resolution =1)
ForONECUT2 = subset(Ukn_C_Clean_Seu, idents = c(2))
ClustPNOC_LHX6 = subset(Ukn_C_Clean_Seu, idents = c(0))
SIX3_Extra = subset(Ukn_C_Clean_Seu, idents = c(1))

set.dim = c(40)
set.res = c(1)
set.kparam = c(5,2)
#ClusterFunc_Neuro_RNA(ClustPNOC_LHX6)
ClustPNOC_LHX6 <- FindNeighbors(ClustPNOC_LHX6, k.param=2, dims=1:40)
ClustPNOC_LHX6 <- FindClusters(ClustPNOC_LHX6, resolution =1)
AddToLHX6 = subset(ClustPNOC_LHX6, idents = c(0))
AddToPNOC = subset(ClustPNOC_LHX6, idents = c(0), invert=T)


CheckZICs = subset(HypothalamicNeurons, cells = c(row.names(RELN_NRN1_Clean), row.names(CCK_RELN_Clean), row.names(PVH_4_Clean), row.names(ExtraNRN1_Clean)))
set.dim = c(40)
set.res = c(1)
set.kparam = c(50)
#ClusterFunc_Neuro_RNA(CheckZICs)
CheckZICs <- FindNeighbors(CheckZICs, k.param=50, dims=1:40)
CheckZICs <- FindClusters(CheckZICs, resolution =1)
ClustForMN = subset(CheckZICs, idents = c(0))

set.dim = c(10,20,30,40)
set.res = c(1)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(ClustForMN)

ClustForMN <- FindNeighbors(ClustForMN, k.param=100, dims=1:40)
ClustForMN <- FindClusters(ClustForMN, resolution =1)
AddToMN = subset(ClustForMN, idents = c(1))

CheckZICs_V2 = subset(CheckZICs, cells = colnames(AddToMN), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(30)
#ClusterFunc_Neuro_RNA(CheckZICs_V2)
CheckZICs_V2 <- FindNeighbors(CheckZICs_V2, k.param=30, dims=1:40)
CheckZICs_V2 <- FindClusters(CheckZICs_V2, resolution =1)
AddToPVH3_V2 = subset(CheckZICs_V2, idents = c(3))

ExtraSIM1 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(CheckZICs_V2) & Fetal.combined_Data$SIM1 > 0)
AddToPVH3_V2pt2 = subset(CheckZICs_V2, cells = unique(c(row.names(ExtraSIM1), colnames(AddToPVH3_V2))))
CheckZICs_V3 = subset(CheckZICs_V2, cells = unique(c(row.names(ExtraSIM1), colnames(AddToPVH3_V2))), invert=T)

set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(CheckZICs_V3)
#ClusterFunc_Neuro_RNA(AddToPVH3_V2pt2)


DefaultAssay(HypothalamicNeurons) = "integrated"
ReclusterPVH = subset(HypothalamicNeurons, cells = c(row.names(NHLH2_Clean),  row.names(PVH_1_Clean)))
set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ReclusterPVH)
ReclusterPVH <- FindNeighbors(ReclusterPVH, k.param=3, dims=1:40)
ReclusterPVH <- FindClusters(ReclusterPVH, resolution =1)
Arc_4 = subset(ReclusterPVH, idents = c(2))
Arc_5 = subset(ReclusterPVH, idents = c(5))

set.dim = c(40)
set.res = c(1)
set.kparam = c(4)
ClusterFunc_Neuro_RNA(Arc_5)

Arc_5_V2 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc_5) & HypothalamicNeurons_UMAP$UMAP_1 > -4.5 & HypothalamicNeurons_UMAP$UMAP_2 < -2)
Arc_5_V2_Seu = subset(Arc_5, cells = row.names(Arc_5_V2))
set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(Arc_5_V2_Seu)

PVHCells = subset(Arc_5, cells = row.names(Arc_5_V2), invert=T)

Arc_4 <- FindNeighbors(Arc_4, k.param=3, dims=1:40)
Arc_4 <- FindClusters(Arc_4, resolution =1)
Arc_4_V2 = subset(Arc_4, idents = c(0,1,3,4,5,7))

Arc_4_V3 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc_4_V2) & 
  HypothalamicNeurons_UMAP$UMAP_2 < -4 & HypothalamicNeurons_UMAP$UMAP_1 > -6 )

set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(Arc_4_V2)

ReclusterPVH_Rest = subset(ReclusterPVH, cells = c(colnames(Arc_5), row.names(Arc_4_V3)), invert=T)
NHLH2_CleanV2 = subset(NHLH2_Clean, row.names(NHLH2_Clean) %in% colnames(ReclusterPVH_Rest))
PVH_1_CleanV2 = subset(PVH_1_Clean, row.names(PVH_1_Clean) %in% colnames(ReclusterPVH_Rest))


ReclusterMN = subset(HypothalamicNeurons, cells =row.names(UKN_PostMM_Clean))
set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ReclusterMN)
ReclusterMN <- FindNeighbors(ReclusterMN, k.param=3, dims=1:40)
ReclusterMN <- FindClusters(ReclusterMN, resolution =2)
AddToPVH3 = subset(ReclusterMN, idents = c(6,13))
KeepMN = subset(ReclusterMN, idents = c(6,13), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(KeepMN)


KeepMN <- FindNeighbors(KeepMN, k.param=2, dims=1:40)
KeepMN <- FindClusters(KeepMN, resolution =1)
AddToPVH3_Pt2 = subset(KeepMN, idents = c(14, 39, 41, 2, 4, 7, 8, 0, 1, 6, 23, 3))
set.dim = c(40)
set.res = c(3)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(AddToPVH3_Pt2)

AddToPVH3_Pt2 <- FindNeighbors(AddToPVH3_Pt2, k.param=2, dims=1:40)
AddToPVH3_Pt2 <- FindClusters(AddToPVH3_Pt2, resolution =3)
AddToPVH3_Pt3 = subset(AddToPVH3_Pt2, idents = c(3, 7, 8, 12, 17, 22, 24, 25, 2, 9, 10, 0, 1, 6, 19))

set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(AddToPVH3_Pt3)


AddToPVH3_Pt3 <- FindNeighbors(AddToPVH3_Pt3, k.param=3, dims=1:40)
AddToPVH3_Pt3 <- FindClusters(AddToPVH3_Pt3, resolution =3)
AddToPVH3_Pt4 = subset(AddToPVH3_Pt3, idents = c(6, 12, 15, 16, 5, 10, 14, 2, 3, 1, 0))
set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(AddToPVH3_Pt4)

AddToPVH3_Clean = subset(HypothalamicNeurons, cells = c(colnames(AddToPVH3_Pt4), colnames(AddToPVH3)))
set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(AddToPVH3_Clean)
MN_Clean = subset(ReclusterMN, cells = c(colnames(AddToPVH3_Pt4), colnames(AddToPVH3)), invert=T)



Check_PVH3 = subset(HypothalamicNeurons, cells = row.names(PVH3_Clean))
set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(Check_PVH3)

Check_PVH3 <- FindNeighbors(Check_PVH3, k.param=3, dims=1:40)
Check_PVH3 <- FindClusters(Check_PVH3, resolution =3)
SIM1Pos = subset(Check_PVH3, idents = c(1,2,5,7,10,12))
Check_PVH3_V2 = subset(Check_PVH3, idents = c(1,2,5,7,10,12), invert=T)
set.dim = c(40)
set.res = c(1,3)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(Check_PVH3_V2)
ExtraSIM1_pt2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(Check_PVH3_V2) & Fetal.combined_Data$SIM1 > 0)

PVH3_Clean_V2 = subset(Check_PVH3, cells = c(row.names(ExtraSIM1_pt2), colnames(SIM1Pos)))
MoveToMN3 = subset(Check_PVH3, cells = c(row.names(ExtraSIM1_pt2), colnames(SIM1Pos)), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(PVH3_Clean_V2)

DefaultAssay(HypothalamicNeurons) = "integrated"
PVH1V2 = subset(HypothalamicNeurons, cells = c(row.names(PVH_1_CleanV2))) #1079 cells

set.dim = c(40)
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(PVH1V2)

PVH1V2 <- FindNeighbors(PVH1V2, k.param=10, dims=1:40)
PVH1V2 <- FindClusters(PVH1V2, resolution =1)
OXT_MC4R = subset(PVH1V2, idents = c(3,8))
PVH_AVP = subset(PVH1V2, idents = c(6))
set.dim = c(40)
set.res = c(5)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(OXT_MC4R)

OXT_MC4R <- FindNeighbors(OXT_MC4R, k.param=2, dims=1:40)
OXT_MC4R <- FindClusters(OXT_MC4R, resolution =5)
OXT_MC4R_V2 = subset(OXT_MC4R, idents = c(1, 2, 4, 9, 10, 11, 15, 18, 22, 24, 27, 30, 5, 8, 13, 14, 7))

set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(OXT_MC4R_V2)


OXT_MC4R_V2 <- FindNeighbors(OXT_MC4R_V2, k.param=2, dims=1:40)
OXT_MC4R_V2 <- FindClusters(OXT_MC4R_V2, resolution =1)
OXT_MC4R_V3 = subset(OXT_MC4R_V2, idents = c(0,7,8,9,10,11,12,13))


OXT_MC4R_V3_pt2  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(OXT_MC4R_V2) & Fetal.combined_Data$MC4R > 0)

OXT_MC4R_V4 = subset(PVH1, cells = unique(c(row.names(OXT_MC4R_V3_pt2), colnames(OXT_MC4R_V3))))
set.dim = c(40)
set.res = c(0.05)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(OXT_MC4R_V4)

OXT_Only = subset(OXT_MC4R, cells = colnames(OXT_MC4R_V4), invert=T)
set.dim = c(40)
set.res = c(0.05)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(OXT_Only)


PVH1_Rest2 = subset(PVH1V2, cells = c(colnames(AVP), colnames(OXT_Only), colnames(OXT_MC4R_V4)), invert=T)

PVH_SST = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PVH1_Rest2) & HypothalamicNeurons_UMAP$UMAP_2 < -5.5)

PVH1_Rest3 = subset(PVH1V2, cells = c(colnames(AVP), colnames(OXT_Only), colnames(OXT_MC4R_V4), row.names(PVH_SST)), invert=T)

set.dim = c(40)
set.res = c(2)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(PVH1_Rest3)

PVH1_Rest3 <- FindNeighbors(PVH1_Rest3, k.param=3, dims=1:40)
PVH1_Rest3 <- FindClusters(PVH1_Rest3, resolution =2)
ClustForMC4R = subset(PVH1_Rest3, idents = c(1, 6, 8, 10, 12, 15, 19, 0, 7, 9, 13, 11))

set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ClustForMC4R)


ClustForMC4R <- FindNeighbors(ClustForMC4R, k.param=3, dims=1:40)
ClustForMC4R <- FindClusters(ClustForMC4R, resolution =3)
ClustForMC4R_V2 = subset(ClustForMC4R, idents = c(0, 2, 3, 7, 10, 15, 6, 9, 5, 13, 4))

set.dim = c(40)
set.res = c(5)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ClustForMC4R_V2)


ClustForMC4R_V2 <- FindNeighbors(ClustForMC4R_V2, k.param=3, dims=1:40)
ClustForMC4R_V2 <- FindClusters(ClustForMC4R_V2, resolution =5)
ClustForMC4R_V3 = subset(ClustForMC4R_V2, idents = c(0, 3, 11, 13, 14, 18, 19, 1, 9, 12, 21, 2, 6))

set.dim = c(40)
set.res = c(5)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ClustForMC4R_V3)


ClustForMC4R_V3 <- FindNeighbors(ClustForMC4R_V3, k.param=3, dims=1:40)
ClustForMC4R_V3 <- FindClusters(ClustForMC4R_V3, resolution =5)
ClustForMC4R_V4 = subset(ClustForMC4R_V3, idents = c(2, 8, 13, 15, 18, 19), invert=T)


set.dim = c(40)
set.res = c(5)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(ClustForMC4R_V4)


ClustForMC4R_V4 <- FindNeighbors(ClustForMC4R_V4, k.param=2, dims=1:40)
ClustForMC4R_V4 <- FindClusters(ClustForMC4R_V4, resolution =5)
ClustForMC4R_V5 = subset(ClustForMC4R_V4, idents = c(4,6,12,13,16,20,9,8,7))

MC4R_Extra  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(PVH1_Rest3) & Fetal.combined_Data$MC4R > 0)

ClustForMC4R_V6 = subset(PVH1_Rest3, cells = unique(c(row.names(MC4R_Extra), colnames(ClustForMC4R_V5))))

set.dim = c(40)
set.res = c(0.5)
set.kparam = c(100)
ClusterFunc_Neuro_RNA(ClustForMC4R_V6)



PVH1_Rest4 = subset(PVH1V2, cells = c(colnames(AVP), colnames(OXT_Only), colnames(OXT_MC4R_V4), row.names(PVH_SST), colnames(ClustForMC4R_V6)), invert=T)


set.dim = c(40)
set.res = c(2)
set.kparam = c(10)
ClusterFunc_Neuro_RNA(PVH1_Rest4)


PVH1_Rest4 <- FindNeighbors(PVH1_Rest4, k.param=10, dims=1:40)
PVH1_Rest4 <- FindClusters(PVH1_Rest4, resolution =2)
ClustForCRH = subset(PVH1_Rest4, idents = c(9))

CRH_Extra  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(PVH1_Rest4) & Fetal.combined_Data$CRH > 0)
CRH_Extra_V2 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(CRH_Extra) & HypothalamicNeurons_UMAP$UMAP_2 < -2.5 & HypothalamicNeurons_UMAP$UMAP_2 > -4.5 &HypothalamicNeurons_UMAP$UMAP_1 < -7.5)


ClustForCRH_V2 = subset(PVH1_Rest4, cells = unique(c(row.names(CRH_Extra_V2), colnames(ClustForCRH))))
set.dim = c(40)
set.res = c(0.5)
set.kparam = c(100)
ClusterFunc_Neuro_RNA(ClustForCRH_V2)

PVH1_Rest5 = subset(PVH1V2, cells = c(colnames(AVP), colnames(OXT_Only), colnames(OXT_MC4R_V4), row.names(PVH_SST), colnames(ClustForMC4R_V6), colnames(ClustForCRH_V2)), invert=T)



set.dim = c(40)
set.res = c(1)
set.kparam = c(10)
ClusterFunc_Neuro_RNA(PVH1_Rest5)

AVP_Extra  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(PVH1_Rest5) & Fetal.combined_Data$AVP > 2)
AVP_Extra_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(AVP_Extra) & HypothalamicNeurons_UMAP$UMAP_2 < -4)
SST_Extra  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in%  colnames(PVH1_Rest5) & Fetal.combined_Data$SST > 0)
SST_Extra_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(SST_Extra) & HypothalamicNeurons_UMAP$UMAP_2 < -3 & row.names(HypothalamicNeurons_UMAP) %notin% c(row.names(AVP_Extra_Clean)))


PVH1_Rest6 = subset(PVH1V2, cells = c(colnames(AVP), colnames(OXT_Only), colnames(OXT_MC4R_V4), row.names(PVH_SST), colnames(ClustForMC4R_V6), colnames(ClustForCRH_V2), row.names(SST_Extra_Clean), row.names(AVP_Extra_Clean)), invert=T)

set.dim = c(40)
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(PVH1_Rest6)


PVH1_Rest6 <- FindNeighbors(PVH1_Rest6, k.param=20, dims=1:40)
PVH1_Rest6 <- FindClusters(PVH1_Rest6, resolution =1)
ClustForTRH = subset(PVH1_Rest6, idents = c(0))
ClustForTH = subset(PVH1_Rest6, idents = c(3))

set.dim = c(40)
set.res = c(1)
set.kparam = c(5,10)
ClusterFunc_Neuro_RNA(ClustForTRH)
ClusterFunc_Neuro_RNA(ClustForTH)

TRH_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ClustForTRH) & HypothalamicNeurons_UMAP$UMAP_2 > -3.2)

TH_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ClustForTH) & HypothalamicNeurons_UMAP$UMAP_1 < -8.5)


PVH1_Rest7 = subset(PVH1V2, cells = c(colnames(AVP), colnames(OXT_Only), colnames(OXT_MC4R_V4), row.names(PVH_SST), colnames(ClustForMC4R_V6), colnames(ClustForCRH_V2), row.names(SST_Extra_Clean), row.names(AVP_Extra_Clean), row.names(TRH_Clean), row.names(TH_Clean)), invert=T)


set.dim = c(40)
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(PVH1_Rest7)

CheckInput = colnames(ClustForCRH_V2)
CheckUMAP(PVH1_Rest4)




PVH2 = subset(HypothalamicNeurons, cells = row.names(PVH_2_Clean)) #No Extra NP clusters
PVH3 = subset(HypothalamicNeurons, cells = c(row.names(MoveToPVH3), colnames(PVH3_Clean_V2), colnames(AddToPVH3_V2pt2))) #No Extra NP clusters


set.dim = c(40)
set.res = c(1)
set.kparam = c(4,5)
ClusterFunc_Neuro_RNA(PVH3)

PVH3 <- FindNeighbors(PVH3, k.param=4, dims=1:40)
PVH3 <- FindClusters(PVH3, resolution =1)
PVH3_OXT = subset(PVH3, idents = c(5,7))
PVH3_TRH = subset(PVH3, idents = c(0,1,4,6))
PVH3_SIM1 = subset(PVH3, idents = c(0,1,4,6,5,7), invert=T)

PVH3_SIM1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(PVH3_SIM1) & HypothalamicNeurons_UMAP$UMAP_2 < 0.5 & HypothalamicNeurons_UMAP$UMAP_1 < -4)
PVH3_Extra = subset(PVH3_SIM1, cells = row.names(PVH3_SIM1_Clean), invert=T)
PVH3_TRH2 = subset(PVH3, cells = c(colnames(PVH3_TRH), colnames(PVH3_Extra)))

####

TM = subset(HypothalamicNeurons, cell = row.names(TM_1_Clean))
Arc1 = subset(HypothalamicNeurons, cell = c(row.names(Arc_1_Clean), row.names(ExtraArc_Clean)))
Arc2 = subset(HypothalamicNeurons, cell = row.names(TBX3_Clean))

set.dim = c(40)
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(TM)
ClusterFunc_Neuro_RNA(Arc1)
ClusterFunc_Neuro_RNA(Arc2)


Arc2 <- FindNeighbors(Arc2, k.param=50, dims=1:40)
Arc2 <- FindClusters(Arc2, resolution =1)
MoveToTM = subset(Arc2, idents = c(1,2))
Arc2_Clean = subset(Arc2, idents = c(0))

NewTM = subset(HypothalamicNeurons, cells = c(colnames(MoveToTM), row.names(TM_1_Clean)))
Arc1_2 = subset(HypothalamicNeurons, cells = c(colnames(Arc2_Clean), colnames(Arc1)))


set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(NewTM)
ClusterFunc_Neuro_RNA(Arc1_2)

NewTM <- FindNeighbors(NewTM, k.param=20, dims=1:40)
NewTM <- FindClusters(NewTM, resolution =1)
TM_NKX21_HDC = subset(NewTM, idents = c(2))
TM_NKX21_HDC_SOX2 = subset(NewTM, idents = c(1))
TM_HDC = subset(NewTM, idents = c(1,2), invert=T)

set.dim = c(40)
set.res = c(3)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(TM_NKX21_HDC) #4,5
ClusterFunc_Neuro_RNA(TM_NKX21_HDC_SOX2)



Arc1_2 <- FindNeighbors(Arc1_2, k.param=20, dims=1:40)
Arc1_2 <- FindClusters(Arc1_2, resolution =1)
Arc_AGRP_NPY = subset(Arc1_2, idents = c(4))

Arc_AGRP_NPY_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc_AGRP_NPY) & HypothalamicNeurons_UMAP$UMAP_2 < 3.5 & HypothalamicNeurons_UMAP$UMAP_1 > 3 & HypothalamicNeurons_UMAP$UMAP_1 < 6)


V2_Arc1_2_Rest1 = subset(Arc1_2, cells = row.names(Arc_AGRP_NPY_Clean), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(V2_Arc1_2_Rest1)                         
                         

V2_Arc1_2_Rest1 <- FindNeighbors(V2_Arc1_2_Rest1, k.param=3, dims=1:40)
V2_Arc1_2_Rest1 <- FindClusters(V2_Arc1_2_Rest1, resolution =1)
ClustForLEPR = subset(V2_Arc1_2_Rest1, idents = c(1,2,6,11))

set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(ClustForLEPR)   

ClustForLEPR <- FindNeighbors(ClustForLEPR, k.param=2, dims=1:40)
ClustForLEPR <- FindClusters(ClustForLEPR, resolution =1)
ClustForLEPR_V2 = subset(ClustForLEPR, idents = c(1,2,4,7,8,13,14,20,21,25))

set.dim = c(40)
set.res = c(3)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(ClustForLEPR_V2) 


ClustForLEPR_V2 <- FindNeighbors(ClustForLEPR_V2, k.param=2, dims=1:40)
ClustForLEPR_V2 <- FindClusters(ClustForLEPR_V2, resolution =3)
ClustForLEPR_V3 = subset(ClustForLEPR_V2, idents = c(6,10,11,12,14,16), invert=T)


set.dim = c(40)
set.res = c(5)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(ClustForLEPR_V3) 

ClustForLEPR_V3 <- FindNeighbors(ClustForLEPR_V3, k.param=2, dims=1:40)
ClustForLEPR_V3 <- FindClusters(ClustForLEPR_V3, resolution =5)
ClustForLEPR_V4 = subset(ClustForLEPR_V3, idents = c(5,8,14), invert=T)

set.dim = c(40)
set.res = c(0.05)
set.kparam = c(200)
ClusterFunc_Neuro_RNA(ClustForLEPR_V4) 

LEPR_Extra  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in%  colnames(V2_Arc1_2_Rest1) & Fetal.combined_Data$LEPR > 0)
LEPR_Extra_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% row.names(LEPR_Extra) & HypothalamicNeurons_UMAP$UMAP_2 > 3)
ClustForLEPR_V5 = subset(V2_Arc1_2_Rest1, cells = unique(c(row.names(LEPR_Extra_Clean), colnames(ClustForLEPR_V4))))

V2_Arc1_2_Rest2 = subset(V2_Arc1_2_Rest1, cells = unique(c(row.names(LEPR_Extra_Clean), colnames(ClustForLEPR_V4))), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(V2_Arc1_2_Rest2) 

V2_Arc1_2_Rest2 <- FindNeighbors(V2_Arc1_2_Rest2, k.param=5, dims=1:40)
V2_Arc1_2_Rest2 <- FindClusters(V2_Arc1_2_Rest2, resolution =1)
ClustForPOMC = subset(V2_Arc1_2_Rest2, idents = c(2,5))


Arc_POMC_Clean_V2 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(ClustForPOMC) & HypothalamicNeurons_UMAP$UMAP_2 > 3 |  row.names(HypothalamicNeurons_UMAP) %in% colnames(V2_Arc1_2_Rest2) & HypothalamicNeurons_UMAP$UMAP_2 > 4.2)
Arc_POMC_Clean_V2Seu = subset(V2_Arc1_2_Rest1, cells = row.names(Arc_POMC_Clean_V2))
set.dim = c(40)
set.res = c(0.05)
set.kparam = c(200)
ClusterFunc_Neuro_RNA(Arc_POMC_Clean_V2Seu) 


Arc1_2_Rest2 = subset(Arc1_2, cells = c(row.names(Arc_POMC_Clean_V2), colnames(ClustForLEPR_V5), row.names(Arc_AGRP_NPY_Clean)), invert=T)
set.dim = c(40)
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(Arc1_2_Rest2) 


Arc_NKX21_PCSK1_Clean = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc1_2_Rest2) & HypothalamicNeurons_UMAP$UMAP_2 < 2 & HypothalamicNeurons_UMAP$UMAP_1 < 5 | row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc1_2_Rest2) & HypothalamicNeurons_UMAP$UMAP_2 < 0 & HypothalamicNeurons_UMAP$UMAP_1 < 7 )


Arc_TBX3_Clean = subset(Arc1_2_Rest2, cells = row.names(Arc_NKX21_PCSK1_Clean), invert=T)


VMH_Seu = subset(HypothalamicNeurons, cells = row.names(VMH_1_Clean))
set.dim = c(40)
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(VMH_Seu) 


ReclustUkns = subset(HypothalamicNeurons, cells = c(colnames(SIX3_SST_OTPpt2),colnames(Ukn_Cells)))
MoveToIDTT  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ReclustUkns) & Fetal.combined_Data$LHX6 > 0)
MoveZI  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in%  colnames(ReclustUkns) & row.names(Fetal.combined_Data) %notin%  row.names(MoveToIDTT) & Fetal.combined_Data$SIX3 > 0)
Ukn48_Clean = subset(ReclustUkns, cells = c(row.names(MoveToIDTT), row.names(MoveZI)), invert=T)

set.dim = c(40)
set.res = c(1)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(Ukn48_Clean)

Ukn48_Clean <- FindNeighbors(Ukn48_Clean, k.param=2, dims=1:40)
Ukn48_Clean <- FindClusters(Ukn48_Clean, resolution =1)
Arc_Cluster = subset(Ukn48_Clean, idents = c(0,1))

set.dim = c(40)
set.res = c(2)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(Arc_Cluster)

Arc_Cluster <- FindNeighbors(Arc_Cluster, k.param=2, dims=1:40)
Arc_Cluster <- FindClusters(Arc_Cluster, resolution =2)
Arc_Cluster_V2 = subset(Arc_Cluster, idents = c(0,1,2,10,11,12))

set.dim = c(40)
set.res = c(3)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(Arc_Cluster_V2)

Arc_Cluster_V2 <- FindNeighbors(Arc_Cluster_V2, k.param=2, dims=1:40)
Arc_Cluster_V2 <- FindClusters(Arc_Cluster_V2, resolution =3)
Arc_Cluster_V3 = subset(Arc_Cluster_V2, idents = c(5,3), invert=T)

EXTRA_Arcs  = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(Ukn48_Clean) & Fetal.combined_Data$TBX3 > 0 | row.names(Fetal.combined_Data) %in% colnames(Ukn48_Clean) & Fetal.combined_Data$TRH > 0 | row.names(Fetal.combined_Data) %in% colnames(Ukn48_Clean) & Fetal.combined_Data$PCSK1 > 0)

Arc_Cluster_V4 = subset(HypothalamicNeurons_UMAP, row.names(HypothalamicNeurons_UMAP) %in% colnames(Arc_Cluster_V3) & HypothalamicNeurons_UMAP$UMAP_2 > -2.7 & HypothalamicNeurons_UMAP$UMAP_1 > 3 | row.names(HypothalamicNeurons_UMAP) %in% row.names(Arc_Cluster) & HypothalamicNeurons_UMAP$UMAP_2 > -3.5 & HypothalamicNeurons_UMAP$UMAP_1 > 3)
Arc_Cluster_V4_Seu = subset(Ukn48_Clean, cells = row.names(Arc_Cluster_V4))

Ukn48_Clean_V2 = subset(Ukn48_Clean, cells = row.names(Arc_Cluster_V4), invert=T)


set.dim = c(40)
set.res = c(0.1)
set.kparam = c(200)
ClusterFunc_Neuro_RNA(Arc_Cluster_V4_Seu)



CheckInput = row.names(Arc_Cluster_V4)
CheckUMAP(Ukn48_Clean)



CheckInput = row.names(MoveToVMH_Comb)
CheckUMAP(UKN_Top_Cleaned)
####




CheckInput = colnames(Ukn48_Clean)
CheckUMAP(All_UKNS)
```

#Check for new genes
```{r}
############ CHECK FOR LOCALIZED GENES ############ 
HypothalamicNeurons_t = as.data.frame(HypothalamicNeurons@assays$RNA@counts)

HypothalamicNeurons_t$RowVar = rowVars(as.matrix(HypothalamicNeurons_t))
GoI_RMNoVar = subset(HypothalamicNeurons_t, HypothalamicNeurons_t$RowVar > 0.3 & row.names(HypothalamicNeurons_t) %notin% HumanTFs$V1)

DefaultAssay(HypothalamicNeurons) = "RNA"
GoI = row.names(GoI_RMNoVar) #HumanTFs$V1 
PlotsPerPage = 50
CompilePlot = list()
num = 0
for( in seq(1,ceiling(length(GoI)/PlotsPerPage),1)){
LowLim = num+1
HighLim =  LowLim+(PlotsPerPage-1)
num = HighLim
Genes = GoI[LowLim:HighLim]
CompilePlot[[x]] = FeaturePlot(HypothalamicNeurons, Genes, ncol=10, reduction="umap")
}
pdf(paste("FPs_HypothalamicNeurons_NonTFsHighVar_18NOV.pdf", sep=""), width=50, height=50)
print(CompilePlot)
dev.off()

pdf(paste("FPs_HypothalamicNeurons_KeyTFs_7NOV.pdf", sep=""), width=30, height=25)
print(FeaturePlot(HypothalamicNeurons, c("HEY1", "FEZF1", "EBF2", "EBF3", "EBF4", "LHX1", "LHX5", "LHX6", "LHX9", "MEIS2", "NPAS4", "NR3C1", "ONECUT2", "ONECUT3", "POU6F2", "PRDM8", "PROX1", "SIX3", "SIX6", "TBR1", "ZIC1", "ZIC2", "ZIC5"), ncol=5, reduction="umap"))
dev.off()

DefaultAssay(HypothalamicNeurons) = "RNA"
pdf(paste("FPs_HypothalamicNeurons_VPHgenes_19NOV.pdf", sep=""), width=30, height=25)
print(FeaturePlot(HypothalamicNeurons, c("TAC2", "PVALB", "FGF1", "CABP7", "FOXB1", "CCK", "GPR83", "SPOCK3", "CXCL14", "NOS1", "CALB1", "NTS", "COL25A1", "SLC24A2", "SYNPR", "STXBP6", "TCF4", "NR4A2", "EBF3", "SLC6A1", "TAC1", "FOXP2", "GLRA3", "HTR2C", "PTHLH", "PTK2B", "DLK1"), ncol=5, reduction="umap"))
dev.off()



Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns"
HypothalamicNeurons_DEGs = DEGs(HypothalamicNeurons)
HypothalamicNeurons_DEGs_subs = subset(HypothalamicNeurons_DEGs, HypothalamicNeurons_DEGs$avg_log2FC > 0.5 & HypothalamicNeurons_DEGs$p_val_adj < 0.05 & HypothalamicNeurons_DEGs$cluster %in% c("MN_1 [ONECUT2]", "UKN_1 [CRABP1]", "UKN_5 [LRTM1]", "UKN_2","UKN_6", "Ukn [SIX3/SST/OTP]", "UKN_4 [NPAS4]"))
DefaultAssay(HypothalamicNeurons) = "RNA"
GoI = unique(HypothalamicNeurons_DEGs_subs$gene) #HumanTFs$V1 
PlotsPerPage = 50
CompilePlot = list()
num = 0
for( in seq(1,ceiling(length(GoI)/PlotsPerPage),1)){
LowLim = num+1
HighLim =  LowLim+(PlotsPerPage-1)
num = HighLim
Genes = GoI[LowLim:HighLim]
CompilePlot[[x]] = FeaturePlot(HypothalamicNeurons, Genes, ncol=10, reduction="umap")
}
pdf(paste("FPs_HypothalamicNeurons_FromDEGs_23NOV.pdf", sep=""), width=50, height=50)
print(CompilePlot)
dev.off()




###### COMPILE ASSIGNMENTS ###### 



```

#Compile Assignments
```{r}
OCT_Neuron_Assigns_Renamed = as.data.frame(c(rep("Neuron_01 [TM]", length(colnames(TM_NKX21_HDC))), #TM_NKX21_HDC
                                 rep("Neuron_02 [TM]", length(colnames(TM_NKX21_HDC_SOX2))), #TM_NKX21_HDC_SOX2
                                 rep("Neuron_03 [TM]", length(colnames(TM_HDC))), #TM_HDC
                                 rep("Neuron_04 [ARC]", length(colnames(Arc_POMC_Clean_V2Seu))), #Arc_POMC
                                 rep("Neuron_05 [ARC]", length(colnames(ClustForLEPR_V5))),       
                                 rep("Neuron_06 [ARC]", length(row.names(Arc_AGRP_NPY_Clean))), #Arc_AGRP_NPY
                                 rep("Neuron_07 [ARC]", length(colnames(Arc_TBX3_Clean))), #Arc_TBX3
                                 rep("Neuron_08 [ARC]", length(row.names(Arc_NKX21_PCSK1_Clean))), #Arc_NKX21_PCSK1
                                 rep("Neuron_09 [ARC]", length(row.names(NKX21TRH_Clean))), 
                                 rep("Neuron_10 [ARC]", length(row.names(Arc_Precursor_Clean))), #Arc_Precursor [DLX1/DLX2/GSX1/HMX2]
                                 rep("Neuron_11 [ARC]", length(row.names(Arc_4_V3))), #
                                 rep("Neuron_12 [ARC]", length(row.names(Arc_5_V2))), 
                                 rep("Neuron_13 [ARC]", length(row.names(ARC_Cells_V3))), #SPLIT INTO ARC and ZI ----- NEW
                                 rep("Neuron_14 [ARC]", length(row.names(NPAS4_Clean))), ####NEW ARC!!!!!
                                 rep("Neuron_15 [ARC]", length(row.names(Arc_Cluster_V4))), ####NEW ARC!!!!!
                                 rep("Neuron_16 [PVH]", length(colnames(OXT_Only))), 
                                 rep("Neuron_17 [PVH]", length(colnames(ClustForCRH_V2))), 
                                 rep("Neuron_18 [PVH]", length(colnames(OXT_MC4R_V4))),   
                                 rep("Neuron_19 [PVH]", length(colnames(ClustForMC4R_V6))), 
                                 rep("Neuron_20 [PVH]", length(row.names(TH_Clean))),
                                 rep("Neuron_22 [PVH]", length(row.names(TRH_Clean))), 
                                 rep("Neuron_22 [PVH]", length(row.names(PVH_SST))),                                 
                                 rep("Neuron_22 [PVH]", length(row.names(SST_Extra_Clean))), 
                                 rep("Neuron_23 [PVH]", length(colnames(AVP))), 
                                 rep("Neuron_23 [PVH]", length(row.names(AVP_Extra_Clean))), 
                                 rep("Neuron_24 [PVH]", length(colnames(PVH1_Rest7))), 
                                 rep("Neuron_25 [PVH]", length(colnames(PVH3_OXT))),
                                 rep("Neuron_26 [PVH]", length(colnames(PVH3_TRH2))), 
                                 rep("Neuron_27 [PVH]", length(row.names(PVH3_SIM1_Clean))), 
                                 rep("Neuron_28 [PVH]", length(colnames(Top_PVH_V4))), 
                                 rep("Neuron_29 [PVH]", length(row.names(PVH_2_Clean))), 
                                 rep("Neuron_30 [PVH]", length(row.names(NHLH2_CleanV2))), 
                                 rep("Neuron_30 [PVH]", length(colnames(PVHCells))), 
                                 rep("Neuron_31 [VMH]", length(row.names(ExtraVMH_Clean))), 
                                 rep("Neuron_31 [VMH]", length(row.names(VMH_1_Clean))),
                                 rep("Neuron_32 [VMH]", length(colnames(UKN_Top_Cleaned))), ####NEW VMH!!!!! ----- NEW
                                 rep("Neuron_33 [LH]", length(row.names(LH_1))), 
                                 rep("Neuron_34 [LH]", length(row.names(UKN_TAC1))), 
                                 rep("Neuron_35 [DMH]", length(row.names(DMH_Clean))), 
                                 rep("Neuron_36 [SCN]", length(row.names(SCN_1_Clean))), 
                                 rep("Neuron_37 [SCN]", length(colnames(CheckZICs_V3))), 
                                 rep("Neuron_38 [MN]", length(colnames(ForONECUT2))), 
                                 rep("Neuron_38 [MN]", length(row.names(ExtraONECUT2))), 
                                 rep("Neuron_38 [MN]", length(row.names(MN_1_Clean))), 
                                 rep("Neuron_38 [MN]", length(colnames(AddToMN))), 
                                 rep("Neuron_39 [MN]", length(row.names(MMPost_Clean))), 
                                 rep("Neuron_40 [MN]", length(colnames(AddToPVH3_Clean))), 
                                 rep("Neuron_40 [MN]", length(colnames(MoveToMN3))), 
                                 rep("Neuron_40 [MN]", length(colnames(MN_Clean))), 
                                 rep("Neuron_41 [SMN]", length(row.names(VMH2_EBF4_Clean))), 
                                 rep("Neuron_42 [ID/TT]", length(row.names(ClustForLHX6_V7))), 
                                 rep("Neuron_42 [ID/TT]", length(colnames(AddToLHX6))), 
                                 rep("Neuron_42 [ID/TT]", length(row.names(MoveToIDTT))), 
                                 rep("Neuron_43 [PO]", length(row.names(PENK_PNOC_Clean))), 
                                 rep("Neuron_44 [PO]", length(row.names(Preoptic_PNOC_NTS_Clean))), 
                                 rep("Neuron_44 [PO]", length(colnames(PNOC_Preoptic))), 
                                 rep("Neuron_44 [PO]", length(colnames(AddToPNOC))), 
                                 rep("Neuron_45 [ZI]", length(row.names(MoveZI))), 
                                 rep("Neuron_45 [ZI]", length(row.names(ZI_Clean))), 
                                 rep("Neuron_46 [ZI]", length(colnames(SIX3_MEIS2_))), 
                                 rep("Neuron_46 [ZI]", length(colnames(SIX3_Extra))), 
                                 rep("Neuron_47 [ZI]", length(row.names(ZI_Cells_V3))), #SPLIT INTO ARC and ZI ----- NEW
                                 rep("Neuron_48", length(colnames(Ukn48_Clean_V2))), #Leave as is!!!!!
                                 rep("Neuron_49", length(row.names(Ukn_A_Clean))), 
                                 rep("Neuron_50", length(row.names(Ukn_B_Clean))))) # LEAVE AS IS
colnames(OCT_Neuron_Assigns_Renamed) = "Pop"
row.names(OCT_Neuron_Assigns_Renamed) = c(colnames(TM_NKX21_HDC), #TM_NKX21_HDC
                                 colnames(TM_NKX21_HDC_SOX2), #TM_NKX21_HDC_SOX2
                                 colnames(TM_HDC), #TM_HDC
                                 colnames(Arc_POMC_Clean_V2Seu), #Arc_POMC
                                 colnames(ClustForLEPR_V5),       
                                 row.names(Arc_AGRP_NPY_Clean), #Arc_AGRP_NPY
                                 colnames(Arc_TBX3_Clean), #Arc_TBX3
                                 row.names(Arc_NKX21_PCSK1_Clean),#Arc_NKX21_PCSK1
                                 row.names(NKX21TRH_Clean), 
                                 row.names(Arc_Precursor_Clean), #Arc_Precursor [DLX1/DLX2/GSX1/HMX2]
                                 row.names(Arc_4_V3), #
                                 row.names(Arc_5_V2), 
                                 row.names(ARC_Cells_V3), #SPLIT INTO ARC and ZI ----- NEW
                                 row.names(NPAS4_Clean), ####NEW ARC!!!!!
                                 row.names(Arc_Cluster_V4),
                                 colnames(OXT_Only), 
                                 colnames(ClustForCRH_V2), 
                                 colnames(OXT_MC4R_V4),   
                                 colnames(ClustForMC4R_V6), 
                                 row.names(TH_Clean),
                                 row.names(TRH_Clean), 
                                 row.names(PVH_SST),                                 
                                 row.names(SST_Extra_Clean), 
                                 colnames(AVP), 
                                 row.names(AVP_Extra_Clean), 
                                 colnames(PVH1_Rest7), 
                                 colnames(PVH3_OXT),
                                 colnames(PVH3_TRH2), 
                                 row.names(PVH3_SIM1_Clean), 
                                 colnames(Top_PVH_V4), 
                                 row.names(PVH_2_Clean), 
                                 row.names(NHLH2_CleanV2), 
                                 colnames(PVHCells), 
                                 row.names(ExtraVMH_Clean), 
                                 row.names(VMH_1_Clean),
                                 colnames(UKN_Top_Cleaned), ####NEW VMH!!!!! ----- NEW
                                 row.names(LH_1), 
                                 row.names(UKN_TAC1), 
                                 row.names(DMH_Clean), 
                                 row.names(SCN_1_Clean), 
                                 colnames(CheckZICs_V3), 
                                 colnames(ForONECUT2), 
                                 row.names(ExtraONECUT2), 
                                 row.names(MN_1_Clean), 
                                 colnames(AddToMN), 
                                 row.names(MMPost_Clean), 
                                 colnames(AddToPVH3_Clean), 
                                 colnames(MoveToMN3), 
                                 colnames(MN_Clean), 
                                 row.names(VMH2_EBF4_Clean), 
                                 row.names(ClustForLHX6_V7), 
                                 colnames(AddToLHX6), 
                                 row.names(MoveToIDTT),
                                 row.names(PENK_PNOC_Clean), 
                                 row.names(Preoptic_PNOC_NTS_Clean), 
                                 colnames(PNOC_Preoptic), 
                                 colnames(AddToPNOC), 
                                 row.names(MoveZI),
                                 row.names(ZI_Clean), 
                                 colnames(SIX3_MEIS2_), 
                                 colnames(SIX3_Extra), 
                                 row.names(ZI_Cells_V3), #SPLIT INTO ARC and ZI ----- NEW
                                 colnames(Ukn48_Clean_V2), #Leave as is!!!!!
                                 row.names(Ukn_A_Clean), ### NEW Mammillary!!!
                                 row.names(Ukn_B_Clean)) # LEAVE AS IS

#OCT_Neuron_Assigns_Renamed$Dup = duplicated(OCT_Neuron_Assigns_Renamed$Barcs) | duplicated(OCT_Neuron_Assigns_Renamed$Barcs, fromLast = T)
#CHECK =  subset(OCT_Neuron_Assigns_Renamed, OCT_Neuron_Assigns_Renamed$Dup == T)
#table(CHECK$Pop)

paste(sort(unique(OCT_Neuron_Assigns_Renamed$Pop)), collapse = ", ")


OCT_Neuron_Assigns_Renamed2 = OCT_Neuron_Assigns_Renamed
OCT_Neuron_Assigns_Renamed2$Pop = factor(OCT_Neuron_Assigns_Renamed2$Pop, levels = c("Neuron_01 [TM]", "Neuron_02 [TM]", "Neuron_03 [TM]", "Neuron_04 [ARC]", "Neuron_05 [ARC]", "Neuron_06 [ARC]", "Neuron_07 [ARC]", "Neuron_08 [ARC]", "Neuron_09 [ARC]", "Neuron_10 [ARC]", "Neuron_11 [ARC]", "Neuron_12 [ARC]", "Neuron_13 [ARC]", "Neuron_14 [ARC]", "Neuron_15 [ARC]", "Neuron_16 [PVH]", "Neuron_17 [PVH]", "Neuron_18 [PVH]", "Neuron_19 [PVH]", "Neuron_20 [PVH]", "Neuron_22 [PVH]", "Neuron_23 [PVH]", "Neuron_24 [PVH]", "Neuron_25 [PVH]", "Neuron_26 [PVH]", "Neuron_27 [PVH]", "Neuron_28 [PVH]", "Neuron_29 [PVH]", "Neuron_30 [PVH]", "Neuron_31 [VMH]", "Neuron_32 [VMH]", "Neuron_33 [LH]", "Neuron_34 [LH]", "Neuron_35 [DMH]", "Neuron_36 [SCN]", "Neuron_37 [SCN]", "Neuron_38 [MN]", "Neuron_39 [MN]", "Neuron_40 [MN]", "Neuron_41 [SMN]", "Neuron_42 [ID/TT]", "Neuron_43 [PO]", "Neuron_44 [PO]", "Neuron_45 [ZI]", "Neuron_46 [ZI]", "Neuron_47 [ZI]", "Neuron_48", "Neuron_49", "Neuron_50"))




write.csv(OCT_Neuron_Assigns_Renamed2, paste("OCT_AllCell_Assigns_", Sys.Date(), ".csv", sep=""))
HypothalamicNeurons = AddMetaData(HypothalamicNeurons, OCT_Neuron_Assigns_Renamed2, "OCT_Neuron_Assigns_Renamed2")
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"

pdf(paste("ALLFETAL_AssignmentsRenamed_UMAP_27OCT21.pdf", sep=""), width=20, height=15)
print(DimPlot(HypothalamicNeurons, reduction = "umap", label=T))
dev.off()



GeneList = c(BroadGenes, HypothalamicProgenSubclass, WellKnownGenes, SuspectedGenes, Zhou_Extras, BlackshawEtc, VariousUkn_1, VariousUkn_2, VariousUkn_3)


Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
DefaultAssay(HypothalamicNeurons) = "RNA"
VLNList = list()
for( in GeneList){
VLNList[[x]] = VlnPlot(HypothalamicNeurons, x, pt.size = 0)
}

pdf("AllNeurons_Vlns.pdf", width = 15, height=5)
print(VLNList)
dev.off()


OCT_Neuron_Assigns_Renamed2$Pop = factor(OCT_Neuron_Assigns_Renamed2$Pop, levels = rev(c("Neuron_01 [TM]", "Neuron_02 [TM]", "Neuron_03 [TM]", "Neuron_04 [ARC]", "Neuron_05 [ARC]", "Neuron_06 [ARC]", "Neuron_07 [ARC]", "Neuron_08 [ARC]", "Neuron_09 [ARC]", "Neuron_10 [ARC]", "Neuron_11 [ARC]", "Neuron_12 [ARC]", "Neuron_13 [ARC]", "Neuron_14 [ARC]", "Neuron_15 [PVH]", "Neuron_16 [PVH]", "Neuron_17 [PVH]", "Neuron_18 [PVH]", "Neuron_19 [PVH]", "Neuron_20 [PVH]", "Neuron_21 [PVH]", "Neuron_22 [PVH]", "Neuron_23 [PVH]", "Neuron_24 [PVH]", "Neuron_25 [PVH]", "Neuron_26 [PVH]", "Neuron_27 [PVH]", "Neuron_28 [PVH]", "Neuron_29 [PVH]", "Neuron_30 [VMH]", "Neuron_31 [VMH]", "Neuron_32 [LH]", "Neuron_33 [LH]", "Neuron_34 [DMH]", "Neuron_35 [SCN]", "Neuron_36 [SCN]", "Neuron_37 [MN]", "Neuron_38 [MN]", "Neuron_39 [MN]", "Neuron_40 [MN]", "Neuron_41 [SMN]", "Neuron_42 [ID/TT]", "Neuron_43 [PO]", "Neuron_44 [PO]", "Neuron_45 [ZI]", "Neuron_46 [ZI]", "Neuron_47 [ZI]", "Neuron_48", "Neuron_49")))

HypothalamicNeurons = AddMetaData(HypothalamicNeurons, OCT_Neuron_Assigns_Renamed2, "OCT_Neuron_Assigns_Renamed2")
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"


AllCells_PlotOrder = c("STMN1", "SOX2", "HDC", "NKX2-1", "TBX3", "POMC", "AGRP", "NPY", "OTP", "PCSK1", "SIM1", "POU3F2", "OXT", "CRH", "MC4R", "GAL", "NHLH2",  "SST", "AVP",  "TRH",  "GABRA5", "NPTX2", "FEZF1", "HCRT", "PMCH", "PPP1R17", "LHX1", "SIX6", "ZIC1", "ZIC2", "ONECUT2", "ONECUT3", "PITX2", "FOXB1",  "EBF2", "LHX6", "PNOC",  "PENK", "SIX3", "MEIS2")

DefaultAssay(HypothalamicNeurons) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(HypothalamicNeurons, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(HypothalamicNeurons, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(2.3, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("NeuronsOnly_MultiVln_29NOV21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(HypothalamicNeurons@active.ident))*0.5)
print(AllVlns)
dev.off()
```

#Pool populations and recluster
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"

All_ARCUATE = subset(HypothalamicNeurons, idents = c("Neuron_04 [ARC]", "Neuron_05 [ARC]", "Neuron_06 [ARC]", "Neuron_07 [ARC]", "Neuron_08 [ARC]", "Neuron_09 [ARC]", "Neuron_10 [ARC]", "Neuron_11 [ARC]", "Neuron_12 [ARC]", "Neuron_13 [ARC]", "Neuron_14 [ARC]", "Neuron_15 [ARC]"))
All_PVH = subset(HypothalamicNeurons, idents = c("Neuron_15 [PVH]", "Neuron_16 [PVH]", "Neuron_17 [PVH]", "Neuron_18 [PVH]", "Neuron_19 [PVH]", "Neuron_20 [PVH]", "Neuron_21 [PVH]", "Neuron_22 [PVH]", "Neuron_23 [PVH]", "Neuron_24 [PVH]", "Neuron_25 [PVH]", "Neuron_26 [PVH]", "Neuron_27 [PVH]", "Neuron_28 [PVH]", "Neuron_29 [PVH]"))
All_MN = subset(HypothalamicNeurons, idents = c("Neuron_37 [MN]", "Neuron_38 [MN]", "Neuron_39 [MN]", "Neuron_40 [MN]"))
All_SCN= subset(HypothalamicNeurons, idents = c("Neuron_35 [SCN]", "Neuron_36 [SCN]"))
All_ZI = subset(HypothalamicNeurons, idents = c("Neuron_45 [ZI]", "Neuron_46 [ZI]", "Neuron_47 [ZI]"))
All_PREOPTIC = subset(HypothalamicNeurons, idents = c("Neuron_43 [PO]", "Neuron_44 [PO]")) #"Neuron_39 [PO]",
All_TM = subset(HypothalamicNeurons, idents = c("Neuron_01 [TM]", "Neuron_02 [TM]", "Neuron_03 [TM]"))
All_LH = subset(HypothalamicNeurons, idents = c("Neuron_32 [LH]", "Neuron_33 [LH]"))
All_VMH = subset(HypothalamicNeurons, idents = c("Neuron_30 [VMH]", "Neuron_31 [VMH]"))
All_DMH = subset(HypothalamicNeurons, idents = c("Neuron_34 [DMH]"))
All_SMN = subset(HypothalamicNeurons, idents = c("Neuron_41 [SMN]"))
All_IDTT = subset(HypothalamicNeurons, idents = c("Neuron_42 [ID/TT]"))
All_UKNS = subset(HypothalamicNeurons, idents = c("Neuron_48", "Neuron_49", "Neuron_50"))
All_KNS = subset(HypothalamicNeurons, idents = c("Neuron_48", "Neuron_49", "Neuron_50"), invert=T)


#("Neuron_01 [TM]", "Neuron_02 [TM]", "Neuron_03 [TM]", "Neuron_04 [ARC]", "Neuron_05 [ARC]", "Neuron_06 [ARC]", "Neuron_07 [ARC]", "Neuron_08 [ARC]", "Neuron_09 [ARC]", "Neuron_10 [ARC]", "Neuron_11 [ARC]", "Neuron_12 [ARC]", "Neuron_13 [ARC]", "Neuron_14 [ARC]", "Neuron_15 [PVH]", "Neuron_16 [PVH]", "Neuron_17 [PVH]", "Neuron_18 [PVH]", "Neuron_19 [PVH]", "Neuron_20 [PVH]", "Neuron_21 [PVH]", "Neuron_22 [PVH]", "Neuron_23 [PVH]", "Neuron_24 [PVH]", "Neuron_25 [PVH]", "Neuron_26 [PVH]", "Neuron_27 [PVH]", "Neuron_28 [PVH]", "Neuron_29 [PVH]", "Neuron_30 [VMH]", "Neuron_31 [VMH]", "Neuron_32 [LH]", "Neuron_33 [LH]", "Neuron_34 [DMH]", "Neuron_35 [SCN]", "Neuron_36 [SCN]", "Neuron_37 [MN]", "Neuron_38 [MN]", "Neuron_39 [MN]", "Neuron_40 [SMN]", "Neuron_41 [ID/TT]", "Neuron_42 [PO]", "Neuron_43 [PO]", "Neuron_44 [ZI]", "Neuron_45 [ZI]", "Neuron_46 [ZI]", 

set.dim = 40
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(All_UKNS) 

DefaultAssay(HypothalamicNeurons) = "RNA"
FeaturePlot(HypothalamicNeurons, "PPP1R17", reduction = "umap")

Check_DEGs_subs = subset(Check_DEGs, Check_DEGs$cluster %in% c("Neuron_45") & Check_DEGs$avg_log2FC > 1 & Check_DEGs$p_val_adj < 0.05)


DefaultAssay(HypothalamicNeurons) = "RNA"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
Check_DEGs = DEGs(HypothalamicNeurons)
Check_DEGs_subs = subset(Check_DEGs, Check_DEGs$cluster %in% c("Neuron_45", "Neuron_46", "Neuron_47", "Neuron_48", "Neuron_49") & Check_DEGs$avg_log2FC > 0.3 & Check_DEGs$p_val_adj < 0.05)

DefaultAssay(HypothalamicNeurons) = "RNA"
for(y in c("Neuron_45", "Neuron_46", "Neuron_47", "Neuron_48", "Neuron_49")){
Subs = subset(Check_DEGs_subs, Check_DEGs_subs$cluster %in%  y)
FPList = list()
for( in unique(Subs$gene)){
FPList[[x]] = FeaturePlot(HypothalamicNeurons, x, reduction = "umap")
}
pdf(paste(y, "_FPs.pdf", sep = ""), width = 8, height = 6)
print(FPList)
dev.off()
}


set.dim = 40
set.res = c(1)
set.kparam = c(100)
ClusterFunc_Neuro_RNA(All_UKNS) 
ClusterFunc_Neuro_RNA(All_ARCUATE) 
##ClusterFunc_Neuro_RNA(All_PVH) 
ClusterFunc_Neuro_RNA(All_LH) 
ClusterFunc_Neuro_RNA(All_MN) 
#ClusterFunc_Neuro_RNA(All_SCN) 
ClusterFunc_Neuro_RNA(All_ZI) 
#ClusterFunc_Neuro_RNA(All_PREOPTIC) 
#ClusterFunc_Neuro_RNA(All_TM) 

CheckInput = colnames(All_UKNS)
CheckUMAP(HypothalamicNeurons)

ClusterList = list(All_UKNS, All_VMH, All_ARCUATE, All_MN, All_ZI, All_IDTT)
names(ClusterList) = c("All_UKNS", "All_VMH", "All_ARCUATE", "All_MN", "All_ZI", "All_IDTT")
  
#for( in names(ClusterList)){
SeuFile = All_UKNS#ClusterList[[x]]
f = "All_UKNS"#x
DPlist2 = list()
DPlist = list()
FPlist = list()
for(y in seq(10,50,10)){
DefaultAssay(SeuFile) = "integrated"
SeuFile = RunPCA(SeuFile, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
SeuFile <- RunUMAP(SeuFile, dims = 1:z, spread= s)
DefaultAssay(SeuFile) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(SeuFile, c("SYT1", "TBX3", "SIM1", "POMC", "NKX2-1", "MC4R", "SST", "KISS1", "GHRH", "TH", "TRH", "OXT", "AVP", "CRH", "BDNF", "PCSK1", "ADCYAP1", "GAL", "NPY", "AGRP", "HDC", "SOX2", "LEPR", "SIX6", "OTP", "POU3F2", "NRN1", "GABRA5", "NPTX2", "FEZF1", "NR5A1", "HCRT", "CARTPT", "SETBP1", "TAC1", "PITX2", "FOXP1", "FOXP2", "FOXB1", "SOX14", "MASP1", "PPP1R17", "DLX1", "DLX2", "GSX1", "HMX2", "NHLH2", "PENK", "LHX1", "PMCH", "NPVF", "ONECUT2", "ONECUT3", "ZIC1", "ZIC2", "MEIS2", "SIX3", "PNOC", "EBF2", "EBF3", "LHX6", "NPAS4"), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(SeuFile, reduction="umap", group.by = "OCT_Neuron_Assigns_Renamed2") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(SeuFile, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", f, "_FeaturePlots.pdf", sep=""), width=20, height=55)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", f, "_OCT_Neuron_Assigns.pdf", sep=""), width=8, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", f, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()
#}
```

#############################################
####### SUBCLUSTERING NEURONAL NUCLEI #######
#############################################

### PREOPTIC: DONE
```{r}
All_PREOPTIC = subset(HypothalamicNeurons, idents = c("Neuron_39 [PO]", "Neuron_40 [PO]")) 

#"PCA 20 _dims 20 _spread 5"
DefaultAssay(All_PREOPTIC) = "integrated"
All_PREOPTIC = RunPCA(All_PREOPTIC, npcs = 20)
All_PREOPTIC <- RunUMAP(All_PREOPTIC, dims = 1:20, spread= 5)
All_PREOPTIC_UMAP = as.data.frame(All_PREOPTIC@reductions$umap@cell.embeddings)

PO_Cluster1 = subset(All_PREOPTIC_UMAP,  All_PREOPTIC_UMAP$UMAP_1 < -20)
PO_Cluster1_Seu = subset(All_PREOPTIC, cells = row.names(PO_Cluster1))
set.dim = 20
set.res = c(1)
set.kparam = c(60)
#ClusterFunc_Neuro_RNA(PO_Cluster1_Seu)  

PO_Cluster1_Seu <- FindNeighbors(PO_Cluster1_Seu, k.param=60, dims=1:20)
PO_Cluster1_Seu <- FindClusters(PO_Cluster1_Seu, resolution =1)
PO_Subcluster_1 = subset(PO_Cluster1_Seu, idents = 0)
PO_Subcluster_2 = subset(PO_Cluster1_Seu, idents = 1)

PO_Rest1 = subset(All_PREOPTIC, cells = row.names(PO_Cluster1), invert=T)


PO_Subcluster_3 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_Rest1) & All_PREOPTIC_UMAP$UMAP_1 < 0 & All_PREOPTIC_UMAP$UMAP_2 < -10)
PO_Subcluster_4 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_Rest1) & All_PREOPTIC_UMAP$UMAP_1 < -2 & All_PREOPTIC_UMAP$UMAP_2 > -10 & All_PREOPTIC_UMAP$UMAP_2 < -4)
PO_Subcluster_5 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_Rest1) & All_PREOPTIC_UMAP$UMAP_1 < -4 & All_PREOPTIC_UMAP$UMAP_2 > -4 & All_PREOPTIC_UMAP$UMAP_2 < 8)


PO_Rest2 = subset(All_PREOPTIC, cells = c(row.names(PO_Cluster1), row.names(PO_Subcluster_3), row.names(PO_Subcluster_4), row.names(PO_Subcluster_5)), invert=T)


set.dim = 20
set.res = c(1)
set.kparam = c(30)
#ClusterFunc_Neuro_RNA(PO_Rest2)  

PO_Rest2 <- FindNeighbors(PO_Rest2, k.param=30, dims=1:20)
PO_Rest2 <- FindClusters(PO_Rest2, resolution =1)
PO_ForSubcluster_6 = subset(PO_Rest2, idents = 2)

set.dim = 20
set.res = c(1)
set.kparam = c(3,10,20)
#ClusterFunc_Neuro_RNA(PO_ForSubcluster_6)  


PO_ForSubcluster_6 <- FindNeighbors(PO_ForSubcluster_6, k.param=5, dims=1:20)
PO_ForSubcluster_6 <- FindClusters(PO_ForSubcluster_6, resolution =1)
PO_ForSubcluster_6_V2 = subset(PO_ForSubcluster_6, idents = c(0,3))

PO_Subcluster_6 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_ForSubcluster_6_V2) & All_PREOPTIC_UMAP$UMAP_2 > 6)

PO_Rest3 = subset(All_PREOPTIC, cells = c(row.names(PO_Cluster1), row.names(PO_Subcluster_3), row.names(PO_Subcluster_4), row.names(PO_Subcluster_5), row.names(PO_Subcluster_6)), invert=T)

PO_Subcluster_6_V2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(PO_Rest3) & Fetal.combined_Data$NTS > 0)
PO_Subcluster_6_V3 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% row.names(PO_Subcluster_6_V2) & All_PREOPTIC_UMAP$UMAP_2 > 2 & All_PREOPTIC_UMAP$UMAP_1 < 4 | row.names(All_PREOPTIC_UMAP) %in% row.names(PO_Subcluster_6_V2) & All_PREOPTIC_UMAP$UMAP_2 > 8 & All_PREOPTIC_UMAP$UMAP_1 < 7 | All_PREOPTIC_UMAP$UMAP_2 > 7 & All_PREOPTIC_UMAP$UMAP_1 < 4 & row.names(All_PREOPTIC_UMAP) %notin% row.names(PO_Subcluster_5))

PO_Subcluster_6_Recombo = subset(All_PREOPTIC, cells = unique(c(row.names(PO_Subcluster_6), row.names(PO_Subcluster_6_V3))))

PO_Rest4 = subset(All_PREOPTIC, cells = c(row.names(PO_Cluster1), row.names(PO_Subcluster_3), row.names(PO_Subcluster_4), row.names(PO_Subcluster_5), row.names(PO_Subcluster_6), row.names(PO_Subcluster_6_V3)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(50)
#ClusterFunc_Neuro_RNA(PO_Rest4)  

PO_Rest4 <- FindNeighbors(PO_Rest4, k.param=50, dims=1:20)
PO_Rest4 <- FindClusters(PO_Rest4, resolution =1)
PO_ForSubcluster_7 = subset(PO_Rest4, idents = c(0))
PO_ForSubcluster_8 = subset(PO_Rest4, idents = c(2))

set.dim = 20
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(PO_ForSubcluster_7) 
#ClusterFunc_Neuro_RNA(PO_ForSubcluster_8) 


PO_ForSubcluster_7 <- FindNeighbors(PO_ForSubcluster_7, k.param=40, dims=1:20)
PO_ForSubcluster_7 <- FindClusters(PO_ForSubcluster_7, resolution =1)
PO_ForSubcluster_7_V2 = subset(PO_ForSubcluster_7, idents = c(2,0))
PO_ForSubcluster_7_V3 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_ForSubcluster_7_V2) & All_PREOPTIC_UMAP$UMAP_2 > 1.5 & All_PREOPTIC_UMAP$UMAP_1 > 3)


PO_ForSubcluster_8 <- FindNeighbors(PO_ForSubcluster_8, k.param=3, dims=1:20)
PO_ForSubcluster_8 <- FindClusters(PO_ForSubcluster_8, resolution =1)
PO_ForSubcluster_8_V2 = subset(PO_ForSubcluster_8, idents = c(2,3,6))
PO_ForSubcluster_8_Rest = subset(PO_ForSubcluster_8, idents = c(2,3,6), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(40)
#ClusterFunc_Neuro_RNA(PO_ForSubcluster_8_Rest) 


PO_Rest5 = subset(All_PREOPTIC, cells = c(row.names(PO_Cluster1), row.names(PO_Subcluster_3), row.names(PO_Subcluster_4), row.names(PO_Subcluster_5), row.names(PO_Subcluster_6), row.names(PO_Subcluster_6_V3), row.names(PO_ForSubcluster_7_V3), colnames(PO_ForSubcluster_8_V2)), invert=T)

#PO_Rest5 = RunPCA(PO_Rest5, npcs = 20)
#PO_Rest5 <- RunUMAP(PO_Rest5, dims = 1:20, spread= 5)
set.dim = 20
set.res = c(1)
set.kparam = c(15)
#ClusterFunc_Neuro_RNA(PO_Rest5)  


PO_Rest5 <- FindNeighbors(PO_Rest5, k.param=15, dims=1:20)
PO_Rest5 <- FindClusters(PO_Rest5, resolution =1)
PO_ForSubcluster_9 = subset(PO_Rest5, idents = c(0))

PO_ForSubcluster_9_Pt2 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_Rest5) & All_PREOPTIC_UMAP$UMAP_1 < 4)
PO_ForSubcluster_9_V2 = subset(PO_Rest5, cells = unique(c(colnames(PO_ForSubcluster_9), row.names(PO_ForSubcluster_9_Pt2))))
PO_ForSubcluster_9_V3 = subset(All_PREOPTIC_UMAP, row.names(All_PREOPTIC_UMAP) %in% colnames(PO_ForSubcluster_9_V2) & All_PREOPTIC_UMAP$UMAP_1 < 8)

CheckInput = row.names(PO_ForSubcluster_9_V3)
CheckUMAP(PO_Rest5)

PO_Rest6 = subset(All_PREOPTIC, cells = c(row.names(PO_Cluster1), row.names(PO_Subcluster_3), row.names(PO_Subcluster_4), row.names(PO_Subcluster_5), row.names(PO_Subcluster_6), row.names(PO_Subcluster_6_V3), row.names(PO_ForSubcluster_7_V3), colnames(PO_ForSubcluster_8_V2), row.names(PO_ForSubcluster_9_V3)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(15)
#ClusterFunc_Neuro_RNA(PO_Rest6)  


OCT_PREOPTIC_Assigns_Renamed = as.data.frame(c(rep("Preoptic_01 [PNOC/PENK]", length(colnames(PO_Subcluster_1))), 
                                 rep("Preoptic_02 [PNOC]", length(colnames(PO_Subcluster_2))),               
                                 rep("Preoptic_07", length(row.names(PO_Subcluster_3))),               
                                 rep("Preoptic_08", length(row.names(PO_Subcluster_4))),               
                                 rep("Preoptic_03 [NTS]", length(row.names(PO_Subcluster_5))),
                                 rep("Preoptic_04 [PNOC/NTS]", length(colnames(PO_Subcluster_6_Recombo))),               
                                 rep("Preoptic_05 [PNOC]", length(row.names(PO_ForSubcluster_7_V3))),               
                                 rep("Preoptic_06 [PNOC]", length(colnames(PO_ForSubcluster_8_V2))),               
                                 rep("Preoptic_09", length(row.names(PO_ForSubcluster_9_V3))),               
                                 rep("Preoptic_10", length(colnames(PO_Rest6))))) 
colnames(OCT_PREOPTIC_Assigns_Renamed) = "Pop"
row.names(OCT_PREOPTIC_Assigns_Renamed) = c(colnames(PO_Subcluster_1), #row.names(OCT_PREOPTIC_Assigns_Renamed) #OCT_PREOPTIC_Assigns_Renamed$Barcs
                                           colnames(PO_Subcluster_2),
                                           row.names(PO_Subcluster_3),
                                           row.names(PO_Subcluster_4), 
                                           row.names(PO_Subcluster_5), 
                                           colnames(PO_Subcluster_6_Recombo), 
                                           row.names(PO_ForSubcluster_7_V3), 
                                           colnames(PO_ForSubcluster_8_V2), 
                                           row.names(PO_ForSubcluster_9_V3),
                                           colnames(PO_Rest6))

#OCT_PREOPTIC_Assigns_Renamed$Dup = duplicated(OCT_PREOPTIC_Assigns_Renamed$Barcs) | duplicated(OCT_PREOPTIC_Assigns_Renamed$Barcs, fromLast = T)
#CHECK =  subset(OCT_PREOPTIC_Assigns_Renamed, OCT_PREOPTIC_Assigns_Renamed$Dup == T)
#table(CHECK$Pop)

All_PREOPTIC = AddMetaData(All_PREOPTIC, OCT_PREOPTIC_Assigns_Renamed, "OCT_PREOPTIC_Assigns_Renamed")
Idents(All_PREOPTIC) = "OCT_PREOPTIC_Assigns_Renamed"

pdf(paste("ALLFETAL_Preoptic_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_PREOPTIC, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_PREOPTIC) = "RNA"
VP = list()
for( in c("PNOC", "NTS", "PENK")){
VP[[x]] = VlnPlot(All_PREOPTIC, x, pt.size = 0)  + NoLegend()
}
pdf("PO_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### LH: DONE
```{r}
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_LH = subset(HypothalamicNeurons, idents = c("Neuron_29 [LH]", "Neuron_30 [LH]"))

#PCA 20 dims 10 spread 10
DefaultAssay(All_LH) = "integrated"
All_LH = RunPCA(All_LH, npcs = 20)
All_LH <- RunUMAP(All_LH, dims = 1:10, spread= 10)
All_LH_UMAP = as.data.frame(All_LH@reductions$umap@cell.embeddings)


LH_Cluster1 = subset(All_LH_UMAP,  All_LH_UMAP$UMAP_1 < -25)
All_LH_Rest1 = subset(All_LH, cells = row.names(LH_Cluster1), invert=T)
set.dim = 20
set.res = c(0.8)
set.kparam = c(80)
#ClusterFunc_Neuro_RNA(All_LH_Rest1)  

All_LH_Rest1 <- FindNeighbors(All_LH_Rest1, k.param=10, dims=1:20)
All_LH_Rest1 <- FindClusters(All_LH_Rest1, resolution =1)
LH_Subcluster_2 = subset(All_LH_Rest1, idents = c(1,2,3))
LH_Subcluster_2_V2 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Subcluster_2) & All_LH_UMAP$UMAP_2 < 5  | row.names(All_LH_UMAP) %in% colnames(All_LH_Rest1) & All_LH_UMAP$UMAP_2 < -3) #All PDYN +
LH_Subcluster_2_V2_Seu = subset(All_LH_Rest1, cells = row.names(LH_Subcluster_2_V2))

set.dim = 20
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(LH_Subcluster_2_V2_Seu)  

LH_Subcluster_2_V2_Seu <- FindNeighbors(LH_Subcluster_2_V2_Seu, k.param=5, dims=1:20)
LH_Subcluster_2_V2_Seu <- FindClusters(LH_Subcluster_2_V2_Seu, resolution =1)
LH_Subcluster_2_V3 = subset(LH_Subcluster_2_V2_Seu, idents = c(2))

LH_Subcluster_2_V4 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Subcluster_2_V3) & All_LH_UMAP$UMAP_2 < -20)
LH_Subcluster_2_V4_Seu = subset(All_LH_Rest1, cells = row.names(LH_Subcluster_2_V4))
LH_Cluster2 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% row.names(LH_Subcluster_2_V4) & All_LH_UMAP$UMAP_2 < -27.5 & All_LH_UMAP$UMAP_1 < 4)
LH_Cluster3 = subset(LH_Subcluster_2_V4_Seu, cells = row.names(LH_Cluster2), invert=T)

LH_Subcluster_2_V5 = subset(LH_Subcluster_2_V2_Seu, cells = c(row.names(LH_Cluster2), colnames(LH_Cluster3)), invert=T)

LH_Cluster4 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Subcluster_2_V5) & All_LH_UMAP$UMAP_2 > -12 & All_LH_UMAP$UMAP_1 < 10)

LH_Subcluster_2_V6 = subset(LH_Subcluster_2_V2_Seu, cells = c(row.names(LH_Cluster2), colnames(LH_Cluster3), row.names(LH_Cluster4)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(LH_Subcluster_2_V6)  

LH_Subcluster_2_V6 <- FindNeighbors(LH_Subcluster_2_V6, k.param=20, dims=1:20)
LH_Subcluster_2_V6 <- FindClusters(LH_Subcluster_2_V6, resolution =1)
LH_Cluster5 = subset(LH_Subcluster_2_V6, idents = c(0))

LH_Cluster5_V2 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Cluster5) & All_LH_UMAP$UMAP_2 < -10 | row.names(All_LH_UMAP) %in% colnames(LH_Subcluster_2_V6) & All_LH_UMAP$UMAP_2 < -15 )


All_LH_Rest2 = subset(All_LH, cells = c(row.names(LH_Cluster1), row.names(LH_Cluster2), colnames(LH_Cluster3), row.names(LH_Cluster4), row.names(LH_Cluster5_V2)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_LH_Rest2)  

All_LH_Rest2 <- FindNeighbors(All_LH_Rest2, k.param=5, dims=1:20)
All_LH_Rest2 <- FindClusters(All_LH_Rest2, resolution =1)
LH_Cluster6 = subset(All_LH_Rest2, idents = c(0))
LH_Cluster6_V2 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Cluster6) & All_LH_UMAP$UMAP_2 < 10)

All_LH_Rest3 = subset(All_LH, cells = c(row.names(LH_Cluster1), row.names(LH_Cluster2), colnames(LH_Cluster3), row.names(LH_Cluster4), row.names(LH_Cluster5_V2), row.names(LH_Cluster6_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20,40)
#ClusterFunc_Neuro_RNA(All_LH_Rest3)  


All_LH_Rest3 <- FindNeighbors(All_LH_Rest3, k.param=40, dims=1:20)
All_LH_Rest3 <- FindClusters(All_LH_Rest3, resolution =1)
LH_Cluster7 = subset(All_LH_Rest3, idents = c(0))
LH_Cluster7_V2 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Cluster7) & All_LH_UMAP$UMAP_2 > 18 & All_LH_UMAP$UMAP_1 < 12)

All_LH_Rest4 = subset(All_LH, cells = c(row.names(LH_Cluster1), row.names(LH_Cluster2), colnames(LH_Cluster3), row.names(LH_Cluster4), row.names(LH_Cluster5_V2), row.names(LH_Cluster6_V2), row.names(LH_Cluster7_V2)), invert=T)

set.dim = c(10,20)
set.res = c(1)
set.kparam = c(20,40)
#ClusterFunc_Neuro_RNA(All_LH_Rest4)  

All_LH_Rest4 <- FindNeighbors(All_LH_Rest4, k.param=40, dims=1:10)
All_LH_Rest4 <- FindClusters(All_LH_Rest4, resolution =1)
LH_Cluster8 = subset(All_LH_Rest4, idents = c(1))
LH_Cluster8_V2 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(LH_Cluster8) & All_LH_UMAP$UMAP_1 < 8)

All_LH_Rest5 = subset(All_LH, cells = c(row.names(LH_Cluster1), row.names(LH_Cluster2), colnames(LH_Cluster3), row.names(LH_Cluster4), row.names(LH_Cluster5_V2), row.names(LH_Cluster6_V2), row.names(LH_Cluster7_V2), row.names(LH_Cluster8_V2)), invert=T)

LH_Cluster9 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(All_LH_Rest5) & All_LH_UMAP$UMAP_2 < 2)
LH_Cluster10 = subset(All_LH_UMAP, row.names(All_LH_UMAP) %in% colnames(All_LH_Rest5) & All_LH_UMAP$UMAP_2 > 2)

set.dim = c(20)
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_LH_Rest5)  

CheckInput = row.names(LH_Cluster8_V2)
CheckUMAP(All_LH_Rest4)

OCT_LH_Assigns_Renamed = as.data.frame(c(rep("LH_01 [PMCH/TAC1]", length(row.names(LH_Cluster1))), 
                                 rep("LH_02 [PDYN/PCSK1]", length(row.names(LH_Cluster2))),               
                                 rep("LH_03 [HCRT/PDYN/PCSK1]", length(colnames(LH_Cluster3))),               
                                 rep("LH_04 [HCRT/LHX9/PDYN]", length(row.names(LH_Cluster4))),               
                                 rep("LH_05 [HCRT/LHX9/PDYN/PCSK1]", length(row.names(LH_Cluster5_V2))),
                                 rep("LH_06 [HCRT/LHX9/PDYN]", length(row.names(LH_Cluster6_V2))),               
                                 rep("LH_07 [LHX9/NPVF]", length(row.names(LH_Cluster7_V2))),               
                                 rep("LH_08 [HCRT]", length(row.names(LH_Cluster8_V2))),               
                                 rep("LH_09 [HCRT/LHX9/PCSK1]", length(row.names(LH_Cluster9))),               
                                 rep("LH_10 [HCRT/LHX9]", length(row.names(LH_Cluster10))))) 
colnames(OCT_LH_Assigns_Renamed) = "Pop"
row.names(OCT_LH_Assigns_Renamed) = c(row.names(LH_Cluster1), 
                                      row.names(LH_Cluster2), 
                                      colnames(LH_Cluster3), 
                                      row.names(LH_Cluster4), 
                                      row.names(LH_Cluster5_V2), 
                                      row.names(LH_Cluster6_V2), 
                                      row.names(LH_Cluster7_V2), 
                                      row.names(LH_Cluster8_V2), 
                                      row.names(LH_Cluster9), 
                                      row.names(LH_Cluster10))

#OCT_LH_Assigns_Renamed$Dup = duplicated(OCT_LH_Assigns_Renamed$Barcs) | duplicated(OCT_LH_Assigns_Renamed$Barcs, fromLast = T)
#CHECK =  subset(OCT_LH_Assigns_Renamed, OCT_LH_Assigns_Renamed$Dup == T)
#table(CHECK$Pop)

All_LH = AddMetaData(All_LH, OCT_LH_Assigns_Renamed, "OCT_LH_Assigns_Renamed")
Idents(All_LH) = "OCT_LH_Assigns_Renamed"

pdf(paste("ALLFETAL_LH_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_LH, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_LH) = "RNA"
VP = list()
for( in c("HCRT", "LHX9", "PMCH", "TAC1", "SST", "NPVF", "PDYN", "PCSK1")){
VP[[x]] = VlnPlot(All_LH, x, pt.size = 0)  + NoLegend()
}
pdf("LH_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### PVH: DONE
```{r}
All_PVH = subset(HypothalamicNeurons, idents = c("Neuron_15 [PVH]", "Neuron_16 [PVH]", "Neuron_17 [PVH]", "Neuron_18 [PVH]", "Neuron_19 [PVH]", "Neuron_20 [PVH]", "Neuron_21 [PVH]", "Neuron_22 [PVH]", "Neuron_23 [PVH]", "Neuron_24 [PVH]", "Neuron_25 [PVH]", "Neuron_26 [PVH]", "Neuron_27 [PVH]", "Neuron_28 [PVH]", "Neuron_29 [PVH]"))

#Use 9 - PCA 20 dims 20 spread 10
DefaultAssay(All_PVH) = "integrated"
All_PVH = RunPCA(All_PVH, npcs = 20)
All_PVH <- RunUMAP(All_PVH, dims = 1:20, spread= 10)
All_PVH_UMAP = as.data.frame(All_PVH@reductions$umap@cell.embeddings)

PVH_1 = subset(All_PVH_UMAP, All_PVH_UMAP$UMAP_1 > 18 & All_PVH_UMAP$UMAP_2 < -16.5)
PVH_2 = subset(All_PVH_UMAP, All_PVH_UMAP$UMAP_1 < -18 & All_PVH_UMAP$UMAP_2 > 16)
PVH_1_Seu = subset(All_PVH, cells = row.names(PVH_1))
PVH_2_Seu = subset(All_PVH, cells = row.names(PVH_2))


set.dim = 20
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(PVH_1_Seu)  
#ClusterFunc_Neuro_RNA(PVH_2_Seu)  

PVH_1_Seu <- FindNeighbors(PVH_1_Seu, k.param=2, dims=1:20)
PVH_1_Seu <- FindClusters(PVH_1_Seu, resolution =1)
PVH_Sub4 = subset(PVH_1_Seu, idents = c(18,20,22,24,26,11,8,6,0))
PVH_Sub4_V2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(PVH_1_Seu) & Fetal.combined_Data$MC4R > 0)

PVH_Sub4_Recombo = subset(PVH_1_Seu, cells = unique(c(colnames(PVH_Sub4), row.names(PVH_Sub4_V2))))
set.dim = 20
set.res = c(0.05)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(PVH_Sub4_Recombo) 

PVH_1_Seu_V2 = subset(PVH_1_Seu, cells = unique(c(colnames(PVH_Sub4), row.names(PVH_Sub4_V2))), invert=T)
PVH_Sub8 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_1_Seu_V2) & All_PVH_UMAP$UMAP_2 > -28 & All_PVH_UMAP$UMAP_1 > 25 )
PVH_Sub5 = subset(PVH_1_Seu_V2, cells = row.names(PVH_Sub8), invert=T)


PVH_2_Seu <- FindNeighbors(PVH_2_Seu, k.param=10, dims=1:20)
PVH_2_Seu <- FindClusters(PVH_2_Seu, resolution =1)
PVH_Sub1 = subset(PVH_2_Seu, idents = c(0))
PVH_Sub3 = subset(PVH_2_Seu, idents = c(3))
PVH_2_Seu_V2 = subset(PVH_2_Seu, idents = c(0,3), invert=T)
set.dim = 10
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(PVH_2_Seu_V2)  

PVH_2_Seu_V2 <- FindNeighbors(PVH_2_Seu_V2, k.param=3, dims=1:10)
PVH_2_Seu_V2 <- FindClusters(PVH_2_Seu_V2, resolution =1)
PVH_Sub1_Pt2 = subset(PVH_2_Seu, idents = c(2))

PVH_Sub1_Recombo = subset(PVH_2_Seu, cells = c(colnames(PVH_Sub1_Pt2), colnames(PVH_Sub1)))
PVH_Sub2 = subset(PVH_2_Seu, cells = c(colnames(PVH_Sub1_Pt2), colnames(PVH_Sub1), colnames(PVH_Sub3)), invert=T)

All_PVH_Rest1 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_Sub8)), invert=T)

PVH_6 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest1) & All_PVH_UMAP$UMAP_1 < -9 & All_PVH_UMAP$UMAP_2 < -12)
PVH_7 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest1) & All_PVH_UMAP$UMAP_1 < 2 & All_PVH_UMAP$UMAP_2 > 12)

All_PVH_Rest2 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(All_PVH_Rest2)  

All_PVH_Rest2 <- FindNeighbors(All_PVH_Rest2, k.param=10, dims=1:20)
All_PVH_Rest2 <- FindClusters(All_PVH_Rest2, resolution =1)
PVH_Sub9 = subset(All_PVH_Rest2, idents = c(0,4,7,1,9))

set.dim = 20
set.res = c(1)
set.kparam = c(10,3)
#ClusterFunc_Neuro_RNA(PVH_Sub9)  

PVH_Sub9 <- FindNeighbors(PVH_Sub9, k.param=3, dims=1:20)
PVH_Sub9 <- FindClusters(PVH_Sub9, resolution =1)
PVH_Sub9_V2 = subset(PVH_Sub9, idents = c(14, 17, 18, 16, 6, 2, 11, 10, 1, 0, 5))

set.dim = 20
set.res = c(1)
set.kparam = c(10,3)
#ClusterFunc_Neuro_RNA(PVH_Sub9_V2) 
PVH_Sub9_V3 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub9_V2) & All_PVH_UMAP$UMAP_1 > 9 & All_PVH_UMAP$UMAP_1 < 16 & All_PVH_UMAP$UMAP_2 > 8 & All_PVH_UMAP$UMAP_2 < 23)
PVH_Sub9_V3_Seu = subset(PVH_Sub9_V2, cells =  row.names(PVH_Sub9_V3))

PVH_Sub10 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub9_V2) & All_PVH_UMAP$UMAP_1 > 13 & All_PVH_UMAP$UMAP_2 > 25.5)
PVH_Sub10_Seu = subset(PVH_Sub9_V2, cells =  row.names(PVH_Sub10))


set.dim = 20
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(PVH_Sub10_Seu) 

PVH_Sub10_Seu <- FindNeighbors(PVH_Sub10_Seu, k.param=2, dims=1:20)
PVH_Sub10_Seu <- FindClusters(PVH_Sub10_Seu, resolution =1)
PVH_Sub10_V2 = subset(PVH_Sub10_Seu, idents = c(1,3,4,5,8))

set.dim = 20
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(PVH_Sub10_V2) 

All_PVH_Rest3 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_PVH_Rest3) 

CRH_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_PVH_Rest3) & Fetal.combined_Data$CRH > 0)
PVH_Sub10_Extra = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% row.names(CRH_Extras) & All_PVH_UMAP$UMAP_1 > 10 & All_PVH_UMAP$UMAP_2 > 21)

All_PVH_Rest4 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_PVH_Rest4) 

All_PVH_Rest4 <- FindNeighbors(All_PVH_Rest4, k.param=20, dims=1:20)
All_PVH_Rest4 <- FindClusters(All_PVH_Rest4, resolution =1)
PVH_Sub11 = subset(All_PVH_Rest4, idents = c(0,5))
set.dim = 20
set.res = c(1)
set.kparam = c(10,20)
#ClusterFunc_Neuro_RNA(PVH_Sub11) 

PVH_Sub11 <- FindNeighbors(PVH_Sub11, k.param=20, dims=1:20)
PVH_Sub11 <- FindClusters(PVH_Sub11, resolution =1)
PVH_Sub11_V2 = subset(PVH_Sub11, idents = c(0,2))

set.dim = 20
set.res = c(1)
set.kparam = c(3,5)
#ClusterFunc_Neuro_RNA(PVH_Sub11_V2) 


PVH_Sub11_V3 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub11_V2) & All_PVH_UMAP$UMAP_1 > 13 & All_PVH_UMAP$UMAP_2 > 2.5 & All_PVH_UMAP$UMAP_2 < 15)
PVH_Sub11_V4 = subset(PVH_Sub11_V2, cells = row.names(PVH_Sub11_V3))

set.dim = 20
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(PVH_Sub11_V4) 

PVH_Sub11_V4 <- FindNeighbors(PVH_Sub11_V4, k.param=2, dims=1:20)
PVH_Sub11_V4 <- FindClusters(PVH_Sub11_V4, resolution =1)
PVH_Sub11_V5 = subset(PVH_Sub11_V4, idents = c(6,7,15,16), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(PVH_Sub11_V5) 


All_PVH_Rest5 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra), colnames(PVH_Sub11_V5)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(40)
#ClusterFunc_Neuro_RNA(All_PVH_Rest5) 

PVH_Sub12 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest5) & All_PVH_UMAP$UMAP_2 > 15)
PVH_Sub12_Seu = subset(All_PVH, cells = row.names(PVH_Sub12))

set.dim = 20
set.res = c(1)
set.kparam = c(10,3)
#ClusterFunc_Neuro_RNA(PVH_Sub12_Seu) 


PVH_Sub12_Seu <- FindNeighbors(PVH_Sub12_Seu, k.param=3, dims=1:20)
PVH_Sub12_Seu <- FindClusters(PVH_Sub12_Seu, resolution =1)
PVH_Sub12_V2 = subset(PVH_Sub12_Seu, idents = c(2,5,8,9))
PVH_Sub13 = subset(PVH_Sub12_Seu, idents = c(2,5,8,9), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(PVH_Sub13) 


PCSK1_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(PVH_Sub13) & Fetal.combined_Data$PCSK1 > 0)
PVH_Sub12_V2_Extras = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% row.names(PCSK1_Extras) & All_PVH_UMAP$UMAP_1 < 10 & All_PVH_UMAP$UMAP_2 > 27 | row.names(All_PVH_UMAP) %in% row.names(PCSK1_Extras) & All_PVH_UMAP$UMAP_1 > 8 & All_PVH_UMAP$UMAP_1 < 12 & All_PVH_UMAP$UMAP_2 > 23 & All_PVH_UMAP$UMAP_2 < 30)

PVH_Sub12_Recombo = subset(PVH_Sub12_Seu, cells = c(colnames(PVH_Sub12_V2), row.names(PVH_Sub12_V2_Extras)))
PVH_Sub13_V2 = subset(PVH_Sub12_Seu, cells = c(colnames(PVH_Sub12_V2), row.names(PVH_Sub12_V2_Extras)), invert=T)

PVH_Sub13_V3 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub13_V2) & All_PVH_UMAP$UMAP_1 < 12 & All_PVH_UMAP$UMAP_2 < 30 )
PVH_Sub14 = subset(PVH_Sub13_V2, cells = row.names(PVH_Sub13_V3), invert=T)



All_PVH_Rest6 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra), colnames(PVH_Sub11_V5), colnames(PVH_Sub12_Recombo), row.names(PVH_Sub13_V3), colnames(PVH_Sub14)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(10)
ClusterFunc_Neuro_RNA(All_PVH_Rest6) 

All_PVH_Rest6 <- FindNeighbors(All_PVH_Rest6, k.param=10, dims=1:20)
All_PVH_Rest6 <- FindClusters(All_PVH_Rest6, resolution =1)
PVH_Sub15 = subset(All_PVH_Rest6, idents = c(2,3,6))

set.dim = 20
set.res = c(1)
set.kparam = c(10)
ClusterFunc_Neuro_RNA(PVH_Sub15) 

PVH_Sub15 <- FindNeighbors(PVH_Sub15, k.param=10, dims=1:20)
PVH_Sub15 <- FindClusters(PVH_Sub15, resolution =1)
PVH_Sub15_V2 = subset(PVH_Sub15, idents = c(1))
PVH_Sub16 = subset(PVH_Sub15, idents = c(0,3,5))

All_PVH_Rest7 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra), colnames(PVH_Sub11_V5), colnames(PVH_Sub12_Recombo), row.names(PVH_Sub13_V3), colnames(PVH_Sub14), colnames(PVH_Sub15_V2), colnames(PVH_Sub16)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(10)
ClusterFunc_Neuro_RNA(All_PVH_Rest7) 


PVH_Sub17 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest7) & All_PVH_UMAP$UMAP_1 < -16.5)
PVH_Sub18 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest7) & All_PVH_UMAP$UMAP_2 < -13 & All_PVH_UMAP$UMAP_1 < 12)

All_PVH_Rest8 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra), colnames(PVH_Sub11_V5), colnames(PVH_Sub12_Recombo), row.names(PVH_Sub13_V3), colnames(PVH_Sub14), colnames(PVH_Sub15_V2), colnames(PVH_Sub16), row.names(PVH_Sub17), row.names(PVH_Sub18)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(All_PVH_Rest8) 

All_PVH_Rest8 <- FindNeighbors(All_PVH_Rest8, k.param=20, dims=1:20)
All_PVH_Rest8 <- FindClusters(All_PVH_Rest8, resolution =1)
PVH_Sub19 = subset(All_PVH_Rest8, idents = c(0,4))
PVH_Sub19_V2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub19) & All_PVH_UMAP$UMAP_1 < 11 & All_PVH_UMAP$UMAP_2 < 8)
PVH_Sub19_pt2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest8) & All_PVH_UMAP$UMAP_1 < 2 & All_PVH_UMAP$UMAP_2 < -2)
PVH_Sub19_Recombo = subset(All_PVH_Rest8, cells = unique(c(row.names(PVH_Sub19_V2), row.names(PVH_Sub19_pt2))))


All_PVH_Rest9 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra), colnames(PVH_Sub11_V5), colnames(PVH_Sub12_Recombo), row.names(PVH_Sub13_V3), colnames(PVH_Sub14), colnames(PVH_Sub15_V2), colnames(PVH_Sub16), row.names(PVH_Sub17), row.names(PVH_Sub18), colnames(PVH_Sub19_Recombo)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(All_PVH_Rest9) 

All_PVH_Rest9 <- FindNeighbors(All_PVH_Rest9, k.param=3, dims=1:20)
All_PVH_Rest9 <- FindClusters(All_PVH_Rest9, resolution =1)
PVH_Sub20 = subset(All_PVH_Rest9, idents = c(11))
PVH_Sub20_pt2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest9) & All_PVH_UMAP$UMAP_2 > 11 & All_PVH_UMAP < 8)
PVH_Sub20_Recombo = subset(All_PVH_Rest9, cells = unique(c(colnames(PVH_Sub20), row.names(PVH_Sub20_pt2))))


All_PVH_Rest10 = subset(All_PVH, cells = c(colnames(PVH_Sub1_Recombo), colnames(PVH_Sub3), colnames(PVH_Sub2), colnames(PVH_Sub4_Recombo), colnames(PVH_Sub5), row.names(PVH_6), row.names(PVH_7), row.names(PVH_Sub8), row.names(PVH_Sub9_V3), colnames(PVH_Sub10_V2), row.names(PVH_Sub10_Extra), colnames(PVH_Sub11_V5), colnames(PVH_Sub12_Recombo), row.names(PVH_Sub13_V3), colnames(PVH_Sub14), colnames(PVH_Sub15_V2), colnames(PVH_Sub16), row.names(PVH_Sub17), row.names(PVH_Sub18), colnames(PVH_Sub19_Recombo), colnames(PVH_Sub20_Recombo)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(5)
ClusterFunc_Neuro_RNA(All_PVH_Rest10) 

All_PVH_Rest10 <- FindNeighbors(All_PVH_Rest10, k.param=5, dims=1:20)
All_PVH_Rest10 <- FindClusters(All_PVH_Rest10, resolution =1)
PVH_Sub21 = subset(All_PVH_Rest10, idents = c(1,2,6))
PVH_Sub22 = subset(All_PVH_Rest10, idents = c(3,4,5))

PVH_Sub21_V2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub21) & All_PVH_UMAP$UMAP_2 > -4)
PVH_Sub21_pt2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest10) & All_PVH_UMAP$UMAP_1 < 15 & All_PVH_UMAP$UMAP_2 > 1)
PVH_Sub21_Recombo = subset(All_PVH_Rest10, cells = unique(c(row.names(PVH_Sub21_V2), row.names(PVH_Sub21_pt2))))

PVH_Sub22_V2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(PVH_Sub22) & All_PVH_UMAP$UMAP_2 < 0)
PVH_Sub22_pt2 = subset(All_PVH_UMAP, row.names(All_PVH_UMAP) %in% colnames(All_PVH_Rest10)  & All_PVH_UMAP$UMAP_2 < -7)
PVH_Sub22_Recombo = subset(All_PVH_Rest10, cells = unique(c(row.names(PVH_Sub22_V2), row.names(PVH_Sub22_pt2))))

PVH_Sub23 = subset(All_PVH_Rest10, cells = c(colnames(PVH_Sub21_Recombo), colnames(PVH_Sub22_Recombo)), invert=T)

CheckInput = colnames(PVH_Sub23)
CheckUMAP(All_PVH_Rest10)




OCT_PVH_Assigns_Renamed = as.data.frame(c(rep("PVH_09 [SIM1/POU3F2/OTP]", length(colnames(PVH_Sub1_Recombo))), 
                                 rep("PVH_10 [SIM1/POU3F2/OTP]", length(colnames(PVH_Sub3))),               
                                 rep("PVH_15 [SIM1/POU3F2/OTP/NHLH2]", length(colnames(PVH_Sub2))),               
                                 rep("PVH_03 [SIM1/POU3F2/OTP/PCSK1/OXT/MC4R/SST]", length(colnames(PVH_Sub4_Recombo))),               
                                 rep("PVH_04 [SIM1/POU3F2/OTP/PCSK1/OXT]", length(colnames(PVH_Sub5))),
                                 rep("PVH_16 [SIM1/OTP/TRH]", length(row.names(PVH_6))),               
                                 rep("PVH_11 [SIM1/POU3F2/OTP]", length(row.names(PVH_7))),               
                                 rep("PVH_05 [SIM1/POU3F2/OTP/PCSK1/SST]", length(row.names(PVH_Sub8))),               
                                 rep("PVH_20 [POU3F2/OTP/PCSK1/SST/CRH]", length(row.names(PVH_Sub9_V3))),               
                                 rep("PVH_19 [POU3F2/SST/AVP/CRH]", length(colnames(PVH_Sub10_V2))),               
                                 rep("PVH_19 [POU3F2/SST/AVP/CRH]", length(row.names(PVH_Sub10_Extra))),               
                                 rep("PVH_07 [SIM1/POU3F2/OTP/PCSK1/AVP]", length(colnames(PVH_Sub11_V5))),               
                                 rep("PVH_06 [SIM1/POU3F2/OTP/PCSK1/SST]", length(colnames(PVH_Sub12_Recombo))),               
                                 rep("PVH_14 [SIM1/POU3F2/OTP/NHLH2]", length(row.names(PVH_Sub13_V3))),               
                                 rep("PVH_22 [POU3F2/OTP/SST]", length(colnames(PVH_Sub14))),               
                                 rep("PVH_23 [NHLH2]", length(colnames(PVH_Sub15_V2))),               
                                 rep("PVH_13 [SIM1/POU3F2/OTP/NHLH2]", length(colnames(PVH_Sub16))),               
                                 rep("PVH_17 [SIM1/NHLH2/TRH]", length(row.names(PVH_Sub17))),               
                                 rep("PVH_18 [SIM1/POU3F2/OTP]", length(row.names(PVH_Sub18))),               
                                 rep("PVH_12 [SIM1/POU3F2/OTP]", length(colnames(PVH_Sub19_Recombo))),               
                                 rep("PVH_08 [SIM1/POU3F2/OTP/PCSK1/TRH]", length(colnames(PVH_Sub20_Recombo))),               
                                 rep("PVH_21 [POU3F2/OTP/PCSK1]", length(colnames(PVH_Sub21_Recombo))),               
                                 rep("PVH_02 [SIM1/POU3F2/OTP/PCSK1]", length(colnames(PVH_Sub22_Recombo))),                  
                                 rep("PVH_01 [SIM1/POU3F2/OTP/PCSK1]", length(colnames(PVH_Sub23))))) 
colnames(OCT_PVH_Assigns_Renamed) = "Pop"
row.names(OCT_PVH_Assigns_Renamed) = c(colnames(PVH_Sub1_Recombo), 
                                            colnames(PVH_Sub3), 
                                            colnames(PVH_Sub2), 
                                            colnames(PVH_Sub4_Recombo), 
                                            colnames(PVH_Sub5), 
                                            row.names(PVH_6), 
                                            row.names(PVH_7), 
                                            row.names(PVH_Sub8), 
                                            row.names(PVH_Sub9_V3), 
                                            colnames(PVH_Sub10_V2), 
                                            row.names(PVH_Sub10_Extra), 
                                            colnames(PVH_Sub11_V5), 
                                            colnames(PVH_Sub12_Recombo), 
                                            row.names(PVH_Sub13_V3), 
                                            colnames(PVH_Sub14), 
                                            colnames(PVH_Sub15_V2), 
                                            colnames(PVH_Sub16), 
                                            row.names(PVH_Sub17), 
                                            row.names(PVH_Sub18), 
                                            colnames(PVH_Sub19_Recombo), 
                                            colnames(PVH_Sub20_Recombo), 
                                            colnames(PVH_Sub21_Recombo),
                                            colnames(PVH_Sub22_Recombo),
                                            colnames(PVH_Sub23))

#OCT_PVH_Assigns_Renamed$Dup = duplicated(OCT_PVH_Assigns_Renamed$Barcs) | duplicated(OCT_PVH_Assigns_Renamed$Barcs, fromLast = T)
#CHECK =  subset(OCT_PVH_Assigns_Renamed, OCT_PVH_Assigns_Renamed$Dup == T)
#table(CHECK$Pop)


OCT_PVH_Assigns_Renamed$Pop = factor(OCT_PVH_Assigns_Renamed$Pop, levels = sort(unique(OCT_PVH_Assigns_Renamed$Pop)))


All_PVH = AddMetaData(All_PVH, OCT_PVH_Assigns_Renamed, "OCT_PVH_Assigns_Renamed")
Idents(All_PVH) = "OCT_PVH_Assigns_Renamed"

pdf(paste("ALLFETAL_PVH_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_PVH, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_PVH) = "RNA"
VP = list()
for( in c("SIM1", "POU3F2", "OTP", "PCSK1", "NHLH2", "OXT", "MC4R", "SST", "AVP", "CRH", "TRH" ,"NPPA")){
VP[[x]] = VlnPlot(All_PVH, x, pt.size = 0)  + NoLegend()
}
pdf("PVH_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### ARCUATE: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_ARCUATE = subset(HypothalamicNeurons, idents = c("Neuron_04 [ARC]", "Neuron_05 [ARC]", "Neuron_06 [ARC]", "Neuron_07 [ARC]", "Neuron_08 [ARC]", "Neuron_09 [ARC]", "Neuron_10 [ARC]", "Neuron_11 [ARC]", "Neuron_12 [ARC]", "Neuron_13 [ARC]", "Neuron_14 [ARC]", "Neuron_15 [ARC]"))

#PCA 40 dims 40 spread 10

DefaultAssay(All_ARCUATE) = "integrated"
All_ARCUATE = RunPCA(All_ARCUATE, npcs = 40)
All_ARCUATE <- RunUMAP(All_ARCUATE, dims = 1:40, spread= 10)
All_ARCUATE_UMAP = as.data.frame(All_ARCUATE@reductions$umap@cell.embeddings)

ARC_1 = subset(All_ARCUATE_UMAP, All_ARCUATE_UMAP$UMAP_1 > 30 )
ARC_2 = subset(All_ARCUATE_UMAP, All_ARCUATE_UMAP$UMAP_1 > 5 & All_ARCUATE_UMAP$UMAP_2 > 15 & All_ARCUATE_UMAP$UMAP_2 < 25)
ARC_3 = subset(All_ARCUATE_UMAP, All_ARCUATE_UMAP$UMAP_1 < -8 & All_ARCUATE_UMAP$UMAP_2 < -5 | All_ARCUATE_UMAP$UMAP_1 < -4 & All_ARCUATE_UMAP$UMAP_2 < -10)
ARC_3_Seu = subset(All_ARCUATE, cells = row.names(ARC_3))

set.dim = c(20,30,40)
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(ARC_3_Seu)

ARC_3_Seu <- FindNeighbors(ARC_3_Seu, k.param=5, dims=1:20)
ARC_3_Seu <- FindClusters(ARC_3_Seu, resolution =1)
ARC_3_V2 = subset(ARC_3_Seu, idents = c(1,2,4))
ARC_3_V3 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_3_V2) & All_ARCUATE_UMAP$UMAP_1 < -15 & All_ARCUATE_UMAP$UMAP_2 > -30 & All_ARCUATE_UMAP$UMAP_2 < -10)

ARC_4 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_3_Seu) & row.names(All_ARCUATE_UMAP) %notin% row.names(ARC_3_V3) & All_ARCUATE_UMAP$UMAP_2 < -25)

ARC_5 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_3_Seu) & row.names(All_ARCUATE_UMAP) %notin% row.names(ARC_3_V3) & All_ARCUATE_UMAP$UMAP_2 > -25)


set.dim = c(10)
set.res = c(0.1)
set.kparam = c(50)
#ClusterFunc_Neuro_RNA(ARC_3_V2)


All_ARCUATE_Rest1 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_ARCUATE_Rest1)  

All_ARCUATE_Rest1 <- FindNeighbors(All_ARCUATE_Rest1, k.param=5, dims=1:10)
All_ARCUATE_Rest1 <- FindClusters(All_ARCUATE_Rest1, resolution =1)
ARC_6 = subset(All_ARCUATE_Rest1, idents = c(0))
ARC_7 = subset(All_ARCUATE_Rest1, idents = c(4))

set.dim = 10
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ARC_6)  
#ClusterFunc_Neuro_RNA(ARC_7)  

ARC_6 <- FindNeighbors(ARC_6, k.param=3, dims=1:10)
ARC_6 <- FindClusters(ARC_6, resolution =1)
ARC_6_V2 = subset(ARC_6, idents = c(0,5,8), invert=T)
POMC_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_ARCUATE_Rest1) & Fetal.combined_Data$POMC > 0)
ARC_6_V3 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_6_V2) & All_ARCUATE_UMAP$UMAP_2 > 20  | row.names(All_ARCUATE_UMAP) %in% row.names(POMC_Extras) & All_ARCUATE_UMAP$UMAP_2 > 20 | row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest1) & All_ARCUATE_UMAP$UMAP_1 < -2 & All_ARCUATE_UMAP$UMAP_2 > 28)

AGRP_NPY_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_ARCUATE_Rest1) & Fetal.combined_Data$AGRP > 0 | row.names(Fetal.combined_Data) %in% colnames(All_ARCUATE_Rest1) & Fetal.combined_Data$NPY > 0)

ARC_7_V2 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_7) & All_ARCUATE_UMAP$UMAP_2 < 20 & All_ARCUATE_UMAP$UMAP_1 < -10 | row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest1) & All_ARCUATE_UMAP$UMAP_1 < -20 | row.names(All_ARCUATE_UMAP) %in% row.names(AGRP_NPY_Extras) & All_ARCUATE_UMAP$UMAP_1 < -10  & All_ARCUATE_UMAP$UMAP_2 < 25)
ARC_7_V2_Seu = subset(All_ARCUATE_Rest1, cells = row.names(ARC_7_V2))

set.dim = 10
set.res = c(1)
set.kparam = c(3,5)
#ClusterFunc_Neuro_RNA(ARC_7_V2_Seu) 

ARC_7_V2_Seu <- FindNeighbors(ARC_7_V2_Seu, k.param=5, dims=1:10)
ARC_7_V2_Seu <- FindClusters(ARC_7_V2_Seu, resolution =1)
ARC_7_V3 = subset(ARC_7_V2_Seu, idents = c(0,4))
PCSK1_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ARC_7_V2_Seu) & Fetal.combined_Data$PCSK1 > 0)

ARC_7_V4 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_7_V3) & All_ARCUATE_UMAP$UMAP_1 < -19  | row.names(All_ARCUATE_UMAP) %in% row.names(PCSK1_Extras) &  All_ARCUATE_UMAP$UMAP_2 < 12 | row.names(All_ARCUATE_UMAP) %in% colnames(ARC_7_V2_Seu) & All_ARCUATE_UMAP$UMAP_1 < -20 & All_ARCUATE_UMAP$UMAP_2 < 7)

ARC_8 = subset(ARC_7_V2_Seu, cells = row.names(ARC_7_V4), invert=T)

All_ARCUATE_Rest2 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5), row.names(ARC_6_V3), row.names(ARC_7_V4), colnames(ARC_8)), invert=T)


ARC_9 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest2) & All_ARCUATE_UMAP$UMAP_2 > 30)

All_ARCUATE_Rest3 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5), row.names(ARC_6_V3), row.names(ARC_7_V4), colnames(ARC_8), row.names(ARC_9)), invert=T)
set.dim = 10
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_ARCUATE_Rest3)  

ARC_10 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest2) & All_ARCUATE_UMAP$UMAP_2 < -6 | row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest2) & All_ARCUATE_UMAP$UMAP_2 < -3 & All_ARCUATE_UMAP$UMAP_1 > 8)
ARC_10_Seu = subset(All_ARCUATE_Rest3, cells = row.names(ARC_10))
set.dim = 10
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(ARC_10_Seu) 

ARC_10_Seu <- FindNeighbors(ARC_10_Seu, k.param=20, dims=1:10)
ARC_10_Seu <- FindClusters(ARC_10_Seu, resolution =1)
ARC_10_V2 = subset(ARC_10_Seu, idents = c(0,1,3))
set.dim = 10
set.res = c(1)
set.kparam = c(4)
#ClusterFunc_Neuro_RNA(ARC_10_V2) 
ARC_10_V2 <- FindNeighbors(ARC_10_V2, k.param=4, dims=1:10)
ARC_10_V2 <- FindClusters(ARC_10_V2, resolution =1)
ARC_10_V3 = subset(ARC_10_V2, idents = c(0,4,6))

ARC_10_V4 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_10_V3) & All_ARCUATE_UMAP$UMAP_2 < -27)
ARC_10_V4_Seu = subset(All_ARCUATE_Rest3, cells = row.names(ARC_10_V4))

set.dim = 10
set.res = c(1)
set.kparam = c(2)
#ClusterFunc_Neuro_RNA(ARC_10_V4_Seu) 

ARC_10_V4_Seu <- FindNeighbors(ARC_10_V4_Seu, k.param=2, dims=1:10)
ARC_10_V4_Seu <- FindClusters(ARC_10_V4_Seu, resolution =1)
ARC_10_V5 = subset(ARC_10_V4_Seu, idents = c(2,6), invert=T)

GHRH_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ARC_10) & Fetal.combined_Data$GHRH > 0)

ARC_10_V6 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_10_V5)  &  All_ARCUATE_UMAP$UMAP_1 < 20 | row.names(All_ARCUATE_UMAP) %in% row.names(PCSK1_Extras) &  All_ARCUATE_UMAP$UMAP_2 < -22 )

ARC_11 = subset(ARC_10_Seu, cells = row.names(ARC_10_V6), invert=T)
set.dim = 10
set.res = c(2)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ARC_11) 

ARC_11 <- FindNeighbors(ARC_11, k.param=3, dims=1:10)
ARC_11 <- FindClusters(ARC_11, resolution =2)
ARC_11_V2 = subset(ARC_11, idents = c(5,19,2))
GSX1_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ARC_11) & Fetal.combined_Data$GSX1 > 0)

ARC_11_V3= subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_11_V2)  &  All_ARCUATE_UMAP$UMAP_2 < -22 | row.names(All_ARCUATE_UMAP) %in% row.names(GSX1_Extras) &  All_ARCUATE_UMAP$UMAP_2 < -20 | row.names(All_ARCUATE_UMAP) %in% colnames(ARC_11)  &  All_ARCUATE_UMAP$UMAP_2 < -25 & All_ARCUATE_UMAP$UMAP_1 < 18 )

ARC_11_V3_Seu = subset(ARC_11, cells = row.names(ARC_11_V3))

set.dim = 10
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(ARC_11_V3_Seu) 



ARC_12 = subset(ARC_10_Seu, cells = c(row.names(ARC_10_V6), row.names(ARC_11_V3)), invert=T)
set.dim = 10
set.res = c(1)
set.kparam = c(5,10)
#ClusterFunc_Neuro_RNA(ARC_12) 

ARC_12 <- FindNeighbors(ARC_12, k.param=5, dims=1:10)
ARC_12 <- FindClusters(ARC_12, resolution =1)
ARC_12_V2 = subset(ARC_12, idents = c(7,8))
HMX2_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ARC_12) & Fetal.combined_Data$HMX2 > 0)

#ARC_12_V3= subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_12_V2)  &  All_ARCUATE_UMAP$UMAP_2 < -12 | row.names(All_ARCUATE_UMAP) %in% row.names(GSX1_Extras) &  All_ARCUATE_UMAP$UMAP_2 < -10  | row.names(All_ARCUATE_UMAP) %in% colnames(ARC_12)  &  All_ARCUATE_UMAP$UMAP_2 < -18 & All_ARCUATE_UMAP$UMAP_1 < 13) #72


ARC_12_V3= subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_12_V2)  &  All_ARCUATE_UMAP$UMAP_2 < -12 | row.names(All_ARCUATE_UMAP) %in% row.names(HMX2_Extras) &  All_ARCUATE_UMAP$UMAP_2 < -10  | row.names(All_ARCUATE_UMAP) %in% colnames(ARC_12)  &  All_ARCUATE_UMAP$UMAP_2 < -18 & All_ARCUATE_UMAP$UMAP_1 < 13) #82
ARC_12_V3_Seu = subset(ARC_12, cells = row.names(ARC_12_V3))


CheckInput = row.names(ARC_12_V3)
CheckUMAP(ARC_12)

set.dim = 10
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(ARC_12_V3_Seu) 

ARC_13 = subset(ARC_10_Seu, cells = c(row.names(ARC_10_V6), row.names(ARC_11_V3), row.names(ARC_12_V3)), invert=T)
set.dim = 20
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(ARC_13) 

ARC_13 <- FindNeighbors(ARC_13, k.param=20, dims=1:20)
ARC_13 <- FindClusters(ARC_13, resolution =1)
ARC_13_V2 = subset(ARC_13, idents = c(1,3))
NKX_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ARC_13) & Fetal.combined_Data$`NKX2-1` > 0)

ARC_13_V3= subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_13_V2)  &  All_ARCUATE_UMAP$UMAP_2 > -20 | row.names(All_ARCUATE_UMAP) %in% row.names(NKX_Extras) &  All_ARCUATE_UMAP$UMAP_2 > -20  | row.names(All_ARCUATE_UMAP) %in% colnames(ARC_13)  &  All_ARCUATE_UMAP$UMAP_2 > -12)

ARC_13_V3_Seu = subset(ARC_13, cells = row.names(ARC_13_V3))

set.dim = 10
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(ARC_13_V3_Seu) 

ARC_14 = subset(ARC_10_Seu, cells = c(row.names(ARC_10_V6), row.names(ARC_11_V3), row.names(ARC_12_V3), row.names(ARC_13_V3)), invert=T)

All_ARCUATE_Rest4 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5), row.names(ARC_6_V3), row.names(ARC_7_V4), colnames(ARC_8), row.names(ARC_9), row.names(ARC_10_V6), row.names(ARC_11_V3), row.names(ARC_12_V3), row.names(ARC_13_V3), colnames(ARC_14)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_ARCUATE_Rest4) 

ARC_15 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest4) & All_ARCUATE_UMAP$UMAP_2 < 8 & All_ARCUATE_UMAP$UMAP_1 > 9)


NPY1R_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_ARCUATE_Rest4) & Fetal.combined_Data$NPY1R > 0)

ARC_16 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest4) & All_ARCUATE_UMAP$UMAP_2 < 10 & All_ARCUATE_UMAP$UMAP_1 < -12 | row.names(All_ARCUATE_UMAP) %in% row.names(NPY1R_Extras) & All_ARCUATE_UMAP$UMAP_2 < 15 & All_ARCUATE_UMAP$UMAP_1 < -8)

ARC_17 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest4) & All_ARCUATE_UMAP$UMAP_2 > 20 & All_ARCUATE_UMAP$UMAP_1 < -10 | row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest4) & All_ARCUATE_UMAP$UMAP_2 > 15 & All_ARCUATE_UMAP$UMAP_1 > -17 & All_ARCUATE_UMAP$UMAP_1 < -10 | row.names(All_ARCUATE_UMAP) %in% row.names(NPY1R_Extras)& All_ARCUATE_UMAP$UMAP_2 > 15 & All_ARCUATE_UMAP$UMAP_1 < -8)

All_ARCUATE_Rest5 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5), row.names(ARC_6_V3), row.names(ARC_7_V4), colnames(ARC_8), row.names(ARC_9), row.names(ARC_10_V6), row.names(ARC_11_V3), row.names(ARC_12_V3), row.names(ARC_13_V3), colnames(ARC_14), row.names(ARC_15), row.names(ARC_16), row.names(ARC_17)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_ARCUATE_Rest5) 


All_ARCUATE_Rest5 <- FindNeighbors(All_ARCUATE_Rest5, k.param=5, dims=1:10)
All_ARCUATE_Rest5 <- FindClusters(All_ARCUATE_Rest5, resolution =1)
ARC_18 = subset(All_ARCUATE_Rest5, idents = c(2))
PCSK1_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_ARCUATE_Rest5) & Fetal.combined_Data$PCSK1 > 0)

ARC_18_V2= subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_18)  &  All_ARCUATE_UMAP$UMAP_2 < 5  &  All_ARCUATE_UMAP$UMAP_1 > -5  &  All_ARCUATE_UMAP$UMAP_1 < 5  | row.names(All_ARCUATE_UMAP) %in% row.names(PCSK1_Extras) &  All_ARCUATE_UMAP$UMAP_2 < 9 &  All_ARCUATE_UMAP$UMAP_1 > -6  &  All_ARCUATE_UMAP$UMAP_1 < 7 )

All_ARCUATE_Rest6 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5), row.names(ARC_6_V3), row.names(ARC_7_V4), colnames(ARC_8), row.names(ARC_9), row.names(ARC_10_V6), row.names(ARC_11_V3), row.names(ARC_12_V3), row.names(ARC_13_V3), colnames(ARC_14), row.names(ARC_15), row.names(ARC_16), row.names(ARC_17), row.names(ARC_18_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_ARCUATE_Rest6) 

ARC_19 = subset(All_ARCUATE_UMAP,  row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest6) &  All_ARCUATE_UMAP$UMAP_2 < 4 &  All_ARCUATE_UMAP$UMAP_1 > 0 )

ARC_20 = subset(All_ARCUATE_UMAP,  row.names(All_ARCUATE_UMAP) %in% colnames(All_ARCUATE_Rest6) &  All_ARCUATE_UMAP$UMAP_2 < 7 &  All_ARCUATE_UMAP$UMAP_1 < 0 )

All_ARCUATE_Rest7 = subset(All_ARCUATE, cells = c(row.names(ARC_1), row.names(ARC_2), row.names(ARC_3_V3), row.names(ARC_4), row.names(ARC_5), row.names(ARC_6_V3), row.names(ARC_7_V4), colnames(ARC_8), row.names(ARC_9), row.names(ARC_10_V6), row.names(ARC_11_V3), row.names(ARC_12_V3), row.names(ARC_13_V3), colnames(ARC_14), row.names(ARC_15), row.names(ARC_16), row.names(ARC_17), row.names(ARC_18_V2), row.names(ARC_19), row.names(ARC_20)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_ARCUATE_Rest7) 

ReclustForLEPR = subset(All_ARCUATE, cells = c(row.names(ARC_6_V3), row.names(ARC_17)))
set.dim = 10
set.res = c(1)
set.kparam = c(3,10)
#ClusterFunc_Neuro_RNA(ReclustForLEPR) 

ReclustForLEPR <- FindNeighbors(ReclustForLEPR, k.param=5, dims=1:10)
ReclustForLEPR <- FindClusters(ReclustForLEPR, resolution =1)
ARC_21 = subset(ReclustForLEPR, idents = c(1))
LEPR_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ReclustForLEPR) & Fetal.combined_Data$LEPR > 0)

ARC_21_V2 = subset(All_ARCUATE_UMAP, row.names(All_ARCUATE_UMAP) %in% colnames(ARC_21) & All_ARCUATE_UMAP$UMAP_2 < 30 & All_ARCUATE_UMAP$UMAP_2 > 20  | row.names(All_ARCUATE_UMAP) %in% row.names(LEPR_Extras) & All_ARCUATE_UMAP$UMAP_2 < 25 & All_ARCUATE_UMAP$UMAP_2 > 20)
ARC_21_V2_Seu = subset(ReclustForLEPR, cells = row.names(ARC_21_V2))
set.dim = 10
set.res = c(0.1)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(ARC_21_V2_Seu) 

ARC_21_V2_Seu <- FindNeighbors(ARC_21_V2_Seu, k.param=3, dims=1:10)
ARC_21_V2_Seu <- FindClusters(ARC_21_V2_Seu, resolution =1)
ARC_21_V3 = subset(ARC_21_V2_Seu, idents = c(3), invert=T)

ARC_6_Cleaned = subset(ReclustForLEPR, cells = unique(c(colnames(ARC_21_V3),  row.names(ARC_17))), invert=T)
ARC_17_Cleaned = subset(ReclustForLEPR, cells = unique(c(colnames(ARC_21_V3),  row.names(ARC_6_V3))), invert=T)

#ClusterFunc_Neuro_RNA(ARC_6_Cleaned) 
#ClusterFunc_Neuro_RNA(ARC_17_Cleaned) 


CheckInput = colnames(ARC_21)
CheckUMAP(ReclustForLEPR)

OCT_ARC_Assigns_Renamed = as.data.frame(c(rep("ARC_13 [NKX2-1/TRH]", length(row.names(ARC_1))), 
                                 rep("ARC_07 [TBX3/NHLH2]", length(row.names(ARC_2))),   
                                 rep("ARC_14 [OTP]", length(row.names(ARC_3_V3))),               
                                 rep("ARC_15 [OTP]", length(row.names(ARC_4))),               
                                 rep("ARC_09 [OTP/NPY]", length(row.names(ARC_5))),
                                 rep("ARC_02 [TBX3/POMC/NHLH2]", length(colnames(ARC_6_Cleaned))),               
                                 rep("ARC_05 [TBX3/PCSK1/OTP/AGRP/NPY]", length(row.names(ARC_7_V4))),               
                                 rep("ARC_06 [TBX3/OTP/AGRP/NPY]", length(colnames(ARC_8))),               
                                 rep("ARC_12 [NKX2-1]", length(row.names(ARC_9))),               
                                 rep("ARC_22 [GHRH/HMX2/DLX1/DLX2]", length(row.names(ARC_10_V6))),               
                                 rep("ARC_21 [GSX1/HMX2/DLX1/DLX2]", length(row.names(ARC_11_V3))),               
                                 rep("ARC_20 [HMX2/DLX1/DLX2]", length(row.names(ARC_12_V3))),               
                                 rep("ARC_11 [NKX2-1]", length(row.names(ARC_13_V3))),               
                                 rep("ARC_19 [DLX1/DLX2]", length(colnames(ARC_14))),               
                                 rep("ARC_18 [NPAS4]", length(row.names(ARC_15))),               
                                 rep("ARC_16 [NPY1R]", length(row.names(ARC_16))),               
                                 rep("ARC_04 [TBX3/NPY1R]", length(colnames(ARC_17_Cleaned))),                    
                                 rep("ARC_17 [PCSK1]", length(row.names(ARC_18_V2))),                    
                                 rep("ARC_10 [NKX2-1]", length(row.names(ARC_19))),                    
                                 rep("ARC_08 [TBX3/NKX2-1]", length(row.names(ARC_20))),   
                                 rep("ARC_03 [TBX3/POMC/LEPR/NPY1R]", length(colnames(ARC_21_V3))),   
                                 rep("ARC_01 [TBX3]", length(colnames(All_ARCUATE_Rest7))))) 
colnames(OCT_ARC_Assigns_Renamed) = "Pop"
row.names(OCT_ARC_Assigns_Renamed) = c(row.names(ARC_1), # #OCT_ARC_Assigns_Renamed$Barcs
                                       row.names(ARC_2),  
                                       row.names(ARC_3_V3), 
                                       row.names(ARC_4), 
                                       row.names(ARC_5), 
                                       colnames(ARC_6_Cleaned), 
                                       row.names(ARC_7_V4), 
                                       colnames(ARC_8), 
                                       row.names(ARC_9), 
                                       row.names(ARC_10_V6), 
                                       row.names(ARC_11_V3), 
                                       row.names(ARC_12_V3), 
                                       row.names(ARC_13_V3),
                                       colnames(ARC_14),
                                       row.names(ARC_15), 
                                       row.names(ARC_16),
                                       colnames(ARC_17_Cleaned),
                                       row.names(ARC_18_V2),
                                       row.names(ARC_19),
                                       row.names(ARC_20),
                                       colnames(ARC_21_V3),
                                       colnames(All_ARCUATE_Rest7))


OCT_ARC_Assigns_Renamed$Pop = factor(OCT_ARC_Assigns_Renamed$Pop, levels = c(sort(unique(OCT_ARC_Assigns_Renamed$Pop))))
 



All_ARCUATE = AddMetaData(All_ARCUATE, OCT_ARC_Assigns_Renamed, "OCT_ARC_Assigns_Renamed")
Idents(All_ARCUATE) = "OCT_ARC_Assigns_Renamed"

pdf(paste("ALLFETAL_ARC_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_ARCUATE, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_ARCUATE) = "RNA"
VP = list()
for( in c("TBX3", "PCSK1", "OTP", "POMC", "AGRP", "NPY", "LEPR", "NHLH2",  "NPY1R", "TRH", "GHRH", "SOX3", "NKX2-1", "GSX1", "HMX2", "DLX1", "DLX2", "NPAS4")){
VP[[x]] = VlnPlot(All_ARCUATE, x, pt.size = 0)  + NoLegend()
}
pdf("ARC_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()


```

### MN: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_MN = subset(HypothalamicNeurons, idents = c("Neuron_34 [MN]", "Neuron_35 [MN]", "Neuron_36 [MN]"))


#PCA 30 dims 10 spread 10

DefaultAssay(All_MN) = "integrated"
All_MN = RunPCA(All_MN, npcs = 30)
All_MN <- RunUMAP(All_MN, dims = 1:10, spread= 10)
All_MN_UMAP = as.data.frame(All_MN@reductions$umap@cell.embeddings)

set.dim = 20
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_MN) 


All_MN <- FindNeighbors(All_MN, k.param=10, dims=1:20)
All_MN <- FindClusters(All_MN, resolution =1)
MN_1 = subset(All_MN, idents = c(4))
MN_1_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_1) & All_MN_UMAP$UMAP_2 > 27 |  row.names(All_MN_UMAP) %in% colnames(All_MN) & All_MN_UMAP$UMAP_2 > 28)
MN_2 = subset(All_MN, idents = c(2,9))
MN_2_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_2) & All_MN_UMAP$UMAP_1 > 19 |  row.names(All_MN_UMAP) %in% colnames(All_MN) & All_MN_UMAP$UMAP_1 > 25)
MN_2_V2_Seu = subset(All_MN, cells = row.names(MN_2_V2))
set.dim = 20
set.res = c(1)
set.kparam = c(10,3)
#ClusterFunc_Neuro_RNA(MN_2_V2_Seu)

MN_2_V2_Seu <- FindNeighbors(MN_2_V2_Seu, k.param=5, dims=1:20)
MN_2_V2_Seu <- FindClusters(MN_2_V2_Seu, resolution =1)
MN_3 = subset(MN_2_V2_Seu, idents = c(1,4), invert=T)
MN_3_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_3) & All_MN_UMAP$UMAP_1 < 42)
MN_2_V3 = subset(MN_2_V2_Seu, cells = row.names(MN_3_V2), invert=T)

All_MN_Rest1 = subset(All_MN, cells = c(row.names(MN_3_V2), colnames(MN_2_V3), row.names(MN_1_V2)), invert=T)

set.dim = 20
set.res = c(2)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_MN_Rest1)

All_MN_Rest1 <- FindNeighbors(All_MN_Rest1, k.param=5, dims=1:20)
All_MN_Rest1 <- FindClusters(All_MN_Rest1, resolution =2)
MN_4 = subset(All_MN_Rest1, idents = c(0,10))
set.dim = 30
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(MN_4)

MN_4_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_4) & All_MN_UMAP$UMAP_2 > 3 & All_MN_UMAP$UMAP_1 > -15 & All_MN_UMAP$UMAP_1 < 10 |  row.names(All_MN_UMAP) %in% colnames(All_MN_Rest1) & All_MN_UMAP$UMAP_2 > 8 & All_MN_UMAP$UMAP_1 > -10 & All_MN_UMAP$UMAP_1 < 4 |  row.names(All_MN_UMAP) %in% colnames(All_MN_Rest1) & All_MN_UMAP$UMAP_2 > 22 )
MN_4_V_Seu = subset(All_MN_Rest1, cells = row.names(MN_4_V2))
set.dim = 30
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(MN_4_V_Seu)

All_MN_Rest2 = subset(All_MN, cells = c(row.names(MN_3_V2), colnames(MN_2_V3), row.names(MN_1_V2), row.names(MN_4_V2)), invert=T)
set.dim = 30
set.res = c(1)
set.kparam = c(10,5)
#ClusterFunc_Neuro_RNA(All_MN_Rest2)


All_MN_Rest2 <- FindNeighbors(All_MN_Rest2, k.param=10, dims=1:30)
All_MN_Rest2 <- FindClusters(All_MN_Rest2, resolution =1)
MN_5 = subset(All_MN_Rest2, idents = c(5))
MN_5_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_5) & All_MN_UMAP$UMAP_2 < -25) 
MN_6 = subset(All_MN_Rest2, idents = c(2))
MN_6_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_6) & All_MN_UMAP$UMAP_1 < -16 | row.names(All_MN_UMAP) %in% colnames(All_MN_Rest2) & All_MN_UMAP$UMAP_1 < -20 & All_MN_UMAP$UMAP_2 > 0) 

All_MN_Rest3 = subset(All_MN, cells = c(row.names(MN_3_V2), colnames(MN_2_V3), row.names(MN_1_V2), row.names(MN_4_V2), row.names(MN_5_V2), row.names(MN_6_V2)), invert=T)
set.dim = 30
set.res = c(2)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_MN_Rest3)


All_MN_Rest3 <- FindNeighbors(All_MN_Rest3, k.param=10, dims=1:30)
All_MN_Rest3 <- FindClusters(All_MN_Rest3, resolution =2)
MN_7 = subset(All_MN_Rest3, idents = c(2))
MN_7_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_7) & All_MN_UMAP$UMAP_1 > 5 & All_MN_UMAP$UMAP_2 > -2 | row.names(All_MN_UMAP) %in% colnames(All_MN_Rest3) & All_MN_UMAP$UMAP_1 > 8 & All_MN_UMAP$UMAP_2 > -2 ) 

MN_6_pt2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(All_MN_Rest3) & All_MN_UMAP$UMAP_1 < 0 & All_MN_UMAP$UMAP_2 > 14 ) 

All_MN_Rest4 = subset(All_MN, cells = c(row.names(MN_3_V2), colnames(MN_2_V3), row.names(MN_1_V2), row.names(MN_4_V2), row.names(MN_5_V2), row.names(MN_6_V2), row.names(MN_6_pt2), row.names(MN_7_V2)), invert=T)

set.dim = 30
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_MN_Rest4)

All_MN_Rest4 <- FindNeighbors(All_MN_Rest4, k.param=5, dims=1:30)
All_MN_Rest4 <- FindClusters(All_MN_Rest4, resolution =1)
MN_4_PT2 = subset(All_MN_Rest4, idents = c(0))
FOXP_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_MN_Rest4) & Fetal.combined_Data$FOXP1 > 0 | row.names(Fetal.combined_Data) %in% colnames(All_MN_Rest4) & Fetal.combined_Data$FOXP2 > 0)

MN_4_PT2_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_4_PT2) & All_MN_UMAP$UMAP_2 > 2 | row.names(All_MN_UMAP) %in% colnames(All_MN_Rest4)  & All_MN_UMAP$UMAP_2 > 5 | row.names(All_MN_UMAP) %in% row.names(FOXP_Extras)  & All_MN_UMAP$UMAP_2 > -5  & All_MN_UMAP$UMAP_1 > -12) 

All_MN_Rest5 = subset(All_MN, cells = c(row.names(MN_3_V2), colnames(MN_2_V3), row.names(MN_1_V2), row.names(MN_4_V2), row.names(MN_5_V2), row.names(MN_6_V2), row.names(MN_6_pt2), row.names(MN_7_V2), row.names(MN_4_PT2_V2)), invert=T)


set.dim = 30
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_MN_Rest5)

MN_8 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(All_MN_Rest5) & All_MN_UMAP$UMAP_1 < -13 & All_MN_UMAP$UMAP_2 > -9.3) 

All_MN_Rest6 = subset(All_MN, cells = c(row.names(MN_3_V2), colnames(MN_2_V3), row.names(MN_1_V2), row.names(MN_4_V2), row.names(MN_5_V2), row.names(MN_6_V2), row.names(MN_6_pt2), row.names(MN_7_V2), row.names(MN_4_PT2_V2), row.names(MN_8)), invert=T)

set.dim = 30
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_MN_Rest6)

All_MN_Rest6 <- FindNeighbors(All_MN_Rest6, k.param=20, dims=1:30)
All_MN_Rest6 <- FindClusters(All_MN_Rest6, resolution =1)
MN_9 = subset(All_MN_Rest6, idents = c(0))
MN_MN_9_V2 = subset(All_MN_UMAP, row.names(All_MN_UMAP) %in% colnames(MN_9) & All_MN_UMAP$UMAP_1 < -3 | row.names(All_MN_UMAP) %in% colnames(All_MN_Rest6)  & All_MN_UMAP$UMAP_1 < -10 ) 
MN_10 = subset(All_MN_Rest6, cells = row.names(MN_MN_9_V2), invert=T)

CheckInput = row.names(MN_MN_9_V2)
CheckUMAP(All_MN_Rest6)



OCT_MN_Assigns_Renamed = as.data.frame(c(rep("MN_01 [ONECUT2/ONECUT3]", length(row.names(MN_3_V2))), 
                                 rep("MN_02 [ONECUT3]", length(colnames(MN_2_V3))),               
                                 rep("MN_03 [TRH]", length(row.names(MN_1_V2))),               
                                 rep("MN_04 [PITX2/FOXP1/FOXP2/TAC1]", length(row.names(MN_4_V2))),               
                                 rep("MN_04 [PITX2/FOXP1/FOXP2/TAC1]", length(row.names(MN_4_PT2_V2))),               
                                 rep("MN_05", length(row.names(MN_5_V2))),
                                 rep("MN_06 [LHX9]", length(row.names(MN_6_V2))),               
                                 rep("MN_06 [LHX9]", length(row.names(MN_6_pt2))),               
                                 rep("MN_07 [ONECUT2]", length(row.names(MN_7_V2))),               
                                 rep("MN_08 [ONECUT2/PITX2/FOXB1/TRH/NKX2-1/TAC1/LHX1]", length(row.names(MN_8))),               
                                 rep("MN_09 [ONECUT2/PITX2/FOXB1/TRH/NKX2-1/TAC1/LHX1]", length(row.names(MN_MN_9_V2))),               
                                 rep("MN_10 [ONECUT2/PITX2/FOXB1/TRH/NKX2-1/TAC1/LHX1]", length(colnames(MN_10))))) 
colnames(OCT_MN_Assigns_Renamed) = "Pop"
row.names(OCT_MN_Assigns_Renamed) = c(row.names(MN_3_V2), #row.names(OCT_MN_Assigns_Renamed) #OCT_MN_Assigns_Renamed$Barcs
                                           colnames(MN_2_V3),
                                           row.names(MN_1_V2),
                                           row.names(MN_4_V2), 
                                           row.names(MN_4_PT2_V2), 
                                           row.names(MN_5_V2), 
                                           row.names(MN_6_V2), 
                                           row.names(MN_6_pt2), 
                                           row.names(MN_7_V2), 
                                           row.names(MN_8),
                                           row.names(MN_MN_9_V2),
                                           colnames(MN_10))


paste(unique(sort(OCT_MN_Assigns_Renamed$Pop)), collapse = ", ")

OCT_MN_Assigns_Renamed$Pop = factor(OCT_MN_Assigns_Renamed$Pop, levels = c("MN_01 [ONECUT2/ONECUT3]", "MN_02 [ONECUT3]", "MN_03 [TRH]", "MN_04 [PITX2/FOXP1/FOXP2/TAC1]", "MN_05", "MN_06 [LHX9]", "MN_07 [ONECUT2]", "MN_08 [ONECUT2/PITX2/FOXB1/TRH/NKX2-1/TAC1/LHX1]", "MN_09 [ONECUT2/PITX2/FOXB1/TRH/NKX2-1/TAC1/LHX1]", "MN_10 [ONECUT2/PITX2/FOXB1/TRH/NKX2-1/TAC1/LHX1]"))

All_MN = AddMetaData(All_MN, OCT_MN_Assigns_Renamed, "OCT_MN_Assigns_Renamed")
Idents(All_MN) = "OCT_MN_Assigns_Renamed"

pdf(paste("ALLFETAL_MN_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_MN, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_MN) = "RNA"
VP = list()
for( in c("ONECUT2", "ONECUT3", "PITX2", "FOXB1", "FOXP1", "FOXP2", "TRH", "NKX2-1", "TAC1", "LHX1", "LHX5")){
VP[[x]] = VlnPlot(All_MN, x, pt.size = 0)  + NoLegend()
}
pdf("MN_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### TM: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_TM = subset(HypothalamicNeurons, idents = c("Neuron_01 [TM]", "Neuron_02 [TM]", "Neuron_03 [TM]"))


#Use PCA 30 dims 10 spread 5

DefaultAssay(All_TM) = "integrated"
All_TM = RunPCA(All_TM, npcs = 30)
All_TM <- RunUMAP(All_TM, dims = 1:10, spread= 5)
All_TM_UMAP = as.data.frame(All_TM@reductions$umap@cell.embeddings)

TM_1 = subset(All_TM_UMAP,  All_TM_UMAP$UMAP_2 < 5 & All_TM_UMAP$UMAP_1 > 6.2)
TM_2 = subset(All_TM_UMAP,  All_TM_UMAP$UMAP_2 < -7 & All_TM_UMAP$UMAP_1 < -14 | All_TM_UMAP$UMAP_2 < -5 & All_TM_UMAP$UMAP_1 < -16 )

All_TM_Rest1 = subset(All_TM, cells = c(row.names(TM_1), row.names(TM_2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(All_TM_Rest1) 

CheckInput = row.names(TM_2)
CheckUMAP(All_TM)



All_TM_Rest1 <- FindNeighbors(All_TM_Rest1, k.param=20, dims=1:10)
All_TM_Rest1 <- FindClusters(All_TM_Rest1, resolution =1)
TM_3 = subset(All_TM_Rest1, idents = c(2))
TM_3_V2 = subset(All_TM_UMAP, row.names(All_TM_UMAP) %in% colnames(TM_3) & All_TM_UMAP$UMAP_2 < 5 & All_TM_UMAP$UMAP_1 < -5 |  row.names(All_TM_UMAP) %in% colnames(All_TM_Rest1) & All_TM_UMAP$UMAP_2 < 5 & All_TM_UMAP$UMAP_1 < -11  |  row.names(All_TM_UMAP) %in% colnames(All_TM_Rest1) & All_TM_UMAP$UMAP_2 < 5 & All_TM_UMAP$UMAP_2 > -2 & All_TM_UMAP$UMAP_1 < -4)

All_TM_Rest2 = subset(All_TM, cells = c(row.names(TM_1), row.names(TM_2), row.names(TM_3_V2)), invert=T)
set.dim = 10
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(All_TM_Rest2) 

All_TM_Rest2 <- FindNeighbors(All_TM_Rest2, k.param=20, dims=1:10)
All_TM_Rest2 <- FindClusters(All_TM_Rest2, resolution =1)
TM_4 = subset(All_TM_Rest2, idents = c(1))
TM_4_V2 = subset(All_TM_UMAP, row.names(All_TM_UMAP) %in% colnames(TM_4) & All_TM_UMAP$UMAP_2 < 2)

TM_4_V2_Seu = subset(All_TM, cells = c(row.names(TM_4_V2)))
set.dim = 10
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(TM_4_V2_Seu) 
HDC_Extras = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(TM_4_V2_Seu) & Fetal.combined_Data$HDC > 0)

MoveToTM3 = subset(All_TM_UMAP, row.names(All_TM_UMAP) %in% row.names(HDC_Extras) & All_TM_UMAP$UMAP_1 < -4 )
Keep_TM4 = subset(TM_4_V2_Seu, cells = c(row.names(MoveToTM3)), invert=T)
TM3_Recombo = subset(All_TM, cells = c(row.names(MoveToTM3), row.names(TM_3_V2)))

TM_5 = subset(All_TM_Rest2, idents = c(2))
TM_5_V2 = subset(All_TM_UMAP, row.names(All_TM_UMAP) %in% colnames(TM_5) & All_TM_UMAP$UMAP_1 < 2 | row.names(All_TM_UMAP) %in% colnames(All_TM_Rest2) & All_TM_UMAP$UMAP_1 < -2 & All_TM_UMAP$UMAP_2 > 2 | row.names(All_TM_UMAP) %in% colnames(All_TM_Rest2) & All_TM_UMAP$UMAP_1 < 1 & All_TM_UMAP$UMAP_2 > 10)

TM_6 = subset(All_TM, cells = c(row.names(TM_1), row.names(TM_2), row.names(TM_3_V2), row.names(TM_4_V2), row.names(TM_5_V2)), invert=T)


CheckInput = c(row.names(MoveToTM3))
CheckUMAP(All_TM)



OCT_TM_Assigns_Renamed = as.data.frame(c(rep("TM_01 [HDC/TBX3/NKX2-1/SST]", length(row.names(TM_1))), 
                                 rep("TM_06", length(row.names(TM_2))),               
                                 rep("TM_03 [HDC/TBX3]", length(colnames(TM3_Recombo))),               
                                 rep("TM_04 [NKX2-1]", length(colnames(Keep_TM4))),               
                                 rep("TM_05 [HDC/TBX3]", length(row.names(TM_5_V2))),        
                                 rep("TM_02 [HDC/TBX3]", length(colnames(TM_6))))) 
colnames(OCT_TM_Assigns_Renamed) = "Pop"
row.names(OCT_TM_Assigns_Renamed) = c(row.names(TM_1), #row.names(OCT_TM_Assigns_Renamed) #OCT_TM_Assigns_Renamed$Barcs
                                           row.names(TM_2),
                                           colnames(TM3_Recombo),
                                           colnames(Keep_TM4), 
                                           row.names(TM_5_V2), 
                                           colnames(TM_6))

OCT_TM_Assigns_Renamed$Pop = factor(OCT_TM_Assigns_Renamed$Pop, levels = paste(unique(sort(OCT_TM_Assigns_Renamed$Pop))))


All_TM = AddMetaData(All_TM, OCT_TM_Assigns_Renamed, "OCT_TM_Assigns_Renamed")
Idents(All_TM) = "OCT_TM_Assigns_Renamed"

pdf(paste("ALLFETAL_TM_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_TM, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_TM) = "RNA"
VP = list()
for( in c("HDC", "TBX3", "NKX2-1", "SST", "SOX2")){
VP[[x]] = VlnPlot(All_TM, x, pt.size = 0)  + NoLegend()
}
pdf("TM_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### SCN: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_SCN= subset(HypothalamicNeurons, idents = c("Neuron_32 [SCN]", "Neuron_33 [SCN]"))


#Use PCA 40 dims 30 spread 10

DefaultAssay(All_SCN) = "integrated"
All_SCN = RunPCA(All_SCN, npcs = 40)
All_SCN <- RunUMAP(All_SCN, dims = 1:30, spread= 10)
All_SCN_UMAP = as.data.frame(All_SCN@reductions$umap@cell.embeddings)

SCN_1 = subset(All_SCN_UMAP,  All_SCN_UMAP$UMAP_2 > 10 & All_SCN_UMAP$UMAP_1 > -22 & All_SCN_UMAP$UMAP_1 < 0)
SCN_2 = subset(All_SCN_UMAP,  All_SCN_UMAP$UMAP_2 > 12 & All_SCN_UMAP$UMAP_1 < -22)
SCN_3 = subset(All_SCN_UMAP,  All_SCN_UMAP$UMAP_2 < -4 & All_SCN_UMAP$UMAP_1 < -19 | All_SCN_UMAP$UMAP_2 > -4 & All_SCN_UMAP$UMAP_2 < 12 & All_SCN_UMAP$UMAP_1 < -10)

All_SCN_Rest1 = subset(All_SCN, cells = c(row.names(SCN_1), row.names(SCN_2), row.names(SCN_3)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(80)
ClusterFunc_Neuro_RNA(All_SCN_Rest1) 


All_SCN_Rest1 <- FindNeighbors(All_SCN_Rest1, k.param=80, dims=1:10)
All_SCN_Rest1 <- FindClusters(All_SCN_Rest1, resolution =1)
SCN_4 = subset(All_SCN_Rest1, idents = c(4))
SCN_4_V2 = subset(All_SCN_UMAP, row.names(All_SCN_UMAP) %in% colnames(SCN_4) & All_SCN_UMAP$UMAP_2 < -8 & All_SCN_UMAP$UMAP_1 > 0 |  row.names(All_SCN_UMAP) %in% colnames(All_SCN_Rest1) & All_SCN_UMAP$UMAP_2 < -12 & All_SCN_UMAP$UMAP_1 > 3 & All_SCN_UMAP$UMAP_1 < 11 )

All_SCN_Rest2 = subset(All_SCN, cells = c(row.names(SCN_1), row.names(SCN_2), row.names(SCN_3), row.names(SCN_4_V2)), invert=T)

SCN_5 = subset(All_SCN_UMAP, row.names(All_SCN_UMAP) %in% colnames(All_SCN_Rest2) & All_SCN_UMAP$UMAP_1 < 7)

All_SCN_Rest3 = subset(All_SCN, cells = c(row.names(SCN_1), row.names(SCN_2), row.names(SCN_3), row.names(SCN_4_V2), row.names(SCN_5)), invert=T)


All_SCN_Rest3 <- FindNeighbors(All_SCN_Rest3, k.param=10, dims=1:10)
All_SCN_Rest3 <- FindClusters(All_SCN_Rest3, resolution =1)
SCN_6 = subset(All_SCN_Rest3, idents = c(6))
SCN_7 = subset(All_SCN_Rest3, idents = c(3,7))
SCN_7_V2 = subset(All_SCN_UMAP, row.names(All_SCN_UMAP) %in% colnames(SCN_7) & All_SCN_UMAP$UMAP_2 > 12 | row.names(All_SCN_UMAP) %in% colnames(All_SCN_Rest3) & All_SCN_UMAP$UMAP_2 > 20 & All_SCN_UMAP$UMAP_1 > 20)
SCN_7_V2_Seu = subset(All_SCN_Rest3, cells = row.names(SCN_7_V2))

set.dim = 10
set.res = c(1)
set.kparam = c(3)
ClusterFunc_Neuro_RNA(SCN_7_V2_Seu) 


SCN_7_V3 = subset(All_SCN_UMAP, row.names(All_SCN_UMAP) %in% colnames(SCN_7_V2_Seu) & All_SCN_UMAP$UMAP_2 > 21.5 & All_SCN_UMAP$UMAP_1 > 17)
SCN_8 = subset(SCN_7_V2_Seu, cells = row.names(SCN_7_V3), invert=T)

All_SCN_Rest4 = subset(All_SCN, cells = c(row.names(SCN_1), row.names(SCN_2), row.names(SCN_3), row.names(SCN_4_V2), row.names(SCN_5), colnames(SCN_6), row.names(SCN_7_V3), colnames(SCN_8)), invert=T)


set.dim = 10
set.res = c(1)
set.kparam = c(20)
ClusterFunc_Neuro_RNA(All_SCN_Rest4) 

All_SCN_Rest4 <- FindNeighbors(All_SCN_Rest4, k.param=20, dims=1:10)
All_SCN_Rest4 <- FindClusters(All_SCN_Rest4, resolution =1)
SCN_9 = subset(All_SCN_Rest4, idents = c(3))
SCN_9_V2 = subset(All_SCN_UMAP, row.names(All_SCN_UMAP) %in% colnames(SCN_9) & All_SCN_UMAP$UMAP_2 > 0 & All_SCN_UMAP$UMAP_1 < 25 |  row.names(All_SCN_UMAP) %in% colnames(All_SCN_Rest4) & All_SCN_UMAP$UMAP_2 > 3 & All_SCN_UMAP$UMAP_1 < 20)


All_SCN_Rest5 = subset(All_SCN, cells = c(row.names(SCN_1), row.names(SCN_2), row.names(SCN_3), row.names(SCN_4_V2), row.names(SCN_5), colnames(SCN_6), row.names(SCN_7_V3), colnames(SCN_8), row.names(SCN_9_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(50,100)
ClusterFunc_Neuro_RNA(All_SCN_Rest5) 

All_SCN_Rest5 <- FindNeighbors(All_SCN_Rest5, k.param=100, dims=1:10)
All_SCN_Rest5 <- FindClusters(All_SCN_Rest5, resolution =1)
SCN_10 = subset(All_SCN_Rest5, idents = c(2))
SCN_10_V2 = subset(All_SCN_UMAP, row.names(All_SCN_UMAP) %in% colnames(SCN_10) & All_SCN_UMAP$UMAP_1 < 23 |  row.names(All_SCN_UMAP) %in% colnames(All_SCN_Rest5)  & All_SCN_UMAP$UMAP_1 < 18)

SCN_11 = subset(All_SCN_Rest5, cells = row.names(SCN_10_V2), invert=T)

CheckInput = row.names(SCN_10_V2)
CheckUMAP(All_SCN_Rest5)

OCT_SCN_Assigns_Renamed = as.data.frame(c(rep("SCN_01 [LHX1/SIX6/SIX3]", length(row.names(SCN_1))), 
                                 rep("SCN_02 [LHX1/SIX6/SIX3]", length(row.names(SCN_2))),               
                                 rep("SCN_03 [LHX1/SIX6/SIX3]", length(row.names(SCN_3))),               
                                 rep("SCN_04 [ZIC1/ZIC2/RELN/SIX3]", length(row.names(SCN_4_V2))),               
                                 rep("SCN_05 [ZIC1/ZIC2]", length(row.names(SCN_5))),        
                                 rep("SCN_06 [LHX1/ZIC1/ZIC2/RELN/CCK/TRH]", length(colnames(SCN_6))),        
                                 rep("SCN_07 [ZIC1/ZIC2/RELN/CCK]", length(row.names(SCN_7_V3))),        
                                 rep("SCN_08 [LHX1/CCK]", length(colnames(SCN_8))),        
                                 rep("SCN_09 [ZIC1/ZIC2/RELN]", length(row.names(SCN_9_V2))),        
                                 rep("SCN_10 [ZIC1/ZIC2]", length(row.names(SCN_10_V2))),        
                                 rep("SCN_11", length(colnames(SCN_11))))) 
colnames(OCT_SCN_Assigns_Renamed) = "Pop"
row.names(OCT_SCN_Assigns_Renamed) = c(row.names(SCN_1), #row.names(OCT_SCN_Assigns_Renamed) #OCT_SCN_Assigns_Renamed$Barcs
                                           row.names(SCN_2),
                                           row.names(SCN_3),
                                           row.names(SCN_4_V2),
                                           row.names(SCN_5), 
                                           colnames(SCN_6), 
                                           row.names(SCN_7_V3), 
                                           colnames(SCN_8), 
                                           row.names(SCN_9_V2), 
                                           row.names(SCN_10_V2), 
                                           colnames(SCN_11))


paste(unique(sort(OCT_SCN_Assigns_Renamed$Pop)), collapse = ", ")

OCT_SCN_Assigns_Renamed$Pop = factor(OCT_SCN_Assigns_Renamed$Pop, levels = c("SCN_01 [LHX1/SIX6/SIX3]", "SCN_02 [LHX1/SIX6/SIX3]", "SCN_03 [LHX1/SIX6/SIX3]", "SCN_04 [ZIC1/ZIC2/RELN/SIX3]", "SCN_05 [ZIC1/ZIC2]", "SCN_06 [LHX1/ZIC1/ZIC2/RELN/CCK/TRH]", "SCN_07 [ZIC1/ZIC2/RELN/CCK]", "SCN_08 [LHX1/CCK]", "SCN_09 [ZIC1/ZIC2/RELN]", "SCN_10 [ZIC1/ZIC2]", "SCN_11"))


All_SCN = AddMetaData(All_SCN, OCT_SCN_Assigns_Renamed, "OCT_SCN_Assigns_Renamed")
Idents(All_SCN) = "OCT_SCN_Assigns_Renamed"

pdf(paste("ALLFETAL_SCN_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_SCN, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_SCN) = "RNA"
VP = list()
for( in c("LHX1", "SIX6", "ZIC1", "ZIC2",  "RELN", "CCK", "SIX3", "OTP", "RELN", "TRH", "PCSK1")){
VP[[x]] = VlnPlot(All_SCN, x, pt.size = 0)  + NoLegend()
}
pdf("SCN_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### ZI: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_ZI = subset(HypothalamicNeurons, idents = c("Neuron_45 [ZI]", "Neuron_46 [ZI]", "Neuron_47 [ZI]"))


#Use PCA 40 dims 20 spread 2

DefaultAssay(All_ZI) = "integrated"
All_ZI = RunPCA(All_ZI, npcs = 40)
All_ZI <- RunUMAP(All_ZI, dims = 1:20, spread= 2)
All_ZI_UMAP = as.data.frame(All_ZI@reductions$umap@cell.embeddings)

set.dim = 10
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_ZI) 

All_ZI <- FindNeighbors(All_ZI, k.param=10, dims=1:10)
All_ZI <- FindClusters(All_ZI, resolution =1)
ZI_1 = subset(All_ZI, idents = c(0,9))
ZI_1_V2 = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% colnames(ZI_1) |  All_ZI_UMAP$UMAP_1 < -4.7)
ZI_1_V2_Seu = subset(All_ZI, cells = c(row.names(ZI_1_V2)))

set.dim = 10
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(ZI_1_V2_Seu) 

ZI_1_V2_Seu <- FindNeighbors(ZI_1_V2_Seu, k.param=5, dims=1:10)
ZI_1_V2_Seu <- FindClusters(ZI_1_V2_Seu, resolution =1)
ZI_1_V3 = subset(ZI_1_V2_Seu, idents = c(2))
ZI_2 = subset(ZI_1_V2_Seu, idents = c(0,4))
ZI_3 = subset(ZI_1_V2_Seu, idents = c(0,4,2), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(ZI_3) 

ZI_3 <- FindNeighbors(ZI_3, k.param=20, dims=1:10)
ZI_3 <- FindClusters(ZI_3, resolution =1)
ZI_3_V2 = subset(ZI_3, idents = c(1))
ZI_4 = subset(ZI_3, idents = c(0,2))

set.dim = 10
set.res = c(0.1)
set.kparam = c(200)
#ClusterFunc_Neuro_RNA(ZI_4) 

All_ZI_Rest1 = subset(All_ZI, cells = c(colnames(ZI_1_V3), colnames(ZI_2), colnames(ZI_3_V2), colnames(ZI_4)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_ZI_Rest1) 

All_ZI_Rest1 <- FindNeighbors(All_ZI_Rest1, k.param=10, dims=1:10)
All_ZI_Rest1 <- FindClusters(All_ZI_Rest1, resolution =1)
ZI_5 = subset(All_ZI_Rest1, idents = c(8))
ExtraMEIS2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_ZI_Rest1) & Fetal.combined_Data$MEIS2 > 0)

ZI_5_V2 = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% colnames(ZI_5) & All_ZI_UMAP$UMAP_2 > 4 | row.names(All_ZI_UMAP) %in% colnames(All_ZI_Rest1) & All_ZI_UMAP$UMAP_2 > 5.5 | row.names(All_ZI_UMAP) %in% row.names(ExtraMEIS2) & All_ZI_UMAP$UMAP_2 > 3.5)

All_ZI_Rest2 = subset(All_ZI, cells = c(colnames(ZI_1_V3), colnames(ZI_2), colnames(ZI_3_V2), colnames(ZI_4), row.names(ZI_5_V2)), invert=T)

All_ZI_Rest2_OLD = All_ZI_Rest2


set.dim = 10
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_ZI_Rest2)



All_ZI_Rest2 <- FindNeighbors(All_ZI_Rest2, k.param=20, dims=1:10)
All_ZI_Rest2 <- FindClusters(All_ZI_Rest2, resolution =1)
ZI_6 = subset(All_ZI_Rest2, idents = c(1,3,5))

set.dim = 10
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(ZI_6)

SSTNeg = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ZI_6) & Fetal.combined_Data$SST == 0)
ZI_6_ToRM = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% row.names(SSTNeg) & All_ZI_UMAP$UMAP_1 < 3)
ZI_6_V2 = subset(ZI_6, cells = row.names(ZI_6_ToRM), invert=T)


#ZI_6 <- FindNeighbors(ZI_6, k.param=3, dims=1:10)
#ZI_6 <- FindClusters(ZI_6, resolution =3)
#ZI_6_V2 = subset(ZI_6, idents = c(18,12, 9, 1, 2), invert=T)

CheckInput = colnames(ZI_6_V2)
CheckUMAP(ZI_6)

set.dim = 10
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(ZI_6_V2)
ZI_6_V2 <- FindNeighbors(ZI_6_V2, k.param=10, dims=1:10)
ZI_6_V2 <- FindClusters(ZI_6_V2, resolution =1)
#ZI_6_V3 = subset(ZI_6_V2, idents = c(0,4))
ZI_6_V3 = subset(ZI_6_V2, idents = c(1,2,3))

ExtraMEIS2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(ZI_6_V3) & Fetal.combined_Data$MEIS2 > 0)

ZI_6_V4 = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% colnames(ZI_6_V3) & All_ZI_UMAP$UMAP_2 > -1 | row.names(All_ZI_UMAP) %in% colnames(ZI_6_V3) & row.names(All_ZI_UMAP) %notin% row.names(ExtraMEIS2) & All_ZI_UMAP$UMAP_2 < -1  & All_ZI_UMAP$UMAP_2 > -2.5 & All_ZI_UMAP$UMAP_1 > -1 | row.names(All_ZI_UMAP) %in% colnames(ZI_6_V2) & All_ZI_UMAP$UMAP_2 > 0)

ZI_7 = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% colnames(ZI_6_V2) & row.names(All_ZI_UMAP) %notin% row.names(ZI_6_V4) & All_ZI_UMAP$UMAP_1 > -1)

CheckInput = row.names(ZI_7)
CheckUMAP(ZI_6_V2)


All_ZI_Rest3 = subset(All_ZI, cells = c(colnames(ZI_1_V3), colnames(ZI_2), colnames(ZI_3_V2), colnames(ZI_4), row.names(ZI_5_V2), row.names(ZI_6_V4), row.names(ZI_7)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_ZI_Rest3)

All_ZI_Rest3 <- FindNeighbors(All_ZI_Rest3, k.param=20, dims=1:10)
All_ZI_Rest3 <- FindClusters(All_ZI_Rest3, resolution =1)
ZI_8 = subset(All_ZI_Rest3, idents = c(2))
ZI_9 = subset(All_ZI_Rest3, idents = c(1,3))
#ZI_10 = subset(All_ZI_Rest3, idents = c(1,3))


ZI_8_V2 = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% colnames(ZI_8) & All_ZI_UMAP$UMAP_2 < 0 | row.names(All_ZI_UMAP) %in% colnames(All_ZI_Rest3) & All_ZI_UMAP$UMAP_2 < -4)


ExtraMEIS2Neg = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_ZI_Rest3) & Fetal.combined_Data$MEIS2 == 0)

ZI_9_V2 = subset(All_ZI_UMAP, row.names(All_ZI_UMAP) %in% colnames(ZI_9) & All_ZI_UMAP$UMAP_2 > 0 | row.names(All_ZI_UMAP) %in% row.names(ExtraMEIS2Neg) & All_ZI_UMAP$UMAP_2 > -1 & All_ZI_UMAP$UMAP_1 > -5)

ZI_10 = subset(All_ZI_Rest3, cells = c(colnames(ZI_1_V3), colnames(ZI_2), colnames(ZI_3_V2), colnames(ZI_4), row.names(ZI_5_V2), row.names(ZI_6_V4), row.names(ZI_7), row.names(ZI_8_V2), row.names(ZI_9_V2)), invert=T)


OCT_ZI_Assigns_Renamed = as.data.frame(c(rep("ZI_01 [MEIS2/SIX3/SST/GAL/TRH]", length(colnames(ZI_1_V3))), 
                                 rep("ZI_02 [SST]", length(colnames(ZI_2))),               
                                 rep("ZI_03 [MEIS2/SIX3/SST]", length(colnames(ZI_3_V2))),               
                                 rep("ZI_04 [MEIS2/SIX3/SST]", length(colnames(ZI_4))),               
                                 rep("ZI_05 [MEIS2/SIX3]", length(row.names(ZI_5_V2))),        
                                 rep("ZI_06 [SIX3/SST]", length(row.names(ZI_6_V4))),        
                                 rep("ZI_07 [MEIS2/SIX3/SST]", length(row.names(ZI_7))),        
                                 rep("ZI_08 [MEIS2/SIX3/GAL]", length(row.names(ZI_8_V2))),        
                                 rep("ZI_09 [SIX3]", length(row.names(ZI_9_V2))),        
                                 rep("ZI_10 [MEIS2/SIX3]", length(colnames(ZI_10))))) 
colnames(OCT_ZI_Assigns_Renamed) = "Pop"
row.names(OCT_ZI_Assigns_Renamed) = c(colnames(ZI_1_V3), #row.names(OCT_ZI_Assigns_Renamed) #OCT_ZI_Assigns_Renamed$Barcs
                                           colnames(ZI_2),
                                           colnames(ZI_3_V2),
                                           colnames(ZI_4),
                                           row.names(ZI_5_V2), 
                                           row.names(ZI_6_V4), 
                                           row.names(ZI_7), 
                                           row.names(ZI_8_V2), 
                                           row.names(ZI_9_V2), 
                                           colnames(ZI_10))


paste(unique(sort(OCT_ZI_Assigns_Renamed$Pop)), collapse = ", ")

OCT_ZI_Assigns_Renamed$Pop = factor(OCT_ZI_Assigns_Renamed$Pop, levels = c("ZI_01 [MEIS2/SIX3/SST/GAL/TRH]", "ZI_02 [SST]", "ZI_03 [MEIS2/SIX3/SST]", "ZI_04 [MEIS2/SIX3/SST]", "ZI_05 [MEIS2/SIX3]", "ZI_06 [SIX3/SST]", "ZI_07 [MEIS2/SIX3/SST]", "ZI_08 [MEIS2/SIX3/GAL]", "ZI_09 [SIX3]", "ZI_10 [MEIS2/SIX3]"))



All_ZI = AddMetaData(All_ZI, OCT_ZI_Assigns_Renamed, "OCT_ZI_Assigns_Renamed")
Idents(All_ZI) = "OCT_ZI_Assigns_Renamed"

pdf(paste("ALLFETAL_ZI_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_ZI, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_ZI) = "RNA"
VP = list()
for( in c("MEIS2", "SIX3", "SIX6", "SST", "GAL", "TRH")){
VP[[x]] = VlnPlot(All_ZI, x, pt.size = 0)  + NoLegend()
}
pdf("ZI_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### VMH: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_VMH = subset(HypothalamicNeurons, idents = c("Neuron_31 [VMH]", "Neuron_32 [VMH]"))

#PCA 20 dims 10 spread 10

DefaultAssay(All_VMH) = "integrated"
All_VMH = RunPCA(All_VMH, npcs = 20)
All_VMH <- RunUMAP(All_VMH, dims = 1:10, spread= 10)
All_VMH_UMAP = as.data.frame(All_VMH@reductions$umap@cell.embeddings)

VMH_1 = subset(All_VMH_UMAP,  All_VMH_UMAP$UMAP_1 > 8 & All_VMH_UMAP$UMAP_2 < -12 | All_VMH_UMAP$UMAP_1 > 18 & All_VMH_UMAP$UMAP_2 < -5)
VMH_1_Seu = subset(All_VMH, cells = row.names(VMH_1))



set.dim = c(20)
set.res = c(1)
set.kparam = c(3,5)
#ClusterFunc_Neuro_RNA(VMH_1_Seu)


VMH_1_Seu <- FindNeighbors(VMH_1_Seu, k.param=3, dims=1:10)
VMH_1_Seu <- FindClusters(VMH_1_Seu, resolution =2)
VMH_1_V2 = subset(VMH_1_Seu, idents = c(3,11,16))
Extra_NR5A1 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(VMH_1_Seu) & Fetal.combined_Data$NR5A1 > 0)
VMH_1_V3 = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_1_V2) & All_VMH_UMAP$UMAP_1 > 20 | row.names(All_VMH_UMAP) %in% row.names(Extra_NR5A1) & All_VMH_UMAP$UMAP_2 < -20  | row.names(All_VMH_UMAP) %in% colnames(VMH_1_Seu) & All_VMH_UMAP$UMAP_1 > 21 & All_VMH_UMAP$UMAP_2 > -26 & All_VMH_UMAP$UMAP_2 < -22)
VMH_1_V3_Seu = subset(All_VMH, cells = row.names(VMH_1_V3))

VMH_1_Rest1 = subset(VMH_1_Seu, cells = row.names(VMH_1_V3), invert=T)

set.dim = c(10,20)
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(VMH_1_Rest1)

VMH_1_Rest1 <- FindNeighbors(VMH_1_Rest1, k.param=3, dims=1:10)
VMH_1_Rest1 <- FindClusters(VMH_1_Rest1, resolution =1)
VMH_2 = subset(VMH_1_Rest1, idents = c(0,3,4))
Extra_ADCYAP1 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(VMH_1_Rest1) & Fetal.combined_Data$ADCYAP1 > 0)
VMH_2_V2 = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_2) & All_VMH_UMAP$UMAP_1 < 22 | row.names(All_VMH_UMAP) %in% row.names(Extra_ADCYAP1) & All_VMH_UMAP$UMAP_1 < 22 | row.names(All_VMH_UMAP) %in% colnames(VMH_1_Rest1) & All_VMH_UMAP$UMAP_1 < 18)
VMH_2_V2_Seu = subset(All_VMH, cells = row.names(VMH_2_V2))


set.dim = c(50)
set.res = c(0.05)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(VMH_2_V2_Seu)

VMH_1_Rest2 = subset(VMH_1_Seu, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2)), invert=T)

set.dim = c(50)
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(VMH_1_Rest2)


VMH_3 = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_1_Rest2) & All_VMH_UMAP$UMAP_2 < -22)
VMH_4 = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_1_Rest2) & All_VMH_UMAP$UMAP_2 > -22)

set.dim = c(50)
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(VMH_3_V2_Seu)
#ClusterFunc_Neuro_RNA(VMH_4)

All_VMH_Rest2 = subset(All_VMH, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2), row.names(VMH_3), row.names(VMH_4)), invert=T)


set.dim = c(10)
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(All_VMH_Rest2)

All_VMH_Rest2 <- FindNeighbors(All_VMH_Rest2, k.param=10, dims=1:10)
All_VMH_Rest2 <- FindClusters(All_VMH_Rest2, resolution =1)
VMH_5 = subset(All_VMH_Rest2, idents = c(2,9))
set.dim = c(10)
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(VMH_5)


VMH_5 <- FindNeighbors(VMH_5, k.param=10, dims=1:10)
VMH_5 <- FindClusters(VMH_5, resolution =1)
VMH_5_V2 = subset(VMH_5, idents = c(0,1))
VMH_5_V3 = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_5_V2) & All_VMH_UMAP$UMAP_2 < 20 & All_VMH_UMAP$UMAP_1 > 7 | row.names(All_VMH_UMAP) %in% colnames(VMH_5) & All_VMH_UMAP$UMAP_1 > 13  & All_VMH_UMAP$UMAP_2 < 20)

VMH_6 = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_5) & All_VMH_UMAP$UMAP_1 < -3)
VMH_7 = subset(VMH_5, cells = c(row.names(VMH_5_V3), row.names(VMH_6)), invert=T)

All_VMH_Rest3 = subset(All_VMH, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2), row.names(VMH_3), row.names(VMH_4), row.names(VMH_5_V3), row.names(VMH_6), colnames(VMH_7)), invert=T)

set.dim = c(10)
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_VMH_Rest3)

CheckInput = row.names(VMH_6)
CheckUMAP(VMH_5)

VMH_7_Cleaned = subset(All_VMH_UMAP, row.names(All_VMH_UMAP) %in% colnames(VMH_7) & All_VMH_UMAP$UMAP_2 < 10)

All_VMH_Rest3 <- FindNeighbors(All_VMH_Rest3, k.param=3, dims=1:10)
All_VMH_Rest3 <- FindClusters(All_VMH_Rest3, resolution =2)
Extra_VMH_8 = subset(All_VMH_Rest3, idents = c(11,13))
Extra_ADCYAP2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_VMH_Rest3) & Fetal.combined_Data$ADCYAP1 > 0)
VMH_8_Cleaned = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(Extra_VMH_8) & All_VMH_UMAP$UMAP_1 > -9  & All_VMH_UMAP$UMAP_1 < 12  & All_VMH_UMAP$UMAP_2 < 7 | row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest3) & All_VMH_UMAP$UMAP_1 > -8  & All_VMH_UMAP$UMAP_1 < 10  & All_VMH_UMAP$UMAP_2 < 6 | row.names(All_VMH_UMAP) %in% row.names(Extra_ADCYAP2) & All_VMH_UMAP$UMAP_1 > -15  & All_VMH_UMAP$UMAP_1 < 10  & All_VMH_UMAP$UMAP_2 < 10)

All_VMH_Rest4 = subset(All_VMH, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2), row.names(VMH_3), row.names(VMH_4), row.names(VMH_5_V3), row.names(VMH_6), row.names(VMH_7_Cleaned), row.names(VMH_8_Cleaned)), invert=T)

set.dim = c(10)
set.res = c(1)
set.kparam = c(10)
#ClusterFunc_Neuro_RNA(All_VMH_Rest4)

All_VMH_Rest4 <- FindNeighbors(All_VMH_Rest4, k.param=10, dims=1:10)
All_VMH_Rest4 <- FindClusters(All_VMH_Rest4, resolution =1)
VMH_9 = subset(All_VMH_Rest4, idents = c(1,5))
VMH_10 = subset(All_VMH_Rest4, idents = c(6))
VMH_11 = subset(All_VMH_Rest4, idents = c(3,4))
All_VMH_Rest5 = subset(All_VMH_Rest4, idents = c(1,5,6,3,4), invert=T)

set.dim = c(10)
set.res = c(3)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(All_VMH_Rest5)

All_VMH_Rest5 <- FindNeighbors(All_VMH_Rest5, k.param=5, dims=1:10)
All_VMH_Rest5 <- FindClusters(All_VMH_Rest5, resolution =2)
VMH_12 = subset(All_VMH_Rest5, idents = c(10))
Extra_NPTX2_NK = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_VMH_Rest5) & Fetal.combined_Data$`NKX2-1` > 0 | row.names(Fetal.combined_Data) %in% colnames(All_VMH_Rest5) & Fetal.combined_Data$NPTX2 > 0)

VMH_12_Cleaned = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(VMH_12) & All_VMH_UMAP$UMAP_2 < -15 & All_VMH_UMAP$UMAP_1 < -22  | row.names(All_VMH_UMAP) %in% row.names(Extra_NPTX2_NKX) & All_VMH_UMAP$UMAP_1 > -32 & All_VMH_UMAP$UMAP_1 < -22 & All_VMH_UMAP$UMAP_2 < -10 | row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest5) & All_VMH_UMAP$UMAP_2 < -17 & All_VMH_UMAP$UMAP_1 > -31)

VMH_12_Seu = subset(All_VMH_Rest5, cells = row.names(VMH_12_Cleaned))

set.dim = c(10)
set.res = c(0.05)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(VMH_12_Seu)

All_VMH_Rest6 = subset(All_VMH, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2), row.names(VMH_3), row.names(VMH_4), row.names(VMH_5_V3), row.names(VMH_6), row.names(VMH_7_Cleaned), row.names(VMH_8_Cleaned), colnames(VMH_9),  colnames(VMH_10), colnames(VMH_11), colnames(VMH_12_Seu)), invert=T)

set.dim = c(10)
set.res = c(1.5)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_VMH_Rest6)


All_VMH_Rest6 <- FindNeighbors(All_VMH_Rest6, k.param=40, dims=1:10)
All_VMH_Rest6 <- FindClusters(All_VMH_Rest6, resolution =1)
VMH_13 = subset(All_VMH_Rest6, idents = c(1,2))
Extra_SST = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_VMH_Rest6) & Fetal.combined_Data$SST > 0 )
VMH_13_V2 = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(VMH_13) & All_VMH_UMAP$UMAP_1 < -24  | row.names(All_VMH_UMAP) %in% row.names(Extra_SST) & All_VMH_UMAP$UMAP_1 < -30)
VMH_13_Seu = subset(All_VMH_Rest6, cells = row.names(VMH_13_V2))


set.dim = c(10)
set.res = c(1)
set.kparam = c(3)
#ClusterFunc_Neuro_RNA(VMH_13_Seu)

VMH_13_Seu <- FindNeighbors(VMH_13_Seu, k.param=3, dims=1:10)
VMH_13_Seu <- FindClusters(VMH_13_Seu, resolution =1)
VMH_13_V3 = subset(VMH_13_Seu, idents = c(0,4,10), invert=T)

set.dim = c(10)
set.res = c(1)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(VMH_13_V3)

VMH_13_V4 = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(VMH_13_V3) & All_VMH_UMAP$UMAP_2 < -10 & All_VMH_UMAP$UMAP_1 < -27 | row.names(All_VMH_UMAP) %in% row.names(Extra_SST) & All_VMH_UMAP$UMAP_1 < -25 & All_VMH_UMAP$UMAP_2 < -8 | row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest6) & All_VMH_UMAP$UMAP_1 < -27 & All_VMH_UMAP$UMAP_2 < -15)
VMH_13_V4_Seu = subset(All_VMH_Rest6, cells = row.names(VMH_13_V4))

set.dim = c(10)
set.res = c(0.1)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(VMH_13_V4_Seu)

All_VMH_Rest7 = subset(All_VMH, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2), row.names(VMH_3), row.names(VMH_4), row.names(VMH_5_V3), row.names(VMH_6), row.names(VMH_7_Cleaned), row.names(VMH_8_Cleaned), colnames(VMH_9),  colnames(VMH_10), colnames(VMH_11), colnames(VMH_12_Seu), row.names(VMH_13_V4)), invert=T)

set.dim = c(10)
set.res = c(1.5)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_VMH_Rest7)

All_VMH_Rest7 <- FindNeighbors(All_VMH_Rest7, k.param=5, dims=1:10)
All_VMH_Rest7 <- FindClusters(All_VMH_Rest7, resolution =1.5)
VMH_14 = subset(All_VMH_Rest7, idents = c(0))
VMH_15 = subset(All_VMH_Rest7, idents = c(1))
Extra_TRH = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_VMH_Rest7) & Fetal.combined_Data$TRH > 0 )



VMH_14_V2 = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(VMH_14) & All_VMH_UMAP$UMAP_2 < -8  | row.names(All_VMH_UMAP) %in% row.names(Extra_TRH) & All_VMH_UMAP$UMAP_1 > -32 & All_VMH_UMAP$UMAP_2 < -7 | row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest7) & All_VMH_UMAP$UMAP_1 > -27 & All_VMH_UMAP$UMAP_2 < -10)
VMH_14_V2_Seu = subset(All_VMH_Rest7, cells = row.names(VMH_14_V2))

VMH_15_V2 = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(VMH_15) & All_VMH_UMAP$UMAP_2 > -10  & All_VMH_UMAP$UMAP_1 < -17 | row.names(All_VMH_UMAP) %in% row.names(Extra_TRH) & All_VMH_UMAP$UMAP_1 > -32 & All_VMH_UMAP$UMAP_1 < -17 & All_VMH_UMAP$UMAP_2 > -10 & All_VMH_UMAP$UMAP_2 < -1  | row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest7) & All_VMH_UMAP$UMAP_1 > -28 & All_VMH_UMAP$UMAP_1 < -22 & All_VMH_UMAP$UMAP_2 > -8 & All_VMH_UMAP$UMAP_2 < -5)
VMH_15_V2_Seu = subset(All_VMH_Rest7, cells = row.names(VMH_15_V2))
VMH_15_V3_Seu = subset(VMH_15_V2_Seu, cells = row.names(VMH_14_V2), invert=T)

set.dim = c(10)
set.res = c(0.1)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(VMH_15_V3_Seu)

All_VMH_Rest9 = subset(All_VMH, cells = c(row.names(VMH_1_V3), row.names(VMH_2_V2), row.names(VMH_3), row.names(VMH_4), row.names(VMH_5_V3), row.names(VMH_6), row.names(VMH_7_Cleaned), row.names(VMH_8_Cleaned), colnames(VMH_9),  colnames(VMH_10), colnames(VMH_11), colnames(VMH_12_Seu), row.names(VMH_13_V4), row.names(VMH_14_V2), colnames(VMH_15_V3_Seu)), invert=T)


set.dim = c(10)
set.res = c(1.5)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(All_VMH_Rest9)


VMH_16 = subset(All_VMH_UMAP,  row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest9) & All_VMH_UMAP$UMAP_2 > 2  & All_VMH_UMAP$UMAP_1 > -20 | row.names(All_VMH_UMAP) %in% colnames(All_VMH_Rest9)   & All_VMH_UMAP$UMAP_1 > -16)
VMH_17 = subset(All_VMH_Rest9, cells = row.names(VMH_16), invert=T)

CheckInput = row.names(VMH_16)
CheckUMAP(All_VMH)



OCT_VMH_Assigns_Renamed = as.data.frame(c(rep("VMH_01 [GABRA5/NPTX2/NKX2-1/FEZF1/NR5A1/SOX14]", length(row.names(VMH_1_V3))), 
                                 rep("VMH_02 [GABRA5/NPTX2/ADCYAP1/FEZF1/SOX14]", length(row.names(VMH_2_V2))),               
                                 rep("VMH_03 [NPTX2/NKX2-1/FEZF1/SOX14]", length(row.names(VMH_3))),               
                                 rep("VMH_04 [NPTX2/NKX2-1/FEZF1/SOX14]", length(row.names(VMH_4))),               
                                 rep("VMH_05 [GABRA5/NPTX2/NKX2-1/ADCYAP1/FEZF1]", length(row.names(VMH_5_V3))),        
                                 rep("VMH_06 [ADCYAP1]", length(row.names(VMH_6))),        
                                 rep("VMH_07 [GABRA5/NPTX2/ADCYAP1/FEZF1]", length(row.names(VMH_7_Cleaned))),        
                                 rep("VMH_08 [GABRA5/NPTX2/NKX2-1/ADCYAP1/FEZF1/SST]", length(row.names(VMH_8_Cleaned))),        
                                 rep("VMH_09 [GABRA5/NPTX2/NKX2-1/FEZF1/PCSK1]", length(colnames(VMH_9))),        
                                 rep("VMH_10 [GABRA5/NPTX2/FEZF1]", length(colnames(VMH_10))),        
                                 rep("VMH_11 [NKX2-1/FEZF1]", length(colnames(VMH_11))),        
                                 rep("VMH_12 [NPTX2/NKX2-1/SST]", length(colnames(VMH_12_Seu))),        
                                 rep("VMH_13 [SST]", length(row.names(VMH_13_V4))),        
                                 rep("VMH_14 [TRH]", length(row.names(VMH_14_V2))),        
                                 rep("VMH_15 [TRH]", length(colnames(VMH_15_V3_Seu))),        
                                 rep("VMH_16 [SST]", length(row.names(VMH_16))),        
                                 rep("VMH_17", length(colnames(VMH_17))))) 
colnames(OCT_VMH_Assigns_Renamed) = "Pop"
row.names(OCT_VMH_Assigns_Renamed) = c(row.names(VMH_1_V3), #row.names(OCT_VMH_Assigns_Renamed) #OCT_VMH_Assigns_Renamed$Barcs
                                           row.names(VMH_2_V2),
                                           row.names(VMH_3),
                                           row.names(VMH_4),
                                           row.names(VMH_5_V3), 
                                           row.names(VMH_6), 
                                           row.names(VMH_7_Cleaned), 
                                           row.names(VMH_8_Cleaned), 
                                           colnames(VMH_9), 
                                           colnames(VMH_10), 
                                           colnames(VMH_11), 
                                           colnames(VMH_12_Seu), 
                                           row.names(VMH_13_V4), 
                                           row.names(VMH_14_V2), 
                                           colnames(VMH_15_V3_Seu), 
                                           row.names(VMH_16), 
                                           colnames(VMH_17))


paste(unique(sort(OCT_VMH_Assigns_Renamed$Pop)), collapse = ", ")

OCT_VMH_Assigns_Renamed$Pop = factor(OCT_VMH_Assigns_Renamed$Pop, levels = c("VMH_01 [GABRA5/NPTX2/NKX2-1/FEZF1/NR5A1/SOX14]", "VMH_02 [GABRA5/NPTX2/ADCYAP1/FEZF1/SOX14]", "VMH_03 [NPTX2/NKX2-1/FEZF1/SOX14]", "VMH_04 [NPTX2/NKX2-1/FEZF1/SOX14]", "VMH_05 [GABRA5/NPTX2/NKX2-1/ADCYAP1/FEZF1]", "VMH_06 [ADCYAP1]", "VMH_07 [GABRA5/NPTX2/ADCYAP1/FEZF1]", "VMH_08 [GABRA5/NPTX2/NKX2-1/ADCYAP1/FEZF1/SST]", "VMH_09 [GABRA5/NPTX2/NKX2-1/FEZF1/PCSK1]", "VMH_10 [GABRA5/NPTX2/FEZF1]", "VMH_11 [NKX2-1/FEZF1]", "VMH_12 [NPTX2/NKX2-1/SST]", "VMH_13 [SST]", "VMH_14 [TRH]", "VMH_15 [TRH]", "VMH_16 [SST]", "VMH_17"))



All_VMH = AddMetaData(All_VMH, OCT_VMH_Assigns_Renamed, "OCT_VMH_Assigns_Renamed")
Idents(All_VMH) = "OCT_VMH_Assigns_Renamed"

pdf(paste("ALLFETAL_VMH_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_VMH, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_VMH) = "RNA"
VP = list()
for( in c("GABRA5", "NPTX2", "NKX2-1", "ADCYAP1", "FEZF1", "NR5A1", "SOX14", "PCSK1", "TRH", "SST", "OTP", "GAL")){
VP[[x]] = VlnPlot(All_VMH, x, pt.size = 0)  + NoLegend()
}
pdf("VMH_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()

Check = as.data.frame(t(table(OCT_VMH_Assigns_Renamed$Pop)))

```

### DMH: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_DMH = subset(HypothalamicNeurons, idents = c("Neuron_31 [DMH]"))

#Use PCA 10 dims 10 spread 10

DefaultAssay(All_DMH) = "integrated"
All_DMH = RunPCA(All_DMH, npcs = 10)
All_DMH <- RunUMAP(All_DMH, dims = 1:10, spread= 10)
All_DMH_UMAP = as.data.frame(All_DMH@reductions$umap@cell.embeddings)

set.dim = c(10)
set.res = c(1)
set.kparam = c(3,5,10,20)
#ClusterFunc_Neuro_RNA(All_DMH)

All_DMH <- FindNeighbors(All_DMH, k.param=3, dims=1:10)
All_DMH <- FindClusters(All_DMH, resolution =1)
DMH_1 = subset(All_DMH, idents = c(0,2))
DMH_2 = subset(All_DMH, idents = c(4,5,9))

DMH_1_V2 = subset(All_DMH_UMAP, row.names(All_DMH_UMAP) %in% colnames(DMH_1) &  All_DMH_UMAP$UMAP_1 > -5  | row.names(All_DMH_UMAP) %in% colnames(All_DMH) &  All_DMH_UMAP$UMAP_1 > 8) 
DMH_1_V2_Seu = subset(All_DMH, cells = row.names(DMH_1_V2))

DMH_2_V2 = subset(All_DMH_UMAP, row.names(All_DMH_UMAP) %in% colnames(DMH_2)   | row.names(All_DMH_UMAP) %in% colnames(All_DMH) &  All_DMH_UMAP$UMAP_1 < 10 & All_DMH_UMAP$UMAP_2 < -10) 
DMH_2_V2_Seu = subset(All_DMH, cells = row.names(DMH_2_V2))

DMH_3 = subset(All_DMH, cells = c(row.names(DMH_1_V2), row.names(DMH_2_V2)), invert=T)



set.dim = c(10)
set.res = c(0.01)
set.kparam = c(100)
#ClusterFunc_Neuro_RNA(DMH_1_V2_Seu)
#ClusterFunc_Neuro_RNA(DMH_2_V2_Seu)
#ClusterFunc_Neuro_RNA(DMH_3)

CheckInput = colnames(DMH_2_V2_Seu)
CheckUMAP(All_DMH)

OCT_DMH_Assigns_Renamed = as.data.frame(c(rep("DMH_01 [PPP1R17]", length(colnames(DMH_1_V2_Seu))), 
                                 rep("DMH_02 [PPP1R17/NKX2-1/SOX2]", length(colnames(DMH_2_V2_Seu))),               
                                 rep("DMH_03 [SOX2]", length(colnames(DMH_3))))) 
colnames(OCT_DMH_Assigns_Renamed) = "Pop"
row.names(OCT_DMH_Assigns_Renamed) = c(colnames(DMH_1_V2_Seu), #row.names(OCT_DMH_Assigns_Renamed) #OCT_DMH_Assigns_Renamed$Barcs
                                           colnames(DMH_2_V2_Seu),
                                           colnames(DMH_3))


paste(unique(sort(OCT_DMH_Assigns_Renamed$Pop)), collapse = ", ")

OCT_DMH_Assigns_Renamed$Pop = factor(OCT_DMH_Assigns_Renamed$Pop, levels = c("DMH_01 [PPP1R17]", "DMH_02 [PPP1R17/NKX2-1/SOX2]", "DMH_03 [SOX2]"))

All_DMH = AddMetaData(All_DMH, OCT_DMH_Assigns_Renamed, "OCT_DMH_Assigns_Renamed")
Idents(All_DMH) = "OCT_DMH_Assigns_Renamed"

pdf(paste("ALLFETAL_DMH_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_DMH, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_DMH) = "RNA"
VP = list()
for( in c("PPP1R17", "NKX2-1", "GHRH", "SOX2")){
VP[[x]] = VlnPlot(All_DMH, x, pt.size = 0)  + NoLegend()
}
pdf("DMH_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()

```

### SMN: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_SMN = subset(HypothalamicNeurons, idents = c("Neuron_37 [SMN]"))

#Use PCA 30 dims 10 spread 2

DefaultAssay(All_SMN) = "integrated"
All_SMN = RunPCA(All_SMN, npcs = 30)
All_SMN <- RunUMAP(All_SMN, dims = 1:10, spread= 2)
All_SMN_UMAP = as.data.frame(All_SMN@reductions$umap@cell.embeddings)

set.dim = c(10)
set.res = c(5)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(All_SMN)

All_SMN <- FindNeighbors(All_SMN, k.param=3, dims=1:10)
All_SMN <- FindClusters(All_SMN, resolution =1)
SMN_1 = subset(All_SMN, idents = c(2,4,6))

set.dim = c(10)
set.res = c(3)
set.kparam = c(2)
ClusterFunc_Neuro_RNA(SMN_1)

SMN_1_Rm = subset(All_SMN_UMAP, row.names(All_SMN_UMAP) %in% colnames(SMN_1) &  All_SMN_UMAP$UMAP_1 > -3.5 & All_SMN_UMAP$UMAP_2 > -2.2)

SMN_1_V2 = subset(SMN_1, cells = row.names(SMN_1_Rm), invert=T)
All_SMN_Rest1 = subset(All_SMN, cells = colnames(SMN_1_V2), invert=T)

set.dim = c(10)
set.res = c(1)
set.kparam = c(10)
ClusterFunc_Neuro_RNA(All_SMN_Rest1)

All_SMN_Rest1 <- FindNeighbors(All_SMN_Rest1, k.param=10, dims=1:10)
All_SMN_Rest1 <- FindClusters(All_SMN_Rest1, resolution =1)
SMN_2 = subset(All_SMN_Rest1, idents = c(0))
SMN_3 = subset(All_SMN_Rest1, idents = c(1))
SMN_4 = subset(All_SMN_Rest1, idents = c(0,1), invert=T)



CheckInput = row.names(SMN_1_Rm)
CheckUMAP(SMN_1)

OCT_SMN_Assigns_Renamed = as.data.frame(c(rep("SMN_01 [TAC1/SOX2]", length(colnames(SMN_1_V2))), 
                                 rep("SMN_02 [TAC1/EBF3/SOX2]", length(colnames(SMN_2))),               
                                 rep("SMN_03 [TAC1/EBF3/EBF2]", length(colnames(SMN_3))),               
                                 rep("SMN_04 [TAC1/EBF3]", length(colnames(SMN_4))))) 
colnames(OCT_SMN_Assigns_Renamed) = "Pop"
row.names(OCT_SMN_Assigns_Renamed) = c(colnames(SMN_1_V2), #row.names(OCT_SMN_Assigns_Renamed) #OCT_SMN_Assigns_Renamed$Barcs
                                           colnames(SMN_2),
                                           colnames(SMN_3),
                                           colnames(SMN_4))


paste(unique(sort(OCT_SMN_Assigns_Renamed$Pop)), collapse = ", ")

OCT_SMN_Assigns_Renamed$Pop = factor(OCT_SMN_Assigns_Renamed$Pop, levels = c("SMN_01 [TAC1/SOX2]", "SMN_02 [TAC1/EBF3/SOX2]", "SMN_03 [TAC1/EBF3/EBF2]", "SMN_04 [TAC1/EBF3]"))

All_SMN = AddMetaData(All_SMN, OCT_SMN_Assigns_Renamed, "OCT_SMN_Assigns_Renamed")
Idents(All_SMN) = "OCT_SMN_Assigns_Renamed"

pdf(paste("ALLFETAL_SMN_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_SMN, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_SMN) = "RNA"
VP = list()
for( in c("EBF2", "EBF3", "TAC1", "SOX2")){
VP[[x]] = VlnPlot(All_SMN, x, pt.size = 0)  + NoLegend()
}
pdf("SMN_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()

```

### IDTT: DONE
```{r}
DefaultAssay(HypothalamicNeurons) = "integrated"
Idents(HypothalamicNeurons) = "OCT_Neuron_Assigns_Renamed2"
All_IDTT = subset(HypothalamicNeurons, idents = c("Neuron_42 [ID/TT]"))

#PCA 30 dims 10 spread 2

DefaultAssay(All_IDTT) = "integrated"
All_IDTT = RunPCA(All_IDTT, npcs = 30)
All_IDTT <- RunUMAP(All_IDTT, dims = 1:10, spread= 2)
All_IDTT_UMAP = as.data.frame(All_IDTT@reductions$umap@cell.embeddings)

set.dim = c(10)
set.res = c(1)
set.kparam = c(20)
#ClusterFunc_Neuro_RNA(All_IDTT)

All_IDTT <- FindNeighbors(All_IDTT, k.param=20, dims=1:10)
All_IDTT <- FindClusters(All_IDTT, resolution =1)
IDTT_1 = subset(All_IDTT, idents = c(0))
IDTT_2 = subset(All_IDTT, idents = c(1))
#IDTT_1_V2 = subset(All_IDTT_UMAP, row.names(All_IDTT_UMAP) %in% colnames(IDTT_1)  | row.names(All_IDTT_UMAP) %in% colnames(All_IDTT) &  All_IDTT_UMAP$UMAP_1 < -5 & All_IDTT_UMAP$UMAP_2 > -10) 
#IDTT_1_V2_Seu = subset(All_IDTT, cells = row.names(IDTT_1_V2))

#All_IDTT_Rest2 = subset(All_IDTT, cells = colnames(IDTT_1), invert=T)
set.dim = c(10)
set.res = c(1)
set.kparam = c(5)
#ClusterFunc_Neuro_RNA(IDTT_2)

IDTT_2 <- FindNeighbors(IDTT_2, k.param=5, dims=1:10)
IDTT_2 <- FindClusters(IDTT_2, resolution =1)
IDTT_2_V2 = subset(IDTT_2, idents = c(1,2))
IDTT_3 = subset(All_IDTT, cells = c(colnames(IDTT_1), colnames(IDTT_2_V2)), invert=T)

CheckInput = colnames(IDTT_3)
CheckUMAP(All_IDTT)

OCT_IDTT_Assigns_Renamed = as.data.frame(c(rep("IDTT_01 [LHX6/SST]", length(colnames(IDTT_1))),   
                                           rep("IDTT_02 [SST]", length(colnames(IDTT_2_V2))),
                                           rep("IDTT_03", length(colnames(IDTT_3))))) 
colnames(OCT_IDTT_Assigns_Renamed) = "Pop"
row.names(OCT_IDTT_Assigns_Renamed) = c(colnames(IDTT_1), #row.names(OCT_IDTT_Assigns_Renamed) #OCT_IDTT_Assigns_Renamed$Barcs
                                           colnames(IDTT_2_V2),
                                           colnames(IDTT_3))


paste(unique(sort(OCT_IDTT_Assigns_Renamed$Pop)), collapse = ", ")

OCT_IDTT_Assigns_Renamed$Pop = factor(OCT_IDTT_Assigns_Renamed$Pop, levels = c("IDTT_01 [LHX6/SST]", "IDTT_02 [SST]", "IDTT_03"))



All_IDTT = AddMetaData(All_IDTT, OCT_IDTT_Assigns_Renamed, "OCT_IDTT_Assigns_Renamed")
Idents(All_IDTT) = "OCT_IDTT_Assigns_Renamed"

pdf(paste("ALLFETAL_IDTT_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_IDTT, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_IDTT) = "RNA"
VP = list()
for( in c("LHX6", "SST")){
VP[[x]] = VlnPlot(All_IDTT, x, pt.size = 0)  + NoLegend()
}
pdf("IDTT_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()

```



#############################################
###### SUBCLUSTERING OTHER POPULATIONS ######
#############################################

########## MICROGLIA ########## 
```{r}
Micro_Seurat = subset(Fetal.combined, cells = row.names(Microglia))

NewMicroList = c("C1QB", "AIF1", "P2RY12", "MRC1", "SELPLG", "CST3", "RPL4", "EEF1A1", "SLC2A5", "CTSB", "APOE", "TMSB4X", "SPARC")

DefaultAssay(Fetal.combined) = "RNA"
GenesNotFound = subset(NewMicroList, NewMicroList %notin% row.names(Fetal.combined))
DefaultAssay(Fetal.combined) = "integrated"


DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "Micro_Seurat"
for(y in seq(10,50,10)){
DefaultAssay(Micro_Seurat) = "integrated"
Micro_Seurat = RunPCA(Micro_Seurat, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Micro_Seurat <- RunUMAP(Micro_Seurat, dims = 1:z, spread= s)
DefaultAssay(Micro_Seurat) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Micro_Seurat, c( "C1QB", "AIF1", "P2RY12", "MRC1", "SELPLG", "CST3", "RPL4", "EEF1A1", "SLC2A5", "CTSB", "APOE", "TMSB4X", "SPARC"), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Micro_Seurat, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Micro_Seurat, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=15)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()



#Use version 32 - "PCA 50 _dims 10 _spread 5"
DefaultAssay(Micro_Seurat) = "integrated"
Micro_Seurat = RunPCA(Micro_Seurat, npcs = 50)
Micro_Seurat <- RunUMAP(Micro_Seurat, dims = 1:10, spread= 5)
Micro_UMAP = as.data.frame(Micro_Seurat@reductions$umap@cell.embeddings)


CheckInput = colnames(Micro_Seurat) #Check Populations
CheckUMAP(Micro_Seurat)

Micro_3 = subset(Micro_UMAP, Micro_UMAP$UMAP_1 < -12)
Micro_1 = subset(Micro_UMAP, Micro_UMAP$UMAP_1 > -12 & Micro_UMAP$UMAP_2 < 8)
Micro_2 = subset(Micro_UMAP, Micro_UMAP$UMAP_1 > -12 & Micro_UMAP$UMAP_2 > 8)


OCTOBER_MICROGLIA_Assigns = as.data.frame(c(rep("Micro_01", length(row.names(Micro_1))),
                                 rep("Micro_02", length(row.names(Micro_2))), 
                                 rep("Micro_03 [Dividing]", length(row.names(Micro_3)))))
colnames(OCTOBER_MICROGLIA_Assigns) = "Pop"
row.names(OCTOBER_MICROGLIA_Assigns) = c(row.names(Micro_1), 
                            row.names(Micro_2),
                            row.names(Micro_3))

write.csv(OCTOBER_MICROGLIA_Assigns, paste("OCTOBER_MICROGLIA_Assigns", Sys.Date(), ".csv", sep=""))
OCTOBER_MICROGLIA_Assigns$Pop = factor(OCTOBER_MICROGLIA_Assigns$Pop, levels = rev(c("Micro_01", "Micro_02", "Micro_03 [Dividing]")))

Micro_Seurat = AddMetaData(Micro_Seurat, OCTOBER_MICROGLIA_Assigns, "OCTOBER_MICROGLIA_Assigns")
Idents(Micro_Seurat) = "OCTOBER_MICROGLIA_Assigns"

pdf(paste("MICRO_Assignments_UMAP_20SEP.pdf", sep=""), width=12, height=8)
print(DimPlot(Micro_Seurat, reduction = "umap", label=T))
dev.off()


AllCells_PlotOrder = c(unique(c("C1QB", "AIF1", "P2RY12",  "MKI67")))

DefaultAssay(Micro_Seurat) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Micro_Seurat, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16),
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Micro_Seurat, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(2, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_Micro_ToSubcluster_MultiVln_25OCT.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Micro_Seurat@active.ident))*1.5)
print(AllVlns)
dev.off()
```

########## VASCULATURE ########## 
```{r}
NewEndoList = unique(c("SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1", "CCNB1",  "CENPE", "CCL19", "MAFB", "LRRC55", "ATP1B1", "CHN2", "DDC", "LYPD1", "CDKN2B", "PLAUR", "MCAM", "PCDH17", "FBLN2", "SEMA3G", "BMX", "ABCB1", "TEK",
                 "ACTA2", "TAGLN", "ADGRF5", "EMCN", "VTN","ATP13A5", "PTH1R", "KCNJ8", "ABCC9",  "CD82", "CHST1", "PLN", "PDGFRB", "RGS5", "IGFBP2", "LUM", "DCN", "PDGFRA", "COL1A1", "COL1A2", "LUM", "DCN", "FBLN1", "TBX18", "VTN", "OGN", "IGFBP2", "IL33", "PTGDS", "NNAT", "RSPO3", "NOV", "SLC47A1", "APOD"))

DefaultAssay(Fetal.combined) = "RNA"
GenesNotFound = subset(NewEndoList, NewEndoList %notin% row.names(Fetal.combined))
DefaultAssay(Fetal.combined) = "integrated"



DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "Vasculature_Seurat"
for(y in seq(10,50,10)){
DefaultAssay(Vasculature_Seurat) = "integrated"
Vasculature_Seurat = RunPCA(Vasculature_Seurat, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Vasculature_Seurat <- RunUMAP(Vasculature_Seurat, dims = 1:z, spread= s)
DefaultAssay(Vasculature_Seurat) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Vasculature_Seurat, c(NewEndoList), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Vasculature_Seurat, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Vasculature_Seurat, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=42)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()


#Use version 40 - PCA 50 _dims 40 _spread 2
DefaultAssay(Vasculature_Seurat) = "integrated"
Vasculature_Seurat = RunPCA(Vasculature_Seurat, npcs = 50)
Vasculature_Seurat <- RunUMAP(Vasculature_Seurat, dims = 1:40, spread= 2)
Vasc_UMAP = as.data.frame(Vasculature_Seurat@reductions$umap@cell.embeddings)

CheckInput = colnames(Vasculature_Seurat) #Check Populations
CheckUMAP(Vasculature_Seurat)

set.dim =50
set.res=1
set.kparam = 50
ClusterFunc_All_RNA(Vasculature_Seurat)

Vasculature_Seurat <- FindNeighbors(Vasculature_Seurat, k.param=50, dims=1:50)
Vasculature_Seurat <- FindClusters(Vasculature_Seurat, resolution =1)
vSMCandPeriToClean = subset(Vasculature_Seurat, idents= c(0))
EndoVenousToClean = subset(Vasculature_Seurat, idents= c(1))
EndoArterialToClean = subset(Vasculature_Seurat, idents= c(2))

CheckInput = colnames(EndoArterialToClean) #Check Populations
CheckUMAP(Vasculature_Seurat)

EndoArterial_Pt2= subset(Vasc_UMAP, row.names(Vasc_UMAP) %in% colnames(EndoVenousToClean) &  Vasc_UMAP$UMAP_1 < 1.5)
EndoVenous_V2= subset(Vasc_UMAP, row.names(Vasc_UMAP) %in% colnames(EndoVenousToClean) &  Vasc_UMAP$UMAP_1 > 1.5)

EndoVenous_Pt2= subset(Vasc_UMAP, row.names(Vasc_UMAP) %in% colnames(vSMCandPeriToClean) &  Vasc_UMAP$UMAP_1 > 1)
EndoArterial_Pt3= subset(Vasc_UMAP, row.names(Vasc_UMAP) %in% colnames(vSMCandPeriToClean) &  Vasc_UMAP$UMAP_1 > -2.5 &  Vasc_UMAP$UMAP_1 < 1)
vSMCandPeri_V2= subset(Vasc_UMAP, row.names(Vasc_UMAP) %in% colnames(vSMCandPeriToClean) &  Vasc_UMAP$UMAP_1 < -2.5)

EndoVenous_Clean = subset(Vasculature_Seurat, cells= c(row.names(EndoVenous_V2), row.names(EndoVenous_Pt2)))
EndoArterial_Clean = subset(Vasculature_Seurat, cells= c(colnames(EndoArterialToClean), row.names(EndoArterial_Pt2), row.names(EndoArterial_Pt3)))
vSMCandPeri_Seu = subset(Vasculature_Seurat, cells= c(row.names(vSMCandPeri_V2)))

CheckInput = colnames(vSMCandPeri_Seu) #Check Populations
CheckUMAP(Vasculature_Seurat)


set.dim =50
set.res=1
set.kparam = c(5,50)
ClusterFunc_All_RNA(vSMCandPeri_Seu)

vSMCandPeri_Seu <- FindNeighbors(vSMCandPeri_Seu, k.param=50, dims=1:50)
vSMCandPeri_Seu <- FindClusters(vSMCandPeri_Seu, resolution =1)
vSMC_Pt1 = subset(vSMCandPeri_Seu, idents= c(2))
Pericytes_ToClean = subset(vSMCandPeri_Seu, idents= c(2), invert=T)


CheckInput = colnames(Pericytes_ToClean) #Check Populations
CheckUMAP(vSMCandPeri_Seu)

vSMC_Pt2= subset(Vasc_UMAP, row.names(Vasc_UMAP) %in% colnames(Pericytes_ToClean) &  Vasc_UMAP$UMAP_1 > -8.5 & Vasc_UMAP$UMAP_1 < -5 & Vasc_UMAP$UMAP_2 < -4)
vSMC_V2 = subset(Vasculature_Seurat, cells= c(row.names(vSMC_Pt2), colnames(vSMC_Pt1)))
set.dim =50
set.res=1
set.kparam = c(2)
ClusterFunc_All_RNA(vSMC_V2)


vSMC_V2 <- FindNeighbors(vSMC_V2, k.param=2, dims=1:50)
vSMC_V2 <- FindClusters(vSMC_V2, resolution =1)
vSMC_Clean = subset(vSMC_V2, idents= c(6,8), invert=T)

Pericytes_Clean = subset(vSMCandPeri_Seu, cells = colnames(vSMC_Clean), invert=T)


CheckInput = colnames(vSMC_Clean) #Check Populations
CheckUMAP(vSMCandPeri_Seu)


OCTOBER_ENDOMURAL_Assigns = as.data.frame(c(rep("Mural_02 [vSMC]", length(colnames(vSMC_Clean))),
                                 rep("Endothelial_01 [Arterial]", length(colnames(EndoArterial_Clean))), 
                                 rep("Endothelial_02 [Venous]", length(colnames(EndoVenous_Clean))), 
                                 rep("Mural_01 [Pericytes]", length(colnames(Pericytes_Clean)))))
colnames(OCTOBER_ENDOMURAL_Assigns) = "Pop"
row.names(OCTOBER_ENDOMURAL_Assigns) = c(colnames(vSMC_Clean), 
                            colnames(EndoArterial_Clean),
                            colnames(EndoVenous_Clean),
                            colnames(Pericytes_Clean))

write.csv(OCTOBER_ENDOMURAL_Assigns, paste("OCTOBER_ENDOMURAL_Assigns", Sys.Date(), ".csv", sep=""))
OCTOBER_ENDOMURAL_Assigns$Pop = factor(OCTOBER_ENDOMURAL_Assigns$Pop, levels = rev(c("Endothelial_01 [Arterial]", "Endothelial_02 [Venous]", "Mural_01 [Pericytes]", "Mural_02 [vSMC]")))

Vasculature_Seurat = AddMetaData(Vasculature_Seurat, OCTOBER_ENDOMURAL_Assigns, "OCTOBER_ENDOMURAL_Assigns")
Idents(Vasculature_Seurat) = "OCTOBER_ENDOMURAL_Assigns"

pdf(paste("Vascu_Assignments_UMAP_20SEP.pdf", sep=""), width=12, height=8)
print(DimPlot(Vasculature_Seurat, reduction = "umap", label=T))
dev.off()






AllCells_PlotOrder = c(unique(c("SLC38A5","CLDN5", "ITM2A", "FLT1","KCNJ8", "ABCC9","ACTA2", "TAGLN","LUM","PDGFRA","PTGDS")))
DefaultAssay(Vasculature_Seurat) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Vasculature_Seurat, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16),
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Vasculature_Seurat, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(3.7, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_Vascu_ToSubcluster_MultiVln_25OCT21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Vasculature_Seurat@active.ident))*1.5)
print(AllVlns)
dev.off()


```

########## VLMC ########## 
```{r}
NewEndoList = unique(c("SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1", "CCNB1",  "CENPE", "CCL19", "MAFB", "LRRC55", "ATP1B1", "CHN2", "DDC", "LYPD1", "CDKN2B", "PLAUR", "MCAM", "PCDH17", "FBLN2", "SEMA3G", "BMX", "ABCB1", "TEK",
                 "ACTA2", "TAGLN", "ADGRF5", "EMCN", "VTN","ATP13A5", "PTH1R", "KCNJ8", "ABCC9",  "CD82", "CHST1", "PLN", "PDGFRB", "RGS5", "IGFBP2", "LUM", "DCN", "PDGFRA", "COL1A1", "COL1A2", "LUM", "DCN", "FBLN1", "TBX18", "VTN", "OGN", "IGFBP2", "IL33", "PTGDS", "NNAT", "RSPO3", "NOV", "SLC47A1", "APOD"))

DefaultAssay(Fetal.combined) = "RNA"
GenesNotFound = subset(NewEndoList, NewEndoList %notin% row.names(Fetal.combined))
DefaultAssay(Fetal.combined) = "integrated"

VLMC_Seurat = subset(Fetal.combined, cells = row.names(VLMC))


DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "VLMC_Seurat"
for(y in seq(10,50,10)){
DefaultAssay(VLMC_Seurat) = "integrated"
VLMC_Seurat = RunPCA(VLMC_Seurat, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
VLMC_Seurat <- RunUMAP(VLMC_Seurat, dims = 1:z, spread= s)
DefaultAssay(VLMC_Seurat) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(VLMC_Seurat, c(NewEndoList), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(VLMC_Seurat, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(VLMC_Seurat, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=42)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()

#Use version 43 - 50,50,2
DefaultAssay(VLMC_Seurat) = "integrated"
VLMC_Seurat = RunPCA(VLMC_Seurat, npcs = 50)
VLMC_Seurat <- RunUMAP(VLMC_Seurat, dims = 1:50, spread= 2)

```

########## OLIGODENDROCYTES ##########
```{r}
DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "Oligodendro_Final"
for(y in seq(10,50,10)){
DefaultAssay(Oligodendro_Final) = "integrated"
Oligodendro_Final = RunPCA(Oligodendro_Final, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Oligodendro_Final <- RunUMAP(Oligodendro_Final, dims = 1:z, spread= s)
DefaultAssay(Oligodendro_Final) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Oligodendro_Final, c("APOD", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9"), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Oligodendro_Final, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Oligodendro_Final, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))

DPlist3[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Oligodendro_Final, reduction="umap", group.by = "SEP_AllCell_Assigns", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=20)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_PRIORASSIGNs.pdf", sep=""), width=8, height=5)
print(DPlist3)
dev.off()


#Use version 7 - PCA 20 _dims 20 _spread 2
DefaultAssay(Oligodendro_Final) = "integrated"
Oligodendro_Final = RunPCA(Oligodendro_Final, npcs = 20)
Oligodendro_Final <- RunUMAP(Oligodendro_Final, dims = 1:20, spread= 2)
Oligo_UMAP = as.data.frame(Oligodendro_Final@reductions$umap@cell.embeddings)


CheckInput = colnames(Oligodendro_Final) #Check Populations
CheckUMAP(Oligodendro_Final)
ImmatureOlig1 = subset(Oligo_UMAP, Oligo_UMAP$UMAP_1 < 0 & Oligo_UMAP$UMAP_2 < -5)

Oligo_Rest1 = subset(Oligodendro_Final, cells = c(row.names(ImmatureOlig1)), invert=T)

set.dim = 20
set.res = c(1)
set.kparam = c(20,50,100)
ClusterFunc_All_RNA(Oligo_Rest1)

DefaultAssay(Oligo_Rest1) = "integrated"
Oligo_Rest1 <- FindNeighbors(Oligo_Rest1, k.param=20, dims=1:20)
Oligo_Rest1 <- FindClusters(Oligo_Rest1, resolution =1)
ImmatureOlig2= subset(Oligo_Rest1, idents = 0) #[PDGFRA/PLLP/FYN/CNP]
ImmatureOlig3= subset(Oligo_Rest1, idents = 3) #[PDGFRA/PLLP/CNP]
ImmatureOlig4= subset(Oligo_Rest1, idents = 4) #[PDGFRA/PLLP/FYN/CNP] #Also SPINT2/SCG2/PNCK
ImmatureOlig5= subset(Oligo_Rest1, idents = 5) #[PDGFRA/PLLP/FYN/CNP]
ImmatureOlig6= subset(Oligo_Rest1, idents = 6) #[PDGFRA/PLLP/TCF7L2] #Also CALCRL
ImmatureOlig7= subset(Oligo_Rest1, idents = 7) #[PDGFRA/PLLP/CNP] #Also PLAT
ImmatureOlig8= subset(Oligo_Rest1, idents = 8) #[PDGFRA/PLLP/FYN/CNP] #AFAP1L2
ImmatureOlig9= subset(Oligo_Rest1, idents = 9) #[PDGFRA/PLLP/CNP] #Also TYMS/ZWINT/CENPU/MAD2L1/CDK1/CENPK/KIAA0101
ImmatureOlig10= subset(Oligo_Rest1, idents = 10) #[PDGFRA/FYN/CNP]
IntermediateOlig = subset(Oligo_Rest1, idents = 1)
MatureOlig = subset(Oligo_Rest1, idents = 2)

#Find new markers
DefaultAssay(Oligo_Rest1) = "RNA"
Oligo_Rest1_DEGs = DEGs(Oligo_Rest1)
Oligo_Rest1_DEGs_Subs = subset(Oligo_Rest1_DEGs, Oligo_Rest1_DEGs$cluster %notin% c(1,2) & Oligo_Rest1_DEGs$p_val_adj < 0.05 & Oligo_Rest1_DEGs$avg_log2FC > 1 & Oligo_Rest1_DEGs$gene %in% HumanTFs$V1) # 


DefaultAssay(Oligo_Rest1) = "RNA"
fplist = list()
for(g in unique(Oligo_Rest1_DEGs_Subs$gene)){
fplist[[g]] = FeaturePlot(Oligo_Rest1, g, reduction="umap")
}
pdf("IndivFeatPlots_Oligo_FromDEGs_TFs.pdf", width = 10, height = 8)
print(fplist)
dev.off()


#Clean assignments
Immat2_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig2) & Oligo_UMAP$UMAP_1 < 9 & Oligo_UMAP$UMAP_1 > 3)
Immat2_MovetoImmat3 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig2) & Oligo_UMAP$UMAP_2 > 2 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < 0)
Immat2_MovetoImmat5 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig2) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_2 < 6 & Oligo_UMAP$UMAP_1 > -4.5 & Oligo_UMAP$UMAP_1 < -1)
Immat2_MovetoImmat6 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig2) & Oligo_UMAP$UMAP_2 < -0.5 & Oligo_UMAP$UMAP_1 < -5)
Immat2_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig2) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 < -5)
Immat2_MovetoImmat9 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig2) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_1 < -4.5)
Immat2_Keep = subset(ImmatureOlig2, cells = c(row.names(Immat2_MovetoImmed), row.names(Immat2_MovetoImmat5), row.names(Immat2_MovetoImmat6), row.names(Immat2_MovetoImmat7), row.names(Immat2_MovetoImmat9), row.names(Immat2_MovetoImmat3)), invert=T)



Immat3_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig3) & Oligo_UMAP$UMAP_1 > 0.5 & Oligo_UMAP$UMAP_1 < 2.5)
Immat3_MovetoImmat6 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig3) & Oligo_UMAP$UMAP_2 < -0.5 & Oligo_UMAP$UMAP_1 < -5)
Immat3_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig3) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 < -6)
Immat3_MovetoImmat8 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig3) & row.names(Oligo_UMAP) %notin% row.names(Immat3_MovetoImmed) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 2.5 & Oligo_UMAP$UMAP_1 > -1)
Immat3_MovetoImmat9 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig3) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_1 < -6)
Immat3_Keep = subset(ImmatureOlig3, cells = c(row.names(Immat3_MovetoImmed), row.names(Immat3_MovetoImmat6), row.names(Immat3_MovetoImmat8), row.names(Immat3_MovetoImmat7), row.names(Immat3_MovetoImmat9)), invert=T)



Immat4_MovetoImmat3 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 > 0 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < -1)
Immat4_MovetoImmat5 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_2 < 4.5 & Oligo_UMAP$UMAP_1 > -4.5 & Oligo_UMAP$UMAP_1 < -1)
Immat4_MovetoImmat6 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 < -0.5 & Oligo_UMAP$UMAP_1 < -5)
Immat4_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 < -5)
Immat4_MovetoImmat8 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 > -1 & Oligo_UMAP$UMAP_2 < 2.5 & Oligo_UMAP$UMAP_1 > -1 & Oligo_UMAP$UMAP_1 < 3)
Immat4_MovetoImmat2 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & row.names(Oligo_UMAP) %notin% row.names(Immat4_MovetoImmat8) & Oligo_UMAP$UMAP_2 > -5 & Oligo_UMAP$UMAP_2 < 0 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < 0.5)
Immat4_MovetoImmat9 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_1 < -4.5)
Immat4_MovetoImmat10 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_2 > 2.5 & Oligo_UMAP$UMAP_2 < 5 & Oligo_UMAP$UMAP_1 > -1 & Oligo_UMAP$UMAP_1 < 3)
Immat4_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig4) & Oligo_UMAP$UMAP_1 < 9 & Oligo_UMAP$UMAP_1 > 3)
Immat4_Keep = subset(ImmatureOlig4, cells = c(row.names(Immat4_MovetoImmat2), row.names(Immat4_MovetoImmat3), row.names(Immat4_MovetoImmat5), row.names(Immat4_MovetoImmat6), row.names(Immat4_MovetoImmat7), row.names(Immat4_MovetoImmat8), row.names(Immat4_MovetoImmat9), row.names(Immat4_MovetoImmat10), row.names(Immat4_MovetoImmed)), invert=T)

Immat5_MovetoImmat3 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig5) & Oligo_UMAP$UMAP_2 > 0 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < -1)
Immat5_MovetoImmat6 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig5) & Oligo_UMAP$UMAP_2 < -0.5 & Oligo_UMAP$UMAP_1 < -5)
Immat5_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig5) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 < -5)
Immat5_MovetoImmat8 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig5) & Oligo_UMAP$UMAP_2 > -1.5 & Oligo_UMAP$UMAP_2 < 2.5 & Oligo_UMAP$UMAP_1 > -1 & Oligo_UMAP$UMAP_1 < 3)
Immat5_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig5) & Oligo_UMAP$UMAP_1 < 9 & Oligo_UMAP$UMAP_1 > 3)
Immat5_Keep = subset(ImmatureOlig5, cells = c(row.names(Immat5_MovetoImmat3), row.names(Immat5_MovetoImmat6), row.names(Immat5_MovetoImmat7), row.names(Immat5_MovetoImmat8), row.names(Immat5_MovetoImmed)), invert=T)


Immat6_MovetoImmat3 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig6) & Oligo_UMAP$UMAP_2 > 0 & Oligo_UMAP$UMAP_2 < 4 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < -1)
Immat6_MovetoImmat4 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig6) & Oligo_UMAP$UMAP_2 > 6)
Immat6_MovetoImmat8 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig6) & Oligo_UMAP$UMAP_2 > -2 & Oligo_UMAP$UMAP_2 < 2.5 & Oligo_UMAP$UMAP_1 > -1 & Oligo_UMAP$UMAP_1 < 3)
Immat6_MovetoImmat9 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig6) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_1 < -4.5)
Immat6_Keep = subset(ImmatureOlig6, cells = c(row.names(Immat6_MovetoImmat3), row.names(Immat6_MovetoImmat4), row.names(Immat6_MovetoImmat8), row.names(Immat6_MovetoImmat9)), invert=T)


Immat7_MovetoImmat8 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig7) & Oligo_UMAP$UMAP_2 > -1 & Oligo_UMAP$UMAP_2 < 2.5 & Oligo_UMAP$UMAP_1 > -1 & Oligo_UMAP$UMAP_1 < 3)
Immat7_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig7) & Oligo_UMAP$UMAP_1 < 9 & Oligo_UMAP$UMAP_1 > 3)
Immat7_Keep = subset(ImmatureOlig7, cells = c(row.names(Immat7_MovetoImmed), row.names(Immat7_MovetoImmat8)), invert=T)


Immat8_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig8) & Oligo_UMAP$UMAP_1 < 9 & Oligo_UMAP$UMAP_1 > 4)
Immat8_MovetoImmat6 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig8) & Oligo_UMAP$UMAP_2 < -0.5 & Oligo_UMAP$UMAP_1 < -4)
Immat8_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig8) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 < -5)
Immat8_MovetoImmat9 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig8) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_1 < -4.5)
Immat8_Keep = subset(ImmatureOlig8, cells = c(row.names(Immat8_MovetoImmed), row.names(Immat8_MovetoImmat6), row.names(Immat8_MovetoImmat7), row.names(Immat8_MovetoImmat9)), invert=T)

Immat9_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig9) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 1.5 & Oligo_UMAP$UMAP_1 < -5)
Immat9_Keep = subset(ImmatureOlig9, cells = c(row.names(Immat9_MovetoImmat7)), invert=T)


Immat10_MovetoImmat3 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_2 > 0 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < -2)
Immat10_MovetoImmat6 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_2 < -0.5 & Oligo_UMAP$UMAP_1 < -5)
Immat10_MovetoImmat7 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_2 > -0.5 & Oligo_UMAP$UMAP_2 < 3 & Oligo_UMAP$UMAP_1 < -5)
Immat10_MovetoImmat8 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_2 > -1 & Oligo_UMAP$UMAP_2 < 1 & Oligo_UMAP$UMAP_1 > -1 & Oligo_UMAP$UMAP_1 < 2)
Immat10_MovetoImmat9 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_1 < -4.5)
Immat10_MovetoImmed = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_1 < 9 & Oligo_UMAP$UMAP_1 > 1.5)
Immat10_MovetoMature = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(ImmatureOlig10) & Oligo_UMAP$UMAP_1 > 9)
Immat10_Keep = subset(ImmatureOlig10, cells = c(row.names(Immat10_MovetoImmat3), row.names(Immat10_MovetoImmat6), row.names(Immat10_MovetoImmat7), row.names(Immat10_MovetoImmat8), row.names(Immat10_MovetoImmat9), row.names(Immat10_MovetoImmed), row.names(Immat10_MovetoMature)), invert=T)

Intermed_MovetoImmat5 = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(IntermediateOlig) & Oligo_UMAP$UMAP_2 > 3 & Oligo_UMAP$UMAP_2 < 5.5 & Oligo_UMAP$UMAP_1 > -5 & Oligo_UMAP$UMAP_1 < -1)
Intermed_MovetoMature = subset(Oligo_UMAP, row.names(Oligo_UMAP) %in% colnames(IntermediateOlig) & Oligo_UMAP$UMAP_1 > 11)
Intermed_Keep = subset(IntermediateOlig, cells = c(row.names(Intermed_MovetoImmat5), row.names(Intermed_MovetoMature)), invert=T)

#MatureOlig = Clean
#ImmatureOlig1 = Clean


###Recombine
ImmatureOlig2_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat2_Keep), row.names(Immat4_MovetoImmat2)))
ImmatureOlig3_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat3_Keep), row.names(Immat2_MovetoImmat3), row.names(Immat4_MovetoImmat3), row.names(Immat5_MovetoImmat3), row.names(Immat6_MovetoImmat3), row.names(Immat10_MovetoImmat3)))
ImmatureOlig4_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat4_Keep), row.names(Immat6_MovetoImmat4)))
ImmatureOlig5_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat5_Keep), row.names(Immat2_MovetoImmat5), row.names(Immat4_MovetoImmat5), row.names(Intermed_MovetoImmat5)))
ImmatureOlig6_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat6_Keep), row.names(Immat2_MovetoImmat6), row.names(Immat3_MovetoImmat6), row.names(Immat4_MovetoImmat6), row.names(Immat5_MovetoImmat6), row.names(Immat8_MovetoImmat6), row.names(Immat10_MovetoImmat6)))
ImmatureOlig7_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat7_Keep), row.names(Immat2_MovetoImmat7), row.names(Immat3_MovetoImmat7), row.names(Immat4_MovetoImmat7), row.names(Immat5_MovetoImmat7), row.names(Immat8_MovetoImmat7), row.names(Immat9_MovetoImmat7), row.names(Immat10_MovetoImmat7)))
ImmatureOlig8_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat8_Keep), row.names(Immat3_MovetoImmat8), row.names(Immat4_MovetoImmat8), row.names(Immat5_MovetoImmat8), row.names(Immat6_MovetoImmat8), row.names(Immat7_MovetoImmat8), row.names(Immat10_MovetoImmat8)))
ImmatureOlig9_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat9_Keep), row.names(Immat2_MovetoImmat9), row.names(Immat3_MovetoImmat9), row.names(Immat4_MovetoImmat9), row.names(Immat6_MovetoImmat9), row.names(Immat8_MovetoImmat9), row.names(Immat10_MovetoImmat9)))
ImmatureOlig10_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Immat10_Keep), row.names(Immat4_MovetoImmat10)))
IntermediateOlig_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(Intermed_Keep), row.names(Immat2_MovetoImmed), row.names(Immat3_MovetoImmed), row.names(Immat4_MovetoImmed), row.names(Immat5_MovetoImmed), row.names(Immat7_MovetoImmed), row.names(Immat8_MovetoImmed), row.names(Immat10_MovetoImmed)))
MatureOlig_RECOMBO = subset(Oligodendro_Final, cells = c(colnames(MatureOlig), row.names(Immat10_MovetoMature), row.names(Intermed_MovetoMature)))





OCTOBER_OLIGO_Assigns = as.data.frame(c(rep("Oligodendrocytes_01 [Immature]", length(row.names(ImmatureOlig1))),
                                 rep("Oligodendrocytes_02 [Immature]", length(colnames(ImmatureOlig2_RECOMBO))), 
                                 rep("Oligodendrocytes_03 [Immature]", length(colnames(ImmatureOlig3_RECOMBO))), 
                                 rep("Oligodendrocytes_04 [Immature]", length(colnames(ImmatureOlig4_RECOMBO))), 
                                 rep("Oligodendrocytes_05 [Immature]", length(colnames(ImmatureOlig5_RECOMBO))), 
                                 rep("Oligodendrocytes_06 [Immature]", length(colnames(ImmatureOlig6_RECOMBO))), 
                                 rep("Oligodendrocytes_07 [Immature]", length(colnames(ImmatureOlig7_RECOMBO))), 
                                 rep("Oligodendrocytes_08 [Immature]", length(colnames(ImmatureOlig8_RECOMBO))), 
                                 rep("Oligodendrocytes_09 [Immature]", length(colnames(ImmatureOlig9_RECOMBO))), 
                                 rep("Oligodendrocytes_10 [Immature]", length(colnames(ImmatureOlig10_RECOMBO))), 
                                 rep("Oligodendrocytes_11 [Intermediate]", length(colnames(IntermediateOlig_RECOMBO))), 
                                 rep("Oligodendrocytes_12 [Mature]", length(colnames(MatureOlig_RECOMBO)))))
colnames(OCTOBER_OLIGO_Assigns) = "Pop"
row.names(OCTOBER_OLIGO_Assigns) = c(row.names(ImmatureOlig1), 
                            colnames(ImmatureOlig2_RECOMBO),
                            colnames(ImmatureOlig3_RECOMBO),
                            colnames(ImmatureOlig4_RECOMBO),
                            colnames(ImmatureOlig5_RECOMBO),
                            colnames(ImmatureOlig6_RECOMBO),
                            colnames(ImmatureOlig7_RECOMBO),
                            colnames(ImmatureOlig8_RECOMBO),
                            colnames(ImmatureOlig9_RECOMBO),
                            colnames(ImmatureOlig10_RECOMBO),
                            colnames(IntermediateOlig_RECOMBO),
                            colnames(MatureOlig_RECOMBO))

#paste(sort(unique(OCTOBER_OLIGO_Assigns$Pop)), collapse=",")

write.csv(OCTOBER_OLIGO_Assigns, paste("OCTOBER_OLIGO_Assigns", Sys.Date(), ".csv", sep=""))
OCTOBER_OLIGO_Assigns$Pop = factor(OCTOBER_OLIGO_Assigns$Pop, levels = sort(unique(OCTOBER_OLIGO_Assigns$Pop)))

Oligodendro_Final = AddMetaData(Oligodendro_Final, OCTOBER_OLIGO_Assigns, "OCTOBER_OLIGO_Assigns")
Idents(Oligodendro_Final) = "OCTOBER_OLIGO_Assigns"

pdf(paste("OLIGO_Assignments_UMAP_26OCT.pdf", sep=""), width=12, height=8)
print(DimPlot(Oligodendro_Final, reduction = "umap"))
dev.off()


AllCells_PlotOrder = c(unique(c("NKX2-2", "RTN1", "HMP19", "SOX6", "FGF12", "MBP", "BCAS1", "KCTD12", "ZNF488", "TOP2A", "MKI67", "MCM6", "CLSPN", "GINS2", "FABP7", "ISG15", "CLDN11")))


AllCells_PlotOrder = c(unique(c("APOD","OLIG2", "OLIG1","CD9", "PDGFRA", "PLLP", "FYN","CSPG4","CNP", "MAG",   "TNS3",  "TCF7L2", "MOG", "PLP1", "SERINC5", "OLIG1")))

DefaultAssay(Oligodendro_Final) = "RNA"
StorePlots = list()
AllVlns =list()
  for(x in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Oligodendro_Final, features = x, pt.size = 0, same.y.lims = F)
            plotA <- plotA + coord_flip()+ theme(axis.ticks.x = element_blank(), axis.title.y=element_blank(), axis.text.x =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16), axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for(x in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Oligodendro_Final, features = x, pt.size = 0, same.y.lims = F)
            plotB <- plotB + coord_flip()+ theme(axis.ticks.x = element_blank(), axis.title.y=element_blank(), axis.text.x =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(3, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_Oligodendro_Final_MultiVln_6OCT21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Oligodendro_Final@active.ident))*1.5)
print(AllVlns)
dev.off()
```

########## ASTROCYTES ##########
```{r}
DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "Astrocytes_Clean"
for(y in seq(10,50,10)){
DefaultAssay(Astrocytes_Clean) = "integrated"
Astrocytes_Clean = RunPCA(Astrocytes_Clean, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Astrocytes_Clean <- RunUMAP(Astrocytes_Clean, dims = 1:z, spread= s)
DefaultAssay(Astrocytes_Clean) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Astrocytes_Clean, unique(c("GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "ALDOC", "SLC1A2", "ID3", "ASCL1", "APOE", "OLIG1", "OLIG2", "ASCL1", "EGFR", "FOXJ1", "RAX", "ADM", "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "NPAS3", "RFX2", "RFX3", "RFX4", "COL23A1", "CD59", "SLC17A8", "VCAN", "FRZB", "PENK", "COL25A1", "CACNA2D2", "STOML3", "TMEM212", "SLC7A11", "PCP4L1", "PLTP", "TGFB2", "NR2E1", "EPHB1", "P3H2", "NELL2", "FGF10", "RGCC", "GRIA2", "RLBP1", "SIX6", "SCN7A", "IGFBP5", "MEST", "RGS7BP")), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Astrocytes_Clean, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Astrocytes_Clean, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))

}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=40)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()


#IterationNames10to50 = names(FPlist)

#Use version 20 - "PCA 40 _dims 10 _spread 5"
DefaultAssay(Astrocytes_Clean) = "integrated"
Astrocytes_Clean = RunPCA(Astrocytes_Clean, npcs = 40)
Astrocytes_Clean <- RunUMAP(Astrocytes_Clean, dims = 1:10, spread= 5)
Astro_UMAP = as.data.frame(Astrocytes_Clean@reductions$umap@cell.embeddings)

set.dim = 10
set.kparam = 20
set.res = 1
#ClusterFunc_All_RNA(Astrocytes_Clean)



Astrocytes_Clean <- FindNeighbors(Astrocytes_Clean, k.param=20, dims=1:10)
Astrocytes_Clean <- FindClusters(Astrocytes_Clean, resolution =1)
Astro_1 = subset(Astrocytes_Clean, idents= c(1,2,5,4), invert=T)
Astro_2 = subset(Astrocytes_Clean, idents= c(1,2,5))
Astro_3 = subset(Astrocytes_Clean, idents= c(4))


SEPTEMBER_ASTRO_Assigns = as.data.frame(c(rep("Astro_01", length(colnames(Astro_1))),
                                 rep("Astro_02", length(colnames(Astro_2))), 
                                 rep("Astro_03", length(colnames(Astro_3)))))
colnames(SEPTEMBER_ASTRO_Assigns) = "Pop"
row.names(SEPTEMBER_ASTRO_Assigns) = c(colnames(Astro_1), 
                            colnames(Astro_2),
                            colnames(Astro_3))

write.csv(SEPTEMBER_ASTRO_Assigns, paste("SEPTEMBER_ASTRO_Assigns", Sys.Date(), ".csv", sep=""))
SEPTEMBER_ASTRO_Assigns$Pop = factor(SEPTEMBER_ASTRO_Assigns$Pop, levels = sort(unique(SEPTEMBER_ASTRO_Assigns$Pop)))
Astrocytes_Clean = AddMetaData(Astrocytes_Clean, SEPTEMBER_ASTRO_Assigns, "SEPTEMBER_ASTRO_Assigns")
Idents(Astrocytes_Clean) = "SEPTEMBER_ASTRO_Assigns"

pdf(paste("ASTRO_Assignments_UMAP_20SEP.pdf", sep=""), width=12, height=8)
print(DimPlot(Astrocytes_Clean, reduction = "umap", label=T))
dev.off()




    #Check
SeuFilez = Astrocytes_Clean 
Cly = colnames(Astro_1_Recombo)
#Cly = row.names(AstroEpendymal_MoveToOlig)
CheckMeta = as.data.frame(c(rep("RoI", length(Cly))))
colnames(CheckMeta) = "Pop"
row.names(CheckMeta) = Cly
SeuFilez = AddMetaData(SeuFilez, CheckMeta, "CheckMeta")
Idents(SeuFilez) = "CheckMeta"
DimPlot(SeuFilez, reduction="umap")





AllCells_PlotOrder = c(unique(c("GFAP", "AQP4","APOE", "SLC1A3", "S100B", "SLC1A2","ALDOC",  "GJA1", "AGT",  "ID3",    "VCAN", "SLC7A11",  "NELL2", "RGCC", "GRIA2")))

DefaultAssay(Astrocytes_Clean) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Astrocytes_Clean, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16),
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Astrocytes_Clean, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(2, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_Astrocytes_Clean_MultiVln_6OCT21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Astrocytes_Clean@active.ident))*1.5)
print(AllVlns)
dev.off()
```

########## EPENDYMAL ##########
```{r}
Ependy_Final_Seu = subset(Fetal.combined, idents = "Ependymocytes")


DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "Ependy_Final_Seu"
for(y in seq(10,50,10)){
DefaultAssay(Ependy_Final_Seu) = "integrated"
Ependy_Final_Seu = RunPCA(Ependy_Final_Seu, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Ependy_Final_Seu <- RunUMAP(Ependy_Final_Seu, dims = 1:z, spread= s)
DefaultAssay(Ependy_Final_Seu) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Ependy_Final_Seu, unique(c("GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "ALDOC", "SLC1A2", "ID3", "ASCL1", "APOE", "FOXJ1", "RAX", "ADM", "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "NPAS3", "RFX2", "RFX3", "RFX4", "COL23A1", "CD59", "SLC17A8", "VCAN", "FRZB", "PENK", "COL25A1", "CACNA2D2", "STOML3", "TMEM212", "SLC7A11", "PCP4L1", "PLTP", "TGFB2", "NR2E1", "EPHB1", "P3H2", "NELL2", "FGF10", "RGCC", "GRIA2", "RLBP1", "SIX6", "SCN7A", "MEST", "IGFBP5", "RGS7BP")), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Ependy_Final_Seu, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Ependy_Final_Seu, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))
}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=40)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()

#Use version 38 - PCA 50 _dims 30 _spread 5
DefaultAssay(Ependy_Final_Seu) = "integrated"
Ependy_Final_Seu = RunPCA(Ependy_Final_Seu, npcs = 50)
Ependy_Final_Seu <- RunUMAP(Ependy_Final_Seu, dims = 1:30, spread= 5)
Ependy_UMAP = as.data.frame(Ependy_Final_Seu@reductions$umap@cell.embeddings)

set.dim = 30
set.kparam = 60
set.res = c(1)
#ClusterFunc_All_RNA(Ependy_Final_Seu)



Ependy_1 = subset(Ependy_UMAP, Ependy_UMAP$UMAP_1 > 0)
Ependy_2 = subset(Ependy_UMAP, Ependy_UMAP$UMAP_1 < 0 & Ependy_UMAP$UMAP_2 < -3.5)
Ependy_3 = subset(Ependy_UMAP, Ependy_UMAP$UMAP_1 < 0 & Ependy_UMAP$UMAP_2 > 14)
Ependy_4 = subset(Ependy_UMAP, Ependy_UMAP$UMAP_1 < 0 & Ependy_UMAP$UMAP_2 < 14 & Ependy_UMAP$UMAP_2 > 0)


CheckInput = row.names(Ependy_4) #Check Populations
CheckUMAP(Ependy_Final_Seu)


OCTOBER_EPENDY_Assigns = as.data.frame(c(rep("Ependy_01", length(row.names(Ependy_1))), 
                                 rep("Ependy_02", length(row.names(Ependy_2))), 
                                 rep("Ependy_03", length(row.names(Ependy_3))), 
                                 rep("Ependy_04", length(row.names(Ependy_4)))))
colnames(OCTOBER_EPENDY_Assigns) = "Pop"
row.names(OCTOBER_EPENDY_Assigns) = c(row.names(Ependy_1),
                            row.names(Ependy_2),
                            row.names(Ependy_3),
                            row.names(Ependy_4))

write.csv(OCTOBER_EPENDY_Assigns, paste("OCTOBER_EPENDY_Assigns", Sys.Date(), ".csv", sep=""))
OCTOBER_EPENDY_Assigns$Pop = factor(OCTOBER_EPENDY_Assigns$Pop, levels = sort(unique(OCTOBER_EPENDY_Assigns$Pop)))

Ependy_Final_Seu = AddMetaData(Ependy_Final_Seu, OCTOBER_EPENDY_Assigns, "OCTOBER_EPENDY_Assigns")
Idents(Ependy_Final_Seu) = "OCTOBER_EPENDY_Assigns"

pdf(paste("EPENDY_Assignments_UMAP_20SEP.pdf", sep=""), width=12, height=8)
print(DimPlot(Ependy_Final_Seu, reduction = "umap", label=T))
dev.off()




#SEPTEMBER_OLIGO_Assigns$Pop = factor(SEPTEMBER_OLIGO_Assigns$Pop, levels = rev(c("Olig_1", "Olig_2", "Olig_3", "Olig_4", "Olig_5", "Olig_6", "Olig_7", "Olig_8", "Olig_9", "Olig_10", "Olig_11", "Olig_12", "Olig_13", "Olig_14")))

AllCells_PlotOrder = c(unique(c("AQP4","SLC1A3","FOXJ1","CCDC153", "NPAS3", "RFX2", "RFX3", "RFX4", "GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "ALDOC", "SLC1A2", "ID3", "ASCL1", "APOE", "FOXJ1", "RAX", "ADM", "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "NPAS3", "RFX2", "RFX3", "RFX4", "COL23A1", "CD59", "SLC17A8", "VCAN", "FRZB", "PENK", "COL25A1", "CACNA2D2", "STOML3", "TMEM212", "SLC7A11", "PCP4L1", "PLTP", "TGFB2", "NR2E1", "EPHB1", "P3H2", "NELL2", "FGF10", "RGCC", "GRIA2", "RLBP1", "SIX6", "SCN7A", "MEST", "IGFBP5", "RGS7BP")))

DefaultAssay(Ependy_Final_Seu) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Ependy_Final_Seu, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16),
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Ependy_Final_Seu, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(2, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_Ependy_ToSubcluster_MultiVln_6OCT21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Ependy_Final_Seu@active.ident))*1.5)
print(AllVlns)
dev.off()
```

########## TANYCYTES ##########
```{r}
DPlist2 = list()
DPlist = list()
FPlist = list()

Filename = "Tany_Final"
for(y in seq(10,50,10)){
DefaultAssay(Tany_Final) = "integrated"
Tany_Final = RunPCA(Tany_Final, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
Tany_Final <- RunUMAP(Tany_Final, dims = 1:z, spread= s)
DefaultAssay(Tany_Final) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(Tany_Final, unique(c("GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "ALDOC", "SLC1A2", "ID3", "ASCL1", "APOE", "FOXJ1", "RAX", "ADM", "EMX2", "LHX2", "TMEM212", "CRYM", "CCDC153", "NPAS3", "RFX2", "RFX3", "RFX4", "COL23A1", "CD59", "SLC17A8", "VCAN", "FRZB", "PENK", "COL25A1", "CACNA2D2", "STOML3", "TMEM212", "SLC7A11", "PCP4L1", "PLTP", "TGFB2", "NR2E1", "EPHB1", "P3H2", "NELL2", "FGF10", "RGCC", "GRIA2", "RLBP1", "SIX6", "SCN7A", "MEST", "IGFBP5", "RGS7BP")), reduction="umap") 

DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Tany_Final, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(Tany_Final, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))

}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=40)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_SPLITUMAP.pdf", sep=""), width=50, height=5)
print(DPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()


#Use version 27 - PCA 40 _dims 30 _spread 10
DefaultAssay(Tany_Final) = "integrated"
Tany_Final = RunPCA(Tany_Final, npcs = 40)
Tany_Final <- RunUMAP(Tany_Final, dims = 1:30, spread= 10)

set.dim = 30
set.kparam = 3
set.res = c(3)
#ClusterFunc_All_RNA(Tany_Final)


Tany_Final <- FindNeighbors(Tany_Final, k.param=10, dims=1:30)
Tany_Final <- FindClusters(Tany_Final, resolution =1)
Tany_1 = subset(Tany_Final, idents= c(0))
Tany_2 = subset(Tany_Final, idents= c(1))
Tany_3 = subset(Tany_Final, idents= c(2))
Tany_4 = subset(Tany_Final, idents= c(3,6))
Tany_5 = subset(Tany_Final, idents= c(4))
Tany_6 = subset(Tany_Final, idents= c(5))




CheckInput = colnames(Tany_1) #Check Populations
CheckUMAP(Tany_Final)



OCTOBER_TANY_Assigns = as.data.frame(c(rep("Tany_01", length(colnames(Tany_1))),
                                 rep("Tany_02", length(colnames(Tany_2))), 
                                 rep("Tany_03", length(colnames(Tany_3))), 
                                 rep("Tany_04", length(colnames(Tany_4))), 
                                 rep("Tany_05", length(colnames(Tany_5))), 
                                 rep("Tany_06", length(colnames(Tany_6)))))
colnames(OCTOBER_TANY_Assigns) = "Pop"
row.names(OCTOBER_TANY_Assigns) = c(colnames(Tany_1), 
                            colnames(Tany_2),
                            colnames(Tany_3),
                            colnames(Tany_4),
                            colnames(Tany_5),
                            colnames(Tany_6))

write.csv(OCTOBER_TANY_Assigns, paste("OCTOBER_TANY_Assigns", Sys.Date(), ".csv", sep=""))
OCTOBER_TANY_Assigns$Pop = factor(OCTOBER_TANY_Assigns$Pop, levels = sort(unique(OCTOBER_TANY_Assigns$Pop)))
Tany_Final = AddMetaData(Tany_Final, OCTOBER_TANY_Assigns, "OCTOBER_TANY_Assigns")
Idents(Tany_Final) = "OCTOBER_TANY_Assigns"

pdf(paste("TANY_Assignments_UMAP_20SEP.pdf", sep=""), width=12, height=8)
print(DimPlot(Tany_Final, reduction = "umap", label=T))
dev.off()




#OCTOBER_OLIGO_Assigns$Pop = factor(OCTOBER_OLIGO_Assigns$Pop, levels = rev(c("Olig_1", "Olig_2", "Olig_3", "Olig_4", "Olig_5", "Olig_6", "Olig_7", "Olig_8", "Olig_9", "Olig_10", "Olig_11", "Olig_12", "Olig_13", "Olig_14")))

AllCells_PlotOrder = c(unique(c("GJA1", "GFAP", "AQP4","CRYM", "FRZB","VCAN","SIX6", "RFX3","RGCC","RFX4",   "NELL2")))
                        

DefaultAssay(Tany_Final) = "RNA"
StorePlots = list()
AllVlns =list()
  for( in AllCells_PlotOrder[1]){
            plotA <- VlnPlot(Tany_Final, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=16),
                                                 axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=12))
            StorePlots[[x]] = plotA 
            }
  for( in AllCells_PlotOrder[2:length(AllCells_PlotOrder)]){
            plotB <- VlnPlot(Tany_Final, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks. = element_blank(), axis.title.y=element_blank(), axis.text. =element_blank(), axis.title.x=element_blank(), 
                      axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=12))
           StorePlots[[x]] = plotB
           }
AllVlns[["Plot1"]] <- ggarrange(plotlist = StorePlots, widths=c(2, rep(1, length(AllCells_PlotOrder)-1)), ncol = length(AllCells_PlotOrder),  nrow = 1)
      
pdf(paste("AllCells_Tany_ToSubcluster_MultiVln_6OCT21.pdf", sep=""), width=length(AllCells_PlotOrder), height=length(unique(Tany_Final@active.ident))*1.5)
print(AllVlns)
dev.off()

```



#############################################
######### SUBCLUSTERING RADIAL GLIA #########
#############################################

########## Iterative cluster parameters for RG ##########
```{r}
Idents(Fetal.combined) = "OCT_AllCell_Assigns" 
All_oRG = subset(Fetal.combined, idents = c("RadialGlia_02 [oRG]", "RadialGlia_03 [oRG]"))
All_vRG = subset(Fetal.combined, idents = c("RadialGlia_01 [vRG]"))
All_IPC = subset(Fetal.combined, idents = c("RadialGlia_04 [IPC]"))
All_GenRG = subset(Fetal.combined, idents = c("RadialGlia_05", "RadialGlia_06"))
#oRG  vRG done

RG_SeuFiles = list(All_oRG, All_vRG,  All_IPC, All_GenRG) 
names(RG_SeuFiles) = c("All_oRG", "All_vRG", "All_IPC", "All_GenRG") 

for( in names(RG_SeuFiles)){
DPlist2 = list()
DPlist = list()
FPlist = list()
SeuFile = RG_SeuFiles[[x]]
Filename = x
for(y in seq(10,50,10)){
DefaultAssay(SeuFile) = "integrated"
SeuFile = RunPCA(SeuFile, npcs = y)
for(z in seq(10,y,10)){
for(s in c(2,5,10)){
SeuFile <- RunUMAP(SeuFile, dims = 1:z, spread= s)
DefaultAssay(SeuFile) = "RNA"

FPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = FeaturePlot(SeuFile, c("HOPX", "NKX2-1", "ASCL1", "STMN2", "EGFR", "PAX6", "NES", "NKX6-2", "SLC17A6", "SLC17A8", "VIM","NES", "HES1", "MKI67", "FABP7", "TTYH1", "HMGA2", "MEIS2", "SLC1A3", "FAM107A",  "NHLH2", "DLX1", "OLIG2", #HPCS
           "SYT1", "SNAP25", "TUBB3","SYP", "SLC32A1", "DLX1", "DLX2", "DLX5", "GAD1", #BROAD NEURONAL
           "GJA1", "GFAP", "AQP4", "SLC7A10", "ALDH1L1", "AGT", "SLC1A3", "S100B", "ALDOC", "SLC1A2", "ID3", "APOE", #ASTROCYTES
           "APOD", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9"), #OLIGODENDROCYTES)
           reduction="umap") 

#DPlist[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(SeuFile, reduction="umap", split.by = "orig.ident") + labs(title = paste("PCA", y, "_dims", z, "_spread", s))

DPlist2[[paste("PCA", y, "_dims", z, "_spread", s)]] = DimPlot(SeuFile, reduction="umap", group.by = "orig.ident", label=T) + labs(title = paste("PCA", y, "dims", z, "spread", s))

}}}  
pdf(paste("ALLFETAL_", Filename, "_FeaturePlots.pdf", sep=""), width=20, height=50)
print(FPlist)
dev.off()

pdf(paste("ALLFETAL_", Filename, "_GROUPUMAP.pdf", sep=""), width=8, height=5)
print(DPlist2)
dev.off()
}

#Change gene lists so we can use the same functions
EndoList = unique(c("SLCO1C1", "SLC38A5", "MYH11", "MRC1", "CLDN5", "ITM2A", "FLT1", "APOD", "PMP22", "MAG", "OLIG1", "PLLP", "CSPG4", "PDGFRA", "NEU4", "TNS3", "FYN", "TCF7L2", "MAL", "MOG", "PLP1", "SERINC5", "TFRC", "OLIG1", "OLIG2", "CNP", "ST8SIA1", "CD9"))

OligList = unique(c("HOPX", "NKX2-1", "ASCL1", "STMN2", "EGFR", "PAX6", "NES", "NKX6-2", "SLC17A6", "SLC17A8", "VIM","NES", "HES1", "MKI67", "FABP7", "TTYH1", "HMGA2", "MEIS2", "SLC1A3", "FAM107A",  "NHLH2", "DLX1", "OLIG2")) #HPCS
```

### oRG: DONE
```{r}
#PCA 40 dims 10 spread 5
DefaultAssay(All_oRG) = "integrated"
All_oRG = RunPCA(All_oRG, npcs = 40)
All_oRG <- RunUMAP(All_oRG, dims = 1:10, spread= 5)
All_oRG_UMAP = as.data.frame(All_oRG@reductions$umap@cell.embeddings)

set.dim = 10
set.res = c(1)
set.kparam = c(20,50)
#ClusterFuncc_All_RNA(All_oRG)

All_oRG <- FindNeighbors(All_oRG, k.param=50, dims=1:10)
All_oRG <- FindClusters(All_oRG, resolution =1)
oRG_1 = subset(All_oRG, idents = c(6))
oRG_2 = subset(All_oRG, idents = c(4))
oRG_3 = subset(All_oRG, idents = c(2))

oRG_2_V2 = subset(All_oRG_UMAP, row.names(All_oRG_UMAP) %in% colnames(oRG_2) & All_oRG_UMAP$UMAP_1 > 3 & All_oRG_UMAP$UMAP_2 > -3   | row.names(All_oRG_UMAP) %in% colnames(All_oRG) & row.names(All_oRG_UMAP) %notin% colnames(oRG_1) & All_oRG_UMAP$UMAP_1 > 11)

oRG_3_V2 = subset(All_oRG_UMAP, row.names(All_oRG_UMAP) %in% colnames(oRG_3) & All_oRG_UMAP$UMAP_2 <  -3   | row.names(All_oRG_UMAP) %in% colnames(All_oRG) & row.names(All_oRG_UMAP) %notin% row.names(oRG_2_V2) & All_oRG_UMAP$UMAP_2 < -13)

All_oRG_Rest1 = subset(All_oRG, cells = c(colnames(oRG_1), row.names(oRG_2_V2), row.names(oRG_3_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(100)
#ClusterFuncc_All_RNA(All_oRG_Rest1)


All_oRG_Rest1 <- FindNeighbors(All_oRG_Rest1, k.param=50, dims=1:10)
All_oRG_Rest1 <- FindClusters(All_oRG_Rest1, resolution =1)
oRG_4 = subset(All_oRG_Rest1, idents = c(2,3))

oRG_4_V2 = subset(All_oRG_UMAP, row.names(All_oRG_UMAP) %in% colnames(oRG_4) & All_oRG_UMAP$UMAP_1 >  -3 & All_oRG_UMAP$UMAP_2 > 8 | row.names(All_oRG_UMAP) %in% colnames(oRG_4) & All_oRG_UMAP$UMAP_2 >  -2 & All_oRG_UMAP$UMAP_2 < 8 | row.names(All_oRG_UMAP) %in% colnames(oRG_4) & All_oRG_UMAP$UMAP_1 > -3 & All_oRG_UMAP$UMAP_2 < -2 | row.names(All_oRG_UMAP) %in% colnames(All_oRG_Rest1) & All_oRG_UMAP$UMAP_1 > 3)

oRG_4_V2_Seu = subset(All_oRG_Rest1, cells = row.names(oRG_4_V2))
set.dim = 10
set.res = c(0.1)
set.kparam = c(100)
#ClusterFuncc_All_RNA(oRG_4_V2_Seu)

oRG_5 = subset(All_oRG_Rest1, cells = row.names(oRG_4_V2), invert=T)

CheckInput = row.names(oRG_4_V2)
CheckUMAP(All_oRG_Rest1)

oRG_SubclusterAssigns = as.data.frame(c(rep("oRG_01 [STMN2]", length(colnames(oRG_1))), 
                                 rep("oRG_02 [HOPX/STMN2/NKX6-2]", length(row.names(oRG_2_V2))),   
                                 rep("oRG_03 [HOPX]", length(row.names(oRG_3_V2))),               
                                 rep("oRG_04 [HOPX/LowSTMN2/NKX6-2]", length(row.names(oRG_4_V2))),               
                                 rep("oRG_05 [HOPX/LowSTMN2]", length(colnames(oRG_5))))) 
colnames(oRG_SubclusterAssigns) = "Pop"
row.names(oRG_SubclusterAssigns) = c(colnames(oRG_1), # #oRG_SubclusterAssigns$BoRGs
                                       row.names(oRG_2_V2),  
                                       row.names(oRG_3_V2), 
                                       row.names(oRG_4_V2),
                                       colnames(oRG_5))

paste(sort(unique(oRG_SubclusterAssigns$Pop)), collapse = ", ")

oRG_SubclusterAssigns$Pop = factor(oRG_SubclusterAssigns$Pop, levels = c(sort(unique(oRG_SubclusterAssigns$Pop))))
 

All_oRG = AddMetaData(All_oRG, oRG_SubclusterAssigns, "oRG_SubclusterAssigns")
Idents(All_oRG) = "oRG_SubclusterAssigns"

pdf(paste("ALLFETAL_oRG_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_oRG, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_oRG) = "RNA"
VP = list()
for( in c("HOPX", "STMN2", "NKX6-2")){
VP[[x]] = VlnPlot(All_oRG, x, pt.size = 0)  + NoLegend()
}
pdf("oRG_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### vRG: DONE
```{r}
#PCA 30 dims 30 spread 5
DefaultAssay(All_vRG) = "integrated"
All_vRG = RunPCA(All_vRG, npcs = 30)
All_vRG <- RunUMAP(All_vRG, dims = 1:30, spread= 5)
All_vRG_UMAP = as.data.frame(All_vRG@reductions$umap@cell.embeddings)

set.dim = 10
set.res = c(1)
set.kparam = c(20,50)
##ClusterFuncc_All_RNA(All_vRG)

vRG_1 = subset(All_vRG_UMAP, All_vRG_UMAP$UMAP_1 < -0.5 & All_vRG_UMAP$UMAP_2 > 10.5 |  All_vRG_UMAP$UMAP_1 < -2 & All_vRG_UMAP$UMAP_2 > 9)
vRG_2 = subset(All_vRG_UMAP, All_vRG_UMAP$UMAP_1 < -9 & All_vRG_UMAP$UMAP_2 > -7)

All_vRG_Rest1 = subset(All_vRG, cells = c(row.names(vRG_1), row.names(vRG_2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20,50)
##ClusterFuncc_All_RNA(All_vRG_Rest1)

All_vRG_Rest1 <- FindNeighbors(All_vRG_Rest1, k.param=50, dims=1:10)
All_vRG_Rest1 <- FindClusters(All_vRG_Rest1, resolution =1)
vRG_3 = subset(All_vRG_Rest1, idents = c(6))
vRG_4 = subset(All_vRG_Rest1, idents = c(3))

vRG_3_V2 = subset(All_vRG_UMAP, row.names(All_vRG_UMAP) %in% colnames(vRG_3) & All_vRG_UMAP$UMAP_1 > 5  | row.names(All_vRG_UMAP) %in% colnames(All_vRG_Rest1) & All_vRG_UMAP$UMAP_1 > 11)


ExtraPAX6 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_vRG_Rest1) & Fetal.combined_Data$PAX6 > 0)
vRG_4_V2 = subset(All_vRG_UMAP, row.names(All_vRG_UMAP) %in% colnames(vRG_4) & row.names(All_vRG_UMAP) %notin% row.names(vRG_3_V2) | row.names(All_vRG_UMAP) %in% row.names(ExtraPAX6) & All_vRG_UMAP$UMAP_2 > 5  & row.names(All_vRG_UMAP) %notin% row.names(vRG_3_V2)| row.names(All_vRG_UMAP) %in% colnames(All_vRG_Rest1) & row.names(All_vRG_UMAP) %notin% row.names(vRG_3_V2) & All_vRG_UMAP$UMAP_2 > 10)

All_vRG_Rest2 = subset(All_vRG, cells = c(row.names(vRG_1), row.names(vRG_2), row.names(vRG_4_V2), row.names(vRG_3_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(50)
##ClusterFuncc_All_RNA(All_vRG_Rest2)

All_vRG_Rest2 <- FindNeighbors(All_vRG_Rest2, k.param=50, dims=1:10)
All_vRG_Rest2 <- FindClusters(All_vRG_Rest2, resolution =1)
vRG_5 = subset(All_vRG_Rest2, idents = c(1,2))

vRG_5 <- FindNeighbors(vRG_5, k.param=5, dims=1:10)
vRG_5 <- FindClusters(vRG_5, resolution =1)
vRG_5_V2 = subset(vRG_5, idents = c(1,3,8), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
##ClusterFuncc_All_RNA(vRG_5_V2)

ExtraNKX62 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_vRG_Rest2) & Fetal.combined_Data$`NKX6-2` > 0)
PAX6_ToRM = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_vRG_Rest2) & Fetal.combined_Data$PAX6 > 0 )

vRG_5_V3 = subset(All_vRG_UMAP, row.names(All_vRG_UMAP) %in% colnames(vRG_5_V2) & All_vRG_UMAP$UMAP_1 > 1 & All_vRG_UMAP$UMAP_2 > 1  | row.names(All_vRG_UMAP) %in% row.names(ExtraNKX62) & All_vRG_UMAP$UMAP_2 > -3  & All_vRG_UMAP$UMAP_1 > -3  | row.names(All_vRG_UMAP) %in% colnames(All_vRG_Rest2) & All_vRG_UMAP$UMAP_1 > 0 & All_vRG_UMAP$UMAP_2 > 3)

vRG_5_V4 = subset(All_vRG_UMAP, row.names(All_vRG_UMAP) %in% row.names(vRG_5_V3) & row.names(All_vRG_UMAP) %notin% row.names(PAX6_ToRM))

vRG_5_V4_Seu = subset(All_vRG_Rest2, cells = row.names(vRG_5_V4))

set.dim = 10
set.res = c(0.1)
set.kparam = c(100)
##ClusterFuncc_All_RNA(vRG_5_V4_Seu)

All_vRG_Rest3 = subset(All_vRG, cells = c(row.names(vRG_1), row.names(vRG_2), row.names(vRG_4_V2), row.names(vRG_3_V2), row.names(vRG_5_V4)), invert=T)

set.dim = c(10,20,30)
set.res = c(1)
set.kparam = c(20,50,100)
##ClusterFuncc_All_RNA(All_vRG_Rest3)

All_vRG_Rest3 <- FindNeighbors(All_vRG_Rest3, k.param=20, dims=1:20)
All_vRG_Rest3 <- FindClusters(All_vRG_Rest3, resolution =1)
vRG_6 = subset(All_vRG_Rest3, idents = c(5,6))

vRG_6_V2 = subset(All_vRG_UMAP, row.names(All_vRG_UMAP) %in% colnames(vRG_6) & All_vRG_UMAP$UMAP_1 < 1 & All_vRG_UMAP$UMAP_2 < -9 | row.names(All_vRG_UMAP) %in% colnames(All_vRG_Rest3) & All_vRG_UMAP$UMAP_1 > -7  & All_vRG_UMAP$UMAP_1 < 2 & All_vRG_UMAP$UMAP_2 < -12)

All_vRG_Rest4 = subset(All_vRG, cells = c(row.names(vRG_1), row.names(vRG_2), row.names(vRG_4_V2), row.names(vRG_3_V2), row.names(vRG_5_V4), row.names(vRG_6_V2)), invert=T)

set.dim = c(10)
set.res = c(1)
set.kparam = c(5,3)
##ClusterFuncc_All_RNA(All_vRG_Rest4)


CheckInput = colnames(vRG_6)
CheckInput = row.names(vRG_6_V2)
CheckUMAP(All_vRG_Rest3)


vRG_SubclusterAssigns = as.data.frame(c(rep("vRG_01 [HOPX/PAX6/NES]", length(row.names(vRG_1))), 
                                 rep("vRG_02 [HOPX/PAX6/NES]", length(row.names(vRG_2))),   
                                 rep("vRG_03 [HOPX/EGFR]", length(row.names(vRG_3_V2))),               
                                 rep("vRG_04 [HOPX/PAX6/NES]", length(row.names(vRG_4_V2))),               
                                 rep("vRG_05 [HOPX/NKX6-2]", length(row.names(vRG_5_V4))),               
                                 rep("vRG_06 [PAX6/NES]", length(row.names(vRG_6_V2))),               
                                 rep("vRG_07 [HOPX/PAX6/NES]", length(colnames(All_vRG_Rest4))))) 
colnames(vRG_SubclusterAssigns) = "Pop"
row.names(vRG_SubclusterAssigns) = c(row.names(vRG_1), # #vRG_SubclusterAssigns$BvRGs
                                       row.names(vRG_2),  
                                       row.names(vRG_3_V2), 
                                       row.names(vRG_4_V2),
                                       row.names(vRG_5_V4),
                                       row.names(vRG_6_V2),
                                       colnames(All_vRG_Rest4))

#paste(sort(unique(vRG_SubclusterAssigns$Pop)), collapse = ", ")

vRG_SubclusterAssigns$Pop = factor(vRG_SubclusterAssigns$Pop, levels = c(sort(unique(vRG_SubclusterAssigns$Pop))))
 

All_vRG = AddMetaData(All_vRG, vRG_SubclusterAssigns, "vRG_SubclusterAssigns")
Idents(All_vRG) = "vRG_SubclusterAssigns"

pdf(paste("ALLFETAL_vRG_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_vRG, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_vRG) = "RNA"
VP = list()
for( in c(OligList)){
VP[[x]] = VlnPlot(All_vRG, x, pt.size = 0)  + NoLegend()
}
pdf("vRG_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### IPC
```{r}
#PCA 40 dims 40 spread 2
DefaultAssay(All_IPC) = "integrated"
All_IPC = RunPCA(All_IPC, npcs = 40)
All_IPC <- RunUMAP(All_IPC, dims = 1:40, spread= 2)
All_IPC_UMAP = as.data.frame(All_IPC@reductions$umap@cell.embeddings)

set.dim = 10
set.res = c(1)
set.kparam = c(50)
##ClusterFuncc_All_RNA(All_IPC)


All_IPC <- FindNeighbors(All_IPC, k.param=50, dims=1:10)
All_IPC <- FindClusters(All_IPC, resolution =1)
IPC_1 = subset(All_IPC, idents = c(0,3))

IPC_1_V2 = subset(All_IPC_UMAP, row.names(All_IPC_UMAP) %in% colnames(IPC_1) & All_IPC_UMAP$UMAP_1 > -2  | row.names(All_IPC_UMAP) %in% colnames(All_IPC) & All_IPC_UMAP$UMAP_1 > 1 & All_IPC_UMAP$UMAP_2 < 2)



All_IPC_Rest1 = subset(All_IPC, cells = row.names(IPC_1_V2), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(20,50)
#ClusterFuncc_All_RNA(All_IPC_Rest1)

All_IPC_Rest1 <- FindNeighbors(All_IPC_Rest1, k.param=50, dims=1:10)
All_IPC_Rest1 <- FindClusters(All_IPC_Rest1, resolution =1)
IPC_2 = subset(All_IPC_Rest1, idents = c(0,2,4))
ExtraOLIG2 = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_IPC_Rest1) & Fetal.combined_Data$OLIG2 > 1)

IPC_2_V2 = subset(All_IPC_UMAP, row.names(All_IPC_UMAP) %in% colnames(IPC_2) & All_IPC_UMAP$UMAP_2 < 0 & All_IPC_UMAP$UMAP_1 < -5 | row.names(All_IPC_UMAP) %in% colnames(IPC_2) & All_IPC_UMAP$UMAP_2 < 1.5 & All_IPC_UMAP$UMAP_1 > -5 | row.names(All_IPC_UMAP) %in% row.names(ExtraOLIG2) &  All_IPC_UMAP$UMAP_2 < 0.5 & All_IPC_UMAP$UMAP_1 < -5  | row.names(All_IPC_UMAP) %in% row.names(ExtraOLIG2) &  All_IPC_UMAP$UMAP_2 < 2  & All_IPC_UMAP$UMAP_1 > -5 | row.names(All_IPC_UMAP) %in% colnames(All_IPC_Rest1) &  All_IPC_UMAP$UMAP_2 < -1)

IPC_2_V2_Seu = subset(All_IPC, cells = row.names(IPC_2_V2))

IPC_4 = subset(All_IPC_UMAP, row.names(All_IPC_UMAP) %in% colnames(IPC_2_V2_Seu) & All_IPC_UMAP$UMAP_2 < -2.6 & All_IPC_UMAP$UMAP_1 < -5)
IPC_2_V3 = subset(IPC_2_V2_Seu, cells = row.names(IPC_4), invert=T)
CheckInput = row.names(IPC_4)
CheckUMAP(IPC_2_V2_Seu)



All_IPC_Rest2 = subset(All_IPC, cells = c(row.names(IPC_1_V2), row.names(IPC_2_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(50)
#ClusterFuncc_All_RNA(All_IPC_Rest2)

All_IPC_Rest2 <- FindNeighbors(All_IPC_Rest2, k.param=50, dims=1:10)
All_IPC_Rest2 <- FindClusters(All_IPC_Rest2, resolution =1)
IPC_3 = subset(All_IPC_Rest2, idents = c(2,3))

IPC_3_V2 = subset(All_IPC_UMAP, row.names(All_IPC_UMAP) %in% colnames(IPC_3) | row.names(All_IPC_UMAP) %in% colnames(All_IPC_Rest2) &  All_IPC_UMAP$UMAP_1 < -6)


All_IPC_Rest3 = subset(All_IPC, cells = c(row.names(IPC_1_V2), row.names(IPC_2_V2), row.names(IPC_3_V2)), invert=T)


set.dim = 10
set.res = c(1)
set.kparam = c(20)
#ClusterFuncc_All_RNA(All_IPC_Rest3)


CheckInput = row.names(IPC_3_V2)
CheckUMAP(All_IPC_Rest2)

IPC_SubclusterAssigns = as.data.frame(c(rep("IPC_01 [HOPX/EGFR/NES/HES1/OLIG2]", length(row.names(IPC_1_V2))), 
                                 rep("IPC_02 [EGFR/OLIG2]", length(colnames(IPC_2_V3))),   
                                 rep("IPC_05", length(row.names(IPC_3_V2))),               
                                 rep("IPC_04 [EGFR/NES/HES1/OLIG2]", length(row.names(IPC_4))),               
                                 rep("IPC_03 [EGFR/HES1]", length(colnames(All_IPC_Rest3))))) 
colnames(IPC_SubclusterAssigns) = "Pop"
row.names(IPC_SubclusterAssigns) = c(row.names(IPC_1_V2), # #IPC_SubclusterAssigns$BIPCs
                                       colnames(IPC_2_V3),  
                                       row.names(IPC_3_V2), 
                                       row.names(IPC_4), 
                                       colnames(All_IPC_Rest3))

#paste(sort(unique(IPC_SubclusterAssigns$Pop)), collapse = ", ")

IPC_SubclusterAssigns$Pop = factor(IPC_SubclusterAssigns$Pop, levels = c(sort(unique(IPC_SubclusterAssigns$Pop))))
 

All_IPC = AddMetaData(All_IPC, IPC_SubclusterAssigns, "IPC_SubclusterAssigns")
Idents(All_IPC) = "IPC_SubclusterAssigns"

pdf(paste("ALLFETAL_IPC_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_IPC, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_IPC) = "RNA"
VP = list()
for( in OligList){
VP[[x]] = VlnPlot(All_IPC, x, pt.size = 0)  + NoLegend()
}
pdf("IPC_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```

### All_GenRG
```{r}
#PCA 40 dims 40 spread 2
DefaultAssay(All_GenRG) = "integrated"
All_GenRG = RunPCA(All_GenRG, npcs = 40)
All_GenRG <- RunUMAP(All_GenRG, dims = 1:40, spread= 2)
All_GenRG_UMAP = as.data.frame(All_GenRG@reductions$umap@cell.embeddings)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
##ClusterFuncc_All_RNA(All_GenRG)

All_GenRG <- FindNeighbors(All_GenRG, k.param=10, dims=1:10)
All_GenRG <- FindClusters(All_GenRG, resolution =1)
GenRG_1 = subset(All_GenRG, idents = c(6,7))

set.dim = c(40)
set.res = c(1)
set.kparam = c(5,3)
##ClusterFuncc_All_RNA(GenRG_1)

GenRG_1 <- FindNeighbors(GenRG_1, k.param=5, dims=1:40)
GenRG_1 <- FindClusters(GenRG_1, resolution =1)
GenRG_1_V2 = subset(GenRG_1, idents = c(0,4))

GenRG_1_V3 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(GenRG_1_V2) & All_GenRG_UMAP$UMAP_2 > 4.5  | row.names(All_GenRG_UMAP) %in% colnames(All_GenRG) &  All_GenRG_UMAP$UMAP_2 > 5.5)

All_GenRG_Rest1 = subset(All_GenRG, cells = c(row.names(GenRG_1_V3)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(5)
##ClusterFuncc_All_RNA(All_GenRG_Rest1)

All_GenRG_Rest1 <- FindNeighbors(All_GenRG_Rest1, k.param=5, dims=1:10)
All_GenRG_Rest1 <- FindClusters(All_GenRG_Rest1, resolution =1)
GenRG_2 = subset(All_GenRG_Rest1, idents = c(4,6))


GenRG_2_V2 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(GenRG_2) & All_GenRG_UMAP$UMAP_1 > 0 & All_GenRG_UMAP$UMAP_2 > -1 | row.names(All_GenRG_UMAP) %in% colnames(All_GenRG_Rest1)  & All_GenRG_UMAP$UMAP_1 > 2 & All_GenRG_UMAP$UMAP_2 > 3 | row.names(All_GenRG_UMAP) %in% colnames(All_GenRG_Rest1)  & All_GenRG_UMAP$UMAP_1 > 1 & All_GenRG_UMAP$UMAP_1 < 3.5 & All_GenRG_UMAP$UMAP_2 < 3 & All_GenRG_UMAP$UMAP_2 > -0.5)
                    
                  
All_GenRG_Rest2 = subset(All_GenRG, cells = c(row.names(GenRG_1_V3), row.names(GenRG_2_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(10)
##ClusterFuncc_All_RNA(All_GenRG_Rest2)


All_GenRG_Rest2 <- FindNeighbors(All_GenRG_Rest2, k.param=10, dims=1:10)
All_GenRG_Rest2 <- FindClusters(All_GenRG_Rest2, resolution =1)
GenRG_3 = subset(All_GenRG_Rest2, idents = c(0,9,1,10,7))


GenRG_3_V2 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(GenRG_3) & All_GenRG_UMAP$UMAP_1 < -1 | row.names(All_GenRG_UMAP) %in% colnames(All_GenRG_Rest2)  & All_GenRG_UMAP$UMAP_1 < -4 & All_GenRG_UMAP$UMAP_2 > -4)

All_GenRG_Rest3 = subset(All_GenRG, cells = c(row.names(GenRG_1_V3), row.names(GenRG_2_V2), row.names(GenRG_3_V2)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(3)
#ClusterFuncc_All_RNA(All_GenRG_Rest3)

All_GenRG_Rest3 <- FindNeighbors(All_GenRG_Rest3, k.param=3, dims=1:10)
All_GenRG_Rest3 <- FindClusters(All_GenRG_Rest3, resolution =1)
GenRG_4 = subset(All_GenRG_Rest3, idents = c(3,6,7,9,17,4))

GenRG_4_V2 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(GenRG_4) & All_GenRG_UMAP$UMAP_2 < 0 & All_GenRG_UMAP$UMAP_1 > 0 | row.names(All_GenRG_UMAP) %in% colnames(All_GenRG_Rest3) & All_GenRG_UMAP$UMAP_2 < 0 & All_GenRG_UMAP$UMAP_1 > 1 & All_GenRG_UMAP$UMAP_1 < 3)

All_GenRG_Rest4 = subset(All_GenRG, cells = c(row.names(GenRG_1_V3), row.names(GenRG_2_V2), row.names(GenRG_3_V2), row.names(GenRG_4_V2)), invert=T)

GenRG_5 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(All_GenRG_Rest4)  & All_GenRG_UMAP$UMAP_2 < 0.5 & All_GenRG_UMAP$UMAP_1 < 0)
GenRG_6 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(All_GenRG_Rest4)  & All_GenRG_UMAP$UMAP_1 > 2)

All_GenRG_Rest5 = subset(All_GenRG, cells = c(row.names(GenRG_1_V3), row.names(GenRG_2_V2), row.names(GenRG_3_V2), row.names(GenRG_4_V2), row.names(GenRG_5), row.names(GenRG_6)), invert=T)

set.dim = 10
set.res = c(1)
set.kparam = c(3)
#ClusterFuncc_All_RNA(All_GenRG_Rest5)

All_GenRG_Rest5 <- FindNeighbors(All_GenRG_Rest5, k.param=3, dims=1:10)
All_GenRG_Rest5 <- FindClusters(All_GenRG_Rest5, resolution =1)
GenRG_7 = subset(All_GenRG_Rest5, idents = c(6,7))
ExtraEGFR = subset(Fetal.combined_Data, row.names(Fetal.combined_Data) %in% colnames(All_GenRG_Rest5) & Fetal.combined_Data$EGFR > 0)

GenRG_7_V2 = subset(All_GenRG_UMAP, row.names(All_GenRG_UMAP) %in% colnames(GenRG_7) | row.names(All_GenRG_UMAP) %in% row.names(ExtraEGFR) & All_GenRG_UMAP$UMAP_2 > 3 & All_GenRG_UMAP$UMAP_1 > -1 )


CheckInput = colnames(GenRG_7)
CheckUMAP(All_GenRG_Rest5)


CheckInput = row.names(GenRG_7_V2)
CheckUMAP(All_GenRG_Rest5)



All_GenRG_Rest5 = subset(All_GenRG, cells = c(row.names(GenRG_1_V3), row.names(GenRG_2_V2), row.names(GenRG_3_V2), row.names(GenRG_4_V2), row.names(GenRG_5), row.names(GenRG_6), row.names(GenRG_7_V2)), invert=T)

GenRG_SubclusterAssigns = as.data.frame(c(rep("GenRG_01 [HOPX/NKX62]", length(row.names(GenRG_1_V3))), 
                                 rep("GenRG_02 [HOPX]", length(row.names(GenRG_2_V2))),   
                                 rep("GenRG_03 [NKX62]", length(row.names(GenRG_3_V2))),               
                                 rep("GenRG_04 [NKX2-1/NES]", length(row.names(GenRG_4_V2))),               
                                 rep("GenRG_05 [NES]", length(row.names(GenRG_5))),               
                                 rep("GenRG_06 [HOPX/NES]", length(row.names(GenRG_6))),               
                                 rep("GenRG_07 [EGFR]", length(row.names(GenRG_7_V2))),               
                                 rep("GenRG_08", length(colnames(All_GenRG_Rest5))))) 
colnames(GenRG_SubclusterAssigns) = "Pop"
row.names(GenRG_SubclusterAssigns) = c(row.names(GenRG_1_V3), # #GenRG_SubclusterAssigns$BGenRGs
                                       row.names(GenRG_2_V2),  
                                       row.names(GenRG_3_V2), 
                                       row.names(GenRG_4_V2),
                                       row.names(GenRG_5),
                                       row.names(GenRG_6),
                                       row.names(GenRG_7_V2),
                                       colnames(All_GenRG_Rest5))

GenRG_SubclusterAssigns$Pop = factor(GenRG_SubclusterAssigns$Pop, levels = c(sort(unique(GenRG_SubclusterAssigns$Pop))))
 

All_GenRG = AddMetaData(All_GenRG, GenRG_SubclusterAssigns, "GenRG_SubclusterAssigns")
Idents(All_GenRG) = "GenRG_SubclusterAssigns"

pdf(paste("ALLFETAL_GenRG_3DEC21_V2.pdf", sep=""), width=20, height=15)
print(DimPlot(All_GenRG, reduction = "umap", label=T))
dev.off()

DefaultAssay(All_GenRG) = "RNA"
VP = list()
for( in OligList){
VP[[x]] = VlnPlot(All_GenRG, x, pt.size = 0)  + NoLegend()
}
pdf("GenRG_CheckVlns.pdf", width = 8, height = 6)
print(VP)
dev.off()
```



#############################################
############### COMPILED DATA ###############
#############################################

#Code to generate UMAPs and violin plots for all data
```{r}
#Seurat File List
All_SeuratFiles = list(Fetal.combined, Astrocytes_Clean, Tany_Final, Ependy_Final_Seu, Vasculature_Seurat, Oligodendro_Final, Micro_Seurat, All_oRG, All_vRG, All_IPC, All_GenRG, HypothalamicNeurons, All_TM, All_ARCUATE, All_PVH, All_VMH, All_LH, All_PREOPTIC, All_SCN, All_DMH, All_IDTT, All_MN, All_SMN, All_ZI)

#Metadata list
All_Metadata = list(OCT_AllCell_Assigns, SEPTEMBER_ASTRO_Assigns, OCTOBER_TANY_Assigns, OCTOBER_EPENDY_Assigns, OCTOBER_ENDOMURAL_Assigns, OCTOBER_OLIGO_Assigns, OCTOBER_MICROGLIA_Assigns, oRG_SubclusterAssigns, vRG_SubclusterAssigns, IPC_SubclusterAssigns, GenRG_SubclusterAssigns, OCT_Neuron_Assigns_Renamed2, OCT_TM_Assigns_Renamed, OCT_ARC_Assigns_Renamed, OCT_PVH_Assigns_Renamed, OCT_VMH_Assigns_Renamed, OCT_LH_Assigns_Renamed, OCT_PREOPTIC_Assigns_Renamed, OCT_SCN_Assigns_Renamed, OCT_DMH_Assigns_Renamed, OCT_IDTT_Assigns_Renamed, OCT_MN_Assigns_Renamed, OCT_SMN_Assigns_Renamed, OCT_ZI_Assigns_Renamed)

#Names list
PlotNames = c("All Data", "Non-Dividing Astrocytes", "Tanycytes", "Ependymocytes", "Vasculature", "Non-Dividing Oligodendrocytes", "Microglia", "Outer Radial Glia", "Ventral Radial Glia", "Intermediate Progenitor Cells", "Other Radial Glia", "Hypothalamic Neurons", "Neurons: Tuberomammillary Nucleus", "Neurons: Arcuate Nucleus", "Neurons: Paraventricular Nucleus", "Neurons: Ventromedial Nucleus", "Neurons: Lateral Hypothalamus", "Neurons: Preoptic Area", "Neurons: Suprachiasmatic Nuclues", "Neurons: Dorsomedial Nucleus", "Neurons: Intrahypothalamic Diagonal", "Neurons: Mammillary Nucleus", "Neurons: Supramammillary Nucleus", "Neurons: Zona Incerta")

#Gene Lists - Max 20 per plot with current setup
GeneList = list()
GeneList[[1]] = c("HBA1", "RGS5", "PDGFRB", "COL1A1","C1QB", "AIF1","CRYM", "FOXJ1", "CCDC153", "ID3", "AQP4", "GFAP", "PDGFRA", "OLIG2", "FYN","SYT1","NKX2-1", "NKX6-2", "HOPX", "EGFR", "PAX6") #
GeneList[[2]] = c(unique(c("GFAP", "AQP4","APOE", "SLC1A3", "S100B", "SLC1A2","ALDOC",  "GJA1", "AGT",  "ID3",    "VCAN", "SLC7A11", "RGCC"))) #Astro
GeneList[[3]] = c(unique(c("GJA1", "GFAP", "AQP4","CRYM", "FRZB","VCAN","SIX6", "RFX3","RGCC","RFX4",  "NELL2"))) #Tanycytes
GeneList[[4]] = c("CRYM", "AQP4", "CCDC153", "FOXJ1", "NPAS3", "RFX2", "RFX3", "RFX4") #Ependy
GeneList[[5]] = c(unique(c("SLC38A5","CLDN5", "ITM2A", "FLT1","KCNJ8","RGS5", "PDGFRB", "ABCC9","ACTA2", "TAGLN"))) #Vasculature
GeneList[[6]] = c(unique(c("APOD","OLIG2", "OLIG1","CD9", "PDGFRA", "PLLP", "FYN","CSPG4", "FGF12", "GINS2", "MBP", "CNP", "MAG",   "TNS3",  "TCF7L2", "MOG", "PLP1", "SERINC5", "OLIG1"))) #Oligodendrocytes
GeneList[[7]] = c(unique(c("C1QB", "AIF1", "P2RY12",  "MKI67"))) #Micro
GeneList[[8]] = c("HOPX", "STMN2", "NKX6-2") #oRG
GeneList[[9]] = c("HOPX", "PAX6", "NES", "EGFR", "NKX6-2") #vRG
GeneList[[10]] = c("HOPX", "EGFR", "NES", "HES1", "OLIG2") #IPC
GeneList[[11]] = c("HOPX", "NKX6-2", "NKX2-1", "NES", "EGFR") #Gen RG
GeneList[[12]] = c("HDC", "NKX2-1", "TBX3", "SIM1", "POU3F2", "GABRA5", "NPTX2", "FEZF1", "HCRT", "PMCH", "PPP1R17", "LHX1", "SIX6", "ZIC1", "ONECUT2",  "FOXB1",  "EBF2", "LHX6", "PNOC", "MEIS2")#All Hypothalamic Neurons
GeneList[[13]] = c("HDC", "TBX3", "NKX2-1", "SST", "SOX2") #TM
GeneList[[14]] = c("TBX3", "POMC","LEPR", "NPY1R", "AGRP", "NPY","PCSK1",  "OTP", "NHLH2", "NKX2-1",  "TRH", "GHRH", "SOX3", "NPAS4", "GSX1", "HMX2", "DLX1", "DLX2") #ARC
GeneList[[15]] = c("SIM1", "POU3F2", "OTP", "PCSK1", "NHLH2", "OXT", "MC4R", "SST", "AVP", "CRH", "TRH") #PVH
GeneList[[16]] = c("FEZF1", "GABRA5", "NPTX2", "NKX2-1", "ADCYAP1", "NR5A1", "SOX14", "PCSK1", "TRH", "SST") #VMH
GeneList[[17]] = c("HCRT", "LHX9", "PMCH", "TAC1", "SST", "NPVF", "PDYN", "PCSK1") #LH
GeneList[[18]] = c("PNOC", "NTS", "PENK") #Preoptic
GeneList[[19]] = c("LHX1", "SIX6", "ZIC1", "ZIC2",  "RELN", "CCK", "SIX3", "OTP", "RELN", "TRH") #SCN
GeneList[[20]] = c("PPP1R17", "NKX2-1", "SOX2") #DMH
GeneList[[21]] = c("LHX6", "SST") #IDTT
GeneList[[22]] = c("ONECUT2", "ONECUT3", "PITX2", "FOXB1", "FOXP1", "FOXP2", "TRH", "NKX2-1", "TAC1", "LHX1", "LHX5") #MN
GeneList[[23]] = c("EBF2", "EBF3", "TAC1", "SOX2") #SMN
GeneList[[24]] = c("MEIS2", "SIX3", "SST", "GAL", "TRH") #ZI


#Generate all UMAPS
Plot_List = list()
for(y in seq(1, length(All_SeuratFiles),1)){
SeuFile =  All_SeuratFiles[[y]]
Meta = All_Metadata[[y]]
Genes = GeneList[[y]]
Meta$Pop = factor(Meta$Pop, levels = rev(c(sort(unique(Meta$Pop)))))
SeuFile = AddMetaData(SeuFile, Meta, "Assigns")
Idents(SeuFile) = "Assigns"

PT = gsub("Neurons: ", "", PlotNames[y])
Plot_List[[paste(PlotNames[y], "UMAP")]] = DimPlot(SeuFile, reduction="umap")  + ggtitle(PT) + NoLegend() + theme(axis.title.x = element_text(size = 8), axis.title.y = element_text(size = 8), axis.text.x = element_text(size = 8), axis.text.y = element_text(size = 8), plot.title = element_text(size = 12))
  #Generate UMAPs


if(y %in% c(6,7,8,9,10,11,13,14,15,16,17,18,19,20,21,22,23,24)){ #Remove labels with gene names
Meta$Pop = gsub(" .*", "", Meta$Pop)
}

if(y %in% c(2,3,4,12)){ #Ensure all violins are in ascending order
  Meta$Pop = factor(Meta$Pop, levels = c(sort(unique(Meta$Pop))))
}else{
  Meta$Pop = factor(Meta$Pop, levels = rev(c(sort(unique(Meta$Pop)))))
}

#Multiple violin plots
SeuFile = AddMetaData(SeuFile, Meta, "Assigns")
Idents(SeuFile) = "Assigns"

DefaultAssay(SeuFile) = "RNA"
StorePlots = list()
BlankPlots = 20 - length(Genes)
AllVlns =list()
  for(x in Genes[1]){
            plotA <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotA <- plotA + coord_flip()+ theme(axis.ticks.x = element_blank(), axis.title.y=element_blank(), axis.text.x =element_blank(), axis.title.x=element_blank(),  axis.text.y = element_text(size=6), axis.ticks.y = element_blank(), legend.position = "none", plot.title = element_text(size=6))
            StorePlots[[x]] = plotA 
            }
  for(x in Genes[2:length(Genes)]){
            plotB <- VlnPlot(SeuFile, features = x, pt.size = 0, same.y.lims = F,)
            plotB <- plotB + coord_flip()+ theme(axis.ticks.x = element_blank(), axis.title.y=element_blank(), axis.text.x =element_blank(), axis.title.x=element_blank(), axis.ticks.y = element_blank(), legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(size=6))
           StorePlots[[x]] = plotB
  }
  for(x in 1:BlankPlots){
  StorePlots[[paste(x, "blank")]] = plot_spacer()
  }
Plot_List[[paste(PlotNames[y], "Violins")]] = wrap_plots(StorePlots, widths=c(2, rep(1, 19)), ncol =24, nrow = 1)
}

#Export all
pdf("FetalData_AllUMAPs_10JAN22_V2.pdf", width = 4, height = 4)
print(Plot_List)
dev.off()


pdf("FetalData_AllNeuronMAPS_2FEB22.pdf", width = 10, height = 8)
print(Plot_List[["Neurons: Tuberomammillary Nucleus UMAP"]] + Plot_List[["Neurons: Arcuate Nucleus UMAP"]] + Plot_List[["Neurons: Paraventricular Nucleus UMAP"]] + Plot_List[["Neurons: Ventromedial Nucleus UMAP"]] + Plot_List[["Neurons: Lateral Hypothalamus UMAP"]] + Plot_List[["Neurons: Preoptic Area UMAP"]] + Plot_List[["Neurons: Suprachiasmatic Nuclues UMAP"]] + Plot_List[[ "Neurons: Dorsomedial Nucleus UMAP"]] + Plot_List[["Neurons: Intrahypothalamic Diagonal UMAP"]] + Plot_List[["Neurons: Mammillary Nucleus UMAP"]] + Plot_List[["Neurons: Supramammillary Nucleus UMAP"]] + Plot_List[["Neurons: Zona Incerta UMAP"]])
dev.off()

```
    
#Compile list of most granular assignmnets
```{r}
All_Subclustered = rbind(SEPTEMBER_ASTRO_Assigns, OCTOBER_TANY_Assigns, OCTOBER_EPENDY_Assigns, OCTOBER_ENDOMURAL_Assigns, OCTOBER_OLIGO_Assigns, OCTOBER_MICROGLIA_Assigns, oRG_SubclusterAssigns, vRG_SubclusterAssigns, IPC_SubclusterAssigns, GenRG_SubclusterAssigns, OCT_TM_Assigns_Renamed, OCT_ARC_Assigns_Renamed, OCT_PVH_Assigns_Renamed, OCT_VMH_Assigns_Renamed, OCT_LH_Assigns_Renamed, OCT_PREOPTIC_Assigns_Renamed, OCT_SCN_Assigns_Renamed, OCT_DMH_Assigns_Renamed, OCT_IDTT_Assigns_Renamed, OCT_MN_Assigns_Renamed, OCT_SMN_Assigns_Renamed, OCT_ZI_Assigns_Renamed)

UknNeurons = subset(OCT_Neuron_Assigns_Renamed2, row.names(OCT_Neuron_Assigns_Renamed2) %notin% row.names(All_Subclustered))
UknNeurons$Pop = gsub("Neuron_48", "UnknownHypothalamicNeurons_01", gsub("Neuron_49", "UnknownHypothalamicNeurons_02", gsub("Neuron_50", "UnknownHypothalamicNeurons_03", UknNeurons$Pop)))

RestCells = subset(OCT_AllCell_Assigns, row.names(OCT_AllCell_Assigns) %notin% c(row.names(All_Subclustered), row.names(All_Subclustered)))
RestCells$Pop = gsub("Neurons \\[Mature\\]", "Non-HypothalamicNeurons", RestCells$Pop)

All_GranularAssignments = rbind(All_Subclustered, UknNeurons, RestCells) ##MOST GRANULAR LIST OF ASSIGNMENTS - 195 populations
```
